<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Quantitative Trait Mapping: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css">
<script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="manifest" href="site.webmanifest">
<link rel="mask-icon" href="safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/aio.html';">Instructor View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="assets/images/incubator-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Quantitative Trait Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Quantitative Trait Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Quantitative Trait Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/aio.html">Instructor View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="intro_data_set.html">1. Introduction to the Data Set</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="input-file-format.html">2. Input File Format</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="calc-genoprob.html">3. Calculating Genotype Probabilities</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="perform-genome-scan.html">4. Performing a genome scan</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="calc-kinship.html">5. Calculating A Kinship Matrix</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="perform-genome-scan-lmm.html">6. Performing a genome scan with a linear mixed model</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="perform-genome-scan-bin.html">7. Performing a genome scan with binary traits</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="perform-perm-test.html">8. Performing a permutation test</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="find-lod-peaks.html">9. Finding QTL peaks</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="est-qtl-effects.html">10. Estimating QTL effects</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="do_qtl_mapping.html">11. QTL Mapping in Diversity Outbred Mice</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Reference</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-intro_data_set"><p>Content from <a href="intro_data_set.html">Introduction to the Data Set</a></p>
<hr>
<p>Last updated on 2024-10-16 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/intro_data_set.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What data will we be using in this workshop?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the experimental design of the data set.</li>
<li>Understand the goals of the experiment.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>In the first part of this lesson, we will be analyzing data from a
mouse experiment involving Type 2 diabetes (T2D). There are two types of
diabetes: type 1, in which the immune system attacks insulin-secreting
cells and prevents insulin production, and type 2, in which the pancreas
makes less insulin and the body becomes less responsive to insulin.</p>
<p><img src="fig/healthy-vs-T2D.png" alt="Figure showing Type 2 diabetes &amp; insulin." class="figure"> Created in
BioRender.com</p>
<p>This study is from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4649649/" class="external-link">Tian et
al</a> and involves an intercross between the diabetes-resistant
C57BL/6J (B6 or B) strain and the diabetes-susceptible BTBR T+ tf/J
(BTBR or R) strain mice carrying a Leptin<sup>ob/ob</sup> mutation.</p>
<figure><img src="fig/intercross.png" alt="Figure showing intercross breeding design." class="figure mx-auto d-block"></figure><p>The <ob> mutation causes the mice to not produce <a href="https://en.wikipedia.org/wiki/Leptin" class="external-link">leptin</a>, a hormone that
regulates hunger and satiety. When leptin levels are low (or missing),
the body does not receive satiety signals and continues to feel hunger.
Leptin<sup>ob/ob</sup> mice continue to eat and become obese. Obesity is
one of the risk factors for T2D and this experiment sought to use
genetic variation between B6 and BTBR strains to identify genes which
influence T2D.</ob></p>
<p>This study measured insulin and glucose levels in mice at 10 weeks,
at which time the mice were euthanized. After euthanasia, the author’s
harvested six tissues, adipose, gastrocnemius muscle, hypothalamus,
pancreatic islets, kidney, and liver, and measured transcript levels via
gene expression microarray.</p>
<p>In this study, we will analyze circulating insulin levels and
pancreatic islet gene expression. We will map circulating insulin levels
to identify genomic loci which influence insulin levels. We will then
use the pancreatic islet gene expression data to identify candidate
genes.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Leptin<sup>ob/ob</sup> mice do now produce insulin and become obese
due to overeating.</li>
<li>This study crossed mice carrying the Leptin<sup>ob/ob</sup> mutation
in C57BL/6J and BTBR T+ tf/J.</li>
<li>C57BL/6J mice are resistant to diabetes and BTBR mice are
susceptible.</li>
<li>By crossing these two strains, the authors aimed to identify genes
which influence susceptibility to T2D.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-input-file-format"><p>Content from <a href="input-file-format.html">Input File Format</a></p>
<hr>
<p>Last updated on 2024-10-16 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/input-file-format.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>“How are the data files formatted for qtl2?”</li>
<li>“Which data files are required for qtl2?”</li>
<li>“Where can I find sample data for mapping with the qtl2
package?”</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>To specify which input files are required for qtl2 and how they
should be formatted.</li>
<li>To locate sample data for qtl mapping.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>QTL mapping data consists of a set of tables of data: sample
genotypes, phenotypes, marker maps, etc. These different tables are in
different comma-separated value (CSV) files. In each file, the first
column is a set of IDs for the rows, and the first row is a set of IDs
for the columns. For example, the genotype data file will have
individual IDs in the first column, marker names for the rest of the
column headers.</p>
<figure><img src="fig/attie_geno_sample.png" alt="Table showing the mouse genotypes as BB, BR, and RR." class="figure mx-auto d-block"><div class="figcaption">Attie Sample Genotypes</div>
</figure><p>The sample genotype file above shows two alleles: B and R. These
represent the founder strains for an intercross, which are C57BL/6 (BB)
and BTBR (RR) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4649649/" class="external-link">Tian et
al</a>. The B and R alleles themselves represent the haplotypes
inherited from the parental strains C57BL/6 and BTBR.</p>
<p>In the Diversity Outbred (DO) and Collaborative Cross (CC), alleles A
to H represent haplotypes of the 8 founder strains.</p>
<figure><img src="fig/cc-founder-alleles.png" alt="Figure showing the colors and letter codes of the CC/DO founders." class="figure mx-auto d-block"><div class="figcaption">CC and DO Founder Alleles</div>
</figure><p>CC lines have very low heterozygosity throughout their genomes. For
most loci, CC lines will be homozygous for one of the founder strains
A-H above, and as such will have one of only 8 genotypes (<em>e.g.</em>
AA, BB, CC, …). In contrast, DO <em>mice</em> (not lines) have high
heterozygosity throughout their genomes. Each locus will have one of 36
possible genotypes (<em>e.g.</em> AA, AB, AC, …, BB, BC, BD,…).</p>
<p><img src="fig/cc-do-genome-comparison.png" alt="Figure showing CC and DO genomes" class="figure"> For the purposes of learning
QTL mapping, this lesson begins with the simplest cases: backcrosses (2
possible genotypes) and intercrosses (3 possible genotypes). Once we
have learned how to use <code>qtl2</code> for these simpler cases, we
will advance to the most complex case involving mapping in DO mice.</p>
<p>R/qtl2 accepts the following files: 1. genotypes 2. phenotypes 3.
phenotype covariates (<em>i.e.</em> tissue type, time points)<br>
4. genetic map<br>
5. physical map (optional)<br>
6. control file (YAML or JSON format, not CSV)</p>
<p>We use both a genetic marker map and a physical map (if available). A
sample from a genetic map of MIT markers is shown here.</p>
<figure><img src="fig/attie_geno_map_sample.png" alt="Table showing the marker, chromosome, and centimorgan position for five markers" class="figure mx-auto d-block"><div class="figcaption">Attie Genetic Map</div>
</figure><p>A physical marker map provides location in bases rather than
centiMorgans.</p>
<figure><img src="fig/attie_phys_map_sample.png" alt="Table showing top five rows of physical marker map." class="figure mx-auto d-block"><div class="figcaption">Attie Physical Map</div>
</figure><p>Numeric phenotypes are separate from the often non-numeric
covariates.</p>
<figure><img src="fig/attie_pheno_sample.png" alt="Table showing top five rows of phenotype table, including insulin" class="figure mx-auto d-block"><div class="figcaption">Attie Phenotypes</div>
</figure><p>Phenotype covariates are <a href="https://en.wikipedia.org/wiki/Metadata" class="external-link">metadata</a> describing
the phenotypes. For example, in the case of a phenotype measured over
time, one column in the phenotype covariate data could be the time of
measurement. For gene expression data, we would have columns
representing chromosome and physical position of genes, as well as gene
IDs.</p>
<figure><img src="fig/attie_covar_sample.png" alt="Table showing the top five rows of covariates table" class="figure mx-auto d-block"><div class="figcaption">Attie Covariates</div>
</figure><p>In addition to the set of CSV files with the primary data, we need a
separate control file with various control parameters (or metadata),
including the names of all of the other data files and the genotype
codes used in the genotype data file. The control file is in a specific
format using either <a href="http://www.yaml.org" class="external-link">YAML</a> or <a href="http://json.org" class="external-link">JSON</a>; these are human-readable text files for
representing relatively complex data.</p>
<p><img src="fig/attie_control_file_sample.png" alt="Attie Control File" class="figure">{alt=‘Figure showing the qtl2 control
file’,width=75%}</p>
<p>A big advantage of this control file scheme is that it greatly
simplifies the function for reading in the data. That function,
<code>read_cross2()</code>, has a <em>single</em> argument: the name
(with path) of the control file.</p>
<p>For further details, see the separate <a href="http://kbroman.org/qtl2/assets/vignettes/input_files.html" class="external-link">vignette
on the input file format</a>.</p>
<div id="challenge-1-what-data-does-qtl2-need" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-what-data-does-qtl2-need" class="callout-inner">
<h3 class="callout-title">Challenge 1: What data does qtl2 need?</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Which data files are required by <code>qtl2</code>?<br>
</li>
<li>Which ones are optional?<br>
</li>
<li>How should they be formatted?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>marker genotypes, phenotypes, genetic map</li>
<li>physical map</li>
<li>csv; JSON or YAML for control file</li>
</ol>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="sample-data-sets">Sample data sets<a class="anchor" aria-label="anchor" href="#sample-data-sets"></a>
</h2>
<hr class="half-width">
<p>In this lesson, we’ll work with data sets included in the
<code>qtl2</code> package. You can find out more about the <a href="http://kbroman.org/qtl2/pages/sampledata.html" class="external-link">sample data
files</a> from the R/qtl2 web site. Zipped versions of these datasets
are included with the <a href="https://github.com/rqtl/qtl2geno" class="external-link">qtl2geno</a> package and can be
loaded into R using the <code>read_cross2()</code> function. Additional
sample data sets, including data on Diversity Outbred (DO) mice, are
available at <a href="https://github.com/rqtl/qtl2data" class="external-link uri">https://github.com/rqtl/qtl2data</a>.</p>
<div id="challenge-2-can-you-do-it" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-can-you-do-it" class="callout-inner">
<h3 class="callout-title">Challenge 2: Can you do it?</h3>
<div class="callout-content">
<p>Go to <a href="https://github.com/rqtl/qtl2data" class="external-link uri">https://github.com/rqtl/qtl2data</a> to view additional
sample data.<br>
1). Find the Recla data and locate the phenotype data file. Open the
file by clicking on the file name. What is in the first column? the
first row?<br>
2). Locate the genotype data file, click on the file name, and view the
raw data. What is in the first column? the first row?<br>
3). Locate the covariates file and open it by clicking on the file name.
What kind of information does this file contain?<br>
4). Locate the control file (YAML or JSON format) and open it. What kind
of information does this file contain?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>1). What is in the first column of the phenotype file? Animal ID. The
first row? Phenotype variable names - OF_distance_first4, OF_distance,
OF_corner_pct, OF_periphery_pct, …<br>
2). What is in the first column of the genotype file? marker ID. the
first row? Animal ID - 1,4,5,6,7,8,9,10, …<br>
3). Locate the covariates file and open it. What kind of information
does this file contain? Animal ID, sex, cohort, group, subgroup, ngen,
and coat color.<br>
4). Locate the control file (YAML or JSON format) and open it. What kind
of information does this file contain? Names of primary data files,
genotype and allele codes, cross type, description, and other
metadata.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="preparing-your-diversity-outbred-do-data-for-qtl2">Preparing your Diversity Outbred (DO) data for qtl2<a class="anchor" aria-label="anchor" href="#preparing-your-diversity-outbred-do-data-for-qtl2"></a>
</h2>
<hr class="half-width">
<p>Karl Broman provides detailed instructions for <a href="https://kbroman.org/qtl2/pages/prep_do_data.html" class="external-link">preparing DO
mouse data for R/qtl2</a>.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>“QTL mapping data consists of a set of tables of data: marker
genotypes, phenotypes, marker maps, etc.”</li>
<li>“These different tables are in separate comma-delimited (CSV)
files.”</li>
<li>“In each file, the first column is a set of IDs for the rows, and
the first row is a set of IDs for the columns.”</li>
<li>“In addition to primary data, a separate file with control
parameters (or metadata) in either <a href="http://www.yaml.org" class="external-link">YAML</a> or <a href="http://json.org" class="external-link">JSON</a> format is required.”</li>
<li>“Published and public data already formatted for QTL mapping are
available on the web.”</li>
<li>“These data can be used as a model for formatting your own QTL
data.”</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-calc-genoprob"><p>Content from <a href="calc-genoprob.html">Calculating Genotype Probabilities</a></p>
<hr>
<p>Last updated on 2024-10-16 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/calc-genoprob.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I calculate QTL at positions between genotyped markers?</li>
<li>How do I calculate QTL genotype probabilities?</li>
<li>How do I calculate allele probabilities?</li>
<li>How can I speed up calculations if I have a large data set?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>To explain why the first step in QTL analysis is to calculate
genotype probabilities.</li>
<li>To calculate genotype probabilities.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>The first task in QTL analysis is to calculate conditional genotype
probabilities, given the observed marker data, at each putative QTL
position. For example, the first step would be to determine the
probabilities for genotypes BR and RR at the locus indicated below.</p>
<figure><img src="fig/unknown_genotype.png" alt="adapted from Broman &amp; Sen, 2009" class="figure mx-auto d-block"><div class="figcaption">adapted from Broman &amp; Sen, 2009</div>
</figure><p>The <code>calc_genoprob()</code> function calculates QTL genotype
probabilities, conditional on the available marker data. These are
needed for most of the QTL mapping functions. The result is returned as
a list of three-dimensional arrays (one per chromosome). Each 3d array
of probabilities is arranged as individuals <span class="math inline">\(\times\)</span> genotypes <span class="math inline">\(\times\)</span> positions.</p>
<figure><img src="fig/threeD_array.png" alt="Figure showing three-dimensional array of genoprobs" class="figure mx-auto d-block"><div class="figcaption">Three-dimensional genotype probabilities
array</div>
</figure><p><img src="http://venus.ifca.unican.es/Rintro/_images/dataStructuresNew.png" alt="See this page for a graphical review of data structures in R" class="figure">.</p>
<p>We’ll use the <a href="https://thejacksonlaboratory.box.com/shared/static/svw7ivp5hhmd7vb8fy26tc53h7r85wez.zip" class="external-link">Attie
BL6/BTBR dataset</a> from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4649649/" class="external-link">Tian et
al</a> (an intercross) as an example. In this study, circulating insulin
levels were measured in an F2 cross between mouse strains C57BL/6J and
BTBTR T+ <tf>.</tf></p>
<p>First, we will load in the <a href="https://kbroman.org/qtl2/" class="external-link">qtl2</a> library, which provides the
functions that we will use for QTL analysis.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cross</span> <span class="op">&lt;-</span> <span class="fu">read_cross2</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_b6btbr_grcm39/attie_control.json'</span><span class="op">)</span></span></code></pre>
</div>
<p>To load your own data from your machine, you would use the file path
to your data files. For example, if the file path to your data files is
<code>/Users/myUserName/qtlProject/data</code>, the command to load your
data would look like this:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myQTLdata</span> <span class="op">&lt;-</span> <span class="fu">read_cross2</span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/Users/myUserName/qtlProject/data/myqtldata.json"</span> <span class="op">)</span></span></code></pre>
</div>
<p>The JSON file contains all control information for your data,
including names of data files, cross type, column specifications for sex
and cross information, and more. This can also be in YAML format.
Alternatively, all data files can be zipped together for loading.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myQTLdata</span> <span class="op">&lt;-</span> <span class="fu">read_cross2</span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/Users/myUserName/qtlProject/data/myqtldata.zip"</span> <span class="op">)</span></span></code></pre>
</div>
<p>Back to the BTBR data. Now look at a summary of the cross data and
the names of each variable within the data.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">cross</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Object of class cross2 (crosstype "f2")

Total individuals             490
No. genotyped individuals     490
No. phenotyped individuals    490
No. with both geno &amp; pheno    490

No. phenotypes                  3
No. covariates                  8
No. phenotype covariates        0

No. chromosomes                20
Total markers                2057

No. markers by chr:
  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19   X
156 135 157 126 125 102 109  91  93 123 124 116 116  91 102  66  60  95  50  20 </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">cross</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] "crosstype"  "geno"       "gmap"       "pmap"       "pheno"
 [6] "covar"      "is_x_chr"   "is_female"  "cross_info" "alleles"   </code></pre>
</div>
<p>Have a look at the markers listed in the genetic map,
<code>gmap</code>. Markers are listed by chromosome and described by cM
position. View only the markers on the first several chromosomes.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">gmap</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>$`1`
rs13475697  rs3681603 rs13475703 rs13475710  rs6367205 rs13475716 rs13475717
 0.1881141  0.1920975  0.4167755  0.6488793  0.6555814  0.6638576  0.6676198
rs13475719 rs13459050  rs3680898 rs13475727 rs13475728 rs13475729 rs13475731
 0.6711377  0.6749344  0.6775292  1.8149573  1.9596637  2.3456569  2.7186389
rs13475737 rs13475744  rs6397513 rs13475747 rs13475748 rs13475749 rs13475750
 3.1059517  3.8222865  4.3094607  4.3120150  4.5098582  4.8154609  4.8853505
rs13475751 rs13475752 rs13475762 rs13475764 rs13475765 rs13475768 rs13475769
 4.8869793  4.8902179  7.2954871  8.2102887  8.3708197  8.7178703  8.8859153
rs13475771  rs6384194 rs13475790  rs3676270 rs13475794 rs13475801  rs4222269
 9.1374722  9.9295192  9.9970634 10.1508878 10.3962716 11.5981956 11.9606369
 rs6387241 rs13475822 rs13475824 rs13475826 rs13475827 rs13475834 rs13475880
16.8770742 16.9815396 17.4434784 18.0866148 18.6276972 19.2288050 27.4056813
rs13475883  rs6239834  rs3162895  rs6212146  rs3022802 rs13475899 rs13475900
28.4641674 30.8427150 31.1526514 31.2751278 31.3428706 31.8493556 31.8518088
rs13475906  rs3022803 rs13475907 rs13475909 rs13475912  rs6209698 rs13475929
32.2967145 32.3074644 32.3683291 32.8001894 33.6026526 36.5341646 37.6881435
rs13475931 rs13475933 rs13475934  rs4222476 rs13475939  rs8253473 rs13475941
37.7429827 38.0416271 38.0430095 38.9647582 39.4116688 39.4192277 39.4871064
rs13475944 rs13475947 rs13475948 rs13475950 rs13475951 rs13475954 rs13475955
39.7672829 40.2599440 40.3380113 40.3417592 40.3439501 41.1407252 41.2887176
rs13475963 rs13475966 rs13475967 rs13475970 rs13475960  rs6250696 rs13475973
42.4744416 42.5667702 42.9736574 43.1427994 43.5985261 43.5992946 43.6014053
 rs3691187 rs13475988 rs13475991 rs13476023 rs13476024  rs3684654  rs6274257
44.6237384 45.7855528 46.0180221 47.8579278 47.8600317 48.2423958 48.9612178
rs13476045 rs13476049  rs6319405 rs13476050 rs13476051 rs13476054 rs13476057
49.2018340 49.3701384 49.4261039 49.4275718 49.4323558 49.4972616 49.5031830
rs13476059 rs13476060 rs13476062 rs13476066 rs13476067  rs6259837 rs13476080
49.5084008 49.5113545 49.6085043 49.6644819 50.1779477 50.8256056 51.0328603
 rs6302966 rs13476085 rs13476089  rs3717360 rs13476090  rs6248251 rs13476091
51.3659553 51.6974451 52.3869798 52.3903517 52.3936241 52.4228715 52.5787388
 rs3088725  rs3022832  rs4222577 rs13476100  rs6263067  rs8256168  rs6327099
53.4044231 53.4129004 53.4189013 54.3267003 54.4193890 55.1459517 55.3274320
rs13476111 rs13476119  rs8236484  rs8270838  rs8236489 rs13476129 rs13476134
55.9050491 56.8936305 56.9852502 57.1870637 58.0248893 58.7605079 59.5401544
rs13476135 rs13476137 rs13476138 rs13476140 rs13476148  rs6202860 rs13476158
59.5426193 59.6023794 60.3355828 60.3439598 61.1791787 61.9905512 61.9930265
rs13476163 rs13476177 rs13476178 rs13476183 rs13476184  rs6194543 rs13476196
62.0039607 62.6243588 62.6269118 63.8101331 64.0856907 66.4047817 66.7425394
rs13476201  rs3685700  rs3022846 rs13476210 rs13476214 rs13459163  rs4222816
67.2638714 68.7230251 68.7246243 69.1209547 70.1550813 75.5548371 75.5593190
 rs4222820  rs3090340  rs8245949 rs13476242 rs13476251 rs13476254  rs6383012
75.5593202 75.5637846 76.7508053 79.0157673 79.7644000 79.8248805 85.3173344
rs13476279  rs6348421 rs13476290 rs13476300 rs13476302 rs13476304  rs3669814
86.7653503 88.2128991 89.0565541 94.6215368 94.8227821 94.8269227 95.5413280
rs13501301 rs13476316
96.0784002 96.9960494

$`2`
rs13476325 rs13476327 rs13476328 rs13476330  rs3695983 rs13476334 rs13476337
  1.329379   1.760872   1.839732   1.950151   1.954566   2.265170   3.619681
rs13476342  rs3696091 rs13476348  rs3681847 rs13476358 rs13476424 rs13476427
  4.113919   5.308337   6.708720   7.382269  10.053730  21.102901  21.547272
rs13476432 rs13476433 rs13476438 rs13476440 rs13476445 rs13476446 rs13476448
 22.457323  22.458829  22.472305  22.502827  22.703755  22.706606  22.809631
rs13476449 rs13476451 rs13476452 rs13476456  rs6333344 rs13476459  rs6288325
 22.811905  22.816989  22.917366  23.496527  23.499872  23.505774  23.764498
rs13476470 rs13476482  rs6203572 rs13476485 rs13476501 rs13476502 rs13476503
 25.671278  26.670288  26.702994  27.001404  29.014864  29.017621  29.019099
rs13476525 rs13476526 rs13476530  rs3660779 rs13476533 rs13476534  rs4223189
 31.635258  31.882499  32.606343  33.182340  33.622217  33.881289  33.886227
 rs6205317 rs13476536  rs3709716 rs13476538 rs13476540  rs6314726 rs13476543
 34.125908  34.182229  34.777129  35.011099  35.196852  35.859566  36.045536
rs13476544  rs6222797 rs13476546 rs13476553 rs13476554  rs8263229 rs13476565
 36.199822  36.656504  36.783097  37.624894  37.628034  38.509801  39.813261
rs13476566  rs3670631 rs13476583 rs13476586 rs13476592 rs13476595 rs13476645
 40.314536  41.215814  43.089828  43.274068  45.681933  46.026997  50.383519
rs13476655 rs13476660  rs6378047 rs13476661 rs13476666 rs13476667 rs13476669
 50.804234  51.471634  51.474692  51.557954  52.083779  52.257580  52.543537
rs13476672  rs3724460  rs3143279 rs13476687  rs6278009 rs13476693  rs3022892
 52.594599  53.100014  54.071835  54.256293  54.721524  55.005598  55.109510
rs13476700 rs13476702 rs13476703 rs13476705  rs4223406  rs3090608 rs13476739
 55.226885  55.233738  55.400903  55.678374  55.949739  57.841429  58.879499
rs13476747 rs13476754  rs6257970 rs13476758  rs3022901 rs13476769  rs4223486
 59.218037  59.231937  59.742629  60.304581  61.226450  61.739777  61.741096
rs13476774  rs4223511 rs13476778 rs13476783  rs6234650 rs13476784 rs13476786
 61.912021  62.703328  63.294674  63.904805  64.118754  64.219897  64.224107
rs13476788  rs3682725 rs13476801 rs13476803 rs13476816  rs6170159  rs3022909
 64.987248  65.971246  66.931607  67.071161  69.496928  69.584633  69.689891
rs13476819 rs13476822 rs13476823  rs3716380  rs6332517 rs13476826 rs13476827
 69.754708  71.191302  71.405084  71.816078  71.959179  72.019045  72.087910
rs13476830 rs13476831 rs13476832  rs4223605  rs3022932 rs13476872  rs3692409
 72.364799  72.368511  72.371502  76.460822  76.621947  78.531760  79.528125
 rs3726342 rs13476882  rs3673613  rs8260429  rs8275858 rs13476892 rs13476894
 80.337464  81.259831  82.295892  82.969957  83.669130  84.046818  84.787594
 rs3024096 rs13476907 rs13476910 rs13476918  rs3673248 rs13476928 rs13476934
 87.473424  87.498263  87.955210  92.672743  94.072805  96.238835  98.601444
rs13476935 rs13476936
 99.057281  99.455950

$`3`
 rs8246929 rs13476972 rs13476973 rs13476974  rs3693600  rs3716641 rs13476982
  1.080189   1.347091   1.470204   1.476022   1.553792   1.553874   1.563144
rs13476983 rs13476984  rs6398851 rs13476992 rs13476993 rs13476994 rs13476995
  1.569943   1.573624   1.576176   2.809096   3.100703   3.300776   3.380060
 rs6168642 rs13477004  rs6410894 rs13477005 rs13477006 rs13477007 rs13477019
  3.552793   4.223627   4.585754   4.691027   4.791228   5.088769   7.798872
rs13477027 rs13477030 rs13477031 rs13477043  rs3677091  rs4223883  rs6371982
  9.282482   9.532585   9.606423  13.791018  14.074175  14.939077  15.157908
rs13477059  rs6246699 rs13477061  rs3022959  rs3140619 rs13477063 rs13477066
 15.918550  15.923947  15.992721  16.802271  16.802276  16.963472  16.970994
 rs6257041 rs13477071 rs13477078 rs13477079 rs13477108 rs13477109 rs13477110
 17.319985  17.755564  18.693494  18.744862  21.080665  21.413925  22.016411
 rs3685081 rs13477111  rs4223969 rs13477121 rs13477134 rs13477137 rs13477138
 22.640148  22.641512  25.049880  25.052026  27.282182  27.588349  27.591228
rs13477140  rs4223979 rs13477142 rs13477143 rs13477144  rs6326655 rs13477153
 27.693007  27.696543  27.696853  27.700581  28.329883  28.503475  28.562688
rs13477154 rs13477155 rs13477156 rs13477160 rs13477168 rs13477169 rs13477174
 28.613511  28.615742  28.694417  28.702636  29.332336  29.349436  30.386145
rs13477176 rs13477179  rs3023594 rs13477183 rs13477187 rs13477188 rs13477189
 30.734338  31.007845  31.105126  31.112997  31.165690  31.167829  31.170044
 rs6351657  rs6289734 rs13477194 rs13477201  rs8273664 rs13477207 rs13477209
 31.173493  31.174332  32.184066  32.341020  32.342736  32.914386  32.973448
rs13477210 rs13477216 rs13477217 rs13477220 rs13477224  rs6196421 rs13477225
 33.009762  33.484516  33.517478  34.047286  34.586747  34.589541  34.590880
rs13477228 rs13477230 rs13477233  rs6183172 rs13477236  rs4224039 rs13477248
 35.026131  35.324560  35.893062  36.115534  36.317250  36.816125  37.403199
rs13477252  rs6224931 rs13477268 rs13477263 rs13477271 rs13477279 rs13477282
 37.639920  38.791882  39.157460  39.160076  39.405423  41.031706  41.504565
rs13477283 rs13477287 rs13477288 rs13477291  rs3159432  rs3159588  rs3162061
 41.544820  41.692166  41.775032  42.169443  42.366378  42.367700  42.369817
rs13477294  rs4224158 rs13477318 rs13477369 rs13477373 rs13477374 rs13477379
 42.957412  44.537215  45.502206  50.866600  51.429778  51.504763  52.411466
 rs3022967 rs13477381 rs13477384 rs13477385 rs13477396 rs13477397 rs13477409
 52.430405  53.262919  53.368879  53.372015  55.570112  55.709649  59.858613
rs13477428 rs13477429 rs13477433 rs13477434 rs13477438 rs13477439  rs3698409
 61.510633  61.670155  61.958144  62.047669  62.546905  62.688373  62.743012
rs13477442 rs13477444 rs13477448  rs6385968 rs13477450 rs13477451 rs13477456
 63.177344  63.279576  63.722373  63.935680  63.936910  63.938561  65.312917
rs13477457 rs13477458 rs13477459 rs13477460 rs13477461 rs13477467 rs13477472
 65.314857  65.317267  65.318925  65.320875  65.322208  67.548098  69.605885
rs13477529 rs13477473  rs3022971 rs13477478 rs13477482 rs13477485  rs3724805
 69.608224  69.609224  70.843094  70.907058  71.149729  74.456309  74.457760
rs13477486 rs13477488 rs13477489 rs13477492 rs13477495  rs4224567 rs13477497
 74.739366  74.745908  74.832780  75.275285  75.481388  75.596886  75.684561
rs13477500 rs13477511 rs13477526
 76.429887  79.219312  80.957067

$`4`
rs13477532 rs13477541 rs13477542 rs13477543 rs13477544 rs13477549 rs13477550
 0.3991292  1.2560214  1.3917929  1.5928861  1.6978705  2.0515920  2.2621396
rs13477552 rs13477553 rs13477554 rs13477556 rs13477558 rs13477559 rs13477561
 2.4352574  2.5251773  2.7585040  3.0530491  3.2150744  3.4585919  3.4626618
 rs6403469  rs4137687 rs13477566 rs13477579 rs13477580 rs13477585 rs13477586
 3.7029303  3.9430172  3.9449684  5.4077682  5.4078174  5.4306980  5.5055395
 rs3709372 rs13477589 rs13477592 rs13459074 rs13477595 rs13477596  rs6163246
 5.6750754  5.6767158  5.9597782  6.1852266  7.1180568  7.1205590  7.1217001
 rs6266094 rs13477605 rs13477606 rs13477611  rs4224426 rs13477616 rs13477619
 7.1234642  8.6068427  8.6549771  8.7910377  8.9901964  9.5194951  9.7572785
rs13477623  rs3679673 rs13477625 rs13477626 rs13477627 rs13477628 rs13477631
11.4021807 11.4058442 11.4073390 11.4114893 11.8595138 11.9573244 12.6677290
rs13477642 rs13477643  rs3703981 rs13477659 rs13477660 rs13477662 rs13477664
15.2447672 15.2467497 15.6652248 18.3775143 18.5369150 18.6189365 18.6215940
rs13477715 rs13477717  rs6239799 rs13477722 rs13477724 rs13477725 rs13477730
27.9120152 28.1256700 29.9945260 30.0009463 30.0051069 30.1318185 30.6953939
 rs4224515 rs13477732 rs13477733  rs6184584 rs13477739  rs6254381 rs13476245
30.7119837 30.8926924 31.1603646 31.6555204 32.3233798 32.4243902 32.4320470
rs13477743  rs8276754 rs13477745 rs13477746 rs13477759  rs3654162 rs13477785
32.4370370 32.5459374 32.7020530 32.9891931 33.4688108 34.3363190 35.7680418
 rs3717147 rs13477793 rs13477794 rs13477795  rs6323325 rs13477814 rs13477821
35.7771234 35.7913993 35.9203487 36.0115374 37.1562808 38.6853237 39.1848617
rs13477823 rs13477824 rs13477835  rs6361351 rs13477854 rs13477860 rs13477863
39.1885557 39.5402592 40.6564169 40.7350605 42.5809515 43.7049055 44.1279677
rs13477865  rs6401724 rs13477919 rs13477888 rs13477907 rs13477908 rs13477910
44.1349533 46.3675418 47.9452479 47.9493262 49.9169893 49.9768431 49.9868206
rs13477912 rs13477914 rs13477915 rs13477916 rs13477920  rs4224727 rs13477931
50.0962204 50.1002715 50.1016595 50.1100592 50.1276921 51.6805005 51.8938626
 rs6173859 rs13477938 rs13477940  rs6365760 rs13477947 rs13477954 rs13477966
52.0518071 53.1881228 53.5004864 53.9066510 55.3233700 55.9983634 59.7633519
rs13477976 rs13477980  rs6355453 rs13477983 rs13477985 rs13459077 rs13477991
61.7882160 61.9716108 63.0252404 63.0303080 63.0330779 64.5217020 64.9138820
rs13477999 rs13478021 rs13459080  rs3023026  rs3711383 rs13478027 rs13478028
66.3759417 70.2574630 71.8991146 72.3118040 73.8896348 74.1570955 76.0540649
rs13478031  rs3023027 rs13478039  rs3668420 rs13478048 rs13478050 rs13478054
76.3810043 77.0004907 77.1047625 77.2968965 79.8090093 79.9367987 81.0633814

$`5`
rs13478092  rs4225033 rs13478097  rs6245801 rs13478103 rs13478104 rs13478106
 0.5148540  0.5601752  1.4578186  1.6459862  1.6508495  1.6521104  1.7168465
 rs8277477 rs13478110 rs13478111 rs13481337 rs13481338 rs13481339 rs13481340
 1.7415309  1.9911696  1.9937269  2.2408610  2.2660419  2.2898547  2.2972730
 rs3669635 rs13481349 rs13478115 rs13478119 rs13478123 rs13478127  rs6256752
 4.5647660  4.6978589  4.9630837  6.4001473  6.9122371  7.6593491  8.1438747
rs13478135  rs6295831 rs13478138  rs3023036 rs13478139 rs13478141 rs13478142
 8.3149190  8.3964728  8.3989242  8.4007825  8.4033521  8.6865375  8.6885014
rs13478143  rs3090417 rs13478145 rs13478146 rs13478148  rs6322061  rs6309009
 8.8977080  9.1431643 10.2406114 10.4115566 10.6617649 11.1715598 11.2373953
rs13478154 rs13478160 rs13478161 rs13478162 rs13478167 rs13459084 rs13478170
11.3740589 12.8940905 12.9954134 13.2099571 15.1865866 15.5507310 15.5578335
rs13459083 rs13478174  rs6196732 rs13478175 rs13478178 rs13478179 rs13459085
15.6270239 15.7081356 15.8280039 16.1317236 16.1364419 16.1481976 16.2543350
rs13478181 rs13478182 rs13478184 rs13478185  rs6346598 rs13478204 rs13478205
16.3646745 16.4752220 17.4437033 17.4455556 17.9704569 20.1679510 20.7422551
rs13478207 rs13478208 rs13478209 rs13478221 rs13478222 rs13478241 rs13478244
20.8151842 20.8793971 20.8804978 22.5458012 22.5484334 24.8718483 25.1324122
rs13478245 rs13478252  rs6237983 rs13478254 rs13478255  rs6263715 rs13478257
25.1347167 25.3480011 25.6909676 26.7835663 27.2330524 27.3672129 27.9932349
rs13478261 rs13478262 rs13478263 rs13478264 rs13478265 rs13478268 rs13478271
28.8161178 28.8414537 28.8728759 29.0427959 29.1322946 29.2399880 29.9682578
rs13478276 rs13478279 rs13478280 rs13478282 rs13478283 rs13478285  rs3684754
29.9813803 30.1772663 30.2076157 30.2106527 30.2140050 30.3329785 34.3825206
rs13478311 rs13478320 rs13478367 rs13478368  rs6167407 rs13478386 rs13478387
34.5022391 35.7707846 41.4115154 41.4669600 42.7614533 42.7644662 43.2203787
rs13478391  rs6232866 rs13478393 rs13478395 rs13478396 rs13478402 rs13478403
43.2305343 44.4209462 45.0370667 45.7058881 45.7104424 45.7465608 45.7527138
 rs4225380  rs6370004 rs13478410 rs13478422 rs13478424 rs13478426 rs13478430
46.0876570 46.2329778 46.2430427 47.0473590 47.0514238 47.4628441 48.7119128
rs13478442 rs13478443 rs13478445 rs13478452 rs13478454 rs13478455 rs13478458
49.9285133 49.9304123 49.9361657 51.7267769 51.7292199 51.7325687 51.9937088
rs13478536 rs13478542 rs13478546 rs13478547  rs8265976  rs4225536 rs13478570
70.3211281 73.2280807 74.3892894 74.3918300 75.2624965 75.4633566 79.8560921
rs13478574  rs3023062  rs4138683  rs3668534 rs13478593 rs13459095
81.1622904 83.1121830 85.1593074 86.9092920 87.4283034 87.4895523

$`6`
rs13459096 rs13478602  rs6172481 rs13478606 rs13478610 rs13478667 rs13478694
 0.2471448  0.2509530  0.2595298  0.2688093  0.7422945 10.1262711 11.5504738
rs13478696 rs13478697 rs13478698 rs13478705  rs6163979 rs13478712  rs3024195
12.4131565 12.5038661 12.5050255 13.5351920 13.5357026 13.9919190 13.9955592
rs13478720 rs13478721 rs13478727  rs3696518 rs13478728  rs6214044 rs13478729
16.3415722 16.3443428 19.9127889 19.9793499 20.1028494 20.1045540 20.1064628
 rs6191358 rs13478730  rs6168553  rs3023068  rs3023069 rs13478756 rs13478757
20.1802678 20.2900546 22.8125569 23.8816735 23.8817887 24.2350441 24.2361490
rs13478758 rs13478759 rs13478761 rs13478762 rs13478768 rs13478802 rs13478819
24.2385415 24.3438438 24.4218302 25.3208587 25.8918938 29.0540752 30.9696170
rs13478827 rs13478829 rs13478830 rs13478833 rs13478834 rs13478843 rs13478845
31.0889580 31.0940937 31.1459713 32.0501760 32.0754991 33.3794730 33.4494137
rs13478852 rs13478856 rs13478858 rs13478863 rs13478864 rs13478865 rs13459097
33.6822587 34.0640838 34.3991792 34.5451575 34.5462936 34.7572579 35.5171427
rs13478871  rs4221995  rs4226063 rs13478876 rs13478897 rs13478901 rs13478904
36.0826404 36.2882063 36.3528729 36.5675727 40.5244122 41.3426572 41.7358722
rs13478905 rs13478906 rs13478913 rs13478915 rs13478916 rs13478917 rs13478919
42.4408121 42.8756819 43.2898350 43.3444653 43.4229502 43.5628361 43.6888274
rs13478927 rs13478932 rs13478933 rs13478934 rs13478935 rs13478936 rs13478941
45.1993835 45.9039023 46.0695345 46.0719750 46.1226745 46.1662896 46.3825629
rs13478948 rs13478950  rs6344812  rs3675615 rs13478966 rs13478971 rs13478977
47.4986393 47.5033268 47.5109362 48.2958455 48.8980754 50.2242802 50.8859454
rs13478978 rs13478983 rs13478985 rs13478990 rs13479058 rs13479059  rs4226339
51.2879832 51.6367222 51.6425091 52.3587248 64.7346868 64.8288933 65.1405567
rs13479070 rs13479071  rs3024135  rs8262456 rs13479085 rs13479086  rs3089737
66.4917745 67.2476353 68.2936963 70.9494089 75.6833494 75.7983622 75.8754886
rs13479087 rs13479088  rs3023105  rs3090435  rs3023103 rs13479099 rs13479091
76.1284889 76.1348324 76.1353741 76.1370335 76.1407613 76.1430157 76.6498568
 rs3023840 rs13479093  rs6155595 rs13479094
76.6518167 76.9251446 76.9270209 76.9292552 </code></pre>
</div>
<p>Next we use <code>calc_genoprob()</code> to calculate the QTL
genotype probabilities.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">probs</span> <span class="op">&lt;-</span> <span class="fu">calc_genoprob</span><span class="op">(</span>cross      <span class="op">=</span> <span class="va">cross</span>, </span>
<span>                       map        <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">gmap</span>, </span>
<span>                       error_prob <span class="op">=</span> <span class="fl">0.002</span><span class="op">)</span></span></code></pre>
</div>
<p>The argument <code>error_prob</code> supplies an assumed genotyping
error probability of 0.002. If a value for <code>error_prob</code> is
not supplied, the default probability is 0.0001.</p>
<p>Recall that the result of <code>calc_genoprob</code>,
<code>probs</code>, is a list of three-dimensional arrays (one per
chromosome).</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">probs</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] "1"  "2"  "3"  "4"  "5"  "6"  "7"  "8"  "9"  "10" "11" "12" "13" "14" "15"
[16] "16" "17" "18" "19" "X" </code></pre>
</div>
<p>Each three-dimensional array of probabilities is arranged as
individuals <span class="math inline">\(\times\)</span> genotypes <span class="math inline">\(\times\)</span> positions. Have a look at the
names of each of the three dimensions for chromosome 19.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dimnames</span><span class="op">(</span><span class="va">probs</span><span class="op">$</span><span class="va">`19`</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[[1]]
  [1] "Mouse3051" "Mouse3551" "Mouse3430" "Mouse3476" "Mouse3414" "Mouse3145"
  [7] "Mouse3656" "Mouse3242" "Mouse3427" "Mouse3527" "Mouse3281" "Mouse3405"
 [13] "Mouse3530" "Mouse3477" "Mouse3498" "Mouse3526" "Mouse3284" "Mouse3160"
 [19] "Mouse3655" "Mouse3615" "Mouse3491" "Mouse3603" "Mouse3191" "Mouse3130"
 [25] "Mouse3330" "Mouse3199" "Mouse3614" "Mouse3577" "Mouse3081" "Mouse3204"
 [31] "Mouse3513" "Mouse3219" "Mouse3331" "Mouse3301" "Mouse3503" "Mouse3083"
 [37] "Mouse3568" "Mouse3189" "Mouse3287" "Mouse3131" "Mouse3311" "Mouse3357"
 [43] "Mouse3149" "Mouse3256" "Mouse3644" "Mouse3217" "Mouse3212" "Mouse3082"
 [49] "Mouse3156" "Mouse3535" "Mouse3481" "Mouse3123" "Mouse3359" "Mouse3555"
 [55] "Mouse3597" "Mouse3624" "Mouse3314" "Mouse3128" "Mouse3531" "Mouse3295"
 [61] "Mouse3231" "Mouse3496" "Mouse3438" "Mouse3183" "Mouse3052" "Mouse3237"
 [67] "Mouse3462" "Mouse3293" "Mouse3543" "Mouse3276" "Mouse3200" "Mouse3502"
 [73] "Mouse3171" "Mouse3364" "Mouse3524" "Mouse3334" "Mouse3355" "Mouse3254"
 [79] "Mouse3358" "Mouse3468" "Mouse3192" "Mouse3214" "Mouse3536" "Mouse3606"
 [85] "Mouse3226" "Mouse3393" "Mouse3415" "Mouse3266" "Mouse3648" "Mouse3224"
 [91] "Mouse3474" "Mouse3381" "Mouse3138" "Mouse3660" "Mouse3616" "Mouse3425"
 [97] "Mouse3554" "Mouse3196" "Mouse3528" "Mouse3312" "Mouse3045" "Mouse3585"
[103] "Mouse3471" "Mouse3308" "Mouse3628" "Mouse3429" "Mouse3324" "Mouse3124"
[109] "Mouse3291" "Mouse3452" "Mouse3373" "Mouse3367" "Mouse3579" "Mouse3647"
[115] "Mouse3169" "Mouse3335" "Mouse3122" "Mouse3635" "Mouse3154" "Mouse3484"
[121] "Mouse3652" "Mouse3612" "Mouse3668" "Mouse3233" "Mouse3175" "Mouse3306"
[127] "Mouse3046" "Mouse3663" "Mouse3165" "Mouse3519" "Mouse3592" "Mouse3127"
[133] "Mouse3184" "Mouse3650" "Mouse3599" "Mouse3494" "Mouse3605" "Mouse3505"
[139] "Mouse3573" "Mouse3561" "Mouse3489" "Mouse3480" "Mouse3186" "Mouse3421"
[145] "Mouse3607" "Mouse3346" "Mouse3375" "Mouse3633" "Mouse3589" "Mouse3094"
[151] "Mouse3611" "Mouse3307" "Mouse3133" "Mouse3152" "Mouse3518" "Mouse3209"
[157] "Mouse3056" "Mouse3320" "Mouse3365" "Mouse3313" "Mouse3441" "Mouse3339"
[163] "Mouse3352" "Mouse3159" "Mouse3619" "Mouse3238" "Mouse3203" "Mouse3137"
[169] "Mouse3509" "Mouse3289" "Mouse3054" "Mouse3432" "Mouse3487" "Mouse3179"
[175] "Mouse3572" "Mouse3285" "Mouse3466" "Mouse3252" "Mouse3517" "Mouse3546"
[181] "Mouse3185" "Mouse3665" "Mouse3537" "Mouse3096" "Mouse3600" "Mouse3349"
[187] "Mouse3098" "Mouse3275" "Mouse3667" "Mouse3342" "Mouse3333" "Mouse3300"
[193] "Mouse3244" "Mouse3478" "Mouse3560" "Mouse3501" "Mouse3315" "Mouse3440"
[199] "Mouse3669" "Mouse3486" "Mouse3632" "Mouse3319" "Mouse3453" "Mouse3172"
[205] "Mouse3121" "Mouse3590" "Mouse3215" "Mouse3447" "Mouse3618" "Mouse3340"
[211] "Mouse3047" "Mouse3666" "Mouse3516" "Mouse3225" "Mouse3167" "Mouse3207"
[217] "Mouse3631" "Mouse3444" "Mouse3168" "Mouse3298" "Mouse3602" "Mouse3309"
[223] "Mouse3416" "Mouse3260" "Mouse3146" "Mouse3374" "Mouse3144" "Mouse3485"
[229] "Mouse3610" "Mouse3348" "Mouse3500" "Mouse3613" "Mouse3253" "Mouse3384"
[235] "Mouse3664" "Mouse3206" "Mouse3426" "Mouse3332" "Mouse3210" "Mouse3283"
[241] "Mouse3670" "Mouse3120" "Mouse3274" "Mouse3461" "Mouse3202" "Mouse3472"
[247] "Mouse3437" "Mouse3434" "Mouse3593" "Mouse3055" "Mouse3234" "Mouse3422"
[253] "Mouse3571" "Mouse3236" "Mouse3049" "Mouse3350" "Mouse3249" "Mouse3326"
[259] "Mouse3134" "Mouse3143" "Mouse3493" "Mouse3361" "Mouse3636" "Mouse3436"
[265] "Mouse3510" "Mouse3117" "Mouse3601" "Mouse3303" "Mouse3497" "Mouse3544"
[271] "Mouse3463" "Mouse3118" "Mouse3354" "Mouse3162" "Mouse3464" "Mouse3181"
[277] "Mouse3188" "Mouse3356" "Mouse3521" "Mouse3591" "Mouse3241" "Mouse3467"
[283] "Mouse3469" "Mouse3262" "Mouse3643" "Mouse3548" "Mouse3372" "Mouse3542"
[289] "Mouse3563" "Mouse3583" "Mouse3584" "Mouse3208" "Mouse3661" "Mouse3659"
[295] "Mouse3195" "Mouse3459" "Mouse3653" "Mouse3649" "Mouse3382" "Mouse3180"
[301] "Mouse3386" "Mouse3084" "Mouse3205" "Mouse3299" "Mouse3515" "Mouse3540"
[307] "Mouse3255" "Mouse3177" "Mouse3523" "Mouse3366" "Mouse3567" "Mouse3557"
[313] "Mouse3114" "Mouse3623" "Mouse3419" "Mouse3580" "Mouse3271" "Mouse3385"
[319] "Mouse3492" "Mouse3119" "Mouse3232" "Mouse3598" "Mouse3150" "Mouse3310"
[325] "Mouse3164" "Mouse3587" "Mouse3050" "Mouse3627" "Mouse3506" "Mouse3413"
[331] "Mouse3435" "Mouse3151" "Mouse3112" "Mouse3630" "Mouse3646" "Mouse3223"
[337] "Mouse3187" "Mouse3263" "Mouse3637" "Mouse3662" "Mouse3508" "Mouse3550"
[343] "Mouse3125" "Mouse3545" "Mouse3570" "Mouse3641" "Mouse3136" "Mouse3626"
[349] "Mouse3166" "Mouse3269" "Mouse3529" "Mouse3218" "Mouse3625" "Mouse3448"
[355] "Mouse3378" "Mouse3227" "Mouse3651" "Mouse3182" "Mouse3304" "Mouse3617"
[361] "Mouse3141" "Mouse3552" "Mouse3479" "Mouse3658" "Mouse3539" "Mouse3190"
[367] "Mouse3093" "Mouse3097" "Mouse3126" "Mouse3170" "Mouse3229" "Mouse3520"
[373] "Mouse3582" "Mouse3351" "Mouse3129" "Mouse3153" "Mouse3450" "Mouse3113"
[379] "Mouse3586" "Mouse3549" "Mouse3538" "Mouse3201" "Mouse3556" "Mouse3247"
[385] "Mouse3455" "Mouse3176" "Mouse3344" "Mouse3343" "Mouse3439" "Mouse3629"
[391] "Mouse3286" "Mouse3216" "Mouse3588" "Mouse3488" "Mouse3221" "Mouse3142"
[397] "Mouse3428" "Mouse3111" "Mouse3353" "Mouse3211" "Mouse3569" "Mouse3280"
[403] "Mouse3325" "Mouse3368" "Mouse3553" "Mouse3245" "Mouse3228" "Mouse3135"
[409] "Mouse3622" "Mouse3095" "Mouse3369" "Mouse3609" "Mouse3410" "Mouse3302"
[415] "Mouse3594" "Mouse3483" "Mouse3197" "Mouse3336" "Mouse3507" "Mouse3305"
[421] "Mouse3532" "Mouse3250" "Mouse3194" "Mouse3449" "Mouse3178" "Mouse3198"
[427] "Mouse3620" "Mouse3596" "Mouse3638" "Mouse3222" "Mouse3147" "Mouse3163"
[433] "Mouse3273" "Mouse3473" "Mouse3578" "Mouse3465" "Mouse3279" "Mouse3558"
[439] "Mouse3443" "Mouse3490" "Mouse3460" "Mouse3248" "Mouse3243" "Mouse3431"
[445] "Mouse3564" "Mouse3347" "Mouse3565" "Mouse3525" "Mouse3574" "Mouse3329"
[451] "Mouse3140" "Mouse3257" "Mouse3328" "Mouse3193" "Mouse3132" "Mouse3220"
[457] "Mouse3235" "Mouse3499" "Mouse3246" "Mouse3270" "Mouse3608" "Mouse3442"
[463] "Mouse3157" "Mouse3642" "Mouse3566" "Mouse3139" "Mouse3282" "Mouse3053"
[469] "Mouse3454" "Mouse3363" "Mouse3213" "Mouse3654" "Mouse3514" "Mouse3341"
[475] "Mouse3401" "Mouse3388" "Mouse3604" "Mouse3161" "Mouse3451" "Mouse3634"
[481] "Mouse3482" "Mouse3559" "Mouse3645" "Mouse3264" "Mouse3155" "Mouse3251"
[487] "Mouse3297" "Mouse3541" "Mouse3158" "Mouse3294"

[[2]]
[1] "BB" "BR" "RR"

[[3]]
 [1] "rs4232073"  "rs13483548" "rs13483549" "rs13483550" "rs13483554"
 [6] "rs13483555" "rs3090321"  "rs3090137"  "rs6309315"  "rs13483577"
[11] "rs3090325"  "rs13483579" "rs13483584" "rs13483586" "rs13483587"
[16] "rs13483589" "rs13483592" "rs13483593" "rs6344448"  "rs13483594"
[21] "rs13483595" "rs3705022"  "rs13483609" "rs13483612" "rs13483648"
[26] "rs13483650" "rs13483654" "rs13483658" "rs13483660" "rs13483664"
[31] "rs13483666" "rs13483667" "rs13483670" "rs8275553"  "rs8275912"
[36] "rs13483677" "rs13483679" "rs13483680" "rs13483681" "rs3660143"
[41] "rs13483682" "rs13483683" "rs13483685" "rs13483686" "rs6355398"
[46] "rs4222106"  "rs13483690" "rs13483693" "rs13483695" "rs13483699"</code></pre>
</div>
<p>View the first three rows of genotype probabilities for the first
genotyped marker on chromosome 19.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">probs</span><span class="op">$</span><span class="va">`19`</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,,<span class="st">"rs4232073"</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                    BB           BR           RR
Mouse3051 1.317728e-11 1.235895e-07 9.999999e-01
Mouse3551 9.999840e-01 1.595361e-05 5.027172e-08
Mouse3430 1.317728e-11 1.235895e-07 9.999999e-01
Mouse3476 9.999999e-01 1.235895e-07 1.317728e-11
Mouse3414 6.179474e-08 9.999999e-01 6.179474e-08</code></pre>
</div>
<p>We can also view the genotype probabilities using <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_genoprob.R" class="external-link">plot_genoprob</a>.
The arguments to this function specify:</p>
<ol style="list-style-type: decimal">
<li>
<code>probs</code>: the genotype probabilities,</li>
<li>
<code>map</code>: the marker map,</li>
<li>
<code>ind</code>: the index of the individual to plot,</li>
<li>
<code>chr</code>: the index of the chromosome to plot.</li>
</ol>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_genoprob</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>              map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>              ind   <span class="op">=</span> <span class="fl">1</span>, </span>
<span>              chr   <span class="op">=</span> <span class="fl">19</span>, </span>
<span>              main  <span class="op">=</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="st">'19'</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/calc-genoprob-rendered-plot_genoprob-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The coordinates along chromosome 19 are shown on the horizontal axis
and the three genotypes are shown on the vertical axis. Higher genotype
probabilities are plotted in darker shades. This mouse has a RR genotype
on the proximal end of the chromosome and transitions to BR.</p>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<!-- DMG: Not sure about this. I'd rather plot more samples or chromosomes. -->
<p>1). Load a second dataset from Arabidopsis recombinant inbred lines
(<a href="https://www.genetics.org/content/195/3/1077" class="external-link">Moore et al,
Genetics, 2013</a>) in a study of plant root response to gravity
(gravitropism).</p>
<p><code>grav &lt;- read_cross2(file = system.file('extdata', 'grav2.zip', package = 'qtl2'))</code><br>
2). How many individuals were in the study? How many phenotypes? How
many chromosomes?<br>
3). Insert pseudomarkers at 1cM intervals and save the results to an
object called <code>gravmap</code>. Have a look at the first
chromosome.<br>
4). Calculate genotype probabilities and save the results to an object
called <code>gravpr</code>. View the genotypes for the first three
markers and pseudomarkers on chromosome 1 for the first five
individuals.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>1).
<code>grav &lt;- read_cross2(file = system.file('extdata', 'grav2.zip', package = 'qtl2'))</code><br>
2). <code>summary(grav)</code><br>
3).
<code>gravmap &lt;- insert_pseudomarkers(map = grav$gmap, step = 1)</code><br>
followed by <code>head(gravmap, n=1)</code><br>
4).
<code>gravpr  &lt;- calc_genoprob(cross = grav, map = gravmap)</code>
followed by<br><code>(gravpr$``1``)[1:5,,"PVV4"]</code>,
<code>(gravpr$</code>1<code>)[1:5,,"c1.loc1"]</code>, and<br><code>(gravpr$</code>1<code>)[1:5,,"c1.loc2"]</code></p>
<!-- DMG: What about this? -->
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span>  <span class="op">=</span> <span class="fu">maxmarg</span><span class="op">(</span><span class="va">probs</span><span class="op">)</span></span>
<span><span class="va">ph</span> <span class="op">=</span> <span class="fu">guess_phase</span><span class="op">(</span><span class="va">cross</span>, <span class="va">m</span><span class="op">)</span></span>
<span><span class="fu">plot_onegeno</span><span class="op">(</span><span class="va">ph</span>, <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/calc-genoprob-rendered-unnamed-chunk-1-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure>
</div>
</div>
</div>
</div>
<div id="challenge-3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3" class="callout-inner">
<h3 class="callout-title">Challenge 3</h3>
<div class="callout-content">
<p>Calculate genotype probabilities for a different data set from the <a href="https://github.com/rqtl/qtl2data" class="external-link">qtl2 data repository</a>, this
one from a study of obesity and diabetes in a C57BL/6 (B6) × BTBR
intercross.<br>
1). Create a new script in RStudio with File -&gt; New File -&gt; R
Script.<br>
2) Download the B6 x BTBR zip file from the <a href="https://github.com/rqtl/qtl2data" class="external-link">qtl2 data repository</a> into an
object called <code>b6btbr</code> by running this code:<br><code>b6btbr &lt;- read_cross2(file = "https://raw.githubusercontent.com/rqtl/qtl2data/master/B6BTBR/b6btbr.zip")</code><br>
3). View a summary of the <code>b6btbr</code> data. How many
individuals? phenotypes? chromosomes? markers?<br>
4). View the genetic map for the <code>b6btbr</code> data.<br>
5). Insert pseudomarkers at 2 cM intervals. Assign the results to an
object called <code>b6btbrmap</code>.<br>
6). Calculate genotype probabilities assuming a genotyping error
probability of 0.001. Assign the results to an object called
<code>b6btbrpr</code>.<br>
7). View the first several rows of genotype probabilities for any marker
on chromosome 18.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>1). Create a new script in RStudio with File -&gt; New File -&gt; R
Script.<br>
2).
<code>b6btbr &lt;- read_cross2(file = "https://raw.githubusercontent.com/rqtl/qtl2data/master/B6BTBR/b6btbr.zip</code><br>
3). <code>summary(b6btbr)</code> shows 544 individuals, 3 phenotypes, 20
chromosomes, 2057 markers.<br>
4). <code>b6btbr$gmap</code><br>
5).
<code>b6btbrmap &lt;- insert_pseudomarkers(map=b6btbr$gmap, step=2)</code><br>
6).
<code>b6btbrpr &lt;- calc_genoprob(cross=b6btbr, map=b6btbrmap, error_prob=0.001)</code><br>
7). <code>dimnames((b6btbrpr$</code>18<code>))</code> shows all marker
names for chromosome 18.
<code>head((b6btbrpr$</code>18<code>)[,,"c18.loc48"])</code> gives
genotype probabilities for an example pseudomarker, while
<code>head((b6btbrpr$</code>18<code>)[,,"rs6338896"])</code> gives
genotype probabilities for a genotyped marker.</p>
</div>
</div>
</div>
</div>
<p><strong>Parallel calculations (optional)</strong> To speed up the
calculations with large datasets on a multi-core machine, you can use
the argument <code>cores</code>. With <code>cores=0</code>, the number
of available cores will be detected via
<code>parallel::detectCores()</code>. Otherwise, specify the number of
cores as a positive integer.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">probs</span> <span class="op">&lt;-</span> <span class="fu">calc_genoprob</span><span class="op">(</span>cross <span class="op">=</span> <span class="va">iron</span>, map <span class="op">=</span> <span class="va">map</span>, error_prob <span class="op">=</span> <span class="fl">0.002</span>, cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<p><strong>Allele probabilities (optional)</strong> The genome scan
functions use genotype probabilities as well as a matrix of phenotypes.
If you wished to perform a genome scan via an additive allele model, you
would first convert the genotype probabilities to allele probabilities,
using the function <code>genoprob_to_alleleprob()</code>.</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">apr</span> <span class="op">&lt;-</span> <span class="fu">genoprob_to_alleleprob</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span><span class="op">)</span></span></code></pre>
</div>
<p>The figure below shows genotype and allele probabilities for 3
samples. In the Diversity Outbred, there are 36 possible genotype states
(AA, AB, AC, …, BB, BC, BD, …, CC, CD, CE, …, DD, DE, DF, …, EE,…) or 8
+ 7 + 6 + 5 + 4 + 3 + 2 + 1. The first SNP below has genotype BB. In the
table describing alleles (8 state founder probabilities), the
probability that this SNP has a B allele is 1. The 2nd SNP has genotype
BH, so the allele table shows a probability of 0.5 for B and 0.5 for H.
The third SNP is either BG or BH, and has a probability of 0.5 for each
of these genotypes. The allele table shows a probability of 0.5 for
allele B, and 0.25 for both G and H.</p>
<p><img src="fig/geno-to-allele-probs.png" class="figure">
::::::::::::::::::::::::::::::::::::: keypoints</p>
<ul>
<li>The first step in QTL analysis is to calculate genotype
probabilities.</li>
<li>Calculate genotype probabilities between genotyped markers with
<code>calc_genoprob()</code>.</li>
</ul>
<p>::::::::::::::::::::::::::::::::::::::::::::::::</p></section><section id="aio-perform-genome-scan"><p>Content from <a href="perform-genome-scan.html">Performing a genome scan</a></p>
<hr>
<p>Last updated on 2024-10-16 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/perform-genome-scan.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I perform a genome scan?</li>
<li>How do I plot a genome scan?</li>
<li>How do additive covariates differ from interactive covariates?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Map one trait using additive covariates.</li>
<li>Map the sample trait using additive and interactive covariates.</li>
<li>Plot a genome scan.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>The freely available chapter on <a href="http://www.rqtl.org/book/rqtlbook_ch04.pdf" class="external-link">single-QTL
analysis</a> from Broman and Sen’s <a href="http://www.rqtl.org/book/" class="external-link">A
Guide to QTL Mapping with R/qtl</a> describes different methods for QTL
analysis. Two of these methods are described here using data from an
experiment on hypertension in the mouse <a href="https://s3.amazonaws.com/academia.edu.documents/45963759/geno.2000.640120160526-29022-36mpgg.pdf?AWSAccessKeyId=AKIAIWOWYYGZ2Y53UL3A&amp;Expires=1513786158&amp;Signature=rtodlYwe0LDmYZFOm1ejvZjZhQ0%3D&amp;response-content-disposition=inline%3B%20filename%3DConcordance_of_murine_quantitative_trait.pdf" class="external-link">Sugiyama
et al., Genomics 71:70-77, 2001</a>. The study employed a backcross
between two mouse strains resulting in two possible genotypes - AA for
homozygotes, and AB for heterozygotes.</p>
<p>Linear regression can be employed to identify presence of QTL in a
cross. To identify QTL using regression, we compare the fit for two
models: 1) the null hypothesis that there are no QTL anywhere in the
genome; and 2) the alternative hypothesis that there is a QTL near a
specific position. A sloped line indicates that there is a difference in
mean phenotype between the two genotype groups, and that a QTL is
present. A line with no slope indicates that there is no difference in
mean phenotype between the two groups, and that no QTL exists.
Regression aims to find the line of best fit to the data. In the case of
a backcross with only two genotypes, a t-test is performed at the marker
to determine whether the difference in phenotype means is zero.</p>
<figure><img src="fig/nullvalt.png" alt="adapted from Broman &amp; Sen, 2009" class="figure mx-auto d-block"><div class="figcaption">adapted from Broman &amp; Sen, 2009</div>
</figure><p>To find the line of best fit, the residuals or errors are calculated,
then squared for each data point.</p>
<p><img src="fig/residual.png" class="figure"><img src="fig/squared-residual.png" class="figure"><img src="fig/null.png" class="figure"></p>
<p>The line of best fit will be the one that minimizes the sum of
squared residuals, which maximizes the likelihood of the data.</p>
<p>Marker regression produces a LOD (logarithm of odds) score comparing
the null hypothesis to the alternative. The LOD score is calculated
using the sum of squared residuals for the null and alternative
hypotheses. The LOD score is the difference between the log10 likelihood
of the null hypothesis and the log10 likelihood of the alternative
hypothesis. It is related to the regression model above by identifying
the line of best fit to the data. A higher LOD score indicates greater
likelihood of the alternative hypothesis. A LOD score closer to zero
favors the null hypothesis.</p>
<p>Marker regression can identify the existence and effect of a QTL by
comparing means between groups, however, it requires known marker
genotypes and can’t identify QTL in between typed markers. To identify
QTL between typed markers, we use Haley-Knott regression. After <a href="https://smcclatchy.github.io/mapping/03-calc-genoprob/" class="external-link">calculating
genotype probabilities</a>, we can regress the phenotypes for animals of
unknown genotype on these conditional genotype probabilities
(conditional on known marker genotypes). In Haley-Knott regression,
phenotype values can be plotted and a regression line drawn through the
phenotype mean for the untyped individuals.</p>
<figure><img src="fig/hk-regress.png" class="figure mx-auto d-block"></figure><p>As shown by the green circle in the figure, an individual of unknown
genotype is placed between known genotypes according to the probability
of its genotype being AA or AB. In this case, the probability of this
individual having genotype AA is 0.6, and the probability of having
genotype AB is 0.4.</p>
<p>To perform a genome scan by Haley-Knott regression (<a href="https://www.ncbi.nlm.nih.gov/pubmed/16718932" class="external-link">Haley and Knott
1992</a>), use the function <code>scan1()</code>. <code>scan1()</code>
takes as input the genotype probabilities, a matrix of phenotypes, and
then optional additive and interactive covariates. Another option is to
provide a vector of weights.</p>
<section><h2 class="section-heading" id="additive-genome-scan">Additive Genome Scan<a class="anchor" aria-label="anchor" href="#additive-genome-scan"></a>
</h2>
<hr class="half-width">
<p>There are two potential covariates in the Attie data set. Let’s look
at the top of the covariates in the <code>cross</code> object.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">covar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             Sex pgm adipose_batch gastroc_batch hypo_batch islet_batch
Mouse3051   Male   1    12/19/2007     8/11/2008 11/26/2007  11/28/2007
Mouse3551   Male   1    12/19/2007     8/12/2008 11/27/2007  12/03/2007
Mouse3430   Male   1    12/19/2007     8/12/2008 11/27/2007  12/03/2007
Mouse3476   Male   1    12/19/2007     8/12/2008 11/27/2007  12/03/2007
Mouse3414   Male   1    12/18/2007     8/11/2008 11/26/2007  11/28/2007
Mouse3145 Female   1    12/19/2007     8/11/2008         NA  11/28/2007
          kidney_batch liver_batch
Mouse3051   07/15/2008       other
Mouse3551   07/16/2008  12/10/2007
Mouse3430   07/15/2008  12/05/2007
Mouse3476   07/16/2008  12/05/2007
Mouse3414   07/15/2008  12/05/2007
Mouse3145   07/15/2008  12/10/2007</code></pre>
</div>
<p><code>Sex</code> is potential covariate. It is a good idea to always
include sex in any analysis. Examples of other covariates might be age,
diet, treatment, or experimental batch. It is worth taking time to
identify covariates that may affect your results.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cross</span><span class="op">$</span><span class="va">covar</span><span class="op">$</span><span class="va">Sex</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">covar</span><span class="op">$</span><span class="va">Sex</span><span class="op">)</span></span>
<span></span>
<span><span class="va">addcovar</span> <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">Sex</span>, data <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">covar</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<p>When we perform a genome scan with additive covariates, we are
searching for loci that have the same effect in both covariate groups.
In this case, we are searching for loci that affect females and male in
the same way.</p>
<!-- DMG: We need a plot of what an additive effect would look like at one marker. -->
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_add</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                 pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">'log10_insulin_10wk'</span><span class="op">]</span>, </span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>On a multi-core machine, you can get some speed-up via the
<code>cores</code> argument, as with <code>calc_genoprob()</code> and
<code>calc_kinship()</code>.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_add</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                 pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">'log10_insulin_10wk'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                 cores     <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<p>The output of <code>scan1()</code> is a matrix of LOD scores,
positions × phenotypes.</p>
<p>Take a look at the first ten rows of the scan object. The numerical
values are the LOD scores for the marker named at the beginning of the
row. LOD values are shown for circulating insulin.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">lod_add</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>               pheno1
rs13475697 0.04674829
rs3681603  0.04674829
rs13475703 0.04680494
rs13475710 0.12953382
rs6367205  0.13734728
rs13475716 0.13735534
rs13475717 0.13735534
rs13475719 0.13735534
rs13459050 0.13735534
rs3680898  0.13735541</code></pre>
</div>
<p>The function <code>plot_scan1()</code> can be used to plot the LOD
curves. Use the argument <code>lodcolumn</code> to indicate which column
to plot.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span><span class="va">lod_add</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>           main <span class="op">=</span> <span class="st">'log(insulin): 10 weeks'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/perform-genome-scan-rendered-plot_add_lod-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The LOD plot for insulin shows several peaks, with the largest peak
on chromosome 2. There are smaller peaks on other chromosomes. Which of
these peaks is significant, and why? We’ll evaluate the significance of
genome scan results in a later episode in <a href="https://smcclatchy.github.io/mapping/10-perform-perm-test/" class="external-link">performing
a permutation test</a>.</p>
<!-- DMG: Suggestion: let's have the exercises have them plot the LOD score for
one chromosome, the find the marker with the maximum LOD score and try to
find it in the physical map. -->
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p>Use the <code>sort()</code> function to sort the LOD scores for
liver. Hint: run <code>dim(out)</code> for the row and column
dimensions, or <code>colnames(out)</code> for the column names.<br>
Which marker has the highest LOD score? Which genotyped marker has the
highest LOD score? What chromosome number are they on?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p><code>sort(out[,1])</code> for column 1 or
<code>sort(out[,"liver"])</code> for the column named “liver”.<br>
The pseudomarker with the largest score is c16.loc29, with a LOD of
7.68. The genotyped marker with the largest LOD is D16Mit30 with a score
of 7.28. Both are located on chromosome 16.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<p>Use the <code>sort()</code> function to sort the LOD scores for
spleen. Which pseudomarker has the highest LOD score? Which genotyped
marker has the highest LOD score? What chromosome number are they
on?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p><code>sort(out[,2])</code> or <code>sort(out[,"spleen"])</code> The
pseudomarker with the largest score is c9.loc57, with a LOD of 12.1. The
genotyped marker with the largest LOD is D9Mit182 with a score of 10.4.
Both are located on chromosome 9.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3" class="callout-inner">
<h3 class="callout-title">Challenge 3</h3>
<div class="callout-content">
<p>Plot the LOD scores for spleen. Does the genome scan for spleen share
any large-ish peaks with the scan for liver?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p><code>plot_scan1(out, map = map, lodcolumn = "spleen")</code> Both
liver and spleen genome scans share a peak on chromosome 8 with a LOD
score near 4.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="interactive-genome-scan">Interactive Genome Scan<a class="anchor" aria-label="anchor" href="#interactive-genome-scan"></a>
</h2>
<hr class="half-width">
<p>Above, we mapped insulin levels using sex as an additive covariate
and searched for loci where both sexes had the same effect. But what if
the two sexes have different effects? The we would like to map in a way
that allows each sex to have different effects. We do this using an
interactive genome scan.</p>
<p>You should always include the interactive covariates as additive
covariates as well. In this case, we only have sex as a covariate, so we
can use the additive covariate matrix. For clarity, we will make a copy
and name it for the interactive covariates.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">intcovar</span> <span class="op">=</span> <span class="va">addcovar</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_int</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                 pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">'log10_insulin_10wk'</span><span class="op">]</span>, </span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                 intcovar  <span class="op">=</span> <span class="va">intcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x <span class="op">=</span> <span class="va">lod_int</span>, map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, main <span class="op">=</span> <span class="st">'log10_insulin_10wk'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/perform-genome-scan-rendered-plot_int-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>It is difficult to tell if there is a difference in LOD scores
between the additive and interactive scans. To resolve this, we can plot
both genome scans in the same plo using the “add = TRUE” argument.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x <span class="op">=</span> <span class="va">lod_int</span>, map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, main <span class="op">=</span> <span class="st">'log10_insulin_10wk'</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x <span class="op">=</span> <span class="va">lod_add</span>, map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, col <span class="op">=</span> <span class="st">'blue'</span>, add <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/perform-genome-scan-rendered-plot_add_int_lod-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>It is still difficult to tell whether any peaks differ by sex.
Another way to view the plot is to plot the difference between the
interactive and additive scans.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x <span class="op">=</span> <span class="va">lod_int</span>, map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, main <span class="op">=</span> <span class="st">'log10_insulin_10wk'</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x <span class="op">=</span> <span class="va">lod_add</span>, map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, col <span class="op">=</span> <span class="st">'blue'</span>, add <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x <span class="op">=</span> <span class="va">lod_int</span> <span class="op">-</span> <span class="va">lod_add</span>, map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, col <span class="op">=</span> <span class="st">'red'</span>, add <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/perform-genome-scan-rendered-plot_add_int_diff_lod-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>While it was important to look at the effect of sex on the trait, in
this experiment, there do not appear to be any sex-specific peaks.</p>
<!-- DMG: Add challenge asking students why we didn't map and plot females and
males separately. Maybe they can even try it. -->
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>A qtl2 genome scan requires genotype probabilities and a phenotype
matrix.</li>
<li>The output from a genome scan contains a LOD score matrix, map
positions, and phenotypes.</li>
<li>LOD curve plots for a genome scan can be viewed with
plot_scan1().</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-calc-kinship"><p>Content from <a href="calc-kinship.html">Calculating A Kinship Matrix</a></p>
<hr>
<p>Last updated on 2024-10-16 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/calc-kinship.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>“Why would I calculate kinship between individuals?”</li>
<li>“How do I calculate kinship between individuals?”</li>
<li>“What does a kinship matrix look like?”</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Explain why and when kinship calculation matters in mapping.</li>
<li>Create a kinship matrix for individuals.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Population structure and kinship are common confounding factors in
genome-wide association studies (GWAS), case-control studies, and other
study types in genetics. They create false positive associations between
genotype and phenotype at genetic markers that differ in genotype
frequencies between subpopulations due to genetic relatedness between
samples. Simple association tests assume statistical independence
between individuals. Population structure and kinship confound
associations when phenotype covariance between individuals results from
genetic similarity. Accounting for relatedness between individuals helps
to distinguish true associations from false positives generated by
population structure or kinship.</p>
<p>As an example see the table below for phenotype and genotype
frequencies between two subpopulations in a case-control study.</p>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="center">subpop1</th>
<th align="center">subpop2</th>
<th align="center">overall pop</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">frequency</td>
<td align="center">0.5</td>
<td align="center">0.5</td>
<td align="center">1</td>
</tr>
<tr class="even">
<td align="left">probability of AA genotype</td>
<td align="center">0.1</td>
<td align="center">0.9</td>
<td align="center">0.5</td>
</tr>
<tr class="odd">
<td align="left">probability of disease</td>
<td align="center">0.9</td>
<td align="center">0.1</td>
<td align="center">0.5</td>
</tr>
<tr class="even">
<td align="left">probability of disease &amp; AA</td>
<td align="center">0.09</td>
<td align="center">0.09</td>
<td align="center">0.09</td>
</tr>
</tbody>
</table>
<p>The full population consists of two equally represented
subpopulations. In the overall population, the probability of the AA
genotype is 0.5, and the probability of disease is also 0.5. The joint
probability of both disease and AA genotype in the population (0.09) is
less than either the probability of disease (0.5) or the probability of
the AA genotype (0.5) alone, and is considerably less than the joint
probability of 0.25 that would be calculated if subpopulations weren’t
taken into account. In a case-control study that fails to recognize
subpopulations, most of the cases will come from subpopulation 1 since
this subpopulation has a disease probability of 0.9. However, this
subpopulation also has a low probability of the AA genotype. So a false
association between AA genotype and disease would occur because only
overall population probabilities would be considered.</p>
<p>Linear mixed models (LMMs) consider genome-wide similarity between
all pairs of individuals to account for population structure, known
kinship and unknown relatedness. They model the covariance between
individuals. Linear mixed models in association mapping studies can
successfully correct for genetic relatedness between individuals in a
population by incorporating kinship into the model. To perform a genome
scan by a linear mixed model, accounting for the relationships among
individuals (in other words, including a random polygenic effect),
you’ll need to calculate a kinship matrix for the individuals. This is
accomplished with the <code>calc_kinship()</code> function. It takes the
genotype probabilities as input.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kinship</span> <span class="op">&lt;-</span> <span class="fu">calc_kinship</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span><span class="op">)</span></span></code></pre>
</div>
<p>Take a look at the kinship values calculated for the first 5
individuals.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kinship</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          Mouse3051 Mouse3551 Mouse3430 Mouse3476 Mouse3414
Mouse3051     0.737     0.501     0.535     0.520     0.510
Mouse3551     0.501     0.758     0.515     0.512     0.499
Mouse3430     0.535     0.515     0.749     0.458     0.493
Mouse3476     0.520     0.512     0.458     0.749     0.434
Mouse3414     0.510     0.499     0.493     0.434     0.696</code></pre>
</div>
<p>We can also look at the first 50 mice in the kinship matrix.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_samples</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="fu">heatmap</span><span class="op">(</span><span class="va">kinship</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span><span class="op">]</span>, symm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/calc-kinship-rendered-plot_kinship-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The mice are listed in the same order on both sides of the matrix.
The comb-like structures are called “dendrograms” and they indicate how
the mice are clustered together. Each cell represents the degree of
allele sharing between mice. Red colors indicate higher kinship and
yellow colors indicate lower kinship. Each mouse is closely related to
itself, so the cells along the diagonal tend to be darker than the other
cells. You can see some evidence of related mice, possibly siblings, in
the orange-shaded blocks along the diagonal.</p>
<p>By default, the genotype probabilities are converted to allele
probabilities, and the kinship matrix is calculated as the proportion of
shared alleles. To use genotype probabilities instead, use
<code>use_allele_probs=FALSE</code> in the call to
<code>calc_kinship()</code>. Further, by default we omit the X
chromosome and only use the autosomes. To include the X chromosome, use
<code>omit_x=FALSE</code>.</p>
<p>On a multi-core machine, you can get some speed-up via the
<code>cores</code> argument, as with <code>calc_genoprob()</code>.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kinship</span> <span class="op">&lt;-</span> <span class="fu">calc_kinship</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                        cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p>Think about what a kinship matrix is and what it represents. Share
your understanding with a neighbor. Write your explanation in the
collaborative document or in your own personal notes.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Kinship matrices account for relationships among individuals.</li>
<li>Kinship is calculated as the proportion of shared alleles between
individuals.</li>
<li>Kinship calculation is a precursor to a genome scan via a linear
mixed model.</li>
</ul>
</div>
</div>
</div></section><section id="aio-perform-genome-scan-lmm"><p>Content from <a href="perform-genome-scan-lmm.html">Performing a genome scan with a linear mixed model</a></p>
<hr>
<p>Last updated on 2024-10-16 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/perform-genome-scan-lmm.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>“How do I use a linear mixed model in a genome scan?”</li>
<li>“How do different mapping and kinship calculation methods
differ?”</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Create a genome scan with a linear mixed model.</li>
<li>Compare LOD plots for Haley-Knott regression and linear mixed model
methods.</li>
<li>Compare LOD plots for the standard kinship matrix with the
leave-one-chromosome-out (LOCO) method.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Genetic mapping in mice presents a good example of why accounting for
population structure is important. Laboratory mouse strains are
descended from a small number of founders (fancy mice) and went through
several population bottlenecks. Wild-derived strains are not descended
from fancy mice and don’t share the same history as laboratory strains.
Linear mixed models were developed to solve problems with population
structure created by differing ancestries, and to handle relatedness
between individuals. Linear mixed models (LMMs) consider genome-wide
similarity between all pairs of individuals to account for population
structure, known kinship and unknown relatedness. Linear mixed models in
mapping studies can successfully correct for genetic relatedness between
individuals in a population by incorporating kinship into the model.
Earlier we <a href="https://smcclatchy.github.io/mapping/04-calc-kinship/" class="external-link">calculated
a kinship matrix</a> for input to a linear mixed model to account for
relationships among individuals. For a current review of mixed models in
genetics, see this <a href="https://www.biorxiv.org/content/early/2017/01/28/092106" class="external-link">preprint
of Martin and Eskin, 2017</a>.</p>
<p>Simple linear regression takes the form:</p>
<p><span class="math inline">\(y = \mu + \beta G + \epsilon\)</span></p>
<p>which describes a line with slope <span class="math inline">\(\beta\)</span> and y-intercept <span class="math inline">\(\mu\)</span>. The error (or residual) is
represented by <span class="math inline">\(\epsilon\)</span>.</p>
<p>To model the relationship between the phenotype <span class="math inline">\(y\)</span> and the genotypes at one marker, we
use:</p>
<p><span class="math inline">\(y_j = \mu + \beta_k G_{jk} +
\epsilon_j\)</span></p>
<p>where $y_{ij} is the phenotype of the <span class="math inline">\(j\)</span>th individual, <span class="math inline">\(\mu\)</span>; is the mean phenotype value, <span class="math inline">\(\beta_k\)</span> is the effect of the <span class="math inline">\(k\)</span>th genotype, $G_{jk} is the genotype for
individual <span class="math inline">\(j\)</span>, and <span class="math inline">\(\epsilon_j\)</span> is the error for the <span class="math inline">\(j\)</span>th individual. In the figure below,
<span class="math inline">\(\mu\)</span>; equals 1, and <span class="math inline">\(\beta\)</span>; equals 0.1 for the alternative
hypothesis (QTL exists). This linear model is <span class="math inline">\(y\)</span> = 1 + 0.1X + <span class="math inline">\(\epsilon\)</span>. The model intersects the
genotype groups at their group means. In contrast, the null hypothesis
would state that there is no difference in group means (no QTL
anywhere). The linear model for the null hypothesis would be <span class="math inline">\(y\)</span> = 94.6 + 0<span class="math inline">\(X\)</span> + <span class="math inline">\(\epsilon\)</span>. This states that the phenotype
is equal to the combined mean (94.6), plus some error (<span class="math inline">\(\epsilon\)</span>). In other words, genotype
doesn’t affect the phenotype.</p>
<figure><img src="fig/nullvalt.png" class="figure mx-auto d-block"></figure><p>The linear models above describe the relationship between genotype
and phenotype but are inadequate for describing the relationship between
genotype and phenotype in large datasets. They don’t account for
relatedness among individuals. In real populations, the effect of a
single genotype is influenced by many other genotypes that affect the
phenotype. A true genetic model takes into account the effect of all
variants on the phenotype.</p>
<p>To model the phenotypes of all individuals in the data, we can adapt
the simple linear model to include all individuals and their variants so
that we capture the effect of all variants shared by individuals on
their phenotypes.</p>
<p><span class="math inline">\(y=\mu+\sum_{i=1}^M{\beta_iX_i}+\epsilon\)</span></p>
<p>Now, <span class="math inline">\(y\)</span> represents the phenotypes
of all individuals. The effect of the <span class="math inline">\(i\)</span>th genotype on the phenotype is <span class="math inline">\(\beta_i\)</span>, the mean is <span class="math inline">\(\mu\)</span> and the error is <span class="math inline">\(\epsilon\)</span>. Here, the number of genotypes
is M.</p>
<p>To model the effect of all genotypes and to account for relatedness,
we test the effect of a single genotype while bringing all other
genotypes into the model.</p>
<figure><img src="fig/all-geno-model.png" class="figure mx-auto d-block"></figure><p><i>β<sub>k</sub></i> is the effect of the genotype
<i>X<sub>k</sub></i>, and
Σ<sub>i≠k</sub><i>β<sub>i</sub>X<sub>i</sub></i> sums the effects of all
other genotypes except genotype k. For the leave one chromosome out
(LOCO) method, <i>β<sub>k</sub></i><i>X<sub>k</sub></i> is the effect of
genotypes on chromosome <i>k</i>, and
<i>β<sub>i</sub></i><i>X<sub>i</sub></i> represents effect of genotypes
on all other chromosomes.</p>
<p>If the sample contains divergent subpopulations, SNPs on different
chromosomes will be correlated because of the difference in allele
frequencies between subpopulations caused by relatedness. To correct for
correlations between chromosomes, we model all genotypes on the other
chromosomes when testing for the association of a SNP.</p>
<p>First, we will create a single kinship matrix using all of the
genoprobs on all chromosomes.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kinship_all</span> <span class="op">&lt;-</span> <span class="fu">calc_kinship</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                            type <span class="op">=</span> <span class="st">"overall"</span><span class="op">)</span></span></code></pre>
</div>
<p>To perform a genome scan using a linear mixed model you also use the
function <code>scan1</code>; you just need to provide the argument
<code>kinship</code>, a kinship matrix (or, for the LOCO method, a list
of kinship matrices).</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_add_pg</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                    pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"log10_insulin_10wk"</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                    kinship   <span class="op">=</span> <span class="va">kinship_all</span>, </span>
<span>                    addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>Again, on a multi-core machine, you can get some speed-up using the
<code>cores</code> argument.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_add_all</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                     pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"log10_insulin_10wk"</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                     kinship   <span class="op">=</span> <span class="va">kinship_all</span>, </span>
<span>                     addcovar  <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                     cores     <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<p>If, for your linear mixed model genome scan, you wish to use the
“leave one chromosome out” (LOCO) method (scan each chromosome using a
kinship matrix that is calculated using data from all other
chromosomes), use <code>type="loco"</code> in the call to
<code>calc_kinship()</code>.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kinship_loco</span> <span class="op">&lt;-</span> <span class="fu">calc_kinship</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                             type  <span class="op">=</span> <span class="st">"loco"</span><span class="op">)</span></span></code></pre>
</div>
<p>For the LOCO (leave one chromosome out) method, provide the list of
kinship matrices as obtained from <code>calc_kinship()</code> with
<code>method="loco"</code>.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_add_loco</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                      pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"log10_insulin_10wk"</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                      kinship   <span class="op">=</span> <span class="va">kinship_loco</span>, </span>
<span>                      addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>To plot the results, we again use <code>plot_scan1()</code>.</p>
<p>Here is a plot of the LOD scores by Haley-Knott regression and the
linear mixed model using either the standard kinship matrix or the LOCO
method.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x     <span class="op">=</span> <span class="va">lod_add</span>,      </span>
<span>           map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           col   <span class="op">=</span> <span class="st">'black'</span>,</span>
<span>           ylim  <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">7.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x     <span class="op">=</span> <span class="va">lod_add_all</span>,  </span>
<span>           map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           col   <span class="op">=</span> <span class="st">'blue'</span>,   </span>
<span>           add   <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'lod_add_all' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x     <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>           map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           col   <span class="op">=</span> <span class="st">'orange'</span>, </span>
<span>           add   <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">ERROR<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="error" tabindex="0"><code>Error in eval(expr, envir, enclos): object 'lod_add_loco' not found</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">legend</span><span class="op">(</span>x <span class="op">=</span> <span class="fl">1500</span>, y <span class="op">=</span> <span class="fl">7.5</span>, legend <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"No kinship"</span>, <span class="st">"All kinship"</span>, <span class="st">"LOCO kinship"</span><span class="op">)</span>,</span>
<span>       lwd <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'black'</span>, <span class="st">'blue'</span>, <span class="st">'orange'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/perform-genome-scan-lmm-rendered-plot_lod_add_loco_all-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>For the circulating insulin, the three methods give quite different
results. The linear mixed model with an overall kinship matrix gives
much lower LOD scores than the other two methods. On chromosomes with
some evidence of a QTL, the LOCO method gives higher LOD scores than
Haley-Knott, except on chromosome 6 where it gives lower LOD scores.</p>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p>What are the benefits and disadvantages of the three methods for
genome scanning (Haley-Knott regression, kinship matrix, and
leave-one-chromosome out (LOCO)?)<br>
Which method would you use to scan, and why?<br>
Think about the advantages and disadvantages of each, discuss with a
neighbor, and share your thoughts in the collaborative document.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<!-- DMG: Ha! we were using the BTBR data anyway. Let's not load a second copy
of the data. Maybe we canb brainstorm about another challend. -->
<p>Pair programming exercise: with your partner, review and carry out
all of the steps in QTL mapping that we have covered so far, using a new
data set. One of you types the code, the other explains what needs to
happen next, finds the relevant code in the lesson, suggests new names
for objects (i.e. NOT the ones you’ve already used, such as “map”, “pr”,
“out”, etc.).</p>
<ol style="list-style-type: decimal">
<li>Run the code above to load the <a href="https://github.com/rqtl/qtl2data/tree/master/B6BTBR" class="external-link">B6 x BTBR
intercross data</a> into an object called b6btbr.</li>
<li>Insert pseudomarkers and calculate genotype probabilities.</li>
<li>Run a genome scan for the log10_insulin_10wk phenotype.</li>
<li>Calculate a kinship matrix.</li>
<li>Calculate a list of kinship matrices with the LOCO method.</li>
<li>Run genome scans with the regular kinship matrix and with the list
of LOCO matrices.</li>
<li>Plot the 3 different genome scans in a single plot in different
colors.</li>
<li>Which chromosomes appear to have peaks with a LOD score greater than
4? Which methods identify these peaks? Which don’t?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p><code>file &lt;- paste0("https://raw.githubusercontent.com/rqtl/", "qtl2data/master/B6BTBR/b6btbr.zip")</code><br><code>b6btbr &lt;- read_cross2(file)</code><br><code>summary(b6btbr)</code><br><code>head(b6btbr$pheno)</code><br><code>colnames(b6btbr$pheno)</code><br><code>b6bmap &lt;- insert_pseudomarkers(map=b6btbr$gmap, step=1)</code><br><code>prb6b &lt;- calc_genoprob(cross=b6btbr, map=b6bmap, error_prob=0.002)</code><br><code>b6baddcovar &lt;- get_x_covar(b6btbr)</code><br><code>b6bout &lt;- scan1(genoprobs = prb6b, pheno = b6btbr$pheno, addcovar=b6baddcovar)</code><br><code>plot(b6bout, map = b6bmap)</code><br><code>b6bkinship &lt;- calc_kinship(probs = prb6b)</code><br><code>out_pg_b6b &lt;- scan1(prb6b, b6btbr$pheno, kinship=b6bkinship, addcovar=b6baddcovar)</code><br><code>kinship_loco_b6b &lt;- calc_kinship(prb6b, "loco")</code><br><code>out_pg_loco_b6b &lt;- scan1(prb6b, b6btbr$pheno, kinship_loco_b6b, addcovar=b6baddcovar)</code><br><code>plot_scan1(out_pg_loco_b6b, map = b6bmap, lodcolumn = "log10_insulin_10wk", col = "black")</code><br><code>plot_scan1(out_pg_b6b, map = b6bmap, lodcolumn = "log10_insulin_10wk", col = "blue", add = TRUE)</code><br><code>plot_scan1(b6bout, map = b6bmap, lodcolumn = "log10_insulin_10wk", col = "green", add = TRUE)</code></p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">file</span> <span class="op">&lt;-</span> <span class="fu">paste0</span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/rqtl/"</span>,</span>
<span>               <span class="st">"qtl2data/master/B6BTBR/b6btbr.zip"</span><span class="op">)</span></span>
<span><span class="va">b6btbr</span> <span class="op">&lt;-</span> <span class="fu">read_cross2</span><span class="op">(</span><span class="va">file</span><span class="op">)</span></span></code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>“To perform a genome scan with a linear mixed model, supply a
kinship matrix.”</li>
<li>“Different mapping and kinship calculation methods give different
results.”</li>
</ul>
</div>
</div>
</div></section><section id="aio-perform-genome-scan-bin"><p>Content from <a href="perform-genome-scan-bin.html">Performing a genome scan with binary traits</a></p>
<hr>
<p>Last updated on 2024-10-16 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/perform-genome-scan-bin.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>“How do I create a genome scan for binary traits?”</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Convert phenotypes to binary values.</li>
<li>Use logistic regression for genome scans with binary traits.</li>
<li>Plot and compare genome scans for binary traits.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>The genome scans above were performed assuming that the residual
variation followed a normal distribution. This will often provide
reasonable results even if the residuals are not normal, but an
important special case is that of a binary trait, with values 0 and 1,
which is best treated differently. The <code>scan1</code> function can
perform a genome scan with binary traits by logistic regression, using
the argument <code>model="binary"</code>. (The default value for the
<code>model</code> argument is <code>"normal"</code>.) At present, we
<em>can not</em> account for relationships among individuals in this
analysis.</p>
<p>Let’s look at the phenotypes in the cross again.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          log10_insulin_10wk agouti_tan tufted
Mouse3051              1.399          1      0
Mouse3551              0.369          1      1
Mouse3430              0.860          0      1
Mouse3476              0.800          1      0
Mouse3414              1.370          0      0
Mouse3145              1.783          1      0</code></pre>
</div>
<p>There are two binary traits called “agouti_tan”, and “tufted” which
are related to coat color and shape.</p>
<p>We perform a binary genome scan in a similar manner to mapping
continuous traits by using <code>scan1</code>. When we mapped insulin,
there was a hidden argument called <code>model</code> which told
<code>qtl2</code> which mapping model to use. There are two options:
<code>normal</code>, the default, and <code>binary</code>. The
<code>normal</code> argument tells
<code>qtl2 to ues a "normal" (least squares) linear model. To map a binary trait, we will  include</code>model
= “binary”` to indicate that the phenotype is a binary trait with values
0 and 1.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_agouti</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                    pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">'agouti_tan'</span><span class="op">]</span>, </span>
<span>                    addcovar  <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                    model     <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s plot the result and see if there is a peak.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_agouti</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           main <span class="op">=</span> <span class="st">'Agouti'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/perform-genome-scan-bin-rendered-plot_bin_scan-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Yes! There is a big peak on chromosome 2. Let’s zoom in on chromosome
2.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_agouti</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           chr  <span class="op">=</span> <span class="fl">2</span>,</span>
<span>           main <span class="op">=</span> <span class="st">'Agouti'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/perform-genome-scan-bin-rendered-plot_bin_scan_chr2-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We can use <code>find_peaks</code> to find the position of the
highest LOD score.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_agouti</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">)</span></span></code></pre>
</div>
<p>This turns out to be a well-known coat color locus for agouti coat
color which contains the <a href="https://www.informatics.jax.org/marker/MGI:87853" class="external-link">nonagouti</a>
gene. Mice carrying two black alleles will have a black coat, and mice
carrying one or no black alleles will have agouti coats.</p>
<div id="challenge-1-how-many-mice-have-black-coats" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-how-many-mice-have-black-coats" class="callout-inner">
<h3 class="callout-title">Challenge 1: How many mice have black coats?</h3>
<div class="callout-content">
<p>Look at the frequency of the black (0) and agouti (1) phenotypes.
What proportion of the mice are black? Can you use what you learned
about how the <code>nonagouti</code> locus works and the cross design to
explain the frequency of black mice?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>First, get the number of black and agouti mice.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl</span> <span class="op">&lt;-</span> <span class="fu">table</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"agouti_tan"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">tbl</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
  0   1
125 356 </code></pre>
</div>
<p>Then use the number of mice to calculate the proportion with each
coat color.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl</span> <span class="op">/</span> <span class="fu">sum</span><span class="op">(</span><span class="va">tbl</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
   0    1
0.26 0.74 </code></pre>
</div>
<p>We can see that the black (0) mice occur about 25 % of the time. If
the <code>A</code> allele causes mice to have black coats when it is
recessive, and if <code>a</code> is the agouti allele, then, when
breeding two heterozygous (<code>Aa</code>) mice together, we expect
mean allele frequencies of:</p>
<table class="table">
<thead><tr class="header">
<th>Allele</th>
<th>Frequency</th>
<th>Coat Color</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>AA</td>
<td>0.25</td>
<td>black</td>
</tr>
<tr class="even">
<td>Aa</td>
<td>0.5</td>
<td>agouti</td>
</tr>
<tr class="odd">
<td>aa</td>
<td>0.25</td>
<td>agouti</td>
</tr>
</tbody>
</table>
<p>From this, we can see that about 25% of the mice should have black
coats.</p>
<!-- DMG: Fill this in with a figure showing the expected frequency of
two allele from F1s. AA = 25%, Aa = 50% & aa = 25%. -->
</div>
</div>
</div>
</div>
<div id="challenge-2-map-the-tufted-phenotype." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-map-the-tufted-phenotype." class="callout-inner">
<h3 class="callout-title">Challenge 2: Map the “tufted” phenotype.</h3>
<div class="callout-content">
<p>Map the tufted phenotype an determine if there are any tall peaks for
this trait.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>First, map the trait.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_tufted</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                    pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"tufted"</span><span class="op">]</span>, </span>
<span>                    addcovar  <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                    model     <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span></code></pre>
</div>
<p>Then, plot the LOD score.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_tufted</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           main <span class="op">=</span> <span class="st">"Tufted"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/perform-genome-scan-bin-rendered-unnamed-chunk-4-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>There is a large peak on chromosome 17. This is a <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3570182/" class="external-link">known
locus</a> associated with the <a href="https://www.informatics.jax.org/marker/MGI:96624" class="external-link">Itpr3</a> gene
near 27.3 Mb on chromsome 17.</p>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>“A genome scan for binary traits (0 and 1) requires special
handling; scans for non-binary traits assume normal variation of the
residuals.”</li>
<li>“A genome scan for binary traits is performed with logistic
regression.”</li>
</ul>
</div>
</div>
</div></section><section id="aio-perform-perm-test"><p>Content from <a href="perform-perm-test.html">Performing a permutation test</a></p>
<hr>
<p>Last updated on 2024-10-16 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/perform-perm-test.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>“How can I evaluate the statistical significance of genome scan
results?”</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Run a permutation test to establish LOD score thresholds.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="using-permutations-to-determine-significance-thresholds">Using Permutations to Determine Significance Thresholds<a class="anchor" aria-label="anchor" href="#using-permutations-to-determine-significance-thresholds"></a>
</h2>
<hr class="half-width">
<p>To establish the statistical significance of the results of a genome
scan, a permutation test can identify the maximum LOD score that can
occur by random chance. A permutation tests shuffles genotypes and
phenotypes, essentially breaking the relationship between the two. The
genome-wide maximum LOD score is then calculated on the permuted data,
and this score used as a threshold of statistical significance. A
genome-wide maximum LOD on shuffled, or permuted, data serves as the
threshold because it represents the highest LOD score generated by
random chance.</p>
<p>The code below shuffles the phenotypes so that they no longer match
up with the genotypes. The purpose of this is to find out how high the
LOD score can be due to random chance alone.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">shuffled_order</span> <span class="op">&lt;-</span> <span class="fu">sample</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pheno_permuted</span> <span class="op">&lt;-</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span></span>
<span><span class="fu">rownames</span><span class="op">(</span><span class="va">pheno_permuted</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">shuffled_order</span></span>
<span><span class="va">xcovar_permuted</span> <span class="op">&lt;-</span> <span class="va">Xcovar</span></span>
<span><span class="fu">rownames</span><span class="op">(</span><span class="va">xcovar_permuted</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">shuffled_order</span></span>
<span><span class="va">out_permuted</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, pheno <span class="op">=</span> <span class="va">pheno_permuted</span>, Xcovar <span class="op">=</span> <span class="va">xcovar_permuted</span><span class="op">)</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">out_permuted</span>, <span class="va">map</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">shuffled_order</span><span class="op">)</span></span></code></pre>
</div>
<p>The <code>scan1perm()</code> function takes the same arguments as
<code>scan1()</code>, plus additional arguments to control the
permutations:</p>
<ul>
<li>
<code>n_perm</code> is the number of permutation replicates.</li>
<li>
<code>perm_Xsp</code> controls whether to perform autosome/X
chromosome specific permutations (with <code>perm_Xsp=TRUE</code>) or
not (the default is FALSE).</li>
<li>
<code>perm_strata</code> is a vector that defines the strata for a
stratified permutation test.</li>
<li>
<code>chr_lengths</code> is a vector of chromosome lengths, used in
the case that <code>perm_Xsp=TRUE</code>.</li>
</ul>
<p>As with <code>scan1()</code>, you may provide a kinship matrix (or
vector of kinship matrices, for the “leave one chromosome out” (loco)
approach), in order to fit linear mixed models. If <code>kinship</code>
is unspecified, the function performs ordinary Haley-Knott
regression.</p>
<p>To perform a permutation test with the insulin phenotype, we run
<code>scan1perm()</code>, provide it with the genotype probabilities,
the phenotype data, X covariates and number of permutations. For
expediency, we’ll use only 10 permutations, although 1000 is
recommended.</p>

<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perm_add</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                      pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">'log10_insulin_10wk'</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                      addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                      Xcovar    <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                      n_perm    <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span> </span></code></pre>
</div>
<p>Note the need to specify special covariates for the X chromosome (via
<code>Xcovar</code>), to be included under the null hypothesis of no
QTL. And note that when these are provided, the default is to perform a
stratified permutation test, using strata defined by the rows in
<code>Xcovar</code>. In general, when the X chromosome is considered,
one will wish to stratify at least by sex.</p>
<p>Also note that, as with <code>scan1()</code>, you can speed up the
calculations on a multi-core machine by specifying the argument
<code>cores</code>. With <code>cores=0</code>, the number of available
cores will be detected via <code>parallel::detectCores()</code>.
Otherwise, specify the number of cores as a positive integer. For large
data sets, be mindful of the amount of memory that will be needed; you
may need to use fewer than the maximum number of cores, to avoid going
beyond the available memory.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode r" tabindex="0"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>perm_add <span class="ot">&lt;-</span> <span class="fu">scan1perm</span>(<span class="at">genoprobs =</span> probs, </span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>                      <span class="at">phenotype =</span> cross<span class="sc">$</span>pheno[,<span class="st">'log10_insulin_10wk'</span>,<span class="at">drop =</span> <span class="cn">FALSE</span>], </span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>                      <span class="at">addcovar  =</span> addcovar</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>                      <span class="at">Xcovar    =</span> Xcovar, </span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>                      <span class="at">n_perm    =</span> <span class="dv">1000</span>, </span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>                      <span class="at">cores     =</span> <span class="dv">0</span>)</span></code></pre>
</div>
<p><code>perm_add</code> now contains the maximum LOD score for each
permutation for the liver and spleen phenotypes. There should be 1000
values for each phenotypes. We can view the insulin permutation LOD
scores by making a histogram.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">perm_add</span>, </span>
<span>     breaks <span class="op">=</span> <span class="fl">50</span>, </span>
<span>     xlab   <span class="op">=</span> <span class="st">"LOD"</span>, </span>
<span>     las    <span class="op">=</span> <span class="fl">1</span>,</span>
<span>     main   <span class="op">=</span> <span class="st">"Empirical distribution of maximum LOD scores under permuation"</span><span class="op">)</span></span>
<span><span class="fu">abline</span><span class="op">(</span>v <span class="op">=</span> <span class="fu">summary</span><span class="op">(</span><span class="va">perm_add</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">'red'</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/perform-perm-test-rendered-hist_perm-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the histogram above, you can see that most of the maximum LOD
scores fall between 1 and 3.5. This means that we expect LOD scores less
than 3.5 to occur by chance fairly often. The red line indicates the
alpha = 0.05 threshold, which means that we only see LOD values by
chance this high or higher, 5% of the time. This is <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1206241/" class="external-link">one way</a>
of estimating a significance threshold for QTL plots.</p>
<p>To get estimated significance thresholds, use the function
<code>summary()</code>.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perm_add</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (1000 permutations)
     log10_insulin_10wk
0.05               3.79</code></pre>
</div>
<p>The default is to return the 5% significance thresholds. Thresholds
for other (or for multiple) significance levels can be obtained via the
<code>alpha</code> argument.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perm_add</span>, </span>
<span>        alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (1000 permutations)
     log10_insulin_10wk
0.2                3.16
0.05               3.79</code></pre>
</div>
</section><section><h2 class="section-heading" id="estimating-an-x-chromosome-specific-threshold">Estimating an X Chromosome Specific Threshold<a class="anchor" aria-label="anchor" href="#estimating-an-x-chromosome-specific-threshold"></a>
</h2>
<hr class="half-width">
<p>To obtain autosome/X chromosome-specific significance thresholds,
specify <code>perm_Xsp=TRUE</code>. In this case, you need to provide
chromosome lengths, which may be obtained with the function
<code>chr_lengths()</code>.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perm_add2</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs   <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                       pheno       <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"log10_insulin_10wk"</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                       addcovar    <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                       n_perm      <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>                       perm_Xsp    <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                       chr_lengths <span class="op">=</span> <span class="fu">chr_lengths</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Separate permutations are performed for the autosomes and X
chromosome, and considerably more permutation replicates are needed for
the X chromosome. The computations take about twice as much time. See <a href="https://www.ncbi.nlm.nih.gov/pubmed/17028340" class="external-link">Broman et al. (2006)
Genetics 174:2151-2158</a>.</p>
<p>The significance thresholds are again derived via
<code>summary()</code>:</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perm_add2</span>, </span>
<span>        alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Autosome LOD thresholds (1000 permutations)
     log10_insulin_10wk
0.2                3.29
0.05               3.87

X chromosome LOD thresholds (14369 permutations)
     log10_insulin_10wk
0.2                3.14
0.05               3.95</code></pre>
</div>
</section><section><h2 class="section-heading" id="estimating-significance-thresholds-with-the-kinship-matrix">Estimating Significance Thresholds with the Kinship Matrix<a class="anchor" aria-label="anchor" href="#estimating-significance-thresholds-with-the-kinship-matrix"></a>
</h2>
<hr class="half-width">
<p>You may have noticed above that we did not include the kinship matrix
as an argument to <code>scan1perm</code>. We can include the LOCO
kinship matrices in our permutations, since this is how we mapped
insulin previously.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perm_add_loco</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                           pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">'log10_insulin_10wk'</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                           kinship   <span class="op">=</span> <span class="va">kinship_loco</span>,</span>
<span>                           addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                           n_perm    <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span> </span></code></pre>
</div>
<p>How does this affect the significance threshold estimates?</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perm_add_loco</span>, </span>
<span>        alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (1000 permutations)
     log10_insulin_10wk
0.2                3.18
0.05               3.90</code></pre>
</div>
<p>There is not a large difference in the thresholds. Currently, we are
on the fence about using the kinship matrices to estimate significance
thresholds. In principle, the kinship matrices should be used because we
used them when mapping the phenotype. However, in practice, we often
find that the significance thresholds are similar to those obtained
without including the kinship matrices. Given that it also takes more
time to run the permutations with the kinship matrices, it seems
reasonable to exclude them.</p>
<p>We ran 1000 permutations of the insulin phenotype and estimated the
0.05 significance threshold with and without the kinship matrices. We
repeated this process 100 times and plotted the thresholds below.</p>
<p><img src="fig/perm_kinship.png" alt="Significance Thresholds with/without Kinship" class="figure">{alt=“Comparison of
significane threholds”,width=“50%”}</p>
<p>The plot shows that the median significance threshold is the same.
The lines connecting the points denote matched simulations in which the
permutation order was the same. While the exact value of the LOD
threshold is different, the median value and the variance are
similar.</p>
</section><section><h2 class="section-heading" id="estimating-binary-model-significance-thresholds">Estimating Binary Model Significance Thresholds<a class="anchor" aria-label="anchor" href="#estimating-binary-model-significance-thresholds"></a>
</h2>
<hr class="half-width">
<p>As with <code>scan1</code>, we can use <code>scan1perm</code> with
binary traits, using the argument <code>model="binary"</code>. Again,
this can’t be used with a kinship matrix, but all of the other arguments
can be applied.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perm_bin</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                      pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"agouti_tan"</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                      addcovar  <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                      n_perm    <span class="op">=</span> <span class="fl">1000</span>, </span>
<span>                      perm_Xsp  <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                      chr_lengths <span class="op">=</span> <span class="fu">chr_lengths</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">)</span>,</span>
<span>                      model     <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span></code></pre>
</div>
<p>Here are the estimated 5% and 20% significance thresholds.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perm_bin</span>, </span>
<span>        alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Autosome LOD thresholds (1000 permutations)
     agouti_tan
0.2        3.25
0.05       3.88

X chromosome LOD thresholds (14369 permutations)
     agouti_tan
0.2        3.09
0.05       3.81</code></pre>
</div>
</section><section><h2 class="section-heading" id="selecting-the-number-of-permutations">Selecting the Number of Permutations<a class="anchor" aria-label="anchor" href="#selecting-the-number-of-permutations"></a>
</h2>
<hr class="half-width">
<p>How do we know how many permutations to perform in order to obtain a
good estimate of the significance threshold? Could we get a good
estimate with 10 permutations? 100? 1000?</p>
<p>When we run more permutations, we decrease the variance of the
threshold estimate.</p>
<p><img src="fig/permutation_simulations.png" alt="Significance Threshold Variance" class="figure">{alt=“Figure showing decreasing
variance of significance threshold estimates with increasing
permutations”,width=75%}</p>
<p>In the figure above, we performed 10, 100, or 1000 permutations 1000
times and recorded the 0.05 significance threshold each time. We plotted
the significance threshold versus the number of permutations and
overlaid violin plots showing the median value. Note that the variance
of the significance threshold estimate is higher at lower numbers of
permutations. With 1000 permutations, the variance decreases. The table
below shows the number of permutations and the mean and standard
deviation of the significance threshold. With 1000 permutations, the
estimate is 3.86 and the standard deviation is 0.064, which is an
acceptable value.</p>
<table class="table">
<thead><tr class="header">
<th>Num. Perm.</th>
<th>Mean</th>
<th>Std. Dev.</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>10</td>
<td>3.63</td>
<td>0.492</td>
</tr>
<tr class="even">
<td>100</td>
<td>3.81</td>
<td>0.195</td>
</tr>
<tr class="odd">
<td>1000</td>
<td>3.86</td>
<td>0.064</td>
</tr>
</tbody>
</table>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1:</h3>
<div class="callout-content">
<p>Run the preceding code to shuffle the phenotype data and plot a
genome scan with this shuffled (permuted) data.</p>
<p>What is the maximum LOD score in the scan from this permuted
data?<br>
How does it compare to the maximum LOD scores obtained from the earlier
scan?<br>
How does it compare to the 5% and 20% LOD thresholds obtained
earlier?<br>
Paste the maximum LOD score in the scan from your permuted data into the
etherpad.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Find the 1% and 10% significance thresholds for the first set of
permutations contained in the object <code>operm</code>.<br>
</li>
<li>What do the 1% and 10% significance thresholds say about LOD
scores?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>
<code>summary(operm, alpha=c(0.01, 0.10))</code><br>
</li>
<li>These LOD thresholds indicate maximum LOD scores that can be
obtained by random chance at the 1% and 10% significance levels. We
expect to see LOD values this high or higher 1% and 10% of the time
respectively.</li>
</ol>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>“A permutation test establishes the statistical significance of a
genome scan.”</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-find-lod-peaks"><p>Content from <a href="find-lod-peaks.html">Finding QTL peaks</a></p>
<hr>
<p>Last updated on 2024-10-16 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/find-lod-peaks.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>“How do I locate QTL peaks above a certain LOD threshold
value?”</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Locate QTL peaks above a LOD threshold value throughout the
genome.</li>
<li>Identify the Bayesian credible interval for a QTL peak.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Once we have LOD scores from a genome scan and a significance
threshold, we can look for QTL peaks associated with the phenotype. High
LOD scores indicate the neighborhood of a QTL but don’t give its precise
position. To find the exact position of a QTL, we define an interval
that is likely to hold the QTL.</p>
<p>We’ll use the Bayesian credible interval, which is a method for
defining QTL intervals. It describes the probability that the interval
contains the true value. Credible intervals make a probabilistic
statement about the true value, for example, a 95% credible interval
states that there is a 95% chance that the true value lies within the
interval.</p>
<p>Let’s remind ourselves how the genome scan for insulin looks.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">thr</span> <span class="op">&lt;-</span> <span class="fu">summary</span><span class="op">(</span><span class="va">perm_add_loco</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_add_loco</span>,</span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>           main <span class="op">=</span> <span class="st">"Insulin"</span><span class="op">)</span></span>
<span><span class="fu">add_threshold</span><span class="op">(</span>map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, thresholdA <span class="op">=</span> <span class="va">thr</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/find-lod-peaks-rendered-plot_scan1-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>To find peaks above a given threshold LOD value, use the function
<code>find_peaks()</code>. It can also provide Bayesian credible
intervals by using the argument <code>prob</code> (the nominal coverage
for the Bayes credible intervals). Set the argument
<code>expand2markers = FALSE</code> to keep from expanding the interval
out to genotyped markers, or exclude this argument if you’d like to
include flanking markers.</p>
<p>You need to provide both the <code>scan1()</code> output, the marker
map and a threshold. We will use the 95% threshold from the permutations
in the previous lesson.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           threshold    <span class="op">=</span> <span class="va">thr</span>, </span>
<span>           prob         <span class="op">=</span> <span class="fl">0.95</span>, </span>
<span>           expand2markers <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr       pos      lod      ci_lo     ci_hi
1        1    pheno1   2 138.94475 7.127351  64.949395 149.57739
2        1    pheno1   7 144.18230 5.724018 139.368290 144.18230
3        1    pheno1  12  25.14494 4.310493  15.834815  29.05053
4        1    pheno1  14  22.24292 3.974322   6.241951  45.93876
5        1    pheno1  16  80.37433 4.114024  10.238134  80.37433
6        1    pheno1  19  54.83012 5.476587  48.370980  55.15007</code></pre>
</div>
<p>In the table above, we have one peak per chromosome because that is
the default behavior of <code>find_peaks()</code>. The
<code>find_peaks()</code> function can also pick out multiple peaks on a
chromosome: each peak must exceed the chosen threshold, and the argument
<code>peakdrop</code> indicates the amount that the LOD curve must drop
between the lowest of two adjacent peaks. Use this feature with
caution.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           threshold    <span class="op">=</span> <span class="va">thr</span>, </span>
<span>           peakdrop     <span class="op">=</span> <span class="fl">1.8</span>, </span>
<span>           prob         <span class="op">=</span> <span class="fl">0.95</span>, </span>
<span>           expand2markers <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr       pos      lod      ci_lo     ci_hi
1        1    pheno1   2 138.94475 7.127351  64.949395 149.57739
2        1    pheno1   7 144.18230 5.724018 139.368290 144.18230
3        1    pheno1  12  25.14494 4.310493  15.834815  29.05053
4        1    pheno1  14  22.24292 3.974322   6.241951  45.93876
5        1    pheno1  16  17.48123 3.995627   5.604167  37.99110
6        1    pheno1  16  80.37433 4.114024  74.250773  80.37433
7        1    pheno1  19  54.83012 5.476587  48.370980  55.15007</code></pre>
</div>
<p>Each row shows a different peak; the columns show the peak location,
LOD score and the lower and upper interval endpoints. Note that we now
have two peaks on chromosome 16, one at 17.5 Mb and one at 80.4 Mb.</p>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p>Find peaks in the genome scan object called <code>lod_add_loco</code>
that meet a threshold of 3 and are in the interval described by a 2
point LOD drop from the peak. How many peaks meet the LOD threshold of 3
and lie within the interval defined by a 2 point LOD drop from the
maximum peaks on each chromosome?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           threshold    <span class="op">=</span> <span class="fl">3</span>, </span>
<span>           drop         <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr       pos      lod      ci_lo     ci_hi
1        1    pheno1   2 138.94475 7.127351  63.943187 156.83772
2        1    pheno1   5 103.41486 3.130862  41.967549 132.28428
3        1    pheno1   7 144.18230 5.724018 129.414016 144.18230
4        1    pheno1   9  83.67606 3.865635  17.504307 111.02206
5        1    pheno1  12  25.14494 4.310493   9.998200  34.23274
6        1    pheno1  14  22.24292 3.974322   6.241951  68.04655
7        1    pheno1  16  80.37433 4.114024   3.804882  96.52406
8        1    pheno1  19  54.83012 5.476587  47.361847  56.37100</code></pre>
</div>
<p>This produces 7 peaks on 6 different chromosomes that meet a LOD
threshold of 3 and are within the interval defined by a 2-LOD drop from
the maximum peak on each chromosome.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<p>1). Calculate the 90% Bayes credible interval. For chromosome 2, what
is the range of the interval that has a 90% chance of containing the
true QTL position?<br>
2). Compare with the 95% Bayes credible interval calculated earlier. How
does the interval change as you increase the probability? Why?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>1). This produces a range from 118.0 to 149.6 Mb.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pks</span> <span class="op">=</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>                 map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>                 prob         <span class="op">=</span> <span class="fl">0.90</span>, </span>
<span>                 expand2markers <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">subset</span><span class="op">(</span><span class="va">pks</span>, <span class="va">chr</span> <span class="op">==</span> <span class="st">'2'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr      pos      lod    ci_lo    ci_hi
1        1    pheno1   2 138.9448 7.127351 117.9557 149.5774</code></pre>
</div>
<p>2). This produces a range from 64.9 to 149.6 Mb, which is much
broader than the 90% interval. The interval widens because the
probability that the interval contains the true QTL position has
increased.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pks</span> <span class="op">=</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>                 map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>                 prob         <span class="op">=</span> <span class="fl">0.95</span>, </span>
<span>                 expand2markers <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">subset</span><span class="op">(</span><span class="va">pks</span>, <span class="va">chr</span> <span class="op">==</span> <span class="st">'2'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr      pos      lod    ci_lo    ci_hi
1        1    pheno1   2 138.9448 7.127351 64.94939 149.5774</code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>LOD peaks and support intervals can be identified with
find_peaks().</li>
<li>The Bayesian Credible Interval estimates the width of the support
interval around a QTL peak.</li>
</ul>
</div>
</div>
</div></section><section id="aio-est-qtl-effects"><p>Content from <a href="est-qtl-effects.html">Estimating QTL effects</a></p>
<hr>
<p>Last updated on 2024-10-16 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/est-qtl-effects.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I find the founder allele effects at a QTL peak?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Estimate the founder allele effects at a QTL peak.</li>
<li>Plot the estimated founder allele effects.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="estimating-founder-allele-effects">Estimating Founder Allele Effects<a class="anchor" aria-label="anchor" href="#estimating-founder-allele-effects"></a>
</h2>
<hr class="half-width">
<!-- DMG: Sue, can you make the equations Latex and remake the figures
with the BTBR data? --><p>Recall that to model data from a cross, we use</p>
<p><span class="math inline">\(y_j = \mu + \beta_k G_{jk} +
\epsilon_j\)</span></p>
<p>where <span class="math inline">\(y_{ij}\)</span> is the phenotype of
the <span class="math inline">\(j\)</span>th individual, <span class="math inline">\(\mu\)</span> is the mean phenotype value, <span class="math inline">\(\beta_k\)</span> is the effect of the <span class="math inline">\(k\)</span>th genotype, <span class="math inline">\(G_{jk}\)</span> is the genotype for individual
<span class="math inline">\(j\)</span>, and <span class="math inline">\(\epsilon_j\)</span> is the error for the <span class="math inline">\(j\)</span>th individual. In the figure below,
<span class="math inline">\(\mu\)</span> equals 1, and <span class="math inline">\(\beta\)</span> equals 0.1 for the alternative
hypothesis (QTL exists).</p>
<figure><img src="fig/nullvalt.png" class="figure mx-auto d-block"></figure><p>This linear model is <i>y</i> = 1 + 0.1X + ε. The model intersects
the genotype groups at their group means, and is based on μ and
<i>β<sub>k</sub></i> for chromosome 2 marker rs13476803 located at
138.944754 Mb.</p>
<p>The effect of genotype RR (the β for the RR genotype) at marker
rs13476803 is 0.1, while the effect of the BB genotype is -0.1 on the
insulin phenotype. The effect of the BR genotype is 0 relative to <span class="math inline">\(\mu\)</span> equals 1.</p>
<p>The <code>scan1()</code> function returns only LOD scores. To obtain
estimated QTL effects, use the function <code>scan1coef()</code>. This
function takes a single phenotype and the genotype probabilities for a
single chromosome and returns a matrix with the estimated coefficients
at each putative QTL location along the chromosome.</p>
<p>For example, to get the estimated QTL effects on chromosome 2 for the
insulin phenotype, we would provide the chromosome 2 genotype
probabilities and the insulin phenotype to the function
<code>scan1coef()</code> as follows:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>      <span class="op">&lt;-</span> <span class="st">"7"</span></span>
<span><span class="va">eff_chr7</span> <span class="op">&lt;-</span> <span class="fu">scan1coef</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                      pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"log10_insulin_10wk"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                      kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                      addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>The result is a matrix of 109 positions <span class="math inline">\(\times\)</span><br>
4 genotypes. An additional column contains the intercept values (<span class="math inline">\(\mu\)</span>).</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">eff_chr7</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 109   5</code></pre>
</div>
</section><section><h2 class="section-heading" id="plotting-founder-allele-effects-along-a-chromosome">Plotting Founder Allele Effects Along a Chromosome<a class="anchor" aria-label="anchor" href="#plotting-founder-allele-effects-along-a-chromosome"></a>
</h2>
<hr class="half-width">
<p>To plot the QTL effects, use the <code>plot_coef()</code> function.
Add the LOD plot to the <code>scan1_output</code> argument to include a
LOD plot at the bottom.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x            <span class="op">=</span> <span class="va">eff_chr7</span>, </span>
<span>          map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>          legend       <span class="op">=</span> <span class="st">"topright"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_coef-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The plot shows effect values on the y-axis and cM values on the
x-axis. The value of the intercept (μ) appears at the top. The effect of
the BR genotype is centered around zero, with the effects of the other
two genotypes above and below. We are usually not directly interested in
how the additive covariates change across the genome, but rather, the
the founder allele effects change.</p>
<p>To plot only the founder allele effects, use the argument
<code>columns</code> to indicate which coefficients to plot. Let’s look
at the columns which contain the founder allele effects.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">eff_chr7</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                   BB           BR          RR    SexMale intercept
rs8252589  0.02792439 -0.002516938 -0.02540745 -0.1753913  1.011125
rs13479104 0.02140644  0.002728978 -0.02413541 -0.1751367  1.009704
rs13479112 0.02010520  0.006609288 -0.02671449 -0.1750756  1.008747
rs13479114 0.01880760  0.007656166 -0.02646377 -0.1750906  1.008527
rs13479120 0.02083594  0.005991180 -0.02682712 -0.1752293  1.008946
rs13479124 0.02061108  0.002498998 -0.02311008 -0.1751596  1.009863</code></pre>
</div>
<p>We would like to plot the columns “BB”, “BR”, and “RR”, which are in
columns 1 through 3. This is what we pass into the <code>columns</code>
argument.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">eff_chr7</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>          main    <span class="op">=</span> <span class="st">"Chromosome 2 QTL effects and LOD scores"</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_effects_chr%202-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Looking at the plot above, which founder allele contributes to higher
insulin levels?</p>
</section><section><h2 class="section-heading" id="estimating-founder-allele-effects-using-blups">Estimating Founder Allele Effects using BLUPs<a class="anchor" aria-label="anchor" href="#estimating-founder-allele-effects-using-blups"></a>
</h2>
<hr class="half-width">
<p>Another option for estimating the founder allele effects is to treat
them as <a href="https://stats.stackexchange.com/questions/4700/what-is-the-difference-between-fixed-effect-random-effect-and-mixed-effect-mode#151800" class="external-link">random
effects</a> and calculate <a href="https://en.wikipedia.org/wiki/Best_linear_unbiased_prediction" class="external-link">Best
Linear Unbiased Predictors</a> (BLUPs). This is particularly valuable
for multi-parent populations such as the Collaborative Cross and
Diversity Outbred mice, where the large number of possible genotypes at
a QTL leads to considerable variability in the effect estimates. To
calculate BLUPs, use <code>scan1blup()</code>; it takes the same
arguments as <code>scan1coef()</code>, including the option of a kinship
matrix to account for a residual polygenic effect.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blup_chr7</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                       pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"log10_insulin_10wk"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                       kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                       addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>We can plot the BLUP effects using <code>plot_coef</code> as
before.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">blup_chr7</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>          main    <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="va">chr</span>, <span class="st">"QTL BLUP effects and LOD scores"</span><span class="op">)</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_blup_chr7-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot below, we plotted the founder allele effects (solid
lines) and the BLUPs (dashed lines). In this case, the effects are not
greatly different, but the effects are “shrunken” toward zero.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">eff_chr7</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          main    <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="va">chr</span>,<span class="st">"QTL BLUP effects and LOD scores"</span><span class="op">)</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span>
<span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">blup_chr7</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,</span>
<span>          lty     <span class="op">=</span> <span class="fl">2</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span>,</span>
<span>          add     <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_blup_chr7_again-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></section><section><h2 class="section-heading" id="plotting-allele-effects-at-one-marker">Plotting Allele Effects at one Marker<a class="anchor" aria-label="anchor" href="#plotting-allele-effects-at-one-marker"></a>
</h2>
<hr class="half-width">
<p>You may also want plot the founder allele effects at the marker with
the highest LOD. To do this, you first need to get the position of the
marker from the peak list.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>,</span>
<span>                    map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>                    threshold    <span class="op">=</span> <span class="va">thr</span><span class="op">)</span></span>
<span><span class="va">peaks</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr       pos      lod
1        1    pheno1   2 138.94475 7.127351
2        1    pheno1   7 144.18230 5.724018
3        1    pheno1  12  25.14494 4.310493
4        1    pheno1  14  22.24292 3.974322
5        1    pheno1  16  80.37433 4.114024
6        1    pheno1  19  54.83012 5.476587</code></pre>
</div>
<p>The position of the maximum LOD on chromosome 7 is 144.182298 Mb. We
can pass this value into the <code>qtl2</code> function
<code>pull_genoprobpos</code> to get the genoprobs at this marker.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">max_pos</span> <span class="op">&lt;-</span> <span class="fu">subset</span><span class="op">(</span><span class="va">peaks</span>, <span class="va">chr</span> <span class="op">==</span> <span class="st">'7'</span><span class="op">)</span><span class="op">$</span><span class="va">pos</span></span>
<span><span class="va">max_mkr</span> <span class="op">&lt;-</span> <span class="fu">find_marker</span><span class="op">(</span>map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>                       chr <span class="op">=</span> <span class="va">chr</span>, </span>
<span>                       pos <span class="op">=</span> <span class="va">max_pos</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pr</span>      <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                            marker    <span class="op">=</span> <span class="va">max_mkr</span><span class="op">)</span></span></code></pre>
</div>
<!-- DMG: Make this a challenge. -->
<p>What does the structure of <code>pr</code> look like?</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">str</span><span class="op">(</span><span class="va">pr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> num [1:490, 1:3] 1.55e-06 1.55e-06 1.55e-06 3.94e-05 1.00 ...
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:490] "Mouse3051" "Mouse3551" "Mouse3430" "Mouse3476" ...
  ..$ : chr [1:3] "BB" "BR" "RR"</code></pre>
</div>
<p><code>pr</code> is a numeric matrix with 490 rows and 3 columns. The
rownames contain the mouse IDs and the column names contain the
genotypes.</p>
<p>We can then pass <code>pr</code> as an argument into the
<code>fit1</code> function, which fits the mapping model at a single
marker.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod</span> <span class="op">=</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr</span>,</span>
<span>           pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"log10_insulin_10wk"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>           kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>           addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>Then, we can plot the founder allele effects and their standard
error.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod_eff</span> <span class="op">=</span> <span class="fu">data.frame</span><span class="op">(</span>eff <span class="op">=</span> <span class="va">mod</span><span class="op">$</span><span class="va">coef</span>, </span>
<span>                     se  <span class="op">=</span> <span class="va">mod</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"genotype"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">filter</span><span class="op">(</span><span class="va">genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"BB"</span>, <span class="st">"BR"</span>, <span class="st">"RR"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data    <span class="op">=</span> <span class="va">mod_eff</span>, </span>
<span>       mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">genotype</span>, y <span class="op">=</span> <span class="va">eff</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_beeswarm</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_pointrange</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>ymin <span class="op">=</span> <span class="va">eff</span> <span class="op">-</span> <span class="va">se</span>, </span>
<span>                                ymax <span class="op">=</span> <span class="va">eff</span> <span class="op">+</span> <span class="va">se</span><span class="op">)</span>,</span>
<span>                  size       <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>                  linewidth  <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Founder Allele Effects on Chr"</span>, <span class="va">chr</span><span class="op">)</span>,</span>
<span>       x     <span class="op">=</span> <span class="st">"Genotype"</span>, y <span class="op">=</span> <span class="st">"Founder Allele Effects"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_fit1-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot above, which founder allele contributes to higher insulin
levels? Is that consistent with the plot created using
<code>plot_coef</code> above?</p>
<p>If instead you want additive and dominance effects, you can provide a
square matrix of <em>contrasts</em>, as follows:</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">c7effB</span> <span class="op">&lt;-</span> <span class="fu">scan1coef</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                    pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"log10_insulin_10wk"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>,</span>
<span>                    kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                    addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                    contrasts <span class="op">=</span> <span class="fu">cbind</span><span class="op">(</span>mu <span class="op">=</span> <span class="fu">c</span><span class="op">(</span>   <span class="fl">1</span>, <span class="fl">1</span>,    <span class="fl">1</span><span class="op">)</span>, </span>
<span>                                      a  <span class="op">=</span> <span class="fu">c</span><span class="op">(</span>  <span class="op">-</span><span class="fl">1</span>, <span class="fl">0</span>,    <span class="fl">1</span><span class="op">)</span>, </span>
<span>                                      d  <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">0.5</span>, <span class="fl">1</span>, <span class="op">-</span><span class="fl">0.5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>The result will then contain the estimates of <code>mu</code>,
<code>a</code> (the additive effect), and <code>d</code> (the dominance
effect).</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">c7effB</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 109   4</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">c7effB</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                 mu           a            d    SexMale
rs8252589  1.011125 -0.02666592 -0.002516938 -0.1753913
rs13479104 1.009704 -0.02277092  0.002728978 -0.1751367
rs13479112 1.008747 -0.02340984  0.006609288 -0.1750756
rs13479114 1.008527 -0.02263568  0.007656166 -0.1750906
rs13479120 1.008946 -0.02383153  0.005991180 -0.1752293
rs13479124 1.009863 -0.02186058  0.002498998 -0.1751596</code></pre>
</div>
<p>For marker rs13479570, <code>mu</code>, <code>a</code>, and
<code>d</code> are 1.0058686, -0.1114157, -0.0201158, -0.164111.</p>
<p>Here’s a plot of the chromosome 7 additive and dominance effects,
which are in the second and third columns.</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">c7effB</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">[</span><span class="va">chr</span><span class="op">]</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          col     <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span><span class="st">'bottomleft'</span>, lty <span class="op">=</span> <span class="fl">1</span>, col <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, legend <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"additive"</span>, <span class="st">"dominant"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-add_dom_contrasts-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></section><section><h2 class="section-heading" id="plotting-phenotypes-versus-genotypes">Plotting Phenotypes versus Genotypes<a class="anchor" aria-label="anchor" href="#plotting-phenotypes-versus-genotypes"></a>
</h2>
<hr class="half-width">
<p>Finally, to plot the raw phenotypes against the genotypes at a single
putative QTL position, you can use the function <code>plot_pxg()</code>.
This takes a vector of genotypes as produced by the
<code>maxmarg()</code> function, which picks the most likely genotype
from a set of genotype probabilities, provided it is greater than some
specified value (the argument <code>minprob</code>). Note that the
“marg” in “maxmarg” stands for “marginal”, as this function is selecting
the genotype at each position that has maximum marginal probability.</p>
<!-- DMG: Change this to Chr 7 -->
<p>For example, we could get inferred genotypes at the chr 7 QTL for the
insulin phenotype (at 28.6 cM) as follows:</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu">maxmarg</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>             map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>             chr   <span class="op">=</span> <span class="va">chr</span>, </span>
<span>             pos   <span class="op">=</span> <span class="va">max_pos</span>, </span>
<span>             return_char <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<p>We use <code>return_char = TRUE</code> to have <code>maxmarg()</code>
return a vector of character strings with the genotype labels.</p>
<p>We then plot the insulin phenotype against these genotypes as
follows:</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_pxg</span><span class="op">(</span>geno   <span class="op">=</span> <span class="va">g</span>, </span>
<span>         pheno  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"log10_insulin_10wk"</span><span class="op">]</span>, </span>
<span>         SEmult <span class="op">=</span> <span class="fl">2</span>,</span>
<span>         main   <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Insulin vs Chr"</span>, <span class="va">chr</span> ,<span class="st">"Genotype"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_pheno_geno_se-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p>Calculate the insulin BLUP effects for chromosome 7.<br>
1) Create an object called <code>blup_chr7</code> to contain the
effects.<br>
2) Plot the chromosome 7 BLUPs and add the LOD plot at bottom. 3) Which
founder allele increases insulin levels?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span> <span class="op">&lt;-</span> <span class="st">'7'</span></span>
<span><span class="va">blup_chr7</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                       pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"log10_insulin_10wk"</span><span class="op">]</span>,</span>
<span>                       addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                       kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>  </span>
<span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">blup_chr7</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,</span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span>,</span>
<span>          main    <span class="op">=</span> <span class="st">"Insulin"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-challenge1-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The C57BL/6J allele on chromosome 7 increases insulin levels.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<p>Calculate the insulin BLUP effects for chromosome 19.<br>
1. Create an object called <code>blup_chr19</code> to contain the
effects.<br>
2. Plot the chromosome 19 BLUPs and add the LOD plot at bottom. 3. Which
founder allele increases insulin levels? 4. Plot insulin versus the
genotype at the marker with the higest LOD on chromosome 19.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span> <span class="op">&lt;-</span> <span class="st">'19'</span></span>
<span><span class="va">blup_chr19</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>, <span class="va">chr</span><span class="op">]</span>, </span>
<span>                        pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>, <span class="st">"log10_insulin_10wk"</span><span class="op">]</span>,</span>
<span>                        addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                        kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>  </span>
<span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">blup_chr19</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,</span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span>,</span>
<span>          main    <span class="op">=</span> <span class="st">"Insulin"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-challenge2-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The BTBR allele on chromosome 19 increases insulin levels.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">max_pos19</span> <span class="op">=</span> <span class="va">peaks</span><span class="op">$</span><span class="va">pos</span><span class="op">[</span><span class="va">peaks</span><span class="op">$</span><span class="va">chr</span> <span class="op">==</span> <span class="va">chr</span><span class="op">]</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu">maxmarg</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>             map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>             chr   <span class="op">=</span> <span class="va">chr</span>, </span>
<span>             pos   <span class="op">=</span> <span class="va">max_pos19</span>, </span>
<span>             return_char <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_pxg</span><span class="op">(</span>geno   <span class="op">=</span> <span class="va">g</span>, </span>
<span>         pheno  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"log10_insulin_10wk"</span><span class="op">]</span>, </span>
<span>         SEmult <span class="op">=</span> <span class="fl">2</span>, </span>
<span>         main   <span class="op">=</span> <span class="st">"Insulin vs Chr 19 Genotype"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-challenge2_b-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>“Estimated founder allele effects can be plotted from the mapping
model coefficients.”</li>
<li>“Additive and dominance effects can be plotted using
contrasts.”</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-do_qtl_mapping"><p>Content from <a href="do_qtl_mapping.html">QTL Mapping in Diversity Outbred Mice</a></p>
<hr>
<p>Last updated on 2024-10-16 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/do_qtl_mapping.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I map traits in Diversity Outbred mice?</li>
<li>How do I interpret the founder allele effects at a QTL peak?</li>
<li>How do I perform association mapping in Diversity Outbred mice?</li>
<li>How do I narrow down the set of candidate genes under a QTL?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the data objects required for QTL mapping.</li>
<li>Use the scan1 family of functions to perform different types of QTL
scans.</li>
<li>Interpret the founder allele effects.</li>
<li>Find significant QTL peaks and identify candidate peaks under a QTL
peak.</li>
<li>Use auxiliary data to identify candidate genes under a QTL
peak.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>This tutorial will take you through the process of mapping a QTL and
searching for candidate genes for DNA damage from benzene exposure. The
adverse health effects of benzene, including leukemia and aplastic
anemia, were first studied in occupational settings in which workers
were exposed to high benzene concentrations. Environmental sources of
benzene exposure typically come from the petrochemical industry,
however, a person’s total exposure can be increased from cigarettes,
consumer products, gas stations, and gasoline powered engines or tools
stored at home.</p>
<p>Exposure thresholds for toxicants are often determined using animal
models that have limited genetic diversity, including standard inbred
and outbred rats and mice. These animal models fail to capture the
influence of genetic diversity on toxicity response, an important
component of human responses to chemicals such as benzene. The Diversity
Outbred (DO) mice reflect a level of genetic diversity similar to that
of humans.</p>
<p>The data comes from a toxicology study in which Diversity Outbred
(DO) mice were exposed to benzene via inhalation for 6 hours a day, 5
days a week for 4 weeks <a href="http://ehp.niehs.nih.gov/1408202/" class="external-link">(French, J. E., et al. (2015)
<i>Environ Health Perspect</i> 123(3): 237-245)</a>. The study was
conducted in two equally sized cohort of 300 male mice each, for a total
of 600 mice. They were then sacrificed and reticulocytes (red blood cell
precursors) were isolated from bone marrow. The number of
micro-nucleated reticulocytes, a measure of DNA damage, was then
measured in each mouse. The goal is to map gene(s) that influence the
level of DNA damage in the bone marrow.</p>
<figure><img src="./fig/benzene_study_design.png" alt="Benzene study dosing showing 6 hours per day, 5 days per week of inhalation." class="figure mx-auto d-block"><div class="figcaption">Benzene Study Dosing</div>
</figure></section><section><h2 class="section-heading" id="do-reference-data">DO Reference Data<a class="anchor" aria-label="anchor" href="#do-reference-data"></a>
</h2>
<hr class="half-width">
<p>As you work with DO data, you may need different reference files.
These are complied on a JAX website at: <a href="https://www.jax.org/research-and-faculty/genetic-diversity-initiative/tools-data/diversity-outbred-reference-data" class="external-link uri">https://www.jax.org/research-and-faculty/genetic-diversity-initiative/tools-data/diversity-outbred-reference-data</a>.
There are links to the markers and founder genotypes as well as the
standard colors that we use for the eight founder strains.</p>
</section><section><h2 class="section-heading" id="open-the-qtl_mapping-project-">Open the qtl_mapping Project.<a class="anchor" aria-label="anchor" href="#open-the-qtl_mapping-project-"></a>
</h2>
<hr class="half-width">
<p>If you have not opened the qtl_mapping project, do this now so that
your script and console will run in the correct directory.</p>
<ol style="list-style-type: decimal">
<li>From the File menu, select “Open Project…”.</li>
<li>Navigate to the project that you created in your Desktop/qtl_mapping
folder and open it.</li>
</ol>
<p>This should reset your RStudio window and close any files.</p>
</section><section><h2 class="section-heading" id="create-a-new-markdown-file">Create a New Markdown File<a class="anchor" aria-label="anchor" href="#create-a-new-markdown-file"></a>
</h2>
<hr class="half-width">
<p>Create a new R Markdown file and save it in the “Desktop/scripts”
folder as “do_qtl_mapping.Rmd”.</p>
</section><section><h2 class="section-heading" id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h2>
<hr class="half-width">
<p>The primary library that we will use is <a href="https://github.com/kbroman/qtl2" class="external-link">qtl2</a>. You should have
<code>qtl2</code> installed from the Setup that you performed before the
workshop.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">ggbeeswarm</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2convert</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="load-and-explore-the-data">Load and Explore the Data<a class="anchor" aria-label="anchor" href="#load-and-explore-the-data"></a>
</h2>
<hr class="half-width">
<p>The data for this tutorial has been saved as an R binary file that
contains several data objects. Load it in now by running the following
command in your new script.</p>


<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="st">"./data/qtl2_demo_grcm39.Rdata"</span><span class="op">)</span></span></code></pre>
</div>
<p>We loaded in four data objects from the *.Rdata file. Look at the
Environment tab to see what was loaded. You should see an object called
<code>pheno</code> which contains the phenotypes. There are two objects
called <code>gmap</code> and <code>pmap</code>, which contain the marker
positions in genetic and physical coordinates. Finally, there is an
object called <code>probs</code> which contains the founder allele
probabilities for each mouse at each marker.</p>
<div class="section level3">
<h3 id="phenotypes">Phenotypes<a class="anchor" aria-label="anchor" href="#phenotypes"></a>
</h3>
<p><code>pheno</code> is a data frame containing the phenotype data.
Click on the blue circle to the left of <code>pheno</code> in the
Environment pane to view its contents.</p>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>The sample IDs <strong>must</strong> be in the rownames of
<code>pheno</code> for qtl2 to work.</p>
</div>
</div>
</div>
<p><code>pheno</code> contains the sample ID, the study cohort, the
concentration of benzene and several blood cell measurements. Note that
the sample IDs are also stored in the rownames of
<code>pheno</code>.</p>
<p>In order to save time for this tutorial, we will only map with 149
samples from the 100 ppm dosing group.</p>
<div id="challenge-1-how-many-mice-are-there-in-the-phenotype-data" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-how-many-mice-are-there-in-the-phenotype-data" class="callout-inner">
<h3 class="callout-title">Challenge 1: How many mice are there in the phenotype data?</h3>
<div class="callout-content">
<p>Look in the Environment tab or use the Console to figure out how many
mice there are in the phenotype data.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Output </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>You can look at the number of observations in the Environment tab or
you can get the number of rows in <code>pheno</code> in the Console.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 598</code></pre>
</div>
<p>There are 598 mice in the phenotype data.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2-how-many-mice-are-there-from-each-sex" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-how-many-mice-are-there-from-each-sex" class="callout-inner">
<h3 class="callout-title">Challenge 2: How many mice are there from each sex?</h3>
<div class="callout-content">
<p>Use an R command to determine how many mice there are from each
sex.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Output </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>You can look at the number of observations in the Environment tab or
you can get the number of rows in <code>pheno</code> in the Console.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">count</span><span class="op">(</span><span class="va">pheno</span>, <span class="va">sex</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>   sex   n
1 male 598</code></pre>
</div>
<p>All of the mice are male.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3-how-is-the-prop.bm.mn.ret-phenotype-distributed" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-how-is-the-prop.bm.mn.ret-phenotype-distributed" class="callout-inner">
<h3 class="callout-title">Challenge 3: How is the “prop.bm.mn.ret” phenotype distributed?</h3>
<div class="callout-content">
<p>Make histogram of the “prop.bm.mn.ret” column and assess whether it
is normally distributed.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Output </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>You can look at the number of observations in the Environment tab or
you can get the number of rows in <code>pheno</code> in the Console.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">pheno</span>, mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span><span class="va">prop.bm.mn.ret</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Histogram of Bone Marrow MN-RETs"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-6-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The bone marrow MN-RET values do not look normally distributed.</p>
</div>
</div>
</div>
</div>
<div id="challenge-4-how-is-the-logprop.bm.mn.ret-phenotype-distributed" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-how-is-the-logprop.bm.mn.ret-phenotype-distributed" class="callout-inner">
<h3 class="callout-title">Challenge 4: How is the log(“prop.bm.mn.ret”) phenotype distributed?</h3>
<div class="callout-content">
<p>Take the log of the “prop.bm.mn.ret” column and make a
histogram.Assess whether it is normally distributed.</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Output </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<p>You can look at the number of observations in the Environment tab or
you can get the number of rows in <code>pheno</code> in the Console.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>log_mnret <span class="op">=</span> <span class="fu">log</span><span class="op">(</span><span class="va">prop.bm.mn.ret</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span><span class="va">log_mnret</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_histogram</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Histogram of log(Bone Marrow MN-RETs)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-7-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The log of the bone marrow MN-RET values look more normally
distributed.</p>
</div>
</div>
</div>
</div>
<p>Let’s add a column containing the log-transformed phenotype to our
<code>pheno</code> object.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="va">pheno</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">mutate</span><span class="op">(</span>log_mnret <span class="op">=</span> <span class="fu">log</span><span class="op">(</span><span class="va">prop.bm.mn.ret</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Some researchers are concerned about the reproducibility of DO
studies. The argument is that each DO mouse is unique and therefore can
never be reproduced. But this misses the point of using the DO. While
mice are the sampling unit, in QTL mapping we are sampling the founder
alleles at each locus. An average of 1/8th of the alleles should come
from each founder at any given locus. Also, DO mice are a
<em>population</em> of mice, not a single strain. While it is true that
results in an individual DO mouse may not be reproducible, results at
the population level should be reproducible. This is similar to the
human population in that results from one individual may not represent
all humans, but results at the population level should be
reproducible.</p>
<p>The benzene inhalation study was conducted in two separate cohorts
(termed <em>study</em> in the <code>pheno</code> object). Let’s plot the
proportion of micronucleated reticulocytes in bone marrow versus the
benzene concentration for each study cohort.</p>

<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>conc <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">conc</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span><span class="va">conc</span>, <span class="va">prop.bm.mn.ret</span> <span class="op">*</span> <span class="fl">1000</span>, color <span class="op">=</span> <span class="va">conc</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_violin</span><span class="op">(</span>draw_quantiles <span class="op">=</span> <span class="fl">0.5</span>, linewidth <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_beeswarm</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Reds"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">study</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Bone Marrow MN-RET by Study Cohort"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>, </span>
<span>          legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-9-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>As you can see, while individual mice have varying micronucleated
reticulocytes, there is a dose-dependent increase in micronucleated
reticulocytes in both cohorts. This is an example of how results in the
DO reproduce at the population level.</p>
</div>
<div class="section level3">
<h3 id="markers">Markers<a class="anchor" aria-label="anchor" href="#markers"></a>
</h3>
<p>The markers are the SNPs on the Mouse Universal Genotyping Array (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4751547/" class="external-link">MUGA</a>)
that were used to reconstruct the haplotypes of the DO mice. This
version of the array had about 8,000 SNPs on it. The latest version
contains over 140,000 SNPs. We are using this smaller data set so that
we can finish the analysis in a timely manner during class.</p>
<p>The markers have been mapped to the latest mouse genome build
(GRCm39) and are provided by Karl Broman on Github at: <a href="https://github.com/kbroman/MUGAarrays/tree/main/UWisc" class="external-link uri">https://github.com/kbroman/MUGAarrays/tree/main/UWisc</a>.
There are four versions of the MUGA platforms. In this study, we used
the MUGA and those marker files are named as “muga_uwisc_vN.csv”, where
“N” is a version number. We will use version 4, which you downloaded
into your “data” directory during the Setup before the workshop.</p>
<p>Open the marker file now:</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers</span> <span class="op">=</span> <span class="fu">read.csv</span><span class="op">(</span><span class="st">"./data/muga_uwisc_v4.csv"</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-5-which-columns-contain-the-marker-positions" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-5-which-columns-contain-the-marker-positions" class="callout-inner">
<h3 class="callout-title">Challenge 5: Which columns contain the marker positions?</h3>
<div class="callout-content">
<p>Look at the structure of the genetic or physical map in the
Environment tab and see which columns contain marker positions in some
coordinate system.</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">str</span><span class="op">(</span><span class="va">markers</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>'data.frame':	7854 obs. of  11 variables:
 $ marker        : chr  "JAX00240603" "UNC010001397" "UNC010515443" "UNC010001943" ...
 $ chr           : chr  "1" "1" "1" "1" ...
 $ bp_mm10       : int  3252796 3336839 3668628 3977130 4430623 4531029 5045931 5840130 6020779 6378154 ...
 $ bp_grcm39     : int  3323019 3407062 3738851 4047353 4500846 4601252 5116154 5910353 6091003 6448378 ...
 $ cM_cox        : num  0.15 0.172 0.194 0.197 0.202 ...
 $ strand        : chr  "minus" "minus" "minus" "plus" ...
 $ snp           : chr  "TC" "AC" "TC" "AC" ...
 $ unique        : logi  TRUE TRUE TRUE TRUE TRUE TRUE ...
 $ unmapped      : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...
 $ probe         : chr  "AAAATATGATTCTTTTAATTAATGCATCAGTGATGAGAAGTGAGAGTGGC" "CATAGTGTCTGGTGAGAAGTCTGGAGTTATTCTAATAGGCCTGCCTTTAT" "CAGGAAATGATGCTGAGAAAGTGAGAAGTAGGAAAACGTGGAGAAAAATA" "TCTATTCCTATCACCTTGTACAAAGCTCAAGTCTTGTAAACCCCCCCCCC" ...
 $ strand_flipped: logi  TRUE FALSE FALSE FALSE FALSE FALSE ...</code></pre>
</div>
<p>The columns labelled “bp_mm10”, “bp_grcm39”, and “cM_cox” contain
marker positions.</p>
</div>
</div>
</div>
</div>
<div id="challenge-6-which-chromosomes-are-in-the-markers" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-which-chromosomes-are-in-the-markers" class="callout-inner">
<h3 class="callout-title">Challenge 6: Which chromosomes are in the markers?</h3>
<div class="callout-content">
<p>Use an R command to determine which chromosomes are in
<code>markers</code>.</p>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">count</span><span class="op">(</span><span class="va">markers</span>, <span class="va">chr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>    chr   n
1     1 493
2    10 375
3    11 335
4    12 318
5    13 327
6    14 331
7    15 265
8    16 266
9    17 227
10   18 216
11   19 162
12    2 474
13    3 438
14    4 412
15    5 382
16    6 414
17    7 379
18    8 341
19    9 342
20    M   3
21    X 414
22    Y   5
23 &lt;NA&gt; 935</code></pre>
</div>
<p>There are 23 chromosomes, including some markers with “NA”. These are
markers that could not be uniquely mapped to one chromosome.</p>
</div>
</div>
</div>
</div>
<div id="challenge-6-what-do-you-think-the-difference-is-between-teh-bp_mm10-and" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-what-do-you-think-the-difference-is-between-teh-bp_mm10-and" class="callout-inner">
<h3 class="callout-title">Challenge 6: What do you think the difference is between teh “bp_mm10” and</h3>
<div class="callout-content">
<p>“bp_grcm39” columns?</p>
<p>Turn to your neighbor and discuss what the differences between the
two column are. Then share your ideas with the class.</p>
</div>
</div>
</div>
<div id="accordionSolution7" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution7" aria-expanded="false" aria-controls="collapseSolution7">
  <h4 class="accordion-header" id="headingSolution7"> Show me the solution </h4>
</button>
<div id="collapseSolution7" class="accordion-collapse collapse" aria-labelledby="headingSolution7" data-bs-parent="#accordionSolution7">
<div class="accordion-body">
<p>There are different builds of the mouse genome. The Genome Resource
Consortium (GRC) named mouse builds from build 1 to 39. The University
of California at Santa Cruz developed their own naming system which
conflicted with the GRC names. “GRCm38” was equivalent to “mm10”. We are
using the GRC naming conventions for genome builds.</p>
</div>
</div>
</div>
</div>
<p><a href="https://github.com/kbroman/qtl2" class="external-link">qtl2</a> needs the markers
to be in a specific format. <code>qtl2</code> requires that the markers
be in a list. The <a href="https://github.com/kbroman/qtl2convert" class="external-link">qtl2convert</a> package
provides functions to make this easier.</p>
<p>First, we will filter the markers to retain ones on the autosomes and
Chr X. We don’t use the mitochondrial or Y chromosomes in mapping, but
they may be important covariates. The MUGA did not contain enough
markers on Chr M and Y to identify the contributing founder strain, but
later versions of the MUGA, such as the GigaMUGA, do contain enough
markers to identify founder contributions.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers</span> <span class="op">&lt;-</span> <span class="va">markers</span> <span class="op">|&gt;</span></span>
<span>             <span class="fu">filter</span><span class="op">(</span><span class="va">chr</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">'X'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will create a column with GRCm39 Mb positions.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers</span> <span class="op">&lt;-</span> <span class="va">markers</span> <span class="op">|&gt;</span></span>
<span>             <span class="fu">mutate</span><span class="op">(</span>pos <span class="op">=</span> <span class="va">bp_grcm39</span> <span class="op">/</span> <span class="fl">1e6</span><span class="op">)</span></span></code></pre>
</div>
<p>Finally, we will use the <a href="">qtl2convert::map_df_to_list()</a>
function to create the list of markers that <code>qtl2</code> requires.
We will call this object the marker “map” to distinguish it from the
marker data.frame.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">map</span> <span class="op">=</span> <span class="fu">qtl2convert</span><span class="fu">::</span><span class="fu">map_df_to_list</span><span class="op">(</span>map <span class="op">=</span> <span class="va">markers</span>, pos_column <span class="op">=</span> <span class="st">"pos"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="founder-allele-probabilities-genoprobs">Founder Allele Probabilities (Genoprobs)<a class="anchor" aria-label="anchor" href="#founder-allele-probabilities-genoprobs"></a>
</h3>
<p>The <code>probs</code> object is a list with 20 elements. Each
element of <code>probs</code> is a three-dimensional array containing
the founder allele dosages for each sample at each marker on one
chromosome. These probabilities have been pre-calculated for you, so you
can skip the step for calculating allele probabilities.</p>
<!-- DMG: We need a figure showing the structure of the genoprobs. It should
show a list of boxes. We can show the first three chromosomes and then ... -->
<p>Let’s look at the dimensions of <code>probs</code> for chromosome
1:</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 590   8 479</code></pre>
</div>
<p><code>probs[[1]]</code> is a three-dimensional array containing the
proportion of each founder haplotype at each marker for each DO sample.
The 590 samples are in the first dimension, the 8 founders in the second
and the 479 markers along chromosome 1 are in the third dimension.</p>
<p>Let’s return to the <code>probs</code> object. Look at the contents
of one sample on chromosome 1.</p>

<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_genoprob</span><span class="op">(</span><span class="va">probs</span>, <span class="va">map</span>, ind <span class="op">=</span> <span class="fl">1</span>, chr <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<p>Uh oh! We have an error which says “Different numbers of positions in
probs and map”. This means that the number of markers in the genoprobs
and the marker map is different. In fact, this is quite important. The
markers in <code>probs</code> and <code>map</code> must match
<strong>exactly</strong> on every chromosome.</p>
<div id="challenge-7-align-the-markers-in-probs-and-map." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-7-align-the-markers-in-probs-and-map." class="callout-inner">
<h3 class="callout-title">Challenge 7: Align the markers in
<code>probs</code> and <code>map</code>.</h3>
<div class="callout-content">
<p>Work with the person next to you to write a short script that will
get the intersection of the marker names in the <code>probs</code> and
<code>map</code> for each chromosome, subset the markers in each object
to only contain the common markers, and make sure that the markers in
<code>probs</code> are in the same order as in <code>map</code>.</p>
<p><strong>Hint</strong>: The markers names for each chromosome in
<code>probs</code> are in dimention 3. i.e. dimnames(probs[[1]])[[3]]
contains the marker names for chromosome 1.</p>
</div>
</div>
</div>
<div id="accordionSolution8" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution8" aria-expanded="false" aria-controls="collapseSolution8">
  <h4 class="accordion-header" id="headingSolution8"> Show me the solution </h4>
</button>
<div id="collapseSolution8" class="accordion-collapse collapse" aria-labelledby="headingSolution8" data-bs-parent="#accordionSolution8">
<div class="accordion-body">
<p>We will write a loop that gets the common markers between probs and
map for each chromosome and then subsets them.</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Verify that probs and map are the same length.</span></span>
<span><span class="fu">stopifnot</span><span class="op">(</span><span class="fu">length</span><span class="op">(</span><span class="va">probs</span><span class="op">)</span> <span class="op">==</span> <span class="fu">length</span><span class="op">(</span><span class="va">map</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Verify that probs and map have their chromsomes in the same order.</span></span>
<span><span class="fu">stopifnot</span><span class="op">(</span><span class="fu">names</span><span class="op">(</span><span class="va">probs</span><span class="op">)</span> <span class="op">==</span> <span class="fu">names</span><span class="op">(</span><span class="va">map</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Loop through each chromosome...</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu">seq_along</span><span class="op">(</span><span class="va">probs</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="co"># Get the intersection of the markers in probs and map.</span></span>
<span>  <span class="va">common_markers</span> <span class="op">&lt;-</span> <span class="fu">intersect</span><span class="op">(</span><span class="fu">dimnames</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span>, <span class="fu">names</span><span class="op">(</span><span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Subset the map to contain the common markers.</span></span>
<span>  <span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">common_markers</span><span class="op">]</span></span>
<span>  </span>
<span>  <span class="co"># Subset the probs to contain the common markers.</span></span>
<span>  <span class="va">probs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">probs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,,<span class="fu">names</span><span class="op">(</span><span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">]</span></span>
<span>  </span>
<span>  <span class="co"># Verify that the markers are identical in probs and map.</span></span>
<span>  <span class="fu">stopifnot</span><span class="op">(</span><span class="fu">dimnames</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="fu">names</span><span class="op">(</span><span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span><span class="op">}</span> <span class="co"># for(i)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>Now that we have aligned the markers in <code>probs</code> and
<code>map</code>, let’s plot the allele probabilities for one sample on
chromosome 1.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_genoprob</span><span class="op">(</span><span class="va">probs</span>, <span class="va">map</span>, ind <span class="op">=</span> <span class="fl">1</span>, chr <span class="op">=</span> <span class="fl">1</span>, main <span class="op">=</span> <span class="st">"Founder Allele Probabilities"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot above, the founder contributions, which range between 0
and 1, are colored from white (= 0) to black (= 1.0). A value of ~0.5 is
grey. The markers are on the X-axis and the eight founders (denoted by
the letters A through H) on the Y-axis. Starting at the left, we see
that this sample has genotype BB because the row for B is black,
indicating values of 1.0. Moving along the genome to the right, the
genotype becomes CE where rows C and E are gray, followed by CD, FH, AG,
GH, etc. The values at each marker sum to 1.0.</p>
</div>
<div class="section level3">
<h3 id="calculating-a-kinship-matrix">Calculating A Kinship Matrix<a class="anchor" aria-label="anchor" href="#calculating-a-kinship-matrix"></a>
</h3>
<p>Next, we need to create a matrix that accounts for the kinship
relationships between the mice. We do this by looking at the correlation
between the founder haplotypes for each sample at each SNP. For each
chromosome, we create a kinship matrix using all markers <em>except</em>
the ones on the current chromosome using the “loco”
(leave-one-chromosome-out) method. Simulations suggest that mapping
using this approach increases the power to detect QTL.</p>
<div id="callout2" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>The sample IDs must be in the rownames of <code>probs</code>. The
sample IDs will be copied to the row and column names in the kinship
matrices.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">K</span> <span class="op">&lt;-</span> <span class="fu">calc_kinship</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, type <span class="op">=</span> <span class="st">"loco"</span><span class="op">)</span></span></code></pre>
</div>
<p>Kinship values between pairs of samples range between 0 (no
relationship) and 1.0 (completely identical). Let’s look at the kinship
matrix for the first 50 samples.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_samples</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="fu">heatmap</span><span class="op">(</span><span class="va">K</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-kinship_probs-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The figure above shows kinship between all pairs of samples. Light
yellow indicates low kinship and dark red indicates higher kinship.
Orange values indicate varying levels of kinship between 0 and 1. The
dark red diagonal of the matrix indicates that each sample is identical
to itself. The orange blocks along the diagonal may indicate close
relatives (i.e. siblings or cousins).</p>
</div>
<div class="section level3">
<h3 id="covariates">Covariates<a class="anchor" aria-label="anchor" href="#covariates"></a>
</h3>
<p>Next, we need to create additive covariates that will be used in the
mapping model. We will use study cohort as a covariate in the mapping
model. This is the same as outbreeding generation since each cohort was
purchased from successive generations. If we were mapping with
<em>all</em> mice, we would also add benzene concentration to the model.
This study contained only male mice, but in most cases, you would
include sex as an additive covariate as well.</p>
<p>To recap, you would normally add sex and outbreeding generation to
the model. In this case, we have one sex and are using
<code>study</code> instead of generation.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">addcovar</span> <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">study</span>, data <span class="op">=</span> <span class="va">pheno</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<p>The code above copies the <code>rownames(pheno)</code> to
<code>rownames(addcovar)</code> as a side-effect.</p>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>The sample IDs <strong>must</strong> be in the rownames of
<code>pheno</code>, <code>addcovar</code>, <code>genoprobs</code> and
<code>K</code>. <code>qtl2</code> uses the sample IDs to align the
samples between objects. For more information about data file format,
see <a href="http://kbroman.org/qtl2/assets/vignettes/input_files.html" class="external-link">Karl
Broman’s vignette on input file format</a>.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="performing-a-genome-scan">Performing a Genome Scan<a class="anchor" aria-label="anchor" href="#performing-a-genome-scan"></a>
</h2>
<hr class="half-width">
<p>Before we perform our first genome scan, let’s look at the mapping
model. At each marker on the genotyping array, we will fit a model that
regresses the phenotype (micronucleated reticulocytes) on covariates and
the founder allele proportions.</p>
<p><span class="math inline">\(y_i =
\beta_ss_i+\sum_{j=1}^8(\beta_jg_{ij}) + \lambda_i +
\epsilon_i\)</span></p>
<p>where:</p>
<p><span class="math inline">\(y_i\)</span> is the phenotype for mouse
<span class="math inline">\(i\)</span>, <span class="math inline">\(\beta_s\)</span> is the effect of study cohort,
<span class="math inline">\(s_i\)</span> is the study cohort for mouse
<span class="math inline">\(i\)</span>, <span class="math inline">\(\beta_j\)</span> is the effect of founder allele
<span class="math inline">\(j\)</span>, <span class="math inline">\(g_{ij}\)</span> is the probability that mouse
<span class="math inline">\(i\)</span> carries an allele from founder
<span class="math inline">\(j\)</span>, <span class="math inline">\(\lambda_;&lt;sub&gt;_i\)</span> is an adjustment
for kinship-induced correlated errors for mouse <span class="math inline">\(i\)</span>, <span class="math inline">\(\epsilon_i\)</span> is the residual error for
mouse <span class="math inline">\(i\)</span>.</p>
<p>Note that this model will give us an estimate of the effect of each
of the eight founder alleles at each marker. This will be important when
we estimate the founder allele effects below.</p>
<p>There are almost 600 samples in this data set and it may take several
minutes to map one trait. In order to save some time, we will map using
only the samples in the 100 ppm concentration group. We will create a
smaller phenotype data.frame.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_100</span> <span class="op">&lt;-</span> <span class="va">pheno</span><span class="op">[</span><span class="va">pheno</span><span class="op">$</span><span class="va">conc</span> <span class="op">==</span> <span class="fl">100</span>,<span class="op">]</span></span></code></pre>
</div>
<div id="challenge-8-how-many-mice-are-we-mapping-with" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-8-how-many-mice-are-we-mapping-with" class="callout-inner">
<h3 class="callout-title">Challenge 8: How many mice are we mapping with?</h3>
<div class="callout-content">
<p>Get the number of rows in <code>pheno_100</code>.</p>
</div>
</div>
</div>
<div id="accordionSolution9" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution9" aria-expanded="false" aria-controls="collapseSolution9">
  <h4 class="accordion-header" id="headingSolution9"> Show me the solution </h4>
</button>
<div id="collapseSolution9" class="accordion-collapse collapse" aria-labelledby="headingSolution9" data-bs-parent="#accordionSolution9">
<div class="accordion-body">
<p>You can do this by looking at the number of observations for
<code>pheno_100</code> in the Environment tab or in the Console.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">pheno_100</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 149</code></pre>
</div>
<p>There are 149 mice in <code>pheno_100</code>.</p>
</div>
</div>
</div>
</div>
<p>In order to map the proportion of bone marrow reticulocytes that were
micro-nucleated, you will use the <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_scan1.R" class="external-link">scan1</a>
function. To see the arguments for <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_scan1.R" class="external-link">scan1</a>,
you can type <code>help(scan1)</code>. First, let’s map the
<em>untransformed</em> phenotype. (Recall that we log-transformed it
above).</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">pheno_100</span><span class="op">)</span> <span class="op">==</span> <span class="st">"prop.bm.mn.ret"</span><span class="op">)</span></span>
<span><span class="va">lod</span>   <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>               pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>               kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>               addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we plot the genome scan.</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">map</span>, </span>
<span>           main <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-qtl_plot-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>It looks like we have a large peak on chromosome 10.</p>
<div id="challenge-9-how-does-a-log-tranformation-change-the-qtl-plot" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-9-how-does-a-log-tranformation-change-the-qtl-plot" class="callout-inner">
<h3 class="callout-title">Challenge 9: How does a log-tranformation change the QTL plot?</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Perform a genome scan on the column called <code>log_mnret</code>.
(Hint: set <code>index</code> to the column index in
<code>pheno</code>.)<br>
</li>
<li>How does the LOD score for the peak on Chr 10 change?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution10" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution10" aria-expanded="false" aria-controls="collapseSolution10">
  <h4 class="accordion-header" id="headingSolution10"> Show me the solution </h4>
</button>
<div id="collapseSolution10" class="accordion-collapse collapse" aria-labelledby="headingSolution10" data-bs-parent="#accordionSolution10">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">pheno_100</span><span class="op">)</span> <span class="op">==</span> <span class="st">"log_mnret"</span><span class="op">)</span></span>
<span><span class="va">lod2</span>  <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>               pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>               kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>               addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod2</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">map</span>, </span>
<span>           main <span class="op">=</span> <span class="st">"(log(Proportion of Micro-nucleated Bone Marrow Reticulocytes)"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-22-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Using a log transformation increases the LOD increase from about 17
to over 25.</p>
</div>
</div>
</div>
</div>
<p>The challenge shows the importance of looking at your data and
transforming it to meet the expectations of the mapping model. In this
case, a log transformation improved the model fit and increased the LOD
score. We will continue the rest of this lesson using the
log-transformed data. Set your <code>index</code> variable equal to the
column index of <code>log.MN.RET</code>. Redo the genome scan with the
log-transformed reticulocytes.</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span> <span class="op">==</span> <span class="st">"log_mnret"</span><span class="op">)</span></span>
<span><span class="va">lod</span>   <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>               pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>, <span class="va">index</span>,  drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>               kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>               addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="assessing-significance-of-lod-scores">Assessing Significance of LOD Scores<a class="anchor" aria-label="anchor" href="#assessing-significance-of-lod-scores"></a>
</h2>
<hr class="half-width">
<p>There is clearly a large peak on Chr 10. But is it significant? In
other words, could we see a LOD score over 25 by chance? And if so, what
is the probability of seeing a LOD of 25 or higher?</p>
<p>These questions are most commonly answered via <a href="http://www.genetics.org/content/178/1/609.long" class="external-link">permutation</a> of
the sample labels and recording the maximum LOD score in each
permutation. This procedure breaks the connection between the phenotypes
and the genotypes of the mice, so the results represent the expected LOD
by chance.</p>
<p>Imagine that you shuffle the sample labels in the phenotype file and
map the resampled trait. Then you record the maximum LOD across the
genome. You would get a table that looks like this:</p>
<table class="table">
<thead><tr class="header">
<th>Permutation</th>
<th>Max LOD</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>1.7</td>
</tr>
<tr class="even">
<td>2</td>
<td>3.2</td>
</tr>
<tr class="odd">
<td>3</td>
<td>2.2</td>
</tr>
<tr class="even">
<td>4</td>
<td>1.8</td>
</tr>
<tr class="odd">
<td>5</td>
<td>2.3</td>
</tr>
<tr class="even">
<td>…</td>
<td>…</td>
</tr>
</tbody>
</table>
<p>Once you ran 1,000 permutations, you could plot a histogram of the
maximum LOD values.</p>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perms</span>   <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">'./data/sim_perm1000.rds'</span><span class="op">)</span></span>
<span><span class="va">max_lod</span> <span class="op">=</span> <span class="fu">apply</span><span class="op">(</span><span class="va">perms</span>, <span class="fl">2</span>, <span class="va">max</span><span class="op">)</span></span>
<span><span class="fu">hist</span><span class="op">(</span><span class="va">max_lod</span>, breaks <span class="op">=</span> <span class="fl">100</span>, main <span class="op">=</span> <span class="st">'Histogram of Max. LOD'</span><span class="op">)</span></span>
<span><span class="va">q95</span> <span class="op">=</span> <span class="fu">quantile</span><span class="op">(</span><span class="va">max_lod</span>, probs <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span>
<span><span class="fu">abline</span><span class="op">(</span>v <span class="op">=</span> <span class="va">q95</span>, col <span class="op">=</span> <span class="st">'red'</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-23-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In this case, the maximum LOD score is around 9. Most of the values
fall between 5 and 6. We have drawn a red line at the 95th percentile of
the distribution. LOD values above this are likely to occur 5% of the
time. That means 1 in 20 times. We often use this 95th percentile as our
significance threshold for one trait.</p>
<p>We advise running at least 1,000 permutations to obtain significance
thresholds. In the interest of time, we perform 100 permutations here
using the <a href="">scan1perm</a> function.</p>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perms</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                   pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                   kinshp    <span class="op">=</span> <span class="va">K</span>,</span>
<span>                   addcovar  <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                   n_perm    <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></code></pre>
</div>
<p>The <code>perms</code> object contains the maximum LOD score from
each genome scan of permuted data.</p>
<div id="challenge-10-how-does-a-log-tranformation-change-the-qtl-plot" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-10-how-does-a-log-tranformation-change-the-qtl-plot" class="callout-inner">
<h3 class="callout-title">Challenge 10: How does a log-tranformation change the QTL plot?</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Create a histogram of the LOD scores <code>perms</code>. Hint: use
the <code>hist()</code> function.<br>
</li>
<li>Estimate the value of the LOD score at the 95th percentile.<br>
</li>
<li>Then find the value of the LOD score at the 95th percentile using
the <code>summary()</code> function.</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution11" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution11" aria-expanded="false" aria-controls="collapseSolution11">
  <h4 class="accordion-header" id="headingSolution11"> Show me the solution </h4>
</button>
<div id="collapseSolution11" class="accordion-collapse collapse" aria-labelledby="headingSolution11" data-bs-parent="#accordionSolution11">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb38">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span>x <span class="op">=</span> <span class="va">perms</span>, breaks <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-24-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perms</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (100 permutations)
     log_mnret
0.05      7.23</code></pre>
</div>
<p>Note that this summary function returns the 95th percentile value of
the LOD distribution.</p>
</div>
</div>
</div>
</div>
<p>We can now add thresholds to the previous QTL plot. We use a
significance threshold of p &lt; 0.05. To do this, we select the 95th
percentile of the permutation LOD distribution.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod</span>, </span>
<span>     map  <span class="op">=</span> <span class="va">map</span>,  </span>
<span>     main <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span><span class="op">)</span></span>
<span><span class="va">thr</span> <span class="op">=</span> <span class="fu">summary</span><span class="op">(</span><span class="va">perms</span><span class="op">)</span></span>
<span><span class="fu">add_threshold</span><span class="op">(</span>map <span class="op">=</span> <span class="va">map</span>, thresholdA <span class="op">=</span> <span class="va">thr</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-25-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The peak on Chr 10 is well above the red significance line.</p>
</section><section><h2 class="section-heading" id="finding-lod-peaks">Finding LOD Peaks<a class="anchor" aria-label="anchor" href="#finding-lod-peaks"></a>
</h2>
<hr class="half-width">
<p>We can find all of the peaks above the significance threshold using
the <a href="https://github.com/rqtl/qtl2/blob/master/R/find_peaks.R" class="external-link">find_peaks</a>
function.</p>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod</span>,</span>
<span>           map          <span class="op">=</span> <span class="va">map</span>, </span>
<span>           threshold    <span class="op">=</span> <span class="va">thr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr      pos      lod
1        1 log_mnret  10 33.52438 27.01701</code></pre>
</div>
<div id="challenge-11-how-does-a-log-tranformation-change-the-qtl-plot" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-11-how-does-a-log-tranformation-change-the-qtl-plot" class="callout-inner">
<h3 class="callout-title">Challenge 11: How does a log-tranformation change the QTL plot?</h3>
<div class="callout-content">
<p>Find all peaks for this scan whether or not they meet the 95%
significance threshold.</p>
</div>
</div>
</div>
<div id="accordionSolution12" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution12" aria-expanded="false" aria-controls="collapseSolution12">
  <h4 class="accordion-header" id="headingSolution12"> Show me the solution </h4>
</button>
<div id="collapseSolution12" class="accordion-collapse collapse" aria-labelledby="headingSolution12" data-bs-parent="#accordionSolution12">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">map</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>   lodindex lodcolumn chr        pos       lod
1         1 log_mnret   1   4.601252  3.594686
2         1 log_mnret   2 141.066196  5.351645
3         1 log_mnret   3 124.914616  3.904994
4         1 log_mnret   4 132.636314  5.501382
5         1 log_mnret   6  85.246537  5.965923
6         1 log_mnret   7  44.136809  4.325292
7         1 log_mnret   8  26.176972  3.073792
8         1 log_mnret   9   8.115893  4.027077
9         1 log_mnret  10  33.524378 27.017008
10        1 log_mnret  11  48.605368  3.563292
11        1 log_mnret  12  28.480475  3.549043
12        1 log_mnret  13 109.752043  3.111400
13        1 log_mnret  16  72.091039  5.263606
14        1 log_mnret  17  12.528523  3.659285
15        1 log_mnret  18  65.977753  3.894702
16        1 log_mnret   X   8.770202  4.059789</code></pre>
</div>
<p>Notice that some peaks are missing because they don’t meet the
default threshold value of 3. See <code>help(find_peaks)</code> for more
information about this function.</p>
</div>
</div>
</div>
</div>
<p>The support interval is determined using the<br><a href="http://www.ncbi.nlm.nih.gov/pubmed/11560912" class="external-link">Bayesian Credible
Interval</a> and represents the region most likely to contain the
causative polymorphism(s). We can obtain this interval by adding a
<code>prob</code> argument to <a href="https://github.com/rqtl/qtl2/blob/master/R/find_peaks.R" class="external-link">find_peaks</a>.
We pass in a value of <code>0.95</code> to request a support interval
that contains the causal SNP 95% of the time.</p>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">map</span>, </span>
<span>           threshold    <span class="op">=</span> <span class="va">thr</span>, </span>
<span>           prob         <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr      pos      lod   ci_lo    ci_hi
1        1 log_mnret  10 33.52438 27.01701 31.7442 34.05311</code></pre>
</div>
<p>From the output above, you can see that the support interval is 2.3
Mb wide (31.7442 to 34.05311 Mb). The location of the maximum LOD score
is at 33.52438 Mb.</p>
</section><section><h2 class="section-heading" id="estimated-founder-allele-effects">Estimated Founder Allele Effects<a class="anchor" aria-label="anchor" href="#estimated-founder-allele-effects"></a>
</h2>
<hr class="half-width">
<p>We will now zoom in on Chr 10 and look at the contribution of each of
the eight founder alleles to the proportion of bone marrow reticulocytes
that were micro-nucleated. Remember, the mapping model above estimates
the effect of each of the eight DO founders. We can plot these effects
(also called ‘coefficients’) across Chr 10 using <a href="https://github.com/rqtl/qtl2/blob/master/R/scan1coef.R" class="external-link">scan1coef</a>.</p>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>    <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">coef10</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                    pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                    kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                    addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>This produces an object containing estimates of each of the eight DO
founder allele effect. These are the <i>β<sub>j</sub></i> values in the
mapping equation above.</p>
<div class="codewrapper sourceCode" id="cb49">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coefCC</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">coef10</span>, </span>
<span>            map  <span class="op">=</span> <span class="va">map</span>, </span>
<span>            scan1_output <span class="op">=</span> <span class="va">lod</span>, </span>
<span>            main <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-27-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The top panel shows the eight founder allele effects (or model
coefficients) along Chr 10. The founder allele effects are centered at
zero and the units are the same as the phenotype. You can see that DO
mice containing the CAST/EiJ allele near 34 Mb have lower levels of
micro-nucleated reticulocytes. This means that the CAST allele is
associated with less DNA damage and has a protective effect. The bottom
panel shows the LOD score, with the support interval for the peak shaded
blue.</p>
</section><section><h2 class="section-heading" id="snp-association-mapping">SNP Association Mapping<a class="anchor" aria-label="anchor" href="#snp-association-mapping"></a>
</h2>
<hr class="half-width">
<p>At this point, we have a 2.3 Mb wide support interval that contains
polymorphism(s) that influence benzene-induced DNA damage. Next, we will
impute the DO founder sequences onto the DO genomes. The <a href="https://www.mousegenomes.org/" class="external-link">Mouse Genomes Project</a> has
sequenced the eight DO founders and provides SNP, insertion-deletion
(indel), and structural variant files for the strains (see <a href="http://www.nature.com/ng/journal/v45/n7/full/ng.2644.html" class="external-link">Baud
et.al., Nat. Gen., 2013</a>). We can impute these SNPs onto the DO
genomes and then perform association mapping. The process involves
several steps and I have provided a function to encapsulate the steps.
To access the Sanger SNPs, we use a SQLlite database provided by <a href="https://github.com/kbroman" class="external-link">Karl Broman</a>. You should have
downloaded this during Setup. It is available from<br><a href="https://figshare.com/ndownloader/files/40157572" class="external-link">figshare</a>,
but the file is 10 GB, so it may take too long to download right
now.</p>
<p><img src="./fig/DO.impute.founders.sm.png" alt="SNP Imputation" class="figure">{alt=“Figure showing haplotypes being used to
impute founder SNPs onto DO genomes.”,width=75%}</p>
<p>Association mapping involves imputing the founder SNPs onto each DO
genome and fitting the mapping model at each SNP. At each marker, we fit
the following model:</p>
<p><span class="math inline">\(y_{im} = \beta_ss_i+\beta_mg_{im} +
\lambda_i + \epsilon_i\)</span></p>
<p>where:</p>
<p><span class="math inline">\(y_i\)</span> is the phenotype for mouse
<span class="math inline">\(i\)</span>, <span class="math inline">\(\beta_s\)</span> is the effect of study cohort,
<span class="math inline">\(s_i\)</span> is the study cohort for mouse
<span class="math inline">\(i\)</span>, <span class="math inline">\(\beta_m\)</span> is the effect of adding one
allele at marker <span class="math inline">\(m\)</span>, <span class="math inline">\(g_{im}\)</span> is the allele call for mouse <span class="math inline">\(i\)</span> at marker <span class="math inline">\(m\)</span>&gt;, <span class="math inline">\(\lambda_i\)</span> is an adjustment for
kinship-induced correlated errors for mouse <span class="math inline">\(i\)</span>, <span class="math inline">\(\epsilon_i\)</span> is the residual error for
mouse <span class="math inline">\(i\)</span>.</p>
<p>We can call <a href="https://github.com/rqtl/qtl2/blob/master/R/scan1snps.R" class="external-link">scan1snps</a>
to perform association mapping in the QTL interval on Chr 10. We first
create variables for the chromosome and support interval where we are
mapping. We then create a function to get the SNPs from the founder SNP
database.</p>
<div id="callout4" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>It is important to use the <code>keep_all_snps = TRUE</code> in order
to return all SNPs.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>   <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">start</span> <span class="op">&lt;-</span> <span class="fl">30</span></span>
<span><span class="va">end</span>   <span class="op">&lt;-</span> <span class="fl">36</span></span>
<span></span>
<span><span class="co"># Create function to query founder SNP database.</span></span>
<span><span class="va">query_snps</span>  <span class="op">&lt;-</span> <span class="fu">create_variant_query_func</span><span class="op">(</span>dbfile   <span class="op">=</span> <span class="st">"./data/fv.2021.snps.db3"</span>,</span>
<span>                                         id_field <span class="op">=</span> <span class="st">"variants_id"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">assoc</span>      <span class="op">&lt;-</span> <span class="fu">scan1snps</span><span class="op">(</span>genoprobs  <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                        map        <span class="op">=</span> <span class="va">map</span>, </span>
<span>                        pheno      <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                        kinship    <span class="op">=</span> <span class="va">K</span>, </span>
<span>                        addcovar   <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                        query_func <span class="op">=</span> <span class="va">query_snps</span>,</span>
<span>                        chr        <span class="op">=</span> <span class="va">chr</span>, </span>
<span>                        start      <span class="op">=</span> <span class="va">start</span>, </span>
<span>                        end        <span class="op">=</span> <span class="va">end</span>, </span>
<span>                        keep_all_snps <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<p>The <code>assoc</code> object is a list containing two objects: the
LOD scores for each unique SNP and a <code>snpinfo</code> object that
maps the LOD scores to each SNP. To plot the association mapping, we
need to provide both objects to the <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_snpasso.R" class="external-link">plot_snpasso</a>
function.</p>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#png('./fig/bm_mnret_assoc_fig1.png', width = 2000, height = 2000, res = 256)</span></span>
<span><span class="fu">plot_snpasso</span><span class="op">(</span>scan1output <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">lod</span>, </span>
<span>             snpinfo     <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">snpinfo</span>, </span>
<span>             main        <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span><span class="op">)</span></span>
<span><span class="co">#dev.off()</span></span></code></pre>
</div>
<p><img src="./fig/bm_mnret_assoc_fig1.png" alt="Bone Marrow MN-RET Association Mapping" class="figure">{alt=“Plot showing LOD of
each SNP in QTL interval”,width=100%}</p>
<p>This plot shows the LOD score for each SNP in the QTL interval. The
SNPs occur in “shelves” because all of the SNPs in a haplotype block
have the same founder strain pattern. The SNPs with the highest LOD
scores are the ones for which CAST/EiJ contributes the alternate
allele.</p>
<p>You might naturally ask what genes are under this QTL peak. We will
query the SNP/gens database for the genes in the interval. As before, we
will create a function to query the genes in a specific interval of the
genome.</p>
<div class="codewrapper sourceCode" id="cb52">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create function to query gene database.</span></span>
<span><span class="va">query_genes</span> <span class="op">&lt;-</span> <span class="fu">create_gene_query_func</span><span class="op">(</span>dbfile      <span class="op">=</span> <span class="st">"./data/fv.2021.snps.db3"</span>,</span>
<span>                                      chr_field   <span class="op">=</span> <span class="st">"chromosome"</span>, </span>
<span>                                      name_field  <span class="op">=</span> <span class="st">"symbol"</span>,</span>
<span>                                      start_field <span class="op">=</span> <span class="st">"start_position"</span>, </span>
<span>                                      stop_field  <span class="op">=</span> <span class="st">"end_position"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu">query_genes</span><span class="op">(</span><span class="va">chr</span>, <span class="va">start</span>, <span class="va">end</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          ensembl_id ensembl_version    Name                     name chromosome start_position end_position strand    start
1 ENSMUSG00000000295              14   Hddc2   HD domain containing 2         10       31189379     31204200      1 31.18938
2 ENSMUSG00000000296               9 Tpd52l1 tumor protein D52-like 1         10       31208372     31321954     -1 31.20837
3 ENSMUSG00000019779              16     Frk       fyn-related kinase         10       34359395     34487274      1 34.35939
4 ENSMUSG00000019782              11   Rwdd1  RWD domain containing 1         10       33872551     33895620     -1 33.87255
5 ENSMUSG00000019785              15   Clvs2               clavesin 2         10       33388282     33500765     -1 33.38828
6 ENSMUSG00000019787              10    Trdn                  triadin         10       32956550     33352705      1 32.95655
      stop
1 31.20420
2 31.32195
3 34.48727
4 33.89562
5 33.50077
6 33.35271</code></pre>
</div>
<p>The <code>genes</code> object contains annotation information for
each gene in the interval.</p>
<p>Next, we will create a plot with two panels: one containing the
association mapping LOD scores and one containing the genes in the QTL
interval. We do this by passing in the <code>genes</code> argument to <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_snpasso.R" class="external-link">plot_snpasso</a>.
We can also adjust the proportion of the plot allocated for SNPs (on the
top) and genes (on the bottom) using the ‘top_panel_prop’ argument.</p>

<div class="codewrapper sourceCode" id="cb54">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#png('./fig/bm_mnret_assoc_fig2.png', width = 2000, height = 2000, res = 256)</span></span>
<span><span class="fu">plot_snpasso</span><span class="op">(</span>scan1output <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">lod</span>, </span>
<span>             snpinfo     <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">snpinfo</span>, </span>
<span>             main        <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span>, </span>
<span>             genes       <span class="op">=</span> <span class="va">genes</span>,</span>
<span>             colors      <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>             top_panel_prop <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span>
<span><span class="co">#dev.off()</span></span></code></pre>
</div>
<p><img src="./fig/bm_mnret_assoc_fig2.png" alt="Bone Marrow MN-RET Association Mapping" class="figure">{alt=“Plot showing LOD of
each SNP in QTL interval with genes below”,width=100%}</p>
<p>Now that we have the genes in this interval, we would like to know
which founders have the minor allele for the SNPs with the highest LOD
scores. To do this, we will highlight SNPs that are withing a 1 LOD drop
of the highest LOD and we will add an argument to show which founder
contributes the minor allele at the highlighted SNPs.</p>

<div class="codewrapper sourceCode" id="cb55">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#png('./fig/bm_mnret_assoc_fig3.png', width = 2000, height = 2000, res = 256)</span></span>
<span><span class="fu">plot_snpasso</span><span class="op">(</span>scan1output <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">lod</span>, </span>
<span>             snpinfo     <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">snpinfo</span>, </span>
<span>             main        <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span>, </span>
<span>             genes       <span class="op">=</span> <span class="va">genes</span>,</span>
<span>             colors      <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>             panel_prop <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>             drop_hilit  <span class="op">=</span> <span class="fl">1.0</span>,</span>
<span>             col_hilit   <span class="op">=</span> <span class="st">'red'</span>,</span>
<span>             sdp_panel   <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#dev.off()</span></span></code></pre>
</div>
<p><img src="./fig/bm_mnret_assoc_fig3.png" alt="Bone Marrow MN-RET Association Mapping" class="figure">{alt=“Plot showing LOD of
each SNP in QTL interval with genes below and strain distribution
pattern above”,width=100%}</p>
<div class="section level3">
<h3 id="searching-for-candidate-genes">Searching for Candidate Genes<a class="anchor" aria-label="anchor" href="#searching-for-candidate-genes"></a>
</h3>
<p>One strategy for finding genes related to a phenotype is to search
for genes with expression QTL (eQTL) in the same location. Ideally, we
would have liver and bone marrow gene expression data in the DO mice
from this experiment. Unfortunately, we did not collect this data.
However, we have liver gene expression for a separate set of untreated
DO mice <a href="https://churchilllab.jax.org/qtlviewer/svenson/DOHFD" class="external-link">Liver eQTL
Viewer</a>. We searched for genes in the QTL interval that had an eQTL
in the same location. Then, we looked at the pattern of founder effects
to see if CAST stood out. We found two genes that met these
criteria.</p>
<p><img src="./fig/French.et.al.Figure3.png" alt="Sult3a1/Sult3a2 Liver eQTL" class="figure">{Figure showing LOD plots of Sult3a1
&amp; 2 and CAST-specific haplotype effects}</p>
<p>The plot above shows the genome scane for two genes: <em>Sult3a1</em>
and <em>Gm4794</em>. <em>Gm4794</em> has been renamed to
<em>Sult3a2</em>. As you can see, both <em>Sult3a1</em> <em>Sult3a2</em>
have eQTL in the same location at the MN-RET QTL on Chr 10. Mice
carrying the CAST allele (in green) express these genes more highly.
<em>Sult3a1</em> is a <a href="https://en.wikipedia.org/wiki/Sulfotransferase" class="external-link">sulfotransferase</a>
that may be involved in adding a sulfate group to phenol, one of the
metabolites of benzene.</p>
<div id="challenge-11-look-for-sult3a1-sult3a2-qtl-in-do-bone-expression" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-11-look-for-sult3a1-sult3a2-qtl-in-do-bone-expression" class="callout-inner">
<h3 class="callout-title">Challenge 11: Look for Sult3a1 &amp; Sult3a2 QTL in DO bone expression</h3>
<div class="callout-content">
<p>Go to the <a href="https://churchilllab.jax.org/qtlviewer/DO/bone" class="external-link">DO
Bone eQTL Viewer</a>, type in <em>Sult3a1</em> and click on the search
button. Then search on <em>Sult3a2</em>.</p>
</div>
</div>
</div>
<div id="accordionSolution13" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution13" aria-expanded="false" aria-controls="collapseSolution13">
  <h4 class="accordion-header" id="headingSolution13"> Output </h4>
</button>
<div id="collapseSolution13" class="accordion-collapse collapse" aria-labelledby="headingSolution13" data-bs-parent="#accordionSolution13">
<div class="accordion-body">
<p>Neither <em>Sult3a1</em> or <em>Sult3a2</em> has an eQTL in the DO
bone marrow data set.</p>
</div>
</div>
</div>
</div>
<p>Next, we will look at the relationship between <em>Sult3a1</em> and
<em>Sult3a2</em>. Go to the Ensembl web page for <a href="https://useast.ensembl.org/Mus_musculus/Gene/Summary?db=core;g=ENSMUSG00000090298;r=10:33642420-33662700" class="external-link">Sult3a2</a>.
In the menu on the left, click on the “Gene Tree (image)” link.</p>
<p><img src="./fig/Ensembl_Mouse_Sult3a1_Sult3a2.png" alt="Ensembl gene tree of Sult3a1 &amp; Sult3a2" class="figure">{Figure showing that
Sult3a1 &amp; 2 are paralogs}</p>
<p>As you can see, <em>Sult3a2</em> is a paralog of <em>Sult3a1</em> and
hence both genes are sulfotransferases. These genes encode enzymes that
attach a sulfate group to other compounds.</p>
<div id="challenge-12-look-for-structural-variants-near-sult3a1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-12-look-for-structural-variants-near-sult3a1" class="callout-inner">
<h3 class="callout-title">Challenge 12: Look for structural variants near Sult3a1</h3>
<div class="callout-content">
<p>Go to the <a href="https://useast.ensembl.org/index.html" class="external-link">Ensembl
website</a> and select “Mouse” from the species dropdown. Then type
“10:33300000-33800000” into the search box and press “Go”. Scroll down
to the detailed view and look for structural variants in this
region.</p>
</div>
</div>
</div>
<div id="accordionSolution14" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution14" aria-expanded="false" aria-controls="collapseSolution14">
  <h4 class="accordion-header" id="headingSolution14"> Output </h4>
</button>
<div id="collapseSolution14" class="accordion-collapse collapse" aria-labelledby="headingSolution14" data-bs-parent="#accordionSolution14">
<div class="accordion-body">
<p>You should see a plot similar to the one below.</p>
<p><img src="./fig/Ensembl_SV_Chr10.png" alt="Ensembl Structural Variants" class="figure">{alt=“Ensembl viewer showing genes
and structural variants under that chromosome 10 QTL
peak”,width=100%}</p>
</div>
</div>
</div>
</div>
<p><img src="./fig/Ensembl_SV_Chr10.png" alt="Ensembl Structural Variants" class="figure">{alt=“Ensembl viewer showing genes
and structural variants under that chromosome 10 QTL
peak”,width=100%}</p>
<p>In order to visualize the size of the copy number gain, we queried
the Mouse Genomes Project <a href="https://www.mousegenomes.org/mouse-strains-sequenced/" class="external-link">alignment
files</a> for the eight founders. We piled up the reads at each base
(which is beyond the scope of this tutorial) and made the figure
below.</p>
<p><img src="./fig/French.et.al.Sup.Figure3.png" alt="DO Founder BAM file pileups" class="figure">{alt=“Figure showing CAST with a
duplication at the Sult3a1/2 locus.”,width=100%}</p>
<p>As you can see, there appears to be a duplicatation in the CAST
founders that covers four genes: <em>Clvs2</em>, <em>Gm15939</em>,
<em>Sult3a1</em> and <em>Sult3a2</em>. <em>Clvs2</em> is expressed in
neurons and <em>Gm15939</em> is a predicted gene that may not produce a
transcript.</p>
<p>Hence, we have three pieces of evidence that narrows our candidate
gene list to <em>Sult3a1</em> and <em>Sult3a2</em>:</p>
<ol style="list-style-type: decimal">
<li>Both genes have a liver eQTL in the same location as the
micronucleated reticulocytes QTL.</li>
<li>Among genes in the micronucleated reticulocytes QTL interval, only
<em>Sult3a1</em> and <em>Sult3a2</em> have differential expression of
the CAST allele in the liver.</li>
<li>There is a copy number gain of these two genes in CAST.</li>
</ol>
<p>Sulfation is a prominent detoxification mechanism for benzene as
well. The diagram below shows the metabolism pathway for benzene <a href="http://europepmc.org/articles/PMC4414400" class="external-link">(Monks, T. J., et
al. (2010). Chem Biol Interact 184(1-2): 201-206.)</a> Hydroquinone,
phenol and catechol are all sulfated and excreted from the body.</p>
<p><img src="./fig/Monks_ChemBiolInter_2010_Fig1.jpg" alt="Benezene metabolism pathways" class="figure">{alt=“Figure showing benezene
metabolism pathways”,width=100%}</p>
<p>This analysis has led us to the following hypothesis. Inhaled benzene
is absorbed by the lungs into the bloodstream and transported to the
liver. There, it is metabolized, and some metabolites are transported to
the bone marrow. One class of genes that is involved in toxicant
metabolism are sulfotransferases. <a href="http://www.informatics.jax.org/marker/MGI:1931469" class="external-link"><i>Sult3a1</i></a>
is a phase II enzyme that conjugates compounds (such as phenol, which is
a metabolite of benzene) with a sulfate group before transport into the
bile. It is possible that a high level of <em>Sult3a1</em> expression
could remove benzene by-products and be protective. Our hypothesis is
that the copy number gain in the CAST allele increases liver gene
expression of <em>Sult3a1</em> and <em>Sult3a2</em>. High liver
expression of these genes allows mice containing the CAST allele to
rapidly conjugate harmful benzene metabolites and excrete them from the
body before they can reach the bone marrow and cause DNA damage. Further
experimental validation is required, but this is a plausible
hypothesis.</p>
<p><img src="./fig/benzene_hypothesis.png" alt="Benzene metabolism hypothesis" class="figure">{alt=“Figure showing mice carrying
the CAST allele being protected from benezene toxicity”,width=100%}</p>
<p>We hope that this tutorial has shown you how the DO can be used to
map QTL and use the founder effects and bioinformatics resources to
narrow down the candidate gene list. Here, we made used of external gene
expression databases and the founder sequence data to build a case for a
pair of genes.</p>
<div id="challenge-13-map-another-trait." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-13-map-another-trait." class="callout-inner">
<h3 class="callout-title">Challenge 13: Map another trait.</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Make a histogram of the column <code>pre.prop.mn.ret</code> in
<code>pheno</code>. Does it look like it should be log transformed? If
so, add a column to <code>pheno</code> that contains the log of
<code>pre.prop.mn.ret</code>.<br>
</li>
<li>Perform a genome scan <em>using all samples</em> on the column
called <code>pre.prop.mn.ret</code>. Since you’re using all samples, the
scan will take longer. You will also need to add the benezene
concentration to the covariates. (Hint: set <code>index</code> to the
column index in <code>pheno</code>.)<br>
</li>
<li>Which chromosome has the highest peak? Use <code>find_peaks</code>
to get the location and support interval for the highest peak.<br>
</li>
<li>Calculate and plot the BLUPs for the chromosome with the highest
peak. (This may take a few minutes.) Which founder contributes an allele
that makes MN-RET values higher?</li>
<li>Perform association mapping in the support interval for the QTL
peak, plot the results and plot the genes beneath the association
mapping plot.</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution15" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution15" aria-expanded="false" aria-controls="collapseSolution15">
  <h4 class="accordion-header" id="headingSolution15"> Output </h4>
</button>
<div id="collapseSolution15" class="accordion-collapse collapse" aria-labelledby="headingSolution15" data-bs-parent="#accordionSolution15">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>Plot the distribution of</li>
</ol>
<div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">pheno</span><span class="op">$</span><span class="va">pre.prop.mn.ret</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-33-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Log transform the pre-dose proportion of MN-RETs.</p>
<div class="codewrapper sourceCode" id="cb57">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span><span class="op">$</span><span class="va">log_pre_prop_mnret</span> <span class="op">&lt;-</span> <span class="fu">log</span><span class="op">(</span><span class="va">pheno</span><span class="op">$</span><span class="va">pre.prop.mn.ret</span><span class="op">)</span></span></code></pre>
</div>
<ol start="2" style="list-style-type: decimal">
<li>Set the mapping index equal to the <code>pre.prop.mn.ret</code>
column.</li>
</ol>
<div class="codewrapper sourceCode" id="cb58">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span> <span class="op">==</span> <span class="st">"log_pre_prop_mnret"</span><span class="op">)</span></span></code></pre>
</div>
<p>Create new covariates which include study and benzene
concentration.</p>
<div class="codewrapper sourceCode" id="cb59">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">addcovar2</span> <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">study</span> <span class="op">+</span> <span class="va">conc</span>, data <span class="op">=</span> <span class="va">pheno</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb60">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_pre</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                 pheno     <span class="op">=</span> <span class="va">pheno</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                 kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar2</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_pre</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">map</span>, </span>
<span>           main <span class="op">=</span> <span class="st">"Log-Transformed Pre-dose micronucleated reticulocytes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-37-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><ol start="3" style="list-style-type: decimal">
<li>Find peaks above a significance threshold.</li>
</ol>
<div class="codewrapper sourceCode" id="cb61">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span><span class="va">lod_pre</span>, <span class="va">map</span>, threshold <span class="op">=</span> <span class="fl">8</span>, prob <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr       pos       lod     ci_lo     ci_hi
1        1 log_pre_prop_mnret   4 132.77875 13.145968 132.77875 134.27938
2        1 log_pre_prop_mnret  12  75.45433  8.404242  74.14949  77.30552</code></pre>
</div>
<ol start="4" style="list-style-type: decimal">
<li>Calculate and plot BLUPs for the highest peak.</li>
</ol>
<div class="codewrapper sourceCode" id="cb63">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>   <span class="op">&lt;-</span> <span class="fl">4</span></span>
<span><span class="va">coef4</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                   pheno     <span class="op">=</span> <span class="va">pheno</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                   kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                   addcovar  <span class="op">=</span> <span class="va">addcovar2</span><span class="op">)</span></span>
<span><span class="fu">plot_coefCC</span><span class="op">(</span>x            <span class="op">=</span> <span class="va">coef4</span>, </span>
<span>            map          <span class="op">=</span> <span class="va">map</span>, </span>
<span>            scan1_output <span class="op">=</span> <span class="va">lod_pre</span>, </span>
<span>            legend       <span class="op">=</span> <span class="st">"topleft"</span>,</span>
<span>            main         <span class="op">=</span> <span class="st">"Log-Transformed Pre-dose micronucleated reticulocytes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-39-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><ol start="5" style="list-style-type: decimal">
<li>Perform association mapping in the QTL interval for the highest
peak.</li>
</ol>
<div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">start</span>  <span class="op">&lt;-</span> <span class="fl">132.8</span></span>
<span><span class="va">end</span>    <span class="op">&lt;-</span> <span class="fl">134.3</span></span>
<span><span class="va">assoc4</span> <span class="op">&lt;-</span> <span class="fu">scan1snps</span><span class="op">(</span>genoprobs  <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                    map        <span class="op">=</span> <span class="va">map</span>, </span>
<span>                    pheno      <span class="op">=</span> <span class="va">pheno</span><span class="op">[</span>,<span class="va">index</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                    kinship    <span class="op">=</span> <span class="va">K</span>, </span>
<span>                    addcovar   <span class="op">=</span> <span class="va">addcovar2</span>, </span>
<span>                    query_func <span class="op">=</span> <span class="va">query_snps</span>, </span>
<span>                    chr        <span class="op">=</span> <span class="va">chr</span>,</span>
<span>                    start      <span class="op">=</span> <span class="va">start</span>, </span>
<span>                    end        <span class="op">=</span> <span class="va">end</span>, </span>
<span>                    keep_all_snps <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu">query_genes</span><span class="op">(</span><span class="va">chr</span>, <span class="va">start</span>, <span class="va">end</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb65">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#png('./fig/pre_mnret_assoc.png', width = 2000, height = 2000, res = 256)</span></span>
<span><span class="fu">plot_snpasso</span><span class="op">(</span>scan1output <span class="op">=</span> <span class="va">assoc4</span><span class="op">$</span><span class="va">lod</span>, </span>
<span>             snpinfo     <span class="op">=</span> <span class="va">assoc4</span><span class="op">$</span><span class="va">snpinfo</span>, </span>
<span>             main        <span class="op">=</span> <span class="st">"Log-Transformed Pre-dose micronucleated reticulocytes"</span>, </span>
<span>             genes       <span class="op">=</span> <span class="va">genes</span>,</span>
<span>             colors      <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>             panel_prop <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>             drop_hilit  <span class="op">=</span> <span class="fl">1.0</span>,</span>
<span>             col_hilit   <span class="op">=</span> <span class="st">'red'</span>,</span>
<span>             sdp_panel   <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#dev.off()</span></span></code></pre>
</div>
<p><a href="./fig/pre_mnret_assoc.png">Pre-dose MN-RET Association
Mapping</a>{alt=“Figure showing pre-dose MN-RET association mapping with
WSB carrying the alternate allele”,width=100%}</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb66">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessionInfo</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
 [1] qtl2convert_0.30 qtl2_0.36        ggbeeswarm_0.7.2 lubridate_1.9.3
 [5] forcats_1.0.0    stringr_1.5.1    dplyr_1.1.4      purrr_1.0.2
 [9] readr_2.1.5      tidyr_1.3.1      tibble_3.2.1     ggplot2_3.5.1
[13] tidyverse_2.0.0

loaded via a namespace (and not attached):
 [1] utf8_1.2.4          generics_0.1.3      renv_1.0.11
 [4] RSQLite_2.3.7       stringi_1.8.4       hms_1.1.3
 [7] magrittr_2.0.3      RColorBrewer_1.1-3  evaluate_0.24.0
[10] grid_4.4.1          timechange_0.3.0    fastmap_1.2.0
[13] blob_1.2.4          DBI_1.2.3           BiocManager_1.30.25
[16] fansi_1.0.6         scales_1.3.0        qtl_1.70
[19] cli_3.6.3           rlang_1.1.4         bit64_4.5.2
[22] munsell_0.5.1       withr_3.0.1         cachem_1.1.0
[25] yaml_2.3.10         parallel_4.4.1      tools_4.4.1
[28] tzdb_0.4.0          memoise_2.0.1       colorspace_2.1-1
[31] vctrs_0.6.5         R6_2.5.1            lifecycle_1.0.4
[34] bit_4.5.0           vipor_0.4.7         pkgconfig_2.0.3
[37] beeswarm_0.4.0      pillar_1.9.0        gtable_0.3.5
[40] glue_1.7.0          data.table_1.16.2   Rcpp_1.0.13
[43] highr_0.11          xfun_0.46           tidyselect_1.2.1
[46] knitr_1.48          farver_2.1.2        labeling_0.4.3
[49] compiler_4.4.1     </code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>There are generally five steps to QTL mapping in DO mice:
<ul>
<li>map the trait,</li>
<li>perform permutations,</li>
<li>find significant peaks,</li>
<li>calculate founder allele effects at the QTL peak,</li>
<li>perform association mapping to narrow the gene candidates.</li>
</ul>
</li>
<li>You may need to bring in outside resources to help narrow your
candidate gene list.</li>
<li>You will need the 10 GB SNP database to perform association
mapping.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.9" class="external-link">sandpaper (0.16.9)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.6" class="external-link">pegboard (0.7.6)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.4" class="external-link">varnish (1.0.4)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-rmd/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries, genetics, statistics, genome, genetic mapping",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/aio.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/aio.html",
  "dateCreated": "2024-09-06",
  "dateModified": "2024-10-16",
  "datePublished": "2024-10-16"
}

  </script><script>
		feather.replace();
	</script>
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

