<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Quantitative Trait Mapping: QTL Mapping in Diversity Outbred Mice</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png"><link rel="manifest" href="site.webmanifest"><link rel="mask-icon" href="safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/do_qtl_mapping.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Quantitative Trait Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Quantitative Trait Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Quantitative Trait Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 76%" class="percentage">
    76%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 76%" aria-valuenow="76" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/do_qtl_mapping.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="introduction.html">1. Using RMarkdown</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="intro_data_set.html">2. Introduction to the Data Set</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="input-file-format.html">3. Input File Format</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="calc-genoprob.html">4. Calculating Genotype Probabilities</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="perform-genome-scan.html">5. Performing a genome scan</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="calc-kinship.html">6. Calculating A Kinship Matrix</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="perform-genome-scan-lmm.html">7. Performing a genome scan with a linear mixed model</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="perform-genome-scan-bin.html">8. Performing a genome scan with binary traits</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="perform-perm-test.html">9. Performing a permutation test</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="find-lod-peaks.html">10. Finding LOD peaks</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="est-qtl-effects.html">11. Estimated QTL effects</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush13">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading13">
        <a href="integrating-gene-expression.html">12. Integrating Gene Expression Data</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        13. QTL Mapping in Diversity Outbred Mice
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#introduction">Introduction</a></li>
<li><a href="#do-reference-data">DO Reference Data</a></li>
<li><a href="#open-the-qtl_mapping-project.">Open the qtl_mapping Project.</a></li>
<li><a href="#create-a-new-markdown-file">Create a New Markdown File</a></li>
<li><a href="#load-libraries">Load Libraries</a></li>
<li><a href="#load-and-explore-the-data">Load and Explore the Data</a></li>
<li><a href="#performing-a-genome-scan">Performing a Genome Scan</a></li>
<li><a href="#assessing-significance-of-lod-scores">Assessing Significance of LOD Scores</a></li>
<li><a href="#finding-lod-peaks">Finding LOD Peaks</a></li>
<li><a href="#estimated-founder-allele-effects">Estimated Founder Allele Effects</a></li>
<li><a href="#snp-association-mapping">SNP Association Mapping</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="integrating-gene-expression.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="index.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="integrating-gene-expression.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Integrating Gene
        </a>
        <a class="chapter-link float-end" href="index.html" rel="next">
          Home
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>QTL Mapping in Diversity Outbred Mice</h1>
        <p>Last updated on 2024-10-09 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/do_qtl_mapping.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How do I map traits in Diversity Outbred mice?</li>
<li>How do I interpret the founder allele effects at a QTL peak?</li>
<li>How do I perform association mapping in Diversity Outbred mice?</li>
<li>How do I narrow down the set of candidate genes under a QTL?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Understand the data objects required for QTL mapping.</li>
<li>Use the scan1 family of functions to perform different types of QTL
scans.</li>
<li>Interpret the founder allele effects.</li>
<li>Find significant QTL peaks and identify candidate peaks under a QTL
peak.</li>
<li>Use auxiliary data to identify candidate genes under a QTL
peak.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a></h2>
<hr class="half-width"><p>This tutorial will take you through the process of mapping a QTL and
searching for candidate genes for DNA damage from benzene exposure. The
adverse health effects of benzene, including leukemia and aplastic
anemia, were first studied in occupational settings in which workers
were exposed to high benzene concentrations. Environmental sources of
benzene exposure typically come from the petrochemical industry,
however, a person’s total exposure can be increased from cigarettes,
consumer products, gas stations, and gasoline powered engines or tools
stored at home.</p>
<p>Exposure thresholds for toxicants are often determined using animal
models that have limited genetic diversity, including standard inbred
and outbred rats and mice. These animal models fail to capture the
influence of genetic diversity on toxicity response, an important
component of human responses to chemicals such as benzene. The Diversity
Outbred (DO) mice reflect a level of genetic diversity similar to that
of humans.</p>
<p>The data comes from a toxicology study in which Diversity Outbred
(DO) mice were exposed to benzene via inhalation for 6 hours a day, 5
days a week for 4 weeks <a href="http://ehp.niehs.nih.gov/1408202/" class="external-link">(French, J. E., et al. (2015)
<i>Environ Health Perspect</i> 123(3): 237-245)</a>. The study was
conducted in two equally sized cohort of 300 male mice each, for a total
of 600 mice. They were then sacrificed and reticulocytes (red blood cell
precursors) were isolated from bone marrow. The number of
micro-nucleated reticulocytes, a measure of DNA damage, was then
measured in each mouse. The goal is to map gene(s) that influence the
level of DNA damage in the bone marrow.</p>
<figure><img src="./fig/benzene_study_design.png" alt="Benzene study dosing showing 6 hours per day, 5 days per week of inhalation." class="figure mx-auto d-block"><div class="figcaption">Benzene Study Dosing</div>
</figure></section><section><h2 class="section-heading" id="do-reference-data">DO Reference Data<a class="anchor" aria-label="anchor" href="#do-reference-data"></a></h2>
<hr class="half-width"><p>As you work with DO data, you may need different reference files.
These are complied on a JAX website at: <a href="https://www.jax.org/research-and-faculty/genetic-diversity-initiative/tools-data/diversity-outbred-reference-data" class="external-link uri">https://www.jax.org/research-and-faculty/genetic-diversity-initiative/tools-data/diversity-outbred-reference-data</a>.
There are links to the markers and founder genotypes as well as the
standard colors that we use for the eight founder strains.</p>
</section><section><h2 class="section-heading" id="open-the-qtl_mapping-project-">Open the qtl_mapping Project.<a class="anchor" aria-label="anchor" href="#open-the-qtl_mapping-project-"></a></h2>
<hr class="half-width"><p>If you have not opened the qtl_mapping project, do this now so that
your script and console will run in the correct directory.</p>
<ol style="list-style-type: decimal"><li>From the File menu, select “Open Project…”.</li>
<li>Navigate to the project that you created in your Desktop/qtl_mapping
folder and open it.</li>
</ol><p>This should reset your RStudio window and close any files.</p>
</section><section><h2 class="section-heading" id="create-a-new-markdown-file">Create a New Markdown File<a class="anchor" aria-label="anchor" href="#create-a-new-markdown-file"></a></h2>
<hr class="half-width"><p>Create a new R Markdown file and save it in the “Desktop/scripts”
folder as “do_qtl_mapping.Rmd”.</p>
</section><section><h2 class="section-heading" id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a></h2>
<hr class="half-width"><p>The primary library that we will use is <a href="https://github.com/kbroman/qtl2" class="external-link">qtl2</a>. You should have
<code>qtl2</code> installed from the Setup that you performed before the
workshop.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">ggbeeswarm</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2convert</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="load-and-explore-the-data">Load and Explore the Data<a class="anchor" aria-label="anchor" href="#load-and-explore-the-data"></a></h2>
<hr class="half-width"><p>The data for this tutorial has been saved as an R binary file that
contains several data objects. Load it in now by running the following
command in your new script.</p>


<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="st">"./data/qtl2_demo_grcm39.Rdata"</span><span class="op">)</span></span></code></pre>
</div>
<p>We loaded in four data objects from the *.Rdata file. Look at the
Environment tab to see what was loaded. You should see an object called
<code>pheno</code> which contains the phenotypes. There are two objects
called <code>gmap</code> and <code>pmap</code>, which contain the marker
positions in genetic and physical coordinates. Finally, there is an
object called <code>probs</code> which contains the founder allele
probabilities for each mouse at each marker.</p>
<div class="section level3">
<h3 id="phenotypes">Phenotypes<a class="anchor" aria-label="anchor" href="#phenotypes"></a></h3>
<p><code>pheno</code> is a data frame containing the phenotype data.
Click on the blue circle to the left of <code>pheno</code> in the
Environment pane to view its contents.</p>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>The sample IDs <strong>must</strong> be in the rownames of
<code>pheno</code> for qtl2 to work.</p>
</div>
</div>
</div>
<p><code>pheno</code> contains the sample ID, the study cohort, the
concentration of benzene and several blood cell measurements. Note that
the sample IDs are also stored in the rownames of
<code>pheno</code>.</p>
<p>In order to save time for this tutorial, we will only map with 149
samples from the 100 ppm dosing group.</p>
<div id="challenge-1-how-many-mice-are-there-in-the-phenotype-data" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-how-many-mice-are-there-in-the-phenotype-data" class="callout-inner">
<h3 class="callout-title">Challenge 1: How many mice are there in the phenotype data?</h3>
<div class="callout-content">
<p>Look in the Environment tab or use the Console to figure out how many
mice there are in the phenotype data.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Output </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>You can look at the number of observations in the Environment tab or
you can get the number of rows in <code>pheno</code> in the Console.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 598</code></pre>
</div>
<p>There are 598 mice in the phenotype data.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2-how-many-mice-are-there-from-each-sex" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-how-many-mice-are-there-from-each-sex" class="callout-inner">
<h3 class="callout-title">Challenge 2: How many mice are there from each sex?</h3>
<div class="callout-content">
<p>Use an R command to determine how many mice there are from each
sex.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Output </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>You can look at the number of observations in the Environment tab or
you can get the number of rows in <code>pheno</code> in the Console.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">count</span><span class="op">(</span><span class="va">pheno</span>, <span class="va">sex</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>   sex   n
1 male 598</code></pre>
</div>
<p>All of the mice are male.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3-how-is-the-prop.bm.mn.ret-phenotype-distributed" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-how-is-the-prop.bm.mn.ret-phenotype-distributed" class="callout-inner">
<h3 class="callout-title">Challenge 3: How is the “prop.bm.mn.ret” phenotype distributed?</h3>
<div class="callout-content">
<p>Make histogram of the “prop.bm.mn.ret” column and assess whether it
is normally distributed.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Output </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>You can look at the number of observations in the Environment tab or
you can get the number of rows in <code>pheno</code> in the Console.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">pheno</span>, mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span><span class="va">prop.bm.mn.ret</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Histogram of Bone Marrow MN-RETs"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-6-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The bone marrow MN-RET values do not look normally distributed.</p>
</div>
</div>
</div>
</div>
<div id="challenge-4-how-is-the-logprop.bm.mn.ret-phenotype-distributed" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-how-is-the-logprop.bm.mn.ret-phenotype-distributed" class="callout-inner">
<h3 class="callout-title">Challenge 4: How is the log(“prop.bm.mn.ret”) phenotype distributed?</h3>
<div class="callout-content">
<p>Take the log of the “prop.bm.mn.ret” column and make a
histogram.Assess whether it is normally distributed.</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Output </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<p>You can look at the number of observations in the Environment tab or
you can get the number of rows in <code>pheno</code> in the Console.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>log_mnret <span class="op">=</span> <span class="fu">log</span><span class="op">(</span><span class="va">prop.bm.mn.ret</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span><span class="va">log_mnret</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_histogram</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Histogram of log(Bone Marrow MN-RETs)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-7-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The log of the bone marrow MN-RET values look more normally
distributed.</p>
</div>
</div>
</div>
</div>
<p>Let’s add a column containing the log-transformed phenotype to our
<code>pheno</code> object.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="va">pheno</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">mutate</span><span class="op">(</span>log_mnret <span class="op">=</span> <span class="fu">log</span><span class="op">(</span><span class="va">prop.bm.mn.ret</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Some researchers are concerned about the reproducibility of DO
studies. The argument is that each DO mouse is unique and therefore can
never be reproduced. But this misses the point of using the DO. While
mice are the sampling unit, in QTL mapping we are sampling the founder
alleles at each locus. An average of 1/8th of the alleles should come
from each founder at any given locus. Also, DO mice are a
<em>population</em> of mice, not a single strain. While it is true that
results in an individual DO mouse may not be reproducible, results at
the population level should be reproducible. This is similar to the
human population in that results from one individual may not represent
all humans, but results at the population level should be
reproducible.</p>
<p>The benzene inhalation study was conducted in two separate cohorts
(termed <em>study</em> in the <code>pheno</code> object). Let’s plot the
proportion of micronucleated reticulocytes in bone marrow versus the
benzene concentration for each study cohort.</p>

<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>conc <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">conc</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span><span class="va">conc</span>, <span class="va">prop.bm.mn.ret</span> <span class="op">*</span> <span class="fl">1000</span>, color <span class="op">=</span> <span class="va">conc</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_violin</span><span class="op">(</span>draw_quantiles <span class="op">=</span> <span class="fl">0.5</span>, linewidth <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_beeswarm</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Reds"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">study</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Bone Marrow MN-RET by Study Cohort"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>, </span>
<span>          legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-9-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>As you can see, while individual mice have varying micronucleated
reticulocytes, there is a dose-dependent increase in micronucleated
reticulocytes in both cohorts. This is an example of how results in the
DO reproduce at the population level.</p>
</div>
<div class="section level3">
<h3 id="markers">Markers<a class="anchor" aria-label="anchor" href="#markers"></a></h3>
<p>The markers are the SNPs on the Mouse Universal Genotyping Array (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4751547/" class="external-link">MUGA</a>)
that were used to reconstruct the haplotypes of the DO mice. This
version of the array had about 8,000 SNPs on it. The latest version
contains over 140,000 SNPs. We are using this smaller data set so that
we can finish the analysis in a timely manner during class.</p>
<p>The markers have been mapped to the latest mouse genome build
(GRCm39) and are provided by Karl Broman on Github at: <a href="https://github.com/kbroman/MUGAarrays/tree/main/UWisc" class="external-link uri">https://github.com/kbroman/MUGAarrays/tree/main/UWisc</a>.
There are four versions of the MUGA platforms. In this study, we used
the MUGA and those marker files are named as “muga_uwisc_vN.csv”, where
“N” is a version number. We will use version 4, which you downloaded
into your “data” directory during the Setup before the workshop.</p>
<p>Open the marker file now:</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers</span> <span class="op">=</span> <span class="fu">read.csv</span><span class="op">(</span><span class="st">"./data/muga_uwisc_v4.csv"</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-5-which-columns-contain-the-marker-positions" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-5-which-columns-contain-the-marker-positions" class="callout-inner">
<h3 class="callout-title">Challenge 5: Which columns contain the marker positions?</h3>
<div class="callout-content">
<p>Look at the structure of the genetic or physical map in the
Environment tab and see which columns contain marker positions in some
coordinate system.</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" data-bs-parent="#accordionSolution5" aria-labelledby="headingSolution5">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">str</span><span class="op">(</span><span class="va">markers</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>'data.frame':	7854 obs. of  11 variables:
 $ marker        : chr  "JAX00240603" "UNC010001397" "UNC010515443" "UNC010001943" ...
 $ chr           : chr  "1" "1" "1" "1" ...
 $ bp_mm10       : int  3252796 3336839 3668628 3977130 4430623 4531029 5045931 5840130 6020779 6378154 ...
 $ bp_grcm39     : int  3323019 3407062 3738851 4047353 4500846 4601252 5116154 5910353 6091003 6448378 ...
 $ cM_cox        : num  0.15 0.172 0.194 0.197 0.202 ...
 $ strand        : chr  "minus" "minus" "minus" "plus" ...
 $ snp           : chr  "TC" "AC" "TC" "AC" ...
 $ unique        : logi  TRUE TRUE TRUE TRUE TRUE TRUE ...
 $ unmapped      : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...
 $ probe         : chr  "AAAATATGATTCTTTTAATTAATGCATCAGTGATGAGAAGTGAGAGTGGC" "CATAGTGTCTGGTGAGAAGTCTGGAGTTATTCTAATAGGCCTGCCTTTAT" "CAGGAAATGATGCTGAGAAAGTGAGAAGTAGGAAAACGTGGAGAAAAATA" "TCTATTCCTATCACCTTGTACAAAGCTCAAGTCTTGTAAACCCCCCCCCC" ...
 $ strand_flipped: logi  TRUE FALSE FALSE FALSE FALSE FALSE ...</code></pre>
</div>
<p>The columns labelled “bp_mm10”, “bp_grcm39”, and “cM_cox” contain
marker positions.</p>
</div>
</div>
</div>
</div>
<div id="challenge-6-which-chromosomes-are-in-the-markers" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-which-chromosomes-are-in-the-markers" class="callout-inner">
<h3 class="callout-title">Challenge 6: Which chromosomes are in the markers?</h3>
<div class="callout-content">
<p>Use an R command to determine which chromosomes are in
<code>markers</code>.</p>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" data-bs-parent="#accordionSolution6" aria-labelledby="headingSolution6">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">count</span><span class="op">(</span><span class="va">markers</span>, <span class="va">chr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>    chr   n
1     1 493
2    10 375
3    11 335
4    12 318
5    13 327
6    14 331
7    15 265
8    16 266
9    17 227
10   18 216
11   19 162
12    2 474
13    3 438
14    4 412
15    5 382
16    6 414
17    7 379
18    8 341
19    9 342
20    M   3
21    X 414
22    Y   5
23 &lt;NA&gt; 935</code></pre>
</div>
<p>There are 23 chromosomes, including some markers with “NA”. These are
markers that could not be uniquely mapped to one chromosome.</p>
</div>
</div>
</div>
</div>
<div id="challenge-6-what-do-you-think-the-difference-is-between-teh-bp_mm10-and" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-what-do-you-think-the-difference-is-between-teh-bp_mm10-and" class="callout-inner">
<h3 class="callout-title">Challenge 6: What do you think the difference is between teh “bp_mm10” and</h3>
<div class="callout-content">
<p>“bp_grcm39” columns?</p>
<p>Turn to your neighbor and discuss what the differences between the
two column are. Then share your ideas with the class.</p>
</div>
</div>
</div>
<div id="accordionSolution7" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution7" aria-expanded="false" aria-controls="collapseSolution7">
  <h4 class="accordion-header" id="headingSolution7"> Show me the solution </h4>
</button>
<div id="collapseSolution7" class="accordion-collapse collapse" data-bs-parent="#accordionSolution7" aria-labelledby="headingSolution7">
<div class="accordion-body">
<p>There are different builds of the mouse genome. The Genome Resource
Consortium (GRC) named mouse builds from build 1 to 39. The University
of California at Santa Cruz developed their own naming system which
conflicted with the GRC names. “GRCm38” was equivalent to “mm10”. We are
using the GRC naming conventions for genome builds.</p>
</div>
</div>
</div>
</div>
<p><a href="https://github.com/kbroman/qtl2" class="external-link">qtl2</a> needs the markers
to be in a specific format. <code>qtl2</code> requires that the markers
be in a list. The <a href="https://github.com/kbroman/qtl2convert" class="external-link">qtl2convert</a> package
provides functions to make this easier.</p>
<p>First, we will filter the markers to retain ones on the autosomes and
Chr X. We don’t use the mitochondrial or Y chromosomes in mapping, but
they may be important covariates. The MUGA did not contain enough
markers on Chr M and Y to identify the contributing founder strain, but
later versions of the MUGA, such as the GigaMUGA, do contain enough
markers to identify founder contributions.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers</span> <span class="op">&lt;-</span> <span class="va">markers</span> <span class="op">|&gt;</span></span>
<span>             <span class="fu">filter</span><span class="op">(</span><span class="va">chr</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">'X'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will create a column with GRCm39 Mb positions.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers</span> <span class="op">&lt;-</span> <span class="va">markers</span> <span class="op">|&gt;</span></span>
<span>             <span class="fu">mutate</span><span class="op">(</span>pos <span class="op">=</span> <span class="va">bp_grcm39</span> <span class="op">/</span> <span class="fl">1e6</span><span class="op">)</span></span></code></pre>
</div>
<p>Finally, we will use the <a href="">qtl2convert::map_df_to_list()</a>
function to create the list of markers that <code>qtl2</code> requires.
We will call this object the marker “map” to distinguish it from the
marker data.frame.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">map</span> <span class="op">=</span> <span class="fu">qtl2convert</span><span class="fu">::</span><span class="fu">map_df_to_list</span><span class="op">(</span>map <span class="op">=</span> <span class="va">markers</span>, pos_column <span class="op">=</span> <span class="st">"pos"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="founder-allele-probabilities-genoprobs">Founder Allele Probabilities (Genoprobs)<a class="anchor" aria-label="anchor" href="#founder-allele-probabilities-genoprobs"></a></h3>
<p>The <code>probs</code> object is a list with 20 elements. Each
element of <code>probs</code> is a three-dimensional array containing
the founder allele dosages for each sample at each marker on one
chromosome. These probabilities have been pre-calculated for you, so you
can skip the step for calculating allele probabilities.</p>
<!-- DMG: We need a figure showing the structure of the genoprobs. It should
show a list of boxes. We can show the first three chromosomes and then ... -->
<p>Let’s look at the dimensions of <code>probs</code> for chromosome
1:</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 590   8 479</code></pre>
</div>
<p><code>probs[[1]]</code> is a three-dimensional array containing the
proportion of each founder haplotype at each marker for each DO sample.
The 590 samples are in the first dimension, the 8 founders in the second
and the 479 markers along chromosome 1 are in the third dimension.</p>
<p>Let’s return to the <code>probs</code> object. Look at the contents
of one sample on chromosome 1.</p>

<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_genoprob</span><span class="op">(</span><span class="va">probs</span>, <span class="va">map</span>, ind <span class="op">=</span> <span class="fl">1</span>, chr <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<p>Uh oh! We have an error which says “Different numbers of positions in
probs and map”. This means that the number of markers in the genoprobs
and the marker map is different. In fact, this is quite important. The
markers in <code>probs</code> and <code>map</code> must match
<strong>exactly</strong> on every chromosome.</p>
<div id="challenge-7-align-the-markers-in-probs-and-map." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-7-align-the-markers-in-probs-and-map." class="callout-inner">
<h3 class="callout-title">Challenge 7: Align the markers in
<code>probs</code> and <code>map</code>.</h3>
<div class="callout-content">
<p>Work with the person next to you to write a short script that will
get the intersection of the marker names in the <code>probs</code> and
<code>map</code> for each chromosome, subset the markers in each object
to only contain the common markers, and make sure that the markers in
<code>probs</code> are in the same order as in <code>map</code>.</p>
<p><strong>Hint</strong>: The markers names for each chromosome in
<code>probs</code> are in dimention 3. i.e. dimnames(probs[[1]])[[3]]
contains the marker names for chromosome 1.</p>
</div>
</div>
</div>
<div id="accordionSolution8" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution8" aria-expanded="false" aria-controls="collapseSolution8">
  <h4 class="accordion-header" id="headingSolution8"> Show me the solution </h4>
</button>
<div id="collapseSolution8" class="accordion-collapse collapse" data-bs-parent="#accordionSolution8" aria-labelledby="headingSolution8">
<div class="accordion-body">
<p>We will write a loop that gets the common markers between probs and
map for each chromosome and then subsets them.</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Verify that probs and map are the same length.</span></span>
<span><span class="fu">stopifnot</span><span class="op">(</span><span class="fu">length</span><span class="op">(</span><span class="va">probs</span><span class="op">)</span> <span class="op">==</span> <span class="fu">length</span><span class="op">(</span><span class="va">map</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Verify that probs and map have their chromsomes in the same order.</span></span>
<span><span class="fu">stopifnot</span><span class="op">(</span><span class="fu">names</span><span class="op">(</span><span class="va">probs</span><span class="op">)</span> <span class="op">==</span> <span class="fu">names</span><span class="op">(</span><span class="va">map</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Loop through each chromosome...</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu">seq_along</span><span class="op">(</span><span class="va">probs</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="co"># Get the intersection of the markers in probs and map.</span></span>
<span>  <span class="va">common_markers</span> <span class="op">&lt;-</span> <span class="fu">intersect</span><span class="op">(</span><span class="fu">dimnames</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span>, <span class="fu">names</span><span class="op">(</span><span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Subset the map to contain the common markers.</span></span>
<span>  <span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">common_markers</span><span class="op">]</span></span>
<span>  </span>
<span>  <span class="co"># Subset the probs to contain the common markers.</span></span>
<span>  <span class="va">probs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">probs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,,<span class="fu">names</span><span class="op">(</span><span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">]</span></span>
<span>  </span>
<span>  <span class="co"># Verify that the markers are identical in probs and map.</span></span>
<span>  <span class="fu">stopifnot</span><span class="op">(</span><span class="fu">dimnames</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="fu">names</span><span class="op">(</span><span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span><span class="op">}</span> <span class="co"># for(i)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>Now that we have aligned the markers in <code>probs</code> and
<code>map</code>, let’s plot the allele probabilities for one sample on
chromosome 1.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_genoprob</span><span class="op">(</span><span class="va">probs</span>, <span class="va">map</span>, ind <span class="op">=</span> <span class="fl">1</span>, chr <span class="op">=</span> <span class="fl">1</span>, main <span class="op">=</span> <span class="st">"Founder Allele Probabilities"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot above, the founder contributions, which range between 0
and 1, are colored from white (= 0) to black (= 1.0). A value of ~0.5 is
grey. The markers are on the X-axis and the eight founders (denoted by
the letters A through H) on the Y-axis. Starting at the left, we see
that this sample has genotype BB because the row for B is black,
indicating values of 1.0. Moving along the genome to the right, the
genotype becomes CE where rows C and E are gray, followed by CD, FH, AG,
GH, etc. The values at each marker sum to 1.0.</p>
</div>
<div class="section level3">
<h3 id="calculating-a-kinship-matrix">Calculating A Kinship Matrix<a class="anchor" aria-label="anchor" href="#calculating-a-kinship-matrix"></a></h3>
<p>Next, we need to create a matrix that accounts for the kinship
relationships between the mice. We do this by looking at the correlation
between the founder haplotypes for each sample at each SNP. For each
chromosome, we create a kinship matrix using all markers <em>except</em>
the ones on the current chromosome using the “loco”
(leave-one-chromosome-out) method. Simulations suggest that mapping
using this approach increases the power to detect QTL.</p>
<div id="callout2" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>The sample IDs must be in the rownames of <code>probs</code>. The
sample IDs will be copied to the row and column names in the kinship
matrices.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">K</span> <span class="op">&lt;-</span> <span class="fu">calc_kinship</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, type <span class="op">=</span> <span class="st">"loco"</span><span class="op">)</span></span></code></pre>
</div>
<p>Kinship values between pairs of samples range between 0 (no
relationship) and 1.0 (completely identical). Let’s look at the kinship
matrix for the first 50 samples.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_samples</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="fu">heatmap</span><span class="op">(</span><span class="va">K</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-kinship_probs-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The figure above shows kinship between all pairs of samples. Light
yellow indicates low kinship and dark red indicates higher kinship.
Orange values indicate varying levels of kinship between 0 and 1. The
dark red diagonal of the matrix indicates that each sample is identical
to itself. The orange blocks along the diagonal may indicate close
relatives (i.e. siblings or cousins).</p>
</div>
<div class="section level3">
<h3 id="covariates">Covariates<a class="anchor" aria-label="anchor" href="#covariates"></a></h3>
<p>Next, we need to create additive covariates that will be used in the
mapping model. We will use study cohort as a covariate in the mapping
model. This is the same as outbreeding generation since each cohort was
purchased from successive generations. If we were mapping with
<em>all</em> mice, we would also add benzene concentration to the model.
This study contained only male mice, but in most cases, you would
include sex as an additive covariate as well.</p>
<p>To recap, you would normally add sex and outbreeding generation to
the model. In this case, we have one sex and are using
<code>study</code> instead of generation.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">addcovar</span> <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">study</span>, data <span class="op">=</span> <span class="va">pheno</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<p>The code above copies the <code>rownames(pheno)</code> to
<code>rownames(addcovar)</code> as a side-effect.</p>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>The sample IDs <strong>must</strong> be in the rownames of
<code>pheno</code>, <code>addcovar</code>, <code>genoprobs</code> and
<code>K</code>. <code>qtl2</code> uses the sample IDs to align the
samples between objects. For more information about data file format,
see <a href="http://kbroman.org/qtl2/assets/vignettes/input_files.html" class="external-link">Karl
Broman’s vignette on input file format</a>.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="performing-a-genome-scan">Performing a Genome Scan<a class="anchor" aria-label="anchor" href="#performing-a-genome-scan"></a></h2>
<hr class="half-width"><p>Before we perform our first genome scan, let’s look at the mapping
model. At each marker on the genotyping array, we will fit a model that
regresses the phenotype (micronucleated reticulocytes) on covariates and
the founder allele proportions.</p>
<p><span class="math inline">\(y_i =
\beta_ss_i+\sum_{j=1}^8(\beta_jg_{ij}) + \lambda_i +
\epsilon_i\)</span></p>
<p>where:</p>
<p><span class="math inline">\(y_i\)</span> is the phenotype for mouse
<span class="math inline">\(i\)</span>, <span class="math inline">\(\beta_s\)</span> is the effect of study cohort,
<span class="math inline">\(s_i\)</span> is the study cohort for mouse
<span class="math inline">\(i\)</span>, <span class="math inline">\(\beta_j\)</span> is the effect of founder allele
<span class="math inline">\(j\)</span>, <span class="math inline">\(g_{ij}\)</span> is the probability that mouse
<span class="math inline">\(i\)</span> carries an allele from founder
<span class="math inline">\(j\)</span>, <span class="math inline">\(\lambda_;&lt;sub&gt;_i\)</span> is an adjustment
for kinship-induced correlated errors for mouse <span class="math inline">\(i\)</span>, <span class="math inline">\(\epsilon_i\)</span> is the residual error for
mouse <span class="math inline">\(i\)</span>.</p>
<p>Note that this model will give us an estimate of the effect of each
of the eight founder alleles at each marker. This will be important when
we estimate the founder allele effects below.</p>
<p>There are almost 600 samples in this data set and it may take several
minutes to map one trait. In order to save some time, we will map using
only the samples in the 100 ppm concentration group. We will create a
smaller phenotype data.frame.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_100</span> <span class="op">&lt;-</span> <span class="va">pheno</span><span class="op">[</span><span class="va">pheno</span><span class="op">$</span><span class="va">conc</span> <span class="op">==</span> <span class="fl">100</span>,<span class="op">]</span></span></code></pre>
</div>
<div id="challenge-8-how-many-mice-are-we-mapping-with" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-8-how-many-mice-are-we-mapping-with" class="callout-inner">
<h3 class="callout-title">Challenge 8: How many mice are we mapping with?</h3>
<div class="callout-content">
<p>Get the number of rows in <code>pheno_100</code>.</p>
</div>
</div>
</div>
<div id="accordionSolution9" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution9" aria-expanded="false" aria-controls="collapseSolution9">
  <h4 class="accordion-header" id="headingSolution9"> Show me the solution </h4>
</button>
<div id="collapseSolution9" class="accordion-collapse collapse" data-bs-parent="#accordionSolution9" aria-labelledby="headingSolution9">
<div class="accordion-body">
<p>You can do this by looking at the number of observations for
<code>pheno_100</code> in the Environment tab or in the Console.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">pheno_100</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 149</code></pre>
</div>
<p>There are 149 mice in <code>pheno_100</code>.</p>
</div>
</div>
</div>
</div>
<p>In order to map the proportion of bone marrow reticulocytes that were
micro-nucleated, you will use the <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_scan1.R" class="external-link">scan1</a>
function. To see the arguments for <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_scan1.R" class="external-link">scan1</a>,
you can type <code>help(scan1)</code>. First, let’s map the
<em>untransformed</em> phenotype. (Recall that we log-transformed it
above).</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">pheno_100</span><span class="op">)</span> <span class="op">==</span> <span class="st">"prop.bm.mn.ret"</span><span class="op">)</span></span>
<span><span class="va">lod</span>   <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>               pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>               kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>               addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we plot the genome scan.</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">map</span>, </span>
<span>           main <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-qtl_plot-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>It looks like we have a large peak on chromosome 10.</p>
<div id="challenge-9-how-does-a-log-tranformation-change-the-qtl-plot" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-9-how-does-a-log-tranformation-change-the-qtl-plot" class="callout-inner">
<h3 class="callout-title">Challenge 9: How does a log-tranformation change the QTL plot?</h3>
<div class="callout-content">
<ol style="list-style-type: decimal"><li>Perform a genome scan on the column called <code>log_mnret</code>.
(Hint: set <code>index</code> to the column index in
<code>pheno</code>.)<br></li>
<li>How does the LOD score for the peak on Chr 10 change?</li>
</ol></div>
</div>
</div>
<div id="accordionSolution10" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution10" aria-expanded="false" aria-controls="collapseSolution10">
  <h4 class="accordion-header" id="headingSolution10"> Show me the solution </h4>
</button>
<div id="collapseSolution10" class="accordion-collapse collapse" data-bs-parent="#accordionSolution10" aria-labelledby="headingSolution10">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">pheno_100</span><span class="op">)</span> <span class="op">==</span> <span class="st">"log_mnret"</span><span class="op">)</span></span>
<span><span class="va">lod2</span>  <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>               pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>               kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>               addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod2</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">map</span>, </span>
<span>           main <span class="op">=</span> <span class="st">"(log(Proportion of Micro-nucleated Bone Marrow Reticulocytes)"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-22-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Using a log transformation increases the LOD increase from about 17
to over 25.</p>
</div>
</div>
</div>
</div>
<p>The challenge shows the importance of looking at your data and
transforming it to meet the expectations of the mapping model. In this
case, a log transformation improved the model fit and increased the LOD
score. We will continue the rest of this lesson using the
log-transformed data. Set your <code>index</code> variable equal to the
column index of <code>log.MN.RET</code>. Redo the genome scan with the
log-transformed reticulocytes.</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span> <span class="op">==</span> <span class="st">"log_mnret"</span><span class="op">)</span></span>
<span><span class="va">lod</span>   <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>               pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>, <span class="va">index</span>,  drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>               kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>               addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="assessing-significance-of-lod-scores">Assessing Significance of LOD Scores<a class="anchor" aria-label="anchor" href="#assessing-significance-of-lod-scores"></a></h2>
<hr class="half-width"><p>There is clearly a large peak on Chr 10. But is it significant? In
other words, could we see a LOD score over 25 by chance? And if so, what
is the probability of seeing a LOD of 25 or higher?</p>
<p>These questions are most commonly answered via <a href="http://www.genetics.org/content/178/1/609.long" class="external-link">permutation</a> of
the sample labels and recording the maximum LOD score in each
permutation. This procedure breaks the connection between the phenotypes
and the genotypes of the mice, so the results represent the expected LOD
by chance.</p>
<p>Imagine that you shuffle the sample labels in the phenotype file and
map the resampled trait. Then you record the maximum LOD across the
genome. You would get a table that looks like this:</p>
<table class="table"><thead><tr class="header"><th>Permutation</th>
<th>Max LOD</th>
</tr></thead><tbody><tr class="odd"><td>1</td>
<td>1.7</td>
</tr><tr class="even"><td>2</td>
<td>3.2</td>
</tr><tr class="odd"><td>3</td>
<td>2.2</td>
</tr><tr class="even"><td>4</td>
<td>1.8</td>
</tr><tr class="odd"><td>5</td>
<td>2.3</td>
</tr><tr class="even"><td>…</td>
<td>…</td>
</tr></tbody></table><p>Once you ran 1,000 permutations, you could plot a histogram of the
maximum LOD values.</p>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perms</span>   <span class="op">=</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">'./data/sim_perm1000.rds'</span><span class="op">)</span></span>
<span><span class="va">max_lod</span> <span class="op">=</span> <span class="fu">apply</span><span class="op">(</span><span class="va">perms</span>, <span class="fl">2</span>, <span class="va">max</span><span class="op">)</span></span>
<span><span class="fu">hist</span><span class="op">(</span><span class="va">max_lod</span>, breaks <span class="op">=</span> <span class="fl">100</span>, main <span class="op">=</span> <span class="st">'Histogram of Max. LOD'</span><span class="op">)</span></span>
<span><span class="va">q95</span> <span class="op">=</span> <span class="fu">quantile</span><span class="op">(</span><span class="va">max_lod</span>, probs <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span>
<span><span class="fu">abline</span><span class="op">(</span>v <span class="op">=</span> <span class="va">q95</span>, col <span class="op">=</span> <span class="st">'red'</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-23-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In this case, the maximum LOD score is around 9. Most of the values
fall between 5 and 6. We have drawn a red line at the 95th percentile of
the distribution. LOD values above this are likely to occur 5% of the
time. That means 1 in 20 times. We often use this 95th percentile as our
significance threshold for one trait.</p>
<p>We advise running at least 1,000 permutations to obtain significance
thresholds. In the interest of time, we perform 100 permutations here
using the <a href="">scan1perm</a> function.</p>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perms</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                   pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                   kinshp    <span class="op">=</span> <span class="va">K</span>,</span>
<span>                   addcovar  <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                   n_perm    <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></code></pre>
</div>
<p>The <code>perms</code> object contains the maximum LOD score from
each genome scan of permuted data.</p>
<div id="challenge-10-how-does-a-log-tranformation-change-the-qtl-plot" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-10-how-does-a-log-tranformation-change-the-qtl-plot" class="callout-inner">
<h3 class="callout-title">Challenge 10: How does a log-tranformation change the QTL plot?</h3>
<div class="callout-content">
<ol style="list-style-type: decimal"><li>Create a histogram of the LOD scores <code>perms</code>. Hint: use
the <code>hist()</code> function.<br></li>
<li>Estimate the value of the LOD score at the 95th percentile.<br></li>
<li>Then find the value of the LOD score at the 95th percentile using
the <code>summary()</code> function.</li>
</ol></div>
</div>
</div>
<div id="accordionSolution11" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution11" aria-expanded="false" aria-controls="collapseSolution11">
  <h4 class="accordion-header" id="headingSolution11"> Show me the solution </h4>
</button>
<div id="collapseSolution11" class="accordion-collapse collapse" data-bs-parent="#accordionSolution11" aria-labelledby="headingSolution11">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb38">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span>x <span class="op">=</span> <span class="va">perms</span>, breaks <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-24-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perms</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (100 permutations)
     log_mnret
0.05      7.59</code></pre>
</div>
<p>Note that this summary function returns the 95th percentile value of
the LOD distribution.</p>
</div>
</div>
</div>
</div>
<p>We can now add thresholds to the previous QTL plot. We use a
significance threshold of p &lt; 0.05. To do this, we select the 95th
percentile of the permutation LOD distribution.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod</span>, </span>
<span>     map  <span class="op">=</span> <span class="va">map</span>,  </span>
<span>     main <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span><span class="op">)</span></span>
<span><span class="va">thr</span> <span class="op">=</span> <span class="fu">summary</span><span class="op">(</span><span class="va">perms</span><span class="op">)</span></span>
<span><span class="fu">add_threshold</span><span class="op">(</span>map <span class="op">=</span> <span class="va">map</span>, thresholdA <span class="op">=</span> <span class="va">thr</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-25-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The peak on Chr 10 is well above the red significance line.</p>
</section><section><h2 class="section-heading" id="finding-lod-peaks">Finding LOD Peaks<a class="anchor" aria-label="anchor" href="#finding-lod-peaks"></a></h2>
<hr class="half-width"><p>We can find all of the peaks above the significance threshold using
the <a href="https://github.com/rqtl/qtl2/blob/master/R/find_peaks.R" class="external-link">find_peaks</a>
function.</p>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod</span>,</span>
<span>           map          <span class="op">=</span> <span class="va">map</span>, </span>
<span>           threshold    <span class="op">=</span> <span class="va">thr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr      pos      lod
1        1 log_mnret  10 33.52438 27.01701</code></pre>
</div>
<div id="challenge-11-how-does-a-log-tranformation-change-the-qtl-plot" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-11-how-does-a-log-tranformation-change-the-qtl-plot" class="callout-inner">
<h3 class="callout-title">Challenge 11: How does a log-tranformation change the QTL plot?</h3>
<div class="callout-content">
<p>Find all peaks for this scan whether or not they meet the 95%
significance threshold.</p>
</div>
</div>
</div>
<div id="accordionSolution12" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution12" aria-expanded="false" aria-controls="collapseSolution12">
  <h4 class="accordion-header" id="headingSolution12"> Show me the solution </h4>
</button>
<div id="collapseSolution12" class="accordion-collapse collapse" data-bs-parent="#accordionSolution12" aria-labelledby="headingSolution12">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">map</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>   lodindex lodcolumn chr        pos       lod
1         1 log_mnret   1   4.601252  3.594686
2         1 log_mnret   2 141.066196  5.351645
3         1 log_mnret   3 124.914616  3.904994
4         1 log_mnret   4 132.636314  5.501382
5         1 log_mnret   6  85.246537  5.965923
6         1 log_mnret   7  44.136809  4.325292
7         1 log_mnret   8  26.176972  3.073792
8         1 log_mnret   9   8.115893  4.027077
9         1 log_mnret  10  33.524378 27.017008
10        1 log_mnret  11  48.605368  3.563292
11        1 log_mnret  12  28.480475  3.549043
12        1 log_mnret  13 109.752043  3.111400
13        1 log_mnret  16  72.091039  5.263606
14        1 log_mnret  17  12.528523  3.659285
15        1 log_mnret  18  65.977753  3.894702
16        1 log_mnret   X   8.770202  4.059789</code></pre>
</div>
<p>Notice that some peaks are missing because they don’t meet the
default threshold value of 3. See <code>help(find_peaks)</code> for more
information about this function.</p>
</div>
</div>
</div>
</div>
<p>The support interval is determined using the<br><a href="http://www.ncbi.nlm.nih.gov/pubmed/11560912" class="external-link">Bayesian Credible
Interval</a> and represents the region most likely to contain the
causative polymorphism(s). We can obtain this interval by adding a
<code>prob</code> argument to <a href="https://github.com/rqtl/qtl2/blob/master/R/find_peaks.R" class="external-link">find_peaks</a>.
We pass in a value of <code>0.95</code> to request a support interval
that contains the causal SNP 95% of the time.</p>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">map</span>, </span>
<span>           threshold    <span class="op">=</span> <span class="va">thr</span>, </span>
<span>           prob         <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr      pos      lod   ci_lo    ci_hi
1        1 log_mnret  10 33.52438 27.01701 31.7442 34.05311</code></pre>
</div>
<p>From the output above, you can see that the support interval is 2.3
Mb wide (31.7442 to 34.05311 Mb). The location of the maximum LOD score
is at 33.52438 Mb.</p>
</section><section><h2 class="section-heading" id="estimated-founder-allele-effects">Estimated Founder Allele Effects<a class="anchor" aria-label="anchor" href="#estimated-founder-allele-effects"></a></h2>
<hr class="half-width"><p>We will now zoom in on Chr 10 and look at the contribution of each of
the eight founder alleles to the proportion of bone marrow reticulocytes
that were micro-nucleated. Remember, the mapping model above estimates
the effect of each of the eight DO founders. We can plot these effects
(also called ‘coefficients’) across Chr 10 using <a href="https://github.com/rqtl/qtl2/blob/master/R/scan1coef.R" class="external-link">scan1coef</a>.</p>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>    <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">coef10</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                    pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                    kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                    addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>This produces an object containing estimates of each of the eight DO
founder allele effect. These are the <i>β<sub>j</sub></i> values in the
mapping equation above.</p>
<div class="codewrapper sourceCode" id="cb49">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coefCC</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">coef10</span>, </span>
<span>            map  <span class="op">=</span> <span class="va">map</span>, </span>
<span>            scan1_output <span class="op">=</span> <span class="va">lod</span>, </span>
<span>            main <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-27-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The top panel shows the eight founder allele effects (or model
coefficients) along Chr 10. The founder allele effects are centered at
zero and the units are the same as the phenotype. You can see that DO
mice containing the CAST/EiJ allele near 34 Mb have lower levels of
micro-nucleated reticulocytes. This means that the CAST allele is
associated with less DNA damage and has a protective effect. The bottom
panel shows the LOD score, with the support interval for the peak shaded
blue.</p>
</section><section><h2 class="section-heading" id="snp-association-mapping">SNP Association Mapping<a class="anchor" aria-label="anchor" href="#snp-association-mapping"></a></h2>
<hr class="half-width"><p>At this point, we have a 2.3 Mb wide support interval that contains
polymorphism(s) that influence benzene-induced DNA damage. Next, we will
impute the DO founder sequences onto the DO genomes. The <a href="https://www.mousegenomes.org/" class="external-link">Mouse Genomes Project</a> has
sequenced the eight DO founders and provides SNP, insertion-deletion
(indel), and structural variant files for the strains (see <a href="http://www.nature.com/ng/journal/v45/n7/full/ng.2644.html" class="external-link">Baud
et.al., Nat. Gen., 2013</a>). We can impute these SNPs onto the DO
genomes and then perform association mapping. The process involves
several steps and I have provided a function to encapsulate the steps.
To access the Sanger SNPs, we use a SQLlite database provided by <a href="https://github.com/kbroman" class="external-link">Karl Broman</a>. You should have
downloaded this during Setup. It is available from<br><a href="https://figshare.com/ndownloader/files/40157572" class="external-link">figshare</a>,
but the file is 10 GB, so it may take too long to download right
now.</p>
<p><img src="./fig/DO.impute.founders.sm.png" alt="SNP Imputation" class="figure">{alt=“Figure showing haplotypes being used to
impute founder SNPs onto DO genomes.”,width=75%}</p>
<p>Association mapping involves imputing the founder SNPs onto each DO
genome and fitting the mapping model at each SNP. At each marker, we fit
the following model:</p>
<p><span class="math inline">\(y_{im} = \beta_ss_i+\beta_mg_{im} +
\lambda_i + \epsilon_i\)</span></p>
<p>where:</p>
<p><span class="math inline">\(y_i\)</span> is the phenotype for mouse
<span class="math inline">\(i\)</span>, <span class="math inline">\(\beta_s\)</span> is the effect of study cohort,
<span class="math inline">\(s_i\)</span> is the study cohort for mouse
<span class="math inline">\(i\)</span>, <span class="math inline">\(\beta_m\)</span> is the effect of adding one
allele at marker <span class="math inline">\(m\)</span>, <span class="math inline">\(g_{im}\)</span> is the allele call for mouse <span class="math inline">\(i\)</span> at marker <span class="math inline">\(m\)</span>&gt;, <span class="math inline">\(\lambda_i\)</span> is an adjustment for
kinship-induced correlated errors for mouse <span class="math inline">\(i\)</span>, <span class="math inline">\(\epsilon_i\)</span> is the residual error for
mouse <span class="math inline">\(i\)</span>.</p>
<p>We can call <a href="https://github.com/rqtl/qtl2/blob/master/R/scan1snps.R" class="external-link">scan1snps</a>
to perform association mapping in the QTL interval on Chr 10. We first
create variables for the chromosome and support interval where we are
mapping. We then create a function to get the SNPs from the founder SNP
database.</p>
<div id="callout4" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>It is important to use the <code>keep_all_snps = TRUE</code> in order
to return all SNPs.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>   <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">start</span> <span class="op">&lt;-</span> <span class="fl">30</span></span>
<span><span class="va">end</span>   <span class="op">&lt;-</span> <span class="fl">36</span></span>
<span></span>
<span><span class="co"># Create function to query founder SNP database.</span></span>
<span><span class="va">query_snps</span>  <span class="op">&lt;-</span> <span class="fu">create_variant_query_func</span><span class="op">(</span>dbfile   <span class="op">=</span> <span class="st">"./data/fv.2021.snps.db3"</span>,</span>
<span>                                         id_field <span class="op">=</span> <span class="st">"variants_id"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">assoc</span>      <span class="op">&lt;-</span> <span class="fu">scan1snps</span><span class="op">(</span>genoprobs  <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                        map        <span class="op">=</span> <span class="va">map</span>, </span>
<span>                        pheno      <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                        kinship    <span class="op">=</span> <span class="va">K</span>, </span>
<span>                        addcovar   <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                        query_func <span class="op">=</span> <span class="va">query_snps</span>,</span>
<span>                        chr        <span class="op">=</span> <span class="va">chr</span>, </span>
<span>                        start      <span class="op">=</span> <span class="va">start</span>, </span>
<span>                        end        <span class="op">=</span> <span class="va">end</span>, </span>
<span>                        keep_all_snps <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<p>The <code>assoc</code> object is a list containing two objects: the
LOD scores for each unique SNP and a <code>snpinfo</code> object that
maps the LOD scores to each SNP. To plot the association mapping, we
need to provide both objects to the <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_snpasso.R" class="external-link">plot_snpasso</a>
function.</p>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#png('./fig/bm_mnret_assoc_fig1.png', width = 2000, height = 2000, res = 256)</span></span>
<span><span class="fu">plot_snpasso</span><span class="op">(</span>scan1output <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">lod</span>, </span>
<span>             snpinfo     <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">snpinfo</span>, </span>
<span>             main        <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span><span class="op">)</span></span>
<span><span class="co">#dev.off()</span></span></code></pre>
</div>
<p><img src="./fig/bm_mnret_assoc_fig1.png" alt="Bone Marrow MN-RET Association Mapping" class="figure">{alt=“Plot showing LOD of
each SNP in QTL interval”,width=100%}</p>
<p>This plot shows the LOD score for each SNP in the QTL interval. The
SNPs occur in “shelves” because all of the SNPs in a haplotype block
have the same founder strain pattern. The SNPs with the highest LOD
scores are the ones for which CAST/EiJ contributes the alternate
allele.</p>
<p>You might naturally ask what genes are under this QTL peak. We will
query the SNP/gens database for the genes in the interval. As before, we
will create a function to query the genes in a specific interval of the
genome.</p>
<div class="codewrapper sourceCode" id="cb52">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create function to query gene database.</span></span>
<span><span class="va">query_genes</span> <span class="op">&lt;-</span> <span class="fu">create_gene_query_func</span><span class="op">(</span>dbfile      <span class="op">=</span> <span class="st">"./data/fv.2021.snps.db3"</span>,</span>
<span>                                      chr_field   <span class="op">=</span> <span class="st">"chromosome"</span>, </span>
<span>                                      name_field  <span class="op">=</span> <span class="st">"symbol"</span>,</span>
<span>                                      start_field <span class="op">=</span> <span class="st">"start_position"</span>, </span>
<span>                                      stop_field  <span class="op">=</span> <span class="st">"end_position"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu">query_genes</span><span class="op">(</span><span class="va">chr</span>, <span class="va">start</span>, <span class="va">end</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          ensembl_id ensembl_version    Name                     name chromosome start_position end_position strand    start
1 ENSMUSG00000000295              14   Hddc2   HD domain containing 2         10       31189379     31204200      1 31.18938
2 ENSMUSG00000000296               9 Tpd52l1 tumor protein D52-like 1         10       31208372     31321954     -1 31.20837
3 ENSMUSG00000019779              16     Frk       fyn-related kinase         10       34359395     34487274      1 34.35939
4 ENSMUSG00000019782              11   Rwdd1  RWD domain containing 1         10       33872551     33895620     -1 33.87255
5 ENSMUSG00000019785              15   Clvs2               clavesin 2         10       33388282     33500765     -1 33.38828
6 ENSMUSG00000019787              10    Trdn                  triadin         10       32956550     33352705      1 32.95655
      stop
1 31.20420
2 31.32195
3 34.48727
4 33.89562
5 33.50077
6 33.35271</code></pre>
</div>
<p>The <code>genes</code> object contains annotation information for
each gene in the interval.</p>
<p>Next, we will create a plot with two panels: one containing the
association mapping LOD scores and one containing the genes in the QTL
interval. We do this by passing in the <code>genes</code> argument to <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_snpasso.R" class="external-link">plot_snpasso</a>.
We can also adjust the proportion of the plot allocated for SNPs (on the
top) and genes (on the bottom) using the ‘top_panel_prop’ argument.</p>

<div class="codewrapper sourceCode" id="cb54">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#png('./fig/bm_mnret_assoc_fig2.png', width = 2000, height = 2000, res = 256)</span></span>
<span><span class="fu">plot_snpasso</span><span class="op">(</span>scan1output <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">lod</span>, </span>
<span>             snpinfo     <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">snpinfo</span>, </span>
<span>             main        <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span>, </span>
<span>             genes       <span class="op">=</span> <span class="va">genes</span>,</span>
<span>             colors      <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>             top_panel_prop <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span>
<span><span class="co">#dev.off()</span></span></code></pre>
</div>
<p><img src="./fig/bm_mnret_assoc_fig2.png" alt="Bone Marrow MN-RET Association Mapping" class="figure">{alt=“Plot showing LOD of
each SNP in QTL interval with genes below”,width=100%}</p>
<p>Now that we have the genes in this interval, we would like to know
which founders have the minor allele for the SNPs with the highest LOD
scores. To do this, we will highlight SNPs that are withing a 1 LOD drop
of the highest LOD and we will add an argument to show which founder
contributes the minor allele at the highlighted SNPs.</p>

<div class="codewrapper sourceCode" id="cb55">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#png('./fig/bm_mnret_assoc_fig3.png', width = 2000, height = 2000, res = 256)</span></span>
<span><span class="fu">plot_snpasso</span><span class="op">(</span>scan1output <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">lod</span>, </span>
<span>             snpinfo     <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">snpinfo</span>, </span>
<span>             main        <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span>, </span>
<span>             genes       <span class="op">=</span> <span class="va">genes</span>,</span>
<span>             colors      <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>             panel_prop <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>             drop_hilit  <span class="op">=</span> <span class="fl">1.0</span>,</span>
<span>             col_hilit   <span class="op">=</span> <span class="st">'red'</span>,</span>
<span>             sdp_panel   <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#dev.off()</span></span></code></pre>
</div>
<p><img src="./fig/bm_mnret_assoc_fig3.png" alt="Bone Marrow MN-RET Association Mapping" class="figure">{alt=“Plot showing LOD of
each SNP in QTL interval with genes below and strain distribution
pattern above”,width=100%}</p>
<div class="section level3">
<h3 id="searching-for-candidate-genes">Searching for Candidate Genes<a class="anchor" aria-label="anchor" href="#searching-for-candidate-genes"></a></h3>
<p>One strategy for finding genes related to a phenotype is to search
for genes with expression QTL (eQTL) in the same location. Ideally, we
would have liver and bone marrow gene expression data in the DO mice
from this experiment. Unfortunately, we did not collect this data.
However, we have liver gene expression for a separate set of untreated
DO mice <a href="https://churchilllab.jax.org/qtlviewer/svenson/DOHFD" class="external-link">Liver eQTL
Viewer</a>. We searched for genes in the QTL interval that had an eQTL
in the same location. Then, we looked at the pattern of founder effects
to see if CAST stood out. We found two genes that met these
criteria.</p>
<p><img src="./fig/French.et.al.Figure3.png" alt="Sult3a1/Sult3a2 Liver eQTL" class="figure">{Figure showing LOD plots of Sult3a1
&amp; 2 and CAST-specific haplotype effects}</p>
<p>The plot above shows the genome scane for two genes: <em>Sult3a1</em>
and <em>Gm4794</em>. <em>Gm4794</em> has been renamed to
<em>Sult3a2</em>. As you can see, both <em>Sult3a1</em> <em>Sult3a2</em>
have eQTL in the same location at the MN-RET QTL on Chr 10. Mice
carrying the CAST allele (in green) express these genes more highly.
<em>Sult3a1</em> is a <a href="https://en.wikipedia.org/wiki/Sulfotransferase" class="external-link">sulfotransferase</a>
that may be involved in adding a sulfate group to phenol, one of the
metabolites of benzene.</p>
<div id="challenge-11-look-for-sult3a1-sult3a2-qtl-in-do-bone-expression" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-11-look-for-sult3a1-sult3a2-qtl-in-do-bone-expression" class="callout-inner">
<h3 class="callout-title">Challenge 11: Look for Sult3a1 &amp; Sult3a2 QTL in DO bone expression</h3>
<div class="callout-content">
<p>Go to the <a href="https://churchilllab.jax.org/qtlviewer/DO/bone" class="external-link">DO
Bone eQTL Viewer</a>, type in <em>Sult3a1</em> and click on the search
button. Then search on <em>Sult3a2</em>.</p>
</div>
</div>
</div>
<div id="accordionSolution13" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution13" aria-expanded="false" aria-controls="collapseSolution13">
  <h4 class="accordion-header" id="headingSolution13"> Output </h4>
</button>
<div id="collapseSolution13" class="accordion-collapse collapse" data-bs-parent="#accordionSolution13" aria-labelledby="headingSolution13">
<div class="accordion-body">
<p>Neither <em>Sult3a1</em> or <em>Sult3a2</em> has an eQTL in the DO
bone marrow data set.</p>
</div>
</div>
</div>
</div>
<p>Next, we will look at the relationship between <em>Sult3a1</em> and
<em>Sult3a2</em>. Go to the Ensembl web page for <a href="https://useast.ensembl.org/Mus_musculus/Gene/Summary?db=core;g=ENSMUSG00000090298;r=10:33642420-33662700" class="external-link">Sult3a2</a>.
In the menu on the left, click on the “Gene Tree (image)” link.</p>
<p><img src="./fig/Ensembl_Mouse_Sult3a1_Sult3a2.png" alt="Ensembl gene tree of Sult3a1 &amp; Sult3a2" class="figure">{Figure showing that
Sult3a1 &amp; 2 are paralogs}</p>
<p>As you can see, <em>Sult3a2</em> is a paralog of <em>Sult3a1</em> and
hence both genes are sulfotransferases. These genes encode enzymes that
attach a sulfate group to other compounds.</p>
<div id="challenge-12-look-for-structural-variants-near-sult3a1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-12-look-for-structural-variants-near-sult3a1" class="callout-inner">
<h3 class="callout-title">Challenge 12: Look for structural variants near Sult3a1</h3>
<div class="callout-content">
<p>Go to the <a href="https://useast.ensembl.org/index.html" class="external-link">Ensembl
website</a> and select “Mouse” from the species dropdown. Then type
“10:33300000-33800000” into the search box and press “Go”. Scroll down
to the detailed view and look for structural variants in this
region.</p>
</div>
</div>
</div>
<div id="accordionSolution14" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution14" aria-expanded="false" aria-controls="collapseSolution14">
  <h4 class="accordion-header" id="headingSolution14"> Output </h4>
</button>
<div id="collapseSolution14" class="accordion-collapse collapse" data-bs-parent="#accordionSolution14" aria-labelledby="headingSolution14">
<div class="accordion-body">
<p>You should see a plot similar to the one below.</p>
<p><img src="./fig/Ensembl_SV_Chr10.png" alt="Ensembl Structural Variants" class="figure">{alt=“Ensembl viewer showing genes
and structural variants under that chromosome 10 QTL
peak”,width=100%}</p>
</div>
</div>
</div>
</div>
<p><img src="./fig/Ensembl_SV_Chr10.png" alt="Ensembl Structural Variants" class="figure">{alt=“Ensembl viewer showing genes
and structural variants under that chromosome 10 QTL
peak”,width=100%}</p>
<p>In order to visualize the size of the copy number gain, we queried
the Mouse Genomes Project <a href="https://www.mousegenomes.org/mouse-strains-sequenced/" class="external-link">alignment
files</a> for the eight founders. We piled up the reads at each base
(which is beyond the scope of this tutorial) and made the figure
below.</p>
<p><img src="./fig/French.et.al.Sup.Figure3.png" alt="DO Founder BAM file pileups" class="figure">{alt=“Figure showing CAST with a
duplication at the Sult3a1/2 locus.”,width=100%}</p>
<p>As you can see, there appears to be a duplicatation in the CAST
founders that covers four genes: <em>Clvs2</em>, <em>Gm15939</em>,
<em>Sult3a1</em> and <em>Sult3a2</em>. <em>Clvs2</em> is expressed in
neurons and <em>Gm15939</em> is a predicted gene that may not produce a
transcript.</p>
<p>Hence, we have three pieces of evidence that narrows our candidate
gene list to <em>Sult3a1</em> and <em>Sult3a2</em>:</p>
<ol style="list-style-type: decimal"><li>Both genes have a liver eQTL in the same location as the
micronucleated reticulocytes QTL.</li>
<li>Among genes in the micronucleated reticulocytes QTL interval, only
<em>Sult3a1</em> and <em>Sult3a2</em> have differential expression of
the CAST allele in the liver.</li>
<li>There is a copy number gain of these two genes in CAST.</li>
</ol><p>Sulfation is a prominent detoxification mechanism for benzene as
well. The diagram below shows the metabolism pathway for benzene <a href="http://europepmc.org/articles/PMC4414400" class="external-link">(Monks, T. J., et
al. (2010). Chem Biol Interact 184(1-2): 201-206.)</a> Hydroquinone,
phenol and catechol are all sulfated and excreted from the body.</p>
<p><img src="./fig/Monks_ChemBiolInter_2010_Fig1.jpg" alt="Benezene metabolism pathways" class="figure">{alt=“Figure showing benezene
metabolism pathways”,width=100%}</p>
<p>This analysis has led us to the following hypothesis. Inhaled benzene
is absorbed by the lungs into the bloodstream and transported to the
liver. There, it is metabolized, and some metabolites are transported to
the bone marrow. One class of genes that is involved in toxicant
metabolism are sulfotransferases. <a href="http://www.informatics.jax.org/marker/MGI:1931469" class="external-link"><i>Sult3a1</i></a>
is a phase II enzyme that conjugates compounds (such as phenol, which is
a metabolite of benzene) with a sulfate group before transport into the
bile. It is possible that a high level of <em>Sult3a1</em> expression
could remove benzene by-products and be protective. Our hypothesis is
that the copy number gain in the CAST allele increases liver gene
expression of <em>Sult3a1</em> and <em>Sult3a2</em>. High liver
expression of these genes allows mice containing the CAST allele to
rapidly conjugate harmful benzene metabolites and excrete them from the
body before they can reach the bone marrow and cause DNA damage. Further
experimental validation is required, but this is a plausible
hypothesis.</p>
<p><img src="./fig/benzene_hypothesis.png" alt="Benzene metabolism hypothesis" class="figure">{alt=“Figure showing mice carrying
the CAST allele being protected from benezene toxicity”,width=100%}</p>
<p>We hope that this tutorial has shown you how the DO can be used to
map QTL and use the founder effects and bioinformatics resources to
narrow down the candidate gene list. Here, we made used of external gene
expression databases and the founder sequence data to build a case for a
pair of genes.</p>
<div id="challenge-13-map-another-trait." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-13-map-another-trait." class="callout-inner">
<h3 class="callout-title">Challenge 13: Map another trait.</h3>
<div class="callout-content">
<ol style="list-style-type: decimal"><li>Make a histogram of the column <code>pre.prop.mn.ret</code> in
<code>pheno</code>. Does it look like it should be log transformed? If
so, add a column to <code>pheno</code> that contains the log of
<code>pre.prop.mn.ret</code>.<br></li>
<li>Perform a genome scan <em>using all samples</em> on the column
called <code>pre.prop.mn.ret</code>. Since you’re using all samples, the
scan will take longer. You will also need to add the benezene
concentration to the covariates. (Hint: set <code>index</code> to the
column index in <code>pheno</code>.)<br></li>
<li>Which chromosome has the highest peak? Use <code>find_peaks</code>
to get the location and support interval for the highest peak.<br></li>
<li>Calculate and plot the BLUPs for the chromosome with the highest
peak. (This may take a few minutes.) Which founder contributes an allele
that makes MN-RET values higher?</li>
<li>Perform association mapping in the support interval for the QTL
peak, plot the results and plot the genes beneath the association
mapping plot.</li>
</ol></div>
</div>
</div>
<div id="accordionSolution15" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution15" aria-expanded="false" aria-controls="collapseSolution15">
  <h4 class="accordion-header" id="headingSolution15"> Output </h4>
</button>
<div id="collapseSolution15" class="accordion-collapse collapse" data-bs-parent="#accordionSolution15" aria-labelledby="headingSolution15">
<div class="accordion-body">
<ol style="list-style-type: decimal"><li>Plot the distribution of</li>
</ol><div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">pheno</span><span class="op">$</span><span class="va">pre.prop.mn.ret</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-33-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Log transform the pre-dose proportion of MN-RETs.</p>
<div class="codewrapper sourceCode" id="cb57">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span><span class="op">$</span><span class="va">log_pre_prop_mnret</span> <span class="op">&lt;-</span> <span class="fu">log</span><span class="op">(</span><span class="va">pheno</span><span class="op">$</span><span class="va">pre.prop.mn.ret</span><span class="op">)</span></span></code></pre>
</div>
<ol start="2" style="list-style-type: decimal"><li>Set the mapping index equal to the <code>pre.prop.mn.ret</code>
column.</li>
</ol><div class="codewrapper sourceCode" id="cb58">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span> <span class="op">==</span> <span class="st">"log_pre_prop_mnret"</span><span class="op">)</span></span></code></pre>
</div>
<p>Create new covariates which include study and benzene
concentration.</p>
<div class="codewrapper sourceCode" id="cb59">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">addcovar2</span> <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">study</span> <span class="op">+</span> <span class="va">conc</span>, data <span class="op">=</span> <span class="va">pheno</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb60">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_pre</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                 pheno     <span class="op">=</span> <span class="va">pheno</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                 kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar2</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_pre</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">map</span>, </span>
<span>           main <span class="op">=</span> <span class="st">"Log-Transformed Pre-dose micronucleated reticulocytes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-37-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><ol start="3" style="list-style-type: decimal"><li>Find peaks above a significance threshold.</li>
</ol><div class="codewrapper sourceCode" id="cb61">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span><span class="va">lod_pre</span>, <span class="va">map</span>, threshold <span class="op">=</span> <span class="fl">8</span>, prob <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr       pos       lod     ci_lo     ci_hi
1        1 log_pre_prop_mnret   4 132.77875 13.145968 132.77875 134.27938
2        1 log_pre_prop_mnret  12  75.45433  8.404242  74.14949  77.30552</code></pre>
</div>
<ol start="4" style="list-style-type: decimal"><li>Calculate and plot BLUPs for the highest peak.</li>
</ol><div class="codewrapper sourceCode" id="cb63">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>   <span class="op">&lt;-</span> <span class="fl">4</span></span>
<span><span class="va">coef4</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                   pheno     <span class="op">=</span> <span class="va">pheno</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                   kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                   addcovar  <span class="op">=</span> <span class="va">addcovar2</span><span class="op">)</span></span>
<span><span class="fu">plot_coefCC</span><span class="op">(</span>x            <span class="op">=</span> <span class="va">coef4</span>, </span>
<span>            map          <span class="op">=</span> <span class="va">map</span>, </span>
<span>            scan1_output <span class="op">=</span> <span class="va">lod_pre</span>, </span>
<span>            legend       <span class="op">=</span> <span class="st">"topleft"</span>,</span>
<span>            main         <span class="op">=</span> <span class="st">"Log-Transformed Pre-dose micronucleated reticulocytes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/do_qtl_mapping-rendered-unnamed-chunk-39-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><ol start="5" style="list-style-type: decimal"><li>Perform association mapping in the QTL interval for the highest
peak.</li>
</ol><div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">start</span>  <span class="op">&lt;-</span> <span class="fl">132.8</span></span>
<span><span class="va">end</span>    <span class="op">&lt;-</span> <span class="fl">134.3</span></span>
<span><span class="va">assoc4</span> <span class="op">&lt;-</span> <span class="fu">scan1snps</span><span class="op">(</span>genoprobs  <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                    map        <span class="op">=</span> <span class="va">map</span>, </span>
<span>                    pheno      <span class="op">=</span> <span class="va">pheno</span><span class="op">[</span>,<span class="va">index</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                    kinship    <span class="op">=</span> <span class="va">K</span>, </span>
<span>                    addcovar   <span class="op">=</span> <span class="va">addcovar2</span>, </span>
<span>                    query_func <span class="op">=</span> <span class="va">query_snps</span>, </span>
<span>                    chr        <span class="op">=</span> <span class="va">chr</span>,</span>
<span>                    start      <span class="op">=</span> <span class="va">start</span>, </span>
<span>                    end        <span class="op">=</span> <span class="va">end</span>, </span>
<span>                    keep_all_snps <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu">query_genes</span><span class="op">(</span><span class="va">chr</span>, <span class="va">start</span>, <span class="va">end</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb65">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#png('./fig/pre_mnret_assoc.png', width = 2000, height = 2000, res = 256)</span></span>
<span><span class="fu">plot_snpasso</span><span class="op">(</span>scan1output <span class="op">=</span> <span class="va">assoc4</span><span class="op">$</span><span class="va">lod</span>, </span>
<span>             snpinfo     <span class="op">=</span> <span class="va">assoc4</span><span class="op">$</span><span class="va">snpinfo</span>, </span>
<span>             main        <span class="op">=</span> <span class="st">"Log-Transformed Pre-dose micronucleated reticulocytes"</span>, </span>
<span>             genes       <span class="op">=</span> <span class="va">genes</span>,</span>
<span>             colors      <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>             panel_prop <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>             drop_hilit  <span class="op">=</span> <span class="fl">1.0</span>,</span>
<span>             col_hilit   <span class="op">=</span> <span class="st">'red'</span>,</span>
<span>             sdp_panel   <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#dev.off()</span></span></code></pre>
</div>
<p><a href="./fig/pre_mnret_assoc.png">Pre-dose MN-RET Association
Mapping</a>{alt=“Figure showing pre-dose MN-RET association mapping with
WSB carrying the alternate allele”,width=100%}</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb66">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessionInfo</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
 [1] qtl2convert_0.30 qtl2_0.36        ggbeeswarm_0.7.2 lubridate_1.9.3
 [5] forcats_1.0.0    stringr_1.5.1    dplyr_1.1.4      purrr_1.0.2
 [9] readr_2.1.5      tidyr_1.3.1      tibble_3.2.1     ggplot2_3.5.1
[13] tidyverse_2.0.0

loaded via a namespace (and not attached):
 [1] utf8_1.2.4         generics_0.1.3     renv_1.0.10        RSQLite_2.3.7
 [5] stringi_1.8.4      hms_1.1.3          magrittr_2.0.3     RColorBrewer_1.1-3
 [9] evaluate_0.24.0    grid_4.4.1         timechange_0.3.0   fastmap_1.2.0
[13] blob_1.2.4         DBI_1.2.3          fansi_1.0.6        scales_1.3.0
[17] qtl_1.70           cli_3.6.3          rlang_1.1.4        bit64_4.5.2
[21] munsell_0.5.1      withr_3.0.1        cachem_1.1.0       yaml_2.3.10
[25] tools_4.4.1        parallel_4.4.1     tzdb_0.4.0         memoise_2.0.1
[29] colorspace_2.1-1   vctrs_0.6.5        R6_2.5.1           lifecycle_1.0.4
[33] bit_4.5.0          vipor_0.4.7        pkgconfig_2.0.3    beeswarm_0.4.0
[37] pillar_1.9.0       gtable_0.3.5       glue_1.7.0         data.table_1.16.0
[41] Rcpp_1.0.13        highr_0.11         xfun_0.46          tidyselect_1.2.1
[45] knitr_1.48         farver_2.1.2       labeling_0.4.3     compiler_4.4.1    </code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>There are generally five steps to QTL mapping in DO mice:
<ul><li>map the trait,</li>
<li>perform permutations,</li>
<li>find significant peaks,</li>
<li>calculate founder allele effects at the QTL peak,</li>
<li>perform association mapping to narrow the gene candidates.</li>
</ul></li>
<li>You may need to bring in outside resources to help narrow your
candidate gene list.</li>
<li>You will need the 10 GB SNP database to perform association
mapping.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="integrating-gene-expression.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="index.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="integrating-gene-expression.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Integrating Gene
        </a>
        <a class="chapter-link float-end" href="index.html" rel="next">
          Home
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/do_qtl_mapping.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.7" class="external-link">sandpaper (0.16.7)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.6" class="external-link">pegboard (0.7.6)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.4" class="external-link">varnish (1.0.4)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-rmd/do_qtl_mapping.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries, genetics, statistics, genome, genetic mapping",
  "name": "QTL Mapping in Diversity Outbred Mice",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/do_qtl_mapping.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/do_qtl_mapping.html",
  "dateCreated": "2024-09-06",
  "dateModified": "2024-10-09",
  "datePublished": "2024-10-10"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

