<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Quantitative Trait Mapping: Estimating QTL effects</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/incubator/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicons/incubator/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicons/incubator/favicon-16x16.png"><link rel="manifest" href="favicons/incubator/site.webmanifest"><link rel="mask-icon" href="favicons/incubator/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="assets/images/incubator-logo.svg"><span class="badge text-bg-info">
          <abbr title="This lesson is in the beta phase, which means that it is ready for teaching by instructors outside of the original author team.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-circle" style="border-radius: 5px"></i>
              Beta
            </a>
            <span class="visually-hidden">This lesson is in the beta phase, which means that it is ready for teaching by instructors outside of the original author team.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/est-qtl-effects.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="assets/images/incubator-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Quantitative Trait Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Quantitative Trait Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Quantitative Trait Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 60%" class="percentage">
    60%
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: 60%" aria-valuenow="60" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/est-qtl-effects.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="intro_data_set.html">1. Introduction to the Data Set</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="input-file-format.html">2. Input File Format</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="calc-genoprob.html">3. Calculating Genotype Probabilities</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="perform-genome-scan.html">4. Performing a Genome Scan</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="calc-kinship.html">5. Calculating A Kinship Matrix</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="perform-genome-scan-lmm.html">6. Performing a genome scan with a linear mixed model</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="perform-genome-scan-bin.html">7. Performing a Genome Scan with Binary Traits</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="perform-perm-test.html">8. Finding Significant Peaks via Permutation</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="find-lod-peaks.html">9. Finding QTL peaks</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        10. Estimating QTL effects
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#estimating-founder-allele-effects">Estimating Founder Allele Effects</a></li>
<li><a href="#plotting-founder-allele-effects-along-a-chromosome">Plotting Founder Allele Effects Along a Chromosome</a></li>
<li><a href="#estimating-founder-allele-effects-using-blups">Estimating Founder Allele Effects using BLUPs</a></li>
<li><a href="#plotting-allele-effects-at-one-marker">Plotting Allele Effects at One Marker</a></li>
<li><a href="#challenge-4-allele-effects">Challenge 4: Allele Effects</a></li>
<li><a href="#challenge-5-allele-effects-on-chromosome-7">Challenge 5: Allele Effects on Chromosome 7</a></li>
<li><a href="#estimating-additive-and-dominant-allele-effects">Estimating Additive and Dominant Allele Effects</a></li>
<li><a href="#plotting-phenotypes-versus-genotypes">Plotting Phenotypes versus Genotypes</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="integrating-gene-expression.html">11. Integrating Gene Expression Data</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush13">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading13">
        <a href="do_qtl_mapping.html">12. QTL Mapping in Diversity Outbred Mice</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="find-lod-peaks.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="integrating-gene-expression.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="find-lod-peaks.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Finding QTL peaks
        </a>
        <a class="chapter-link float-end" href="integrating-gene-expression.html" rel="next">
          Next: Integrating Gene...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Estimating QTL effects</h1>
        <p>Last updated on 2025-07-29 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/est-qtl-effects.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How do I find the founder allele effects at a QTL peak?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Estimate the founder allele effects at a QTL peak.</li>
<li>Plot the estimated founder allele effects.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="estimating-founder-allele-effects">Estimating Founder Allele Effects<a class="anchor" aria-label="anchor" href="#estimating-founder-allele-effects"></a></h2>
<hr class="half-width"><p>Recall that to model data from a cross, we use</p>
<p><span class="math inline">\(y_j = \mu + \beta_k G_{jk} +
\epsilon_j\)</span></p>
<p>where <span class="math inline">\(y_{ij}\)</span> is the phenotype of
the <span class="math inline">\(j\)</span>th individual, <span class="math inline">\(\mu\)</span> is the mean phenotype value, <span class="math inline">\(\beta_k\)</span> is the effect of the <span class="math inline">\(k\)</span>th genotype, <span class="math inline">\(G_{jk}\)</span> is the genotype for individual
<span class="math inline">\(j\)</span>, and <span class="math inline">\(\epsilon_j\)</span> is the error for the <span class="math inline">\(j\)</span>th individual. In the figure below,
<span class="math inline">\(\mu\)</span> equals 94.6 and <span class="math inline">\(\beta\)</span> equals 15 for the alternative
hypothesis (QTL exists).</p>
<figure><img src="fig/nullvalt.png" alt="Null and alternative hypotheses" class="figure mx-auto d-block"></figure><p>This linear model is <span class="math inline">\(y\)</span> = 94.6 +
15<span class="math inline">\(G\)</span> + <span class="math inline">\(\epsilon\)</span>. The model intersects the
genotype groups at their group means, and is based <span class="math inline">\(\mu\)</span> and <span class="math inline">\(\beta_k\)</span>.</p>
<p>The effect of genotype RR (the <span class="math inline">\(\beta\)</span> for the RR genotype) at the marker
is 15, while the effect of the BB genotype is -15 on the phenotype. The
effect of the BR genotype is 0 relative to <span class="math inline">\(\mu\)</span> equals 94.6.</p>
<p>The <code>scan1()</code> function returns only LOD scores. To obtain
estimated QTL effects, use the function <code>scan1coef()</code>. This
function takes a single phenotype and the genotype probabilities for a
single chromosome and returns a matrix with the estimated coefficients
at each putative QTL location along the chromosome.</p>
<p>For example, to get the estimated QTL effects on chromosome 19 for
the insulin phenotype, we would provide the chromosome 19 genotype
probabilities and the insulin phenotype to the function
<code>scan1coef()</code> as follows:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>       <span class="op">&lt;-</span> <span class="st">"19"</span></span>
<span><span class="va">eff_chr19</span> <span class="op">&lt;-</span> <span class="fu">scan1coef</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                       pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                       kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                       addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>The result is a matrix of 50 positions <span class="math inline">\(\times\)</span> 3 genotypes. Additional columns
contain the sex and intercept values (<span class="math inline">\(\mu\)</span>).</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">eff_chr19</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                BB      BR     RR SexMale intercept
rs4232073  -0.0387 -0.0275 0.0661  -0.169      1.01
rs13483548 -0.0386 -0.0272 0.0658  -0.168      1.01
rs13483549 -0.0388 -0.0272 0.0660  -0.169      1.01
rs13483550 -0.0413 -0.0275 0.0688  -0.169      1.01
rs13483554 -0.0374 -0.0325 0.0698  -0.170      1.01
rs13483555 -0.0367 -0.0288 0.0655  -0.170      1.01</code></pre>
</div>
</section><section><h2 class="section-heading" id="plotting-founder-allele-effects-along-a-chromosome">Plotting Founder Allele Effects Along a Chromosome<a class="anchor" aria-label="anchor" href="#plotting-founder-allele-effects-along-a-chromosome"></a></h2>
<hr class="half-width"><p>To plot the QTL effects, use the <code>plot_coef()</code> function.
Add the LOD plot to the <code>scan1_output</code> argument to include a
LOD plot at the bottom.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x            <span class="op">=</span> <span class="va">eff_chr19</span>, </span>
<span>          map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>          legend       <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_coef-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The plot shows effect values on the y-axis and cM values on the
x-axis. The value of the intercept (<span class="math inline">\(\mu\)</span>) appears at the top. The effect of
the BR genotype is centered around zero, with the effects of the other
two genotypes above and below. We are usually not directly interested in
how the additive covariates change across the genome, but rather, how
the founder allele effects change.</p>
<p>To plot only the founder allele effects, use the argument
<code>columns</code> to indicate which coefficients to plot. Let’s look
at the columns which contain the founder allele effects.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">eff_chr19</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                BB      BR     RR SexMale intercept
rs4232073  -0.0387 -0.0275 0.0661  -0.169      1.01
rs13483548 -0.0386 -0.0272 0.0658  -0.168      1.01
rs13483549 -0.0388 -0.0272 0.0660  -0.169      1.01
rs13483550 -0.0413 -0.0275 0.0688  -0.169      1.01
rs13483554 -0.0374 -0.0325 0.0698  -0.170      1.01
rs13483555 -0.0367 -0.0288 0.0655  -0.170      1.01</code></pre>
</div>
<p>We would like to plot the columns “BB”, “BR”, and “RR”, which are in
columns 1 through 3. This is what we pass into the <code>columns</code>
argument.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">eff_chr19</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>          main    <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="va">chr</span>, <span class="st">"QTL effects and LOD scores"</span><span class="op">)</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_effects_chr19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="challenge-1-founder-allele-effects" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-founder-allele-effects" class="callout-inner">
<h3 class="callout-title">Challenge 1: Founder Allele Effects</h3>
<div class="callout-content">
<p>Looking at the plot above, which founder allele contributes to higher
insulin levels?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>The BTBR allele contributes to higher insulin levels because the “RR”
allele effect (green line) is higher than the other allele effect
lines.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="estimating-founder-allele-effects-using-blups">Estimating Founder Allele Effects using BLUPs<a class="anchor" aria-label="anchor" href="#estimating-founder-allele-effects-using-blups"></a></h2>
<hr class="half-width"><p>Another option for estimating the founder allele effects is to treat
them as <a href="https://stats.stackexchange.com/questions/4700/what-is-the-difference-between-fixed-effect-random-effect-and-mixed-effect-mode#151800" class="external-link">random
effects</a> and calculate <a href="https://en.wikipedia.org/wiki/Best_linear_unbiased_prediction" class="external-link">Best
Linear Unbiased Predictors</a> (BLUPs). This is particularly valuable
for multi-parent populations such as the Collaborative Cross and
Diversity Outbred mice, where the large number of possible genotypes at
a QTL leads to considerable variability in the effect estimates. To
calculate BLUPs, use <code>scan1blup()</code>; it takes the same
arguments as <code>scan1coef()</code>, including the option of a kinship
matrix to account for a residual polygenic effect.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blup_chr19</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                        pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                        kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                        addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>We can plot the BLUP effects using <code>plot_coef</code> as
before.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">blup_chr19</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>          main    <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="va">chr</span>, <span class="st">"QTL BLUP effects and LOD scores"</span><span class="op">)</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_blup_chr19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot below, we plotted the founder allele effects (solid
lines) and the BLUPs (dashed lines). In this case, the effects are not
greatly different, but the effects are “shrunken” toward zero.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">eff_chr19</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          main    <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="va">chr</span>, <span class="st">"QTL BLUP effects and LOD scores"</span><span class="op">)</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span>
<span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">blup_chr19</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,</span>
<span>          lty     <span class="op">=</span> <span class="fl">2</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span>,</span>
<span>          add     <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_blup_chr19_again-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></section><section><h2 class="section-heading" id="plotting-allele-effects-at-one-marker">Plotting Allele Effects at One Marker<a class="anchor" aria-label="anchor" href="#plotting-allele-effects-at-one-marker"></a></h2>
<hr class="half-width"><p>You may also want plot the founder allele effects at the marker with
the highest LOD. To do this, you first need to get the position of the
marker from the peak list.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>,</span>
<span>                    map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>                    threshold    <span class="op">=</span> <span class="va">thr</span><span class="op">)</span></span>
<span><span class="va">peaks</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr   pos  lod
1        1 log10_insulin_10wk   2 138.9 7.13
2        1 log10_insulin_10wk   7 144.2 5.72
3        1 log10_insulin_10wk  12  25.1 4.31
4        1 log10_insulin_10wk  14  22.2 3.97
5        1 log10_insulin_10wk  16  80.4 4.11
6        1 log10_insulin_10wk  19  54.8 5.48</code></pre>
</div>
<p>The position of the maximum LOD on chromosome 19 is 54.83 Mb. We can
pass this value into the <code>qtl2</code> function
<code>pull_genoprobpos</code> to get the genoprobs at this marker.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">max_pos</span> <span class="op">&lt;-</span> <span class="fu">subset</span><span class="op">(</span><span class="va">peaks</span>, <span class="va">chr</span> <span class="op">==</span> <span class="st">'19'</span><span class="op">)</span><span class="op">$</span><span class="va">pos</span></span>
<span><span class="va">max_mkr</span> <span class="op">&lt;-</span> <span class="fu">find_marker</span><span class="op">(</span>map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>                       chr <span class="op">=</span> <span class="va">chr</span>, </span>
<span>                       pos <span class="op">=</span> <span class="va">max_pos</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pr</span>      <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                            marker    <span class="op">=</span> <span class="va">max_mkr</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-2-structure-of-genoprobs-at-one-marker" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-structure-of-genoprobs-at-one-marker" class="callout-inner">
<h3 class="callout-title">Challenge 2: Structure of Genoprobs at One Marker</h3>
<div class="callout-content">
<p>What does the structure of <code>pr</code> look like? How many rows
and columns does it have?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>The <a href="https://www.rdocumentation.org/packages/utils/versions/3.6.2/topics/str" class="external-link">str</a>
function provides the structure of an R object.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">str</span><span class="op">(</span><span class="va">pr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> num [1:490, 1:3] 1.09e-08 1.09e-08 3.06e-14 1.09e-08 1.09e-08 ...
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:490] "Mouse3051" "Mouse3551" "Mouse3430" "Mouse3476" ...
  ..$ : chr [1:3] "BB" "BR" "RR"</code></pre>
</div>
<p>In this case, we can see that <code>pr</code> is a numeric matrix
because the first line starts with “num”. The next item on the first
line provides the dimensions of the object. <code>pr</code> is a matrix
because there are two dimensions, rows and columns. There are 490 rows
and 3 columns in <code>pr</code>.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3-genoprobs-at-one-marker" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-genoprobs-at-one-marker" class="callout-inner">
<h3 class="callout-title">Challenge 3: Genoprobs at One Marker</h3>
<div class="callout-content">
<p>Look at the top few rows of <code>pr</code>. You may want to round
the values to make them easier to read. What is the genotype of
<code>Mouse3051</code>? What about <code>Mouse3430</code>?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>We can use the <a href="">head</a> function to see the top few rows
of <code>pr</code>.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="fu">round</span><span class="op">(</span><span class="va">pr</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          BB BR RR
Mouse3051  0  1  0
Mouse3551  0  1  0
Mouse3430  0  0  1
Mouse3476  0  1  0
Mouse3414  0  1  0
Mouse3145  0  1  0</code></pre>
</div>
<p>From the listing above, the probability that Mouse3051 carries the
“BR” genotype at this marker is close to 1, so this is the most likely
genotype.</p>
<p>the probability that Mouse3430 carries the “RR” genotype at this
marker is close to 1.</p>
</div>
</div>
</div>
</div>
<p><code>pr</code> is a numeric matrix with 490 rows and 3 columns. The
rownames contain the mouse IDs and the column names contain the
genotypes.</p>
<p>We can then pass <code>pr</code> as an argument into the
<code>fit1</code> function, which fits the mapping model at a single
marker.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod</span> <span class="op">=</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr</span>,</span>
<span>           pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>           kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>           addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>Then, we can plot the founder allele effects and their standard
error.</p>

<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod_eff</span> <span class="op">=</span> <span class="fu">data.frame</span><span class="op">(</span>eff <span class="op">=</span> <span class="va">mod</span><span class="op">$</span><span class="va">coef</span>, </span>
<span>                     se  <span class="op">=</span> <span class="va">mod</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"genotype"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">filter</span><span class="op">(</span><span class="va">genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"BB"</span>, <span class="st">"BR"</span>, <span class="st">"RR"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data    <span class="op">=</span> <span class="va">mod_eff</span>, </span>
<span>       mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">genotype</span>, y <span class="op">=</span> <span class="va">eff</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_pointrange</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>ymin <span class="op">=</span> <span class="va">eff</span> <span class="op">-</span> <span class="va">se</span>, </span>
<span>                                ymax <span class="op">=</span> <span class="va">eff</span> <span class="op">+</span> <span class="va">se</span><span class="op">)</span>,</span>
<span>                  size       <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>                  linewidth  <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Founder Allele Effects on Chr"</span>, <span class="va">chr</span><span class="op">)</span>,</span>
<span>       x     <span class="op">=</span> <span class="st">"Genotype"</span>, y <span class="op">=</span> <span class="st">"Founder Allele Effects"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_fit1-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p><img src="./fig/mapping-workflow-estimate-effects.png" alt="A diagram showing mapping steps up to estimating QTL effects." class="figure">
::::::::::::::::::::::::::::::::::::: challenge</p>
</section><section><h2 class="section-heading" id="challenge-4-allele-effects">Challenge 4: Allele Effects<a class="anchor" aria-label="anchor" href="#challenge-4-allele-effects"></a></h2>
<hr class="half-width"><ol style="list-style-type: decimal"><li>In the plot above, which founder allele contributes to higher
insulin levels?</li>
<li>Is that consistent with the plot created using
<code>plot_coef</code> above?</li>
</ol><div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<ol style="list-style-type: decimal"><li>The R (BTBR) allele contributes to higher insulin levels on
chromosome 19.</li>
<li>Yes, the BTBR allele also contributed to higher insulin level in the
founder allele effects plot.</li>
</ol></div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="challenge-5-allele-effects-on-chromosome-7">Challenge 5: Allele Effects on Chromosome 7<a class="anchor" aria-label="anchor" href="#challenge-5-allele-effects-on-chromosome-7"></a></h2>
<hr class="half-width"><ol style="list-style-type: decimal"><li>Use <code>scan1_blup</code> and <code>plot_coef</code> to look at
the founder allele effects at the chromosome 7 QTL peak. Does the
pattern of allele effects look different? Which founder allele
contributes to higher insulin levels?</li>
<li>Get the genoprobs at the marker with the highest LOD on chromosome 7
and use <code>fit1</code> and the plotting code for <code>fit1</code>
above to examine the founder allele effects at the peak marker. Again,
how do these allele effects differ from the allele effects on chromosome
19? Which founder allele contributes to higher insulin levels?</li>
</ol><p>Be careful to rename the ouput objects so that you don’t overwrite
the values from chromosome 19.</p>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">
<ol style="list-style-type: decimal"><li>We will use <code>scan1_blup</code> and <code>plot_coef</code> to
estimate the founder allele effects on chromosome 7.</li>
</ol><div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blup_chr7</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="st">"7"</span><span class="op">]</span>,</span>
<span>                       pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                       kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="st">"7"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                       addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">blup_chr7</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>          main    <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="st">"7"</span>, <span class="st">"QTL BLUP effects and LOD scores"</span><span class="op">)</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-challenge5a-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>On chromosome 19, the BB and BR genotypes seems to have the same
effect. On chromosome 7, there is a clear split between the BB, BR, and
RR genotypes. Each additional B allele raises insulin levels. The
C57BL/6J (B) allele contributes to higher insulin levels.</p>
<ol start="2" style="list-style-type: decimal"><li>We will get the marker with the highest LOD on chromosome 7, then
get the genoprobs at that marker and use them in <code>fit1</code> to
estimate the founder allele effects.</li>
</ol><div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">max_pos</span> <span class="op">&lt;-</span> <span class="fu">subset</span><span class="op">(</span><span class="va">peaks</span>, <span class="va">chr</span> <span class="op">==</span> <span class="st">"7"</span><span class="op">)</span><span class="op">$</span><span class="va">pos</span></span>
<span><span class="va">max_mkr</span> <span class="op">&lt;-</span> <span class="fu">find_marker</span><span class="op">(</span>map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>                       chr <span class="op">=</span> <span class="st">"7"</span>, </span>
<span>                       pos <span class="op">=</span> <span class="va">max_pos</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pr7</span>     <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                            marker    <span class="op">=</span> <span class="va">max_mkr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod7</span> <span class="op">=</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr7</span>,</span>
<span>            pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>            kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="st">"7"</span><span class="op">]</span><span class="op">]</span>, </span>
<span>            addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod_eff7</span> <span class="op">=</span> <span class="fu">data.frame</span><span class="op">(</span>eff <span class="op">=</span> <span class="va">mod7</span><span class="op">$</span><span class="va">coef</span>, </span>
<span>                      se  <span class="op">=</span> <span class="va">mod7</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"genotype"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">filter</span><span class="op">(</span><span class="va">genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"BB"</span>, <span class="st">"BR"</span>, <span class="st">"RR"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data    <span class="op">=</span> <span class="va">mod_eff7</span>, </span>
<span>       mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">genotype</span>, y <span class="op">=</span> <span class="va">eff</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_pointrange</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>ymin <span class="op">=</span> <span class="va">eff</span> <span class="op">-</span> <span class="va">se</span>, </span>
<span>                                ymax <span class="op">=</span> <span class="va">eff</span> <span class="op">+</span> <span class="va">se</span><span class="op">)</span>,</span>
<span>                  size       <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>                  linewidth  <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Founder Allele Effects on Chr"</span>, <span class="st">"7"</span><span class="op">)</span>,</span>
<span>       x     <span class="op">=</span> <span class="st">"Genotype"</span>, y <span class="op">=</span> <span class="st">"Founder Allele Effects"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-challenge5d-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>On chromosome 19, mice carrying the BB and BR genotypes had almost
the same levels of insulin. On chromosome 7, the BB genotype has the
highest insulin levels, the BR genotype has the next highest, and the RR
genotype has the lowest insulin levels.</p>
</div>
</div>
</div>
</div>
<p>::::::::::::::::::::::::::::::::::::::::::::::::</p>
</section><section><h2 class="section-heading" id="estimating-additive-and-dominant-allele-effects">Estimating Additive and Dominant Allele Effects<a class="anchor" aria-label="anchor" href="#estimating-additive-and-dominant-allele-effects"></a></h2>
<hr class="half-width"><p>You may have noticed the difference in the pattern of allele effects
between chromosome 7 and 19. On chromosome 7, adding each “B” allele
increased insulin levels by the same amount. On chromosome 19, adding
one “R” allele did not change insulin levels. We saw an increase in
insulin only when we added two “R” alleles.</p>
<p>What we are seeing between these two QTL are “additive” and
“dominant” effects. At an additive QTL (which is different from additive
covariates), adding (or subtracting) one allele produces the same effect
between each genotype. At a dominant QTL, adding (or subtracting) two
alleles is required to see the effect. You can also have QTL which are a
mixture of additive and dominant effects.</p>
<p>We can map the additive and dominance effects by providing a matrix
of <em>contrasts</em> to indicate the different effects. Let’s look at
the additive and dominant effects on chromosome 7 first.</p>
<figure><img src="fig/add_dom_qtl_effect.png" alt="Figure showing additive &amp; dominance QTL effects" class="figure mx-auto d-block"><div class="figcaption">Additive &amp; Dominance Effects</div>
</figure><p>In the figure above, we have simulated a QTL with a dominant effect.
Genotypes are shown on the X-axis and the phenotype on the Y-axis. Each
point represents the phenotype value for one mouse. The grey line is the
mean of the two homozygote groups. The additive effect is defined as the
difference between the homozygote mean (grey line) and the individual
homozygote group means (red lines). The blue line shows the heterozygote
mean. The dominance effect is the difference between the homozygote mean
(grey line) and the heterozygote mean (blue line).</p>
<p>First, we create a <em>contrasts</em> matrix to indicate the mean,
additive, and dominance effects that we want to estimate.</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">add_dom_contr</span> <span class="op">&lt;-</span> <span class="fu">cbind</span><span class="op">(</span>mu <span class="op">=</span> <span class="fu">c</span><span class="op">(</span> <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                       a  <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                       d  <span class="op">=</span> <span class="fu">c</span><span class="op">(</span> <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">c7effB</span> <span class="op">&lt;-</span> <span class="fu">scan1coef</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="st">"7"</span><span class="op">]</span>, </span>
<span>                    pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                    kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="st">"7"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                    addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                    contrasts <span class="op">=</span> <span class="va">add_dom_contr</span><span class="op">)</span></span></code></pre>
</div>
<p>The result will then contain the estimates of <code>mu</code>,
<code>a</code> (the additive effect), and <code>d</code> (the dominance
effect).</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">c7effB</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 109   4</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">c7effB</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             mu       a        d SexMale
rs8252589  1.01 -0.0267 -0.00378  -0.175
rs13479104 1.01 -0.0228  0.00409  -0.175
rs13479112 1.01 -0.0234  0.00991  -0.175
rs13479114 1.00 -0.0226  0.01148  -0.175
rs13479120 1.01 -0.0238  0.00899  -0.175
rs13479124 1.01 -0.0219  0.00375  -0.175</code></pre>
</div>
<p>For marker rs13479570, <code>mu</code>, <code>a</code>, and
<code>d</code> are 1.016, -0.111, -0.03, -0.164.</p>
<p>Here’s a plot of the chromosome 7 additive and dominance effects,
which are in the second and third columns.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">c7effB</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">[</span><span class="st">"7"</span><span class="op">]</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          col     <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,</span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>,</span>
<span>          main    <span class="op">=</span> <span class="st">"log10(insulin): Chromosome 7"</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span><span class="st">'bottomleft'</span>, lwd <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, legend <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"additive"</span>, <span class="st">"dominance"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_add_dom_contrasts-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot above, the dominance effects remain near zero across the
chromosome. The additive effects decrease as we move</p>
<div id="challenge-6-estimating-additive-and-dominance-effects-on-chromsome-19" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-estimating-additive-and-dominance-effects-on-chromsome-19" class="callout-inner">
<h3 class="callout-title">Challenge 6: Estimating Additive and Dominance Effects on Chromsome 19</h3>
<div class="callout-content">
<p>Use the code above to estimate and plot the additive and dominance
effects along chromosome 19. You can use the same contrasts matrix as
before. How dow they differ from the effects on chromosome 7?</p>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">
<p>First we estimate the effects using <code>scan1coef</code>.</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">c19effB</span> <span class="op">&lt;-</span> <span class="fu">scan1coef</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="st">"19"</span><span class="op">]</span>, </span>
<span>                     pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                     kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="st">"19"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                     addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                     contrasts <span class="op">=</span> <span class="va">add_dom_contr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">c19effB</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">[</span><span class="st">"19"</span><span class="op">]</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          col     <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span>,</span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>,</span>
<span>          main    <span class="op">=</span> <span class="st">"log10(insulin): Chromosome 19"</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span><span class="st">'bottomleft'</span>, lwd <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span>, </span>
<span>       legend <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"additive"</span>, <span class="st">"dominance"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-challenge6b-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In this case, we see that the additive effect adds about 0.094 and
the dominance effect adds about -0.113.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="plotting-phenotypes-versus-genotypes">Plotting Phenotypes versus Genotypes<a class="anchor" aria-label="anchor" href="#plotting-phenotypes-versus-genotypes"></a></h2>
<hr class="half-width"><p>Finally, to plot the raw phenotypes against the genotypes at a single
putative QTL position, you can use the function <code>plot_pxg()</code>.
This takes a vector of genotypes as produced by the
<code>maxmarg()</code> function, which picks the most likely genotype
from a set of genotype probabilities, provided it is greater than some
specified value (the argument <code>minprob</code>). Note that the
<code>marg</code> in <code>maxmarg</code> stands for “marginal”, as this
function is selecting the genotype at each position that has maximum
marginal probability.</p>
<p>For example, we could get inferred genotypes at the chr 19 QTL for
the insulin phenotype (at 28.6 cM) as follows:</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu">maxmarg</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>             map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>             chr   <span class="op">=</span> <span class="va">chr</span>, </span>
<span>             pos   <span class="op">=</span> <span class="va">max_pos</span>, </span>
<span>             return_char <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<p>We use <code>return_char = TRUE</code> to have <code>maxmarg()</code>
return a vector of character strings with the genotype labels.</p>
<p>We then plot the insulin phenotype against these genotypes as
follows:</p>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_pxg</span><span class="op">(</span>geno   <span class="op">=</span> <span class="va">g</span>, </span>
<span>         pheno  <span class="op">=</span> <span class="va">insulin</span>, </span>
<span>         SEmult <span class="op">=</span> <span class="fl">2</span>,</span>
<span>         main   <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Insulin vs Chr"</span>, <span class="va">chr</span> ,<span class="st">"Genotype"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="fig/est-qtl-effects-rendered-plot_pheno_geno_se-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>Estimated founder allele effects can be plotted from the mapping
model coefficients.</li>
<li>Additive and dominance effects can be plotted using contrasts.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="find-lod-peaks.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="integrating-gene-expression.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="find-lod-peaks.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Finding QTL peaks
        </a>
        <a class="chapter-link float-end" href="integrating-gene-expression.html" rel="next">
          Next: Integrating Gene...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/est-qtl-effects.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.12" class="external-link">sandpaper (0.16.12)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.6" class="external-link">varnish (1.0.6)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "https://carpentries.github.io/workbench-template-rmd/est-qtl-effects.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries, genetics, statistics, genome, genetic mapping",
  "name": "Estimating QTL effects",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/est-qtl-effects.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/est-qtl-effects.html",
  "dateCreated": "2024-09-06",
  "dateModified": "2025-07-29",
  "datePublished": "2025-07-29"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

