<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Quantitative Trait Mapping: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/incubator/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicons/incubator/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicons/incubator/favicon-16x16.png">
<link rel="manifest" href="../favicons/incubator/site.webmanifest">
<link rel="mask-icon" href="../favicons/incubator/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="../assets/images/incubator-logo.svg"><span class="badge text-bg-danger">
          <abbr title="This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#early-development-pre-alpha-through-alpha" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-octagon" style="border-radius: 5px"></i>
              Pre-Alpha
            </a>
            <span class="visually-hidden">This lesson is in the pre-alpha phase, which means that it is in early development, but has not yet been taught.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../aio.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="../assets/images/incubator-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Quantitative Trait Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Quantitative Trait Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Quantitative Trait Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../aio.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="intro_data_set.html">1. Introduction to the Data Set</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="input-file-format.html">2. Input File Format</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="calc-genoprob.html">3. Calculating Genotype Probabilities</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="perform-genome-scan.html">4. Performing a Genome Scan</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="calc-kinship.html">5. Calculating A Kinship Matrix</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="perform-genome-scan-lmm.html">6. Performing a genome scan with a linear mixed model</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="perform-genome-scan-bin.html">7. Performing a Genome Scan with Binary Traits</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="perform-perm-test.html">8. Finding Significant Peaks via Permutation</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="find-lod-peaks.html">9. Finding QTL peaks</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="est-qtl-effects.html">10. Estimating QTL effects</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="integrating-gene-expression.html">11. Integrating Gene Expression Data</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush13">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading13">
        <a href="do_qtl_mapping.html">12. QTL Mapping in Diversity Outbred Mice</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-intro_data_set"><p>Content from <a href="intro_data_set.html">Introduction to the Data Set</a></p>
<hr>
<p>Last updated on 2024-11-13 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/intro_data_set.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 20 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What data will we be using in this workshop?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the experimental design of the data set.</li>
<li>Understand the goals of the experiment.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>In the first part of this lesson, we will be analyzing data from a
mouse experiment involving Type 2 diabetes (T2D). There are two types of
diabetes: type 1, in which the immune system attacks insulin-secreting
cells and prevents insulin production, and type 2, in which the pancreas
makes less insulin and the body becomes less responsive to insulin.</p>
<p><img src="../fig/healthy-vs-T2D.png" alt="Figure showing Type 2 diabetes &amp; insulin." class="figure"> Created in
BioRender.com</p>
<p>This study is from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4649649/" class="external-link">Tian et
al</a> and involves an intercross between the diabetes-resistant
C57BL/6J (B6 or B) strain and the diabetes-susceptible BTBR T+ tf/J
(BTBR or R) strain mice carrying a Leptin<sup>ob/ob</sup> mutation.</p>
<figure><img src="../fig/intercross.png" alt="Figure showing intercross breeding design." class="figure mx-auto d-block"></figure><p>The <ob> mutation prevents the production of <a href="https://en.wikipedia.org/wiki/Leptin" class="external-link">leptin</a>, a hormone that
regulates hunger and satiety. When leptin levels are low (or missing),
the body does not receive satiety signals and continues to feel hunger.
Leptin<sup>ob/ob</sup> mice continue to eat and become obese. Obesity is
one of the risk factors for T2D and this experiment sought to use
genetic variation between B6 and BTBR strains to identify genes which
influence T2D.</ob></p>
<p>This study measured insulin and glucose levels in mice at 10 weeks,
at which time the mice were euthanized. After euthanasia, the authors
harvested six tissues, adipose, gastrocnemius muscle, hypothalamus,
pancreatic islets, kidney, and liver, and measured transcript levels via
gene expression microarray.</p>
<p>In this study, we will analyze circulating insulin levels and
pancreatic islet gene expression. We will map circulating insulin levels
to identify genomic loci which influence insulin levels. We will then
use SNPs that differ between C57BL/6J and BTBR and pancreatic islet gene
expression data to identify candidate genes.</p>
<div id="challenge-1-research-question-and-study-design" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-research-question-and-study-design" class="callout-inner">
<h3 class="callout-title">Challenge 1: Research question and study design</h3>
<div class="callout-content">
<p>Turn to a partner and describe:<br>
1. the research question that the study addresses, and<br>
2. how the study is designed to address this question.</p>
<p>Share your description with your partner, and then listen to them
describe their understanding of the study. When you are finished, write
your responses into the collaborative document.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Leptin<sup>ob/ob</sup> mice do now produce insulin and become obese
due to overeating.</li>
<li>This study crossed mice carrying the Leptin<sup>ob/ob</sup> mutation
in C57BL/6J and BTBR T+ tf/J.</li>
<li>C57BL/6J mice are resistant to diabetes and BTBR mice are
susceptible.</li>
<li>By crossing these two strains, the authors aimed to identify genes
which influence susceptibility to T2D.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-input-file-format"><p>Content from <a href="input-file-format.html">Input File Format</a></p>
<hr>
<p>Last updated on 2024-11-13 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/input-file-format.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 25 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How are the data files formatted for qtl2?</li>
<li>Which data files are required for qtl2?</li>
<li>Where can I find sample data for mapping with the qtl2 package?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>To specify which input files are required for qtl2 and how they
should be formatted.</li>
<li>To locate sample data for qtl mapping.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>QTL mapping data consists of a set of tables of data: sample
genotypes, phenotypes, marker maps, etc. These different tables are in
different comma-separated value (CSV) files. In each file, the first
column is a set of IDs for the rows, and the first row is a set of IDs
for the columns. For example, the genotype data file will have
individual IDs in the first column, marker names for the rest of the
column headers.</p>
<figure><img src="../fig/attie_geno_sample.png" alt="Table showing the mouse genotypes as BB, BR, and RR." class="figure mx-auto d-block"><div class="figcaption">Attie Sample Genotypes</div>
</figure><p>The sample genotype file above shows two alleles: B and R. These
represent the founder strains for an intercross, which are C57BL/6 (BB)
and BTBR (RR) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4649649/" class="external-link">Tian et
al</a>. The B and R alleles themselves represent the haplotypes
inherited from the parental strains C57BL/6 and BTBR.</p>
<p>For the purposes of learning QTL mapping, this lesson begins with an
intercross that has only 3 possible genotypes instead of 8 or 36. Once
we have learned how to use <code>qtl2</code> for the simpler case, we
will advance to the most complex case involving mapping in DO mice.</p>
<p>R/qtl2 accepts the following files:<br>
1. genotypes<br>
2. phenotypes<br>
3. phenotype covariates (<em>i.e.</em> tissue type, time points)<br>
4. genetic map<br>
5. physical map (optional)<br>
6. control file (YAML or JSON format, not CSV).</p>
<p>We use both a genetic marker map and a physical map (if available). A
sample from a genetic map of SNP markers is shown here.</p>
<figure><img src="../fig/attie_geno_map_sample.png" alt="Table showing the marker, chromosome, and centimorgan position for five markers" class="figure mx-auto d-block"><div class="figcaption">Attie Genetic Map</div>
</figure><p>A physical marker map provides location in bases rather than
centiMorgans.</p>
<figure><img src="../fig/attie_phys_map_sample.png" alt="Table showing top five rows of physical marker map." class="figure mx-auto d-block"><div class="figcaption">Attie Physical Map</div>
</figure><p>Numeric phenotypes are separate from the often non-numeric
covariates.</p>
<figure><img src="../fig/attie_pheno_sample.png" alt="Table showing top five rows of phenotype table, including insulin" class="figure mx-auto d-block"><div class="figcaption">Attie Phenotypes</div>
</figure><p>Phenotype covariates are <a href="https://en.wikipedia.org/wiki/Metadata" class="external-link">metadata</a> describing
the phenotypes. For example, in the case of a phenotype measured over
time, one column in the phenotype covariate data could be the time of
measurement. For gene expression data, we would have columns
representing chromosome and physical position of genes, as well as gene
IDs. The covariates shown below include sex and parental grandmother
(pgm).</p>
<figure><img src="../fig/attie_covar_sample.png" alt="Table showing the top five rows of covariates table" class="figure mx-auto d-block"><div class="figcaption">Attie Covariates</div>
</figure><p>In addition to the set of CSV files with the primary data, we need a
separate control file with various control parameters (or metadata),
including the names of all of the other data files and the genotype
codes used in the genotype data file. The control file is in a specific
format using either <a href="https://www.yaml.org" class="external-link">YAML</a> or <a href="https://json.org" class="external-link">JSON</a>; these are human-readable text files
for representing relatively complex data.</p>
<figure><img src="../fig/attie_control_file_sample.png" alt="Figure showing the qtl2 control file" class="figure mx-auto d-block"><div class="figcaption">Attie Control File</div>
</figure><p>A big advantage of this control file scheme is that it greatly
simplifies the function for reading in the data. That function,
<code>read_cross2()</code>, has a <em>single</em> argument: the name
(with path) of the control file.</p>
<p>For further details, see the separate <a href="https://kbroman.org/qtl2/assets/vignettes/input_files.html" class="external-link">vignette
on the input file format</a>.</p>
<div id="challenge-1-what-data-does-qtl2-need" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-what-data-does-qtl2-need" class="callout-inner">
<h3 class="callout-title">Challenge 1: What data does qtl2 need?</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Which data files are required by <code>qtl2</code>?<br>
</li>
<li>Which ones are optional?<br>
</li>
<li>How should they be formatted?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>genotypes, phenotypes, genetic map</li>
<li>physical map</li>
<li>csv; JSON or YAML for control file</li>
</ol>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="sample-data-sets">Sample data sets<a class="anchor" aria-label="anchor" href="#sample-data-sets"></a>
</h2>
<hr class="half-width">
<p>In this lesson, we will not work with data sets included in the
<code>qtl2</code> package, though you may want to explore them to learn
more. You can find out more about the <a href="https://kbroman.org/qtl2/pages/sampledata.html" class="external-link">sample data
files</a> from the R/qtl2 web site. Zipped versions of these datasets
are included with the <a href="https://github.com/rqtl/qtl2geno" class="external-link">qtl2geno</a> package and can be
loaded into R using the <code>read_cross2()</code> function. Additional
sample data sets, including data on Diversity Outbred (DO) mice, are
available at <a href="https://github.com/rqtl/qtl2data" class="external-link uri">https://github.com/rqtl/qtl2data</a>.</p>
<div id="challenge-2-additional-rqtl2-datasets" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-additional-rqtl2-datasets" class="callout-inner">
<h3 class="callout-title">Challenge 2: Additional R/qtl2 datasets</h3>
<div class="callout-content">
<p>Go to <a href="https://github.com/rqtl/qtl2data" class="external-link uri">https://github.com/rqtl/qtl2data</a> to view additional
sample data.<br>
1). Find the Recla data and locate the phenotype data file. Open the
file by clicking on the file name. What is in the first column? the
first row?<br>
2). Locate the genotype data file, click on the file name, and view the
raw data. What is in the first column? the first row?<br>
3). Locate the covariates file and open it by clicking on the file name.
What kind of information does this file contain?<br>
4). Locate the control file (YAML or JSON format) and open it. What kind
of information does this file contain?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>1). What is in the first column of the phenotype file? Animal ID. The
first row? Phenotype variable names - OF_distance_first4, OF_distance,
OF_corner_pct, OF_periphery_pct, …<br>
2). What is in the first column of the genotype file? marker ID. the
first row? Animal ID - 1,4,5,6,7,8,9,10, …<br>
3). Locate the covariates file and open it. What kind of information
does this file contain? Animal ID, sex, cohort, group, subgroup, ngen,
and coat color.<br>
4). Locate the control file (YAML or JSON format) and open it. What kind
of information does this file contain? Names of primary data files,
genotype and allele codes, cross type, description, and other
metadata.</p>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>QTL mapping data consists of a set of tables of data: genotypes,
phenotypes, marker maps, etc.</li>
<li>These different tables are in separate comma-delimited (CSV)
files.</li>
<li>In each file, the first column is a set of IDs for the rows, and the
first row is a set of IDs for the columns.</li>
<li>In addition to primary data, a separate file with control parameters
(or metadata) in either <a href="https://www.yaml.org" class="external-link">YAML</a> or <a href="https://json.org" class="external-link">JSON</a> format is required.</li>
<li>Published and public data already formatted for QTL mapping are
available on the web.</li>
<li>These data can be used as a model for formatting your own QTL
data.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-calc-genoprob"><p>Content from <a href="calc-genoprob.html">Calculating Genotype Probabilities</a></p>
<hr>
<p>Last updated on 2024-11-13 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/calc-genoprob.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 35 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I calculate QTL at positions between genotyped markers?</li>
<li>How do I calculate QTL genotype probabilities?</li>
<li>How do I calculate allele probabilities?</li>
<li>How can I speed up calculations if I have a large data set?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>To explain why the first step in QTL analysis is to calculate
genotype probabilities.</li>
<li>To calculate genotype probabilities.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>The first task in QTL analysis is to calculate conditional genotype
probabilities, given the observed marker data, at each putative QTL
position. For example, the first step would be to determine the
probabilities for genotypes BR and RR at the locus indicated below.</p>
<figure><img src="../fig/unknown_genotype.png" alt="a chromosome with two typed markers labeled BR and RR with a locus of unknown genotype between them" class="figure mx-auto d-block"><div class="figcaption">Genotype probabilities must be calculated
between typed markers; adapted from Broman &amp; Sen, 2009</div>
</figure><p>The <code>calc_genoprob()</code> function calculates QTL genotype
probabilities conditional on the available marker data. These are needed
for most of the QTL mapping functions. The result is returned as a list
of three-dimensional arrays (one per chromosome). Each 3d array of
probabilities is arranged as individuals <span class="math inline">\(\times\)</span> genotypes <span class="math inline">\(\times\)</span> positions.</p>
<figure><img src="../fig/threeD_array.png" alt="Figure showing three-dimensional array of genotype probabilities (genoprobs)" class="figure mx-auto d-block"><div class="figcaption">Three-dimensional genotype probabilities
array</div>
</figure><p><img src="https://devopedia.org/images/article/46/9213.1526125966.png" alt="a web page showing R data structures including one-dimensional vectors and lists, two dimensional dataframes and matrices, and n-dimensional arrays" class="figure">
Notice that arrays in R require data to be all of the same type - all
numeric, all character, all Boolean, etc. If you are familiar with data
frames in R you know that you can mix different kinds of data in that
structure. The first column might contain numeric data, the second
column character data, the third Boolean (True / False), and so on.
Arrays won’t accept mixed data types though.</p>
<p>We’ll use the <a href="https://thejacksonlaboratory.box.com/shared/static/svw7ivp5hhmd7vb8fy26tc53h7r85wez.zip" class="external-link">Attie
BL6/BTBR dataset</a> from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4649649/" class="external-link">Tian et
al</a> (an intercross) as an example. In this study, circulating insulin
levels were measured in an F2 cross between mouse strains C57BL/6J and
BTBTR T+ <tf>.</tf></p>
<p>First, we will load in the <a href="https://kbroman.org/qtl2/" class="external-link">qtl2</a> library, which provides the
functions that we will use for QTL analysis.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span></span></code></pre>
</div>
<p>The function <code>read_cross2()</code> has a single argument: the
name (with path) of the control file, or alternatively a zip file
containing all the required data. We read in the data with a JSON
control file like this:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cross</span> <span class="op">&lt;-</span> <span class="fu">read_cross2</span><span class="op">(</span>file <span class="op">=</span> <span class="st">'data/attie_control.json'</span><span class="op">)</span></span></code></pre>
</div>
<div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor1" aria-labelledby="headingInstructor1">
<div class="accordion-body">
<p>We need the following block for the site to build on Github. The
students do not need to see or run the next block.</p>
</div>
</div>
</div>
</div>
<p>To load your own data from your machine, you would use the file path
to your data files. For example, if the file path to your data files is
<code>/Users/myUserName/qtlProject/data</code>, the command to load your
data would look like this:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myQTLdata</span> <span class="op">&lt;-</span> <span class="fu">read_cross2</span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/Users/myUserName/qtlProject/data/myqtldata.json"</span> <span class="op">)</span></span></code></pre>
</div>
<p>The JSON file contains all control information for your data,
including names of data files, cross type, column specifications for sex
and cross information, and more. This can also be in YAML format.
Alternatively, all data files can be zipped together for loading.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myQTLdata</span> <span class="op">&lt;-</span> <span class="fu">read_cross2</span><span class="op">(</span>file <span class="op">=</span> <span class="st">"/Users/myUserName/qtlProject/data/myqtldata.zip"</span> <span class="op">)</span></span></code></pre>
</div>
<p>Back to the BTBR data. Now look at a summary of the cross data and
the names of each variable within the data.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">cross</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Object of class cross2 (crosstype "f2")

Total individuals             490
No. genotyped individuals     490
No. phenotyped individuals    490
No. with both geno &amp; pheno    490

No. phenotypes                  3
No. covariates                  8
No. phenotype covariates        0

No. chromosomes                20
Total markers                2057

No. markers by chr:
  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19   X
156 135 157 126 125 102 109  91  93 123 124 116 116  91 102  66  60  95  50  20 </code></pre>
</div>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">cross</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] "crosstype"  "geno"       "gmap"       "pmap"       "pheno"
 [6] "covar"      "is_x_chr"   "is_female"  "cross_info" "alleles"   </code></pre>
</div>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p>1). How many mice are in this study?<br>
2). How many phenotypes are there?<br>
3). How many markers?<br>
4). How many markers are on chr 11?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>The output of <code>summary(cross)</code> provides this
information.<br>
1). There are 490 individuals in the cross.<br>
2). 3 phenotypes<br>
3). 2,057 markers<br>
4). 124 markers on chromosome 11</p>
</div>
</div>
</div>
</div>
<p>Have a look at the markers listed in the genetic map,
<code>gmap</code>. Markers are listed by chromosome and described by cM
position. View only the markers on the first two chromosomes.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">gmap</span>, n<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>$`1`
rs13475697  rs3681603 rs13475703 rs13475710  rs6367205 rs13475716 rs13475717
 0.1881141  0.1920975  0.4167755  0.6488793  0.6555814  0.6638576  0.6676198
rs13475719 rs13459050  rs3680898 rs13475727 rs13475728 rs13475729 rs13475731
 0.6711377  0.6749344  0.6775292  1.8149573  1.9596637  2.3456569  2.7186389
rs13475737 rs13475744  rs6397513 rs13475747 rs13475748 rs13475749 rs13475750
 3.1059517  3.8222865  4.3094607  4.3120150  4.5098582  4.8154609  4.8853505
rs13475751 rs13475752 rs13475762 rs13475764 rs13475765 rs13475768 rs13475769
 4.8869793  4.8902179  7.2954871  8.2102887  8.3708197  8.7178703  8.8859153
rs13475771  rs6384194 rs13475790  rs3676270 rs13475794 rs13475801  rs4222269
 9.1374722  9.9295192  9.9970634 10.1508878 10.3962716 11.5981956 11.9606369
 rs6387241 rs13475822 rs13475824 rs13475826 rs13475827 rs13475834 rs13475880
16.8770742 16.9815396 17.4434784 18.0866148 18.6276972 19.2288050 27.4056813
rs13475883  rs6239834  rs3162895  rs6212146  rs3022802 rs13475899 rs13475900
28.4641674 30.8427150 31.1526514 31.2751278 31.3428706 31.8493556 31.8518088
rs13475906  rs3022803 rs13475907 rs13475909 rs13475912  rs6209698 rs13475929
32.2967145 32.3074644 32.3683291 32.8001894 33.6026526 36.5341646 37.6881435
rs13475931 rs13475933 rs13475934  rs4222476 rs13475939  rs8253473 rs13475941
37.7429827 38.0416271 38.0430095 38.9647582 39.4116688 39.4192277 39.4871064
rs13475944 rs13475947 rs13475948 rs13475950 rs13475951 rs13475954 rs13475955
39.7672829 40.2599440 40.3380113 40.3417592 40.3439501 41.1407252 41.2887176
rs13475963 rs13475966 rs13475967 rs13475970 rs13475960  rs6250696 rs13475973
42.4744416 42.5667702 42.9736574 43.1427994 43.5985261 43.5992946 43.6014053
 rs3691187 rs13475988 rs13475991 rs13476023 rs13476024  rs3684654  rs6274257
44.6237384 45.7855528 46.0180221 47.8579278 47.8600317 48.2423958 48.9612178
rs13476045 rs13476049  rs6319405 rs13476050 rs13476051 rs13476054 rs13476057
49.2018340 49.3701384 49.4261039 49.4275718 49.4323558 49.4972616 49.5031830
rs13476059 rs13476060 rs13476062 rs13476066 rs13476067  rs6259837 rs13476080
49.5084008 49.5113545 49.6085043 49.6644819 50.1779477 50.8256056 51.0328603
 rs6302966 rs13476085 rs13476089  rs3717360 rs13476090  rs6248251 rs13476091
51.3659553 51.6974451 52.3869798 52.3903517 52.3936241 52.4228715 52.5787388
 rs3088725  rs3022832  rs4222577 rs13476100  rs6263067  rs8256168  rs6327099
53.4044231 53.4129004 53.4189013 54.3267003 54.4193890 55.1459517 55.3274320
rs13476111 rs13476119  rs8236484  rs8270838  rs8236489 rs13476129 rs13476134
55.9050491 56.8936305 56.9852502 57.1870637 58.0248893 58.7605079 59.5401544
rs13476135 rs13476137 rs13476138 rs13476140 rs13476148  rs6202860 rs13476158
59.5426193 59.6023794 60.3355828 60.3439598 61.1791787 61.9905512 61.9930265
rs13476163 rs13476177 rs13476178 rs13476183 rs13476184  rs6194543 rs13476196
62.0039607 62.6243588 62.6269118 63.8101331 64.0856907 66.4047817 66.7425394
rs13476201  rs3685700  rs3022846 rs13476210 rs13476214 rs13459163  rs4222816
67.2638714 68.7230251 68.7246243 69.1209547 70.1550813 75.5548371 75.5593190
 rs4222820  rs3090340  rs8245949 rs13476242 rs13476251 rs13476254  rs6383012
75.5593202 75.5637846 76.7508053 79.0157673 79.7644000 79.8248805 85.3173344
rs13476279  rs6348421 rs13476290 rs13476300 rs13476302 rs13476304  rs3669814
86.7653503 88.2128991 89.0565541 94.6215368 94.8227821 94.8269227 95.5413280
rs13501301 rs13476316
96.0784002 96.9960494 </code></pre>
</div>
<p>Next we use <code>calc_genoprob()</code> to calculate the QTL
genotype probabilities.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">probs</span> <span class="op">&lt;-</span> <span class="fu">calc_genoprob</span><span class="op">(</span>cross      <span class="op">=</span> <span class="va">cross</span>, </span>
<span>                       map        <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">gmap</span>, </span>
<span>                       error_prob <span class="op">=</span> <span class="fl">0.002</span><span class="op">)</span></span></code></pre>
</div>
<p>The argument <code>error_prob</code> supplies an assumed genotyping
error probability of 0.002. If a value for <code>error_prob</code> is
not supplied, the default probability is 0.0001.</p>
<p>Recall that the result of <code>calc_genoprob</code>,
<code>probs</code>, is a list of three-dimensional arrays (one per
chromosome).</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">names</span><span class="op">(</span><span class="va">probs</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] "1"  "2"  "3"  "4"  "5"  "6"  "7"  "8"  "9"  "10" "11" "12" "13" "14" "15"
[16] "16" "17" "18" "19" "X" </code></pre>
</div>
<p>Each three-dimensional array of probabilities is arranged as
individuals <span class="math inline">\(\times\)</span> genotypes <span class="math inline">\(\times\)</span> positions. Have a look at the
names of each of the three dimensions for chromosome 19.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="fu">dimnames</span><span class="op">(</span><span class="va">probs</span><span class="op">$</span><span class="va">`19`</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "Mouse3051" "Mouse3551" "Mouse3430" "Mouse3476" "Mouse3414" "Mouse3145"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dimnames</span><span class="op">(</span><span class="va">probs</span><span class="op">$</span><span class="va">`19`</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[[1]]
[1] "BB" "BR" "RR"</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dimnames</span><span class="op">(</span><span class="va">probs</span><span class="op">$</span><span class="va">`19`</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[[1]]
 [1] "rs4232073"  "rs13483548" "rs13483549" "rs13483550" "rs13483554"
 [6] "rs13483555" "rs3090321"  "rs3090137"  "rs6309315"  "rs13483577"
[11] "rs3090325"  "rs13483579" "rs13483584" "rs13483586" "rs13483587"
[16] "rs13483589" "rs13483592" "rs13483593" "rs6344448"  "rs13483594"
[21] "rs13483595" "rs3705022"  "rs13483609" "rs13483612" "rs13483648"
[26] "rs13483650" "rs13483654" "rs13483658" "rs13483660" "rs13483664"
[31] "rs13483666" "rs13483667" "rs13483670" "rs8275553"  "rs8275912"
[36] "rs13483677" "rs13483679" "rs13483680" "rs13483681" "rs3660143"
[41] "rs13483682" "rs13483683" "rs13483685" "rs13483686" "rs6355398"
[46] "rs4222106"  "rs13483690" "rs13483693" "rs13483695" "rs13483699"</code></pre>
</div>
<p>View the first three rows of genotype probabilities for the first
genotyped marker on chromosome 19.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">probs</span><span class="op">$</span><span class="va">`19`</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, , <span class="st">"rs4232073"</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                    BB           BR           RR
Mouse3051 1.317728e-11 1.235895e-07 9.999999e-01
Mouse3551 9.999840e-01 1.595361e-05 5.027172e-08
Mouse3430 1.317728e-11 1.235895e-07 9.999999e-01
Mouse3476 9.999999e-01 1.235895e-07 1.317728e-11
Mouse3414 6.179474e-08 9.999999e-01 6.179474e-08</code></pre>
</div>
<p>We can also view the genotype probabilities using <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_genoprob.R" class="external-link">plot_genoprob</a>.
The arguments to this function specify:</p>
<ol style="list-style-type: decimal">
<li>
<code>probs</code>: the genotype probabilities,</li>
<li>
<code>map</code>: the marker map,</li>
<li>
<code>ind</code>: the index of the individual to plot,</li>
<li>
<code>chr</code>: the index of the chromosome to plot.</li>
</ol>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_genoprob</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>              map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>              ind   <span class="op">=</span> <span class="fl">1</span>, </span>
<span>              chr   <span class="op">=</span> <span class="fl">19</span>, </span>
<span>              main  <span class="op">=</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="st">'19'</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/calc-genoprob-rendered-plot_genoprob-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The coordinates along chromosome 19 are shown on the horizontal axis
and the three genotypes are shown on the vertical axis. Higher genotype
probabilities are plotted in darker shades. This mouse has a RR genotype
on the proximal end of the chromosome and transitions to BR.</p>
<figure><img src=".././fig/mapping-workflow-calc-genoprob.png" alt="a workflow diagram showing genotypes and genetic maps used to calculate genotype probabilities" class="figure mx-auto d-block"><div class="figcaption">We have completed the first steps of the mapping
workflow.</div>
</figure><div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<p>1). Load a second dataset from the <a href="https://github.com/rqtl/qtl2data" class="external-link">qtl2data repository</a>. Locate
the <a href="https://github.com/rqtl/qtl2data/tree/main/BXD" class="external-link">BXD
directory</a> and load the data directly from the web using this code
given at the bottom of the <code>ReadMe.md</code> file.</p>
<p><code>file &lt;- paste0("https://raw.githubusercontent.com/rqtl/", "qtl2data/main/BXD/bxd.zip")</code>.</p>
<p><code>bxd &lt;- read_cross2(file)</code></p>
<p>2). How many individuals were in the study? How many phenotypes? How
many markers?</p>
<p>3). Calculate genotype probabilities and save the results to an
object called <code>bxdpr</code>. View the genotypes for the first three
markers on chromosome 1 for the first five individuals.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>1).
<code>file &lt;- paste0("https://raw.githubusercontent.com/rqtl/", "qtl2data/main/BXD/bxd.zip")</code>.<br><code>bxd &lt;- read_cross2(file)</code><br>
2). <code>summary(bxd)</code> gives 198 individuals, 5,806 phenotypes
and 7,320 markers.<br>
3). <code>bxdpr  &lt;- calc_genoprob(cross = bxd, map = bxd$gmap)</code>
followed by<br><code>(bxdpr$</code>1<code>)[1:5, , 1:3]</code></p>
</div>
</div>
</div>
</div>
<div id="challenge-3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3" class="callout-inner">
<h3 class="callout-title">Challenge 3</h3>
<div class="callout-content">
<p>Plot the genotype probabilities for individual number 3 for
chromosome 1.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p><code>plot_genoprob(probs = bxdpr,                  map   = bxd$pmap,                   ind   = 3,                  chr   = 1,                  main  = rownames(bxdpr[['1']])[3])</code></p>
</div>
</div>
</div>
</div>
<p><strong>Parallel calculations (optional)</strong> To speed up the
calculations with large datasets on a multi-core machine, you can use
the argument <code>cores</code>. With <code>cores=0</code>, the number
of available cores will be detected via
<code>parallel::detectCores()</code>. Otherwise, specify the number of
cores as a positive integer.</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">probs</span> <span class="op">&lt;-</span> <span class="fu">calc_genoprob</span><span class="op">(</span>cross <span class="op">=</span> <span class="va">iron</span>, </span>
<span>                         map <span class="op">=</span> <span class="va">map</span>, </span>
<span>                  error_prob <span class="op">=</span> <span class="fl">0.002</span>, </span>
<span>                       cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<p><strong>Allele probabilities (optional)</strong> The genome scan
functions use genotype probabilities as well as a matrix of phenotypes.
If you wished to perform a genome scan via an additive allele model, you
would first convert the genotype probabilities to allele probabilities,
using the function <code>genoprob_to_alleleprob()</code>.</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">apr</span> <span class="op">&lt;-</span> <span class="fu">genoprob_to_alleleprob</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span><span class="op">)</span></span></code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>The first step in QTL analysis is to calculate genotype
probabilities.</li>
<li>Calculate genotype probabilities between genotyped markers with
<code>calc_genoprob()</code>.</li>
</ul>
</div>
</div>
</div></section><section id="aio-perform-genome-scan"><p>Content from <a href="perform-genome-scan.html">Performing a Genome Scan</a></p>
<hr>
<p>Last updated on 2024-11-13 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/perform-genome-scan.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 60 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I perform a genome scan?</li>
<li>How do I plot a genome scan?</li>
<li>How do additive covariates differ from interactive covariates?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Map one trait using additive covariates.</li>
<li>Map the sample trait using additive and interactive covariates.</li>
<li>Plot a genome scan.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>The freely available chapter on <a href="https://www.rqtl.org/book/rqtlbook_ch04.pdf" class="external-link">single-QTL
analysis</a> from Broman and Sen’s <a href="https://www.rqtl.org/book/" class="external-link">A Guide to QTL Mapping with R/qtl</a>
describes different methods for QTL analysis. We will present two of
these methods here - marker regression and Haley-Knott regression. The
chapter provides the statistical background behind these and other QTL
mapping methods.</p>
<p>Linear regression can be employed to identify presence of QTL in a
cross. To identify QTL using regression, we compare the fit for two
models: 1) the null hypothesis that there are no QTL anywhere in the
genome; and 2) the alternative hypothesis that there is a QTL near a
specific position. A sloped line indicates that there is a difference in
mean phenotype between genotype groups, and that a QTL is present. A
line with no slope indicates that there is no difference in mean
phenotype between genotype groups, and that no QTL exists. Regression
aims to find the line of best fit to the data.</p>
<figure><img src="../fig/nullvalt.png" alt="Null and alternative hypotheses" class="figure mx-auto d-block"><div class="figcaption">Linear regression can identify QTL; adapted from
Broman &amp; Sen, 2009</div>
</figure><p>To find the line of best fit, the residuals or errors are calculated,
then squared for each data point. A residual, also known as an error, is
the distance from the line to a data point.</p>
<p><img src="../fig/althypothesis.png" alt="Alternative hypothesis" class="figure"><img src="../fig/nullhypothesis.png" alt="Null hypothesis" class="figure"></p>
<p>The line of best fit will be the one that minimizes the sum of
squared residuals, which maximizes the likelihood of the data.</p>
<p>Marker regression produces a LOD (logarithm of odds) score comparing
the null hypothesis to the alternative. The LOD score is calculated
using the sum of squared residuals (RSS) for the null and alternative
hypotheses. The LOD score is the difference between the log10 likelihood
of the null hypothesis and the log10 likelihood of the alternative
hypothesis. It is related to the regression model above by identifying
the line of best fit to the data.</p>
<p>LOD = <span class="math inline">\(n/2 \times
log10(RSS_0/RSS_1)\)</span></p>
<p>A higher LOD score indicates greater likelihood of the alternative
hypothesis. A LOD score closer to zero favors the null hypothesis.</p>
<p>Marker regression can identify the existence and effect of a QTL by
comparing means between groups, however, it requires known marker
genotypes and can’t identify QTL in between typed markers. To identify
QTL between typed markers, we use Haley-Knott regression. After <a href="https://smcclatchy.github.io/qtl-mapping/calc-genoprob.html" class="external-link">calculating
genotype probabilities</a>, we can regress the phenotypes for animals of
unknown genotype on these conditional genotype probabilities
(conditional on known marker genotypes). In Haley-Knott regression,
phenotype values can be plotted and a regression line drawn through the
phenotype mean for the untyped individuals.</p>
<figure><img src="../fig/hk-regress.png" alt="Haley Knott regression" class="figure mx-auto d-block"><div class="figcaption">Haley-Knott regression for missing
genotypes</div>
</figure><p>As shown by the green circle in the figure, an individual of unknown
genotype is placed between known genotypes according to the probability
of its genotype being BB or BR. In this case, the probability of this
individual having genotype BB is 0.6, and the probability of having
genotype BR is 0.4.</p>
<p>To perform a genome scan by Haley-Knott regression (<a href="https://www.ncbi.nlm.nih.gov/pubmed/16718932" class="external-link">Haley and Knott
1992</a>), use the function <code>scan1()</code>. <code>scan1()</code>
takes as input the genotype probabilities, a matrix of phenotypes, and
then optional additive and interactive covariates. Another option is to
provide a vector of weights.</p>
<section><h2 class="section-heading" id="additive-genome-scan">Additive Genome Scan<a class="anchor" aria-label="anchor" href="#additive-genome-scan"></a>
</h2>
<hr class="half-width">
<p>There are two potential covariates in the Attie data set. Let’s look
at the top of the covariates in the <code>cross</code> object.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">covar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             Sex pgm adipose_batch gastroc_batch hypo_batch islet_batch
Mouse3051   Male   1    12/19/2007     8/11/2008 11/26/2007  11/28/2007
Mouse3551   Male   1    12/19/2007     8/12/2008 11/27/2007  12/03/2007
Mouse3430   Male   1    12/19/2007     8/12/2008 11/27/2007  12/03/2007
Mouse3476   Male   1    12/19/2007     8/12/2008 11/27/2007  12/03/2007
Mouse3414   Male   1    12/18/2007     8/11/2008 11/26/2007  11/28/2007
Mouse3145 Female   1    12/19/2007     8/11/2008         NA  11/28/2007
          kidney_batch liver_batch
Mouse3051   07/15/2008       other
Mouse3551   07/16/2008  12/10/2007
Mouse3430   07/15/2008  12/05/2007
Mouse3476   07/16/2008  12/05/2007
Mouse3414   07/15/2008  12/05/2007
Mouse3145   07/15/2008  12/10/2007</code></pre>
</div>
<p><code>Sex</code> is a potential covariate. It is a good idea to
always include sex in any analysis. Even if you perform an ANOVA and
think that sex is not important, it doesn’t hurt to add in one extra
degree of freedom to your model. Examples of other covariates might be
age, diet, treatment, or experimental batch. It is worth taking time to
identify covariates that may affect your results.</p>
<p>First, we will make sex a <code>factor</code>, which is the term that
R uses for categorical variables. This is required for the next function
to work correctly. Then we will use <code>model.matrix</code> to create
a matrix of “dummy” variables which encode the sex of each mouse.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cross</span><span class="op">$</span><span class="va">covar</span><span class="op">$</span><span class="va">Sex</span> <span class="op">&lt;-</span> <span class="fu">factor</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">covar</span><span class="op">$</span><span class="va">Sex</span><span class="op">)</span></span>
<span></span>
<span><span class="va">addcovar</span> <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">Sex</span>, data <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">covar</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<p>When we perform a genome scan with additive covariates, we are
searching for loci that have the same effect in both covariate groups.
In this case, we are searching for loci that affect females and males in
the same way.</p>
<figure><img src="../fig/additive_qtl_effect_w_sex.png" alt="Figure showing sex and additive QTL effects." class="figure mx-auto d-block"><div class="figcaption">Additive Sex Effect</div>
</figure><p>In the figure above, we plotted simulated phenotype values versus the
genotypes BB, BR and RR. Each point represents one mouse, with females
shown in red and males in blue. We show the sex effect at the mean of
each sex group. The difference between sexes is the same for all three
genotype groups. This is different from the QTL effect, which is the
difference between the mean of each sex and the homozygous group(s).</p>
<p>When we add sex into the model as an additive covariate, we regress
the sex effect out of the phenotype data and then estimate the overall
QTL effect.</p>
<figure><img src="../fig/additive_qtl_effect_wo_sex.png" alt="Figure showing sex and additive QTL effects." class="figure mx-auto d-block"><div class="figcaption">Additive Sex Effect</div>
</figure><p>In the figure above, we have plotted the same simulated phenotype
with sex regressed out. Now the female and male means are the same, but
the QTL effect remains.</p>
<p>With that introduction to additive covariates, let’s map the insulin
phenotype.</p>
<p>First, we will make a <code>data.frame</code> called <em>insulin</em>
so that we don’t have to type quite as many characters every time that
we map insulin.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">insulin</span> <span class="op">&lt;-</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">'log10_insulin_10wk'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<p>Next, we will use the qtl2 function <code>scan1</code> to map insulin
values across the genome with sex as an additive covariate.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_add</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                 pheno     <span class="op">=</span> <span class="va">insulin</span>, </span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>On a multi-core machine, you can get some speed-up via the
<code>cores</code> argument, as with <code>calc_genoprob()</code> and
<code>calc_kinship()</code>.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_add</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                 pheno     <span class="op">=</span> <span class="va">insulin</span>, </span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                 cores     <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<p>The output of <code>scan1()</code> is a matrix of LOD scores, with
markers in rows and phenotypes in columns.</p>
<p>Take a look at the first ten rows of the scan object. The numerical
values are the LOD scores for the marker named at the beginning of the
row. LOD values are shown for circulating insulin.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">lod_add</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>           log10_insulin_10wk
rs13475697         0.04674829
rs3681603          0.04674829
rs13475703         0.04680494
rs13475710         0.12953382
rs6367205          0.13734728
rs13475716         0.13735534
rs13475717         0.13735534
rs13475719         0.13735534
rs13459050         0.13735534
rs3680898          0.13735541</code></pre>
</div>
<p>The function <code>plot_scan1()</code> can be used to plot the LOD
curves. If you have more than one phenotype, use the
<code>lodcolumn</code> argument to indicate which column to plot. In
this case, we have only one column and <code>plot_scan</code> will plot
it by default.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span><span class="va">lod_add</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>           main <span class="op">=</span> <span class="st">'log(insulin): 10 weeks'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/perform-genome-scan-rendered-plot_add_lod-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The LOD plot for insulin shows several peaks, with the largest peak
on chromosome 2. There are smaller peaks on other chromosomes. Which of
these peaks is significant, and why? We’ll evaluate the significance of
genome scan results in a later episode in <a href="https://smcclatchy.github.io/qtl-mapping/perform-perm-test/" class="external-link">Finding
Significant Peaks via Permutation</a>.</p>
<figure><img src=".././fig/mapping-workflow-genome-scan.png" alt="A workflow diagram showing genotypes and genetic map used to calculate genotype probabilities which are then used to perform a genome scan along with phenotypes, covariates, and an optional physical map" class="figure mx-auto d-block"><div class="figcaption">We have completed these steps in the mapping
workflow.</div>
</figure><div id="challenge-1-find-the-marker-with-the-highest-lod." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-find-the-marker-with-the-highest-lod." class="callout-inner">
<h3 class="callout-title">Challenge 1: Find the marker with the highest LOD.</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>What is the highest LOD score for insulin?</li>
<li>Which marker does the highest LOD score occur at?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>You can find the maximum LOD score using the <code>max</code>
function.</li>
</ol>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">max</span><span class="op">(</span><span class="va">lod_add</span><span class="op">[</span>,<span class="st">'log10_insulin_10wk'</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 6.668835</code></pre>
</div>
<ol start="2" style="list-style-type: decimal">
<li>You can find the marker name with the maximum LOD score using the
<code>which.max</code> function.</li>
</ol>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rownames</span><span class="op">(</span><span class="va">lod_add</span><span class="op">)</span><span class="op">[</span><span class="fu">which.max</span><span class="op">(</span><span class="va">lod_add</span><span class="op">[</span>,<span class="st">'log10_insulin_10wk'</span><span class="op">]</span><span class="op">)</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] "rs13476803"</code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="challenge-2-find-the-chromosome-and-position-of-the-maximum-lod-score." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-find-the-chromosome-and-position-of-the-maximum-lod-score." class="callout-inner">
<h3 class="callout-title">Challenge 2: Find the chromosome and position of the maximum LOD score.</h3>
<div class="callout-content">
<p>Look up the <code>qtl2</code> function <code>find_markerpos</code> in
the Help and find the chromosome and Mb position of the marker with the
maximum LOD.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">max_mkr</span> <span class="op">&lt;-</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">lod_add</span><span class="op">)</span><span class="op">[</span><span class="fu">which.max</span><span class="op">(</span><span class="va">lod_add</span><span class="op">[</span>,<span class="st">'log10_insulin_10wk'</span><span class="op">]</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu">find_markerpos</span><span class="op">(</span>cross <span class="op">=</span> <span class="va">cross</span>, markers <span class="op">=</span> <span class="va">max_mkr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>           chr     gmap     pmap
rs13476803   2 67.07116 138.9448</code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="challenge-3-plot-the-lod-on-specific-chromosomes." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-plot-the-lod-on-specific-chromosomes." class="callout-inner">
<h3 class="callout-title">Challenge 3: Plot the LOD on specific chromosomes.</h3>
<div class="callout-content">
<p>The <code>plot_scan1</code> function has a <code>chr</code> argument
that allows you to only plot specific chromosomes. Use this to plot the
insulin LOD on chromosomes 2, 7, 12, and 19. Add a title to the plot
using the <code>main</code> argument, which is part of the basic
plotting function.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span><span class="va">lod_add</span>, map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, chr <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">2</span>, <span class="fl">7</span>, <span class="fl">12</span>, <span class="fl">19</span><span class="op">)</span>,</span>
<span>           main <span class="op">=</span> <span class="st">"log(insulin): 10 weeks"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/perform-genome-scan-rendered-challenge3-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="interactive-genome-scan">Interactive Genome Scan<a class="anchor" aria-label="anchor" href="#interactive-genome-scan"></a>
</h2>
<hr class="half-width">
<p>Above, we mapped insulin levels using sex as an additive covariate
and searched for loci where both sexes had the same QTL effect. But what
if the two sexes have different effects? You might think that we could
map each sex separately. But this approach reduces your sample size, and
hence statistical power, in each sex. A better way is to use all of the
data and map with sex as an additive and an interactive covariate.
Mapping with an interactive covariate allows each sex to have different
effects. We do this by including sex as an interactive covariate in the
genome scan.</p>
<figure><img src="../fig/interactive_qtl_effect_w_sex.png" alt="Figure showing interactive sex by QTL effects" class="figure mx-auto d-block"><div class="figcaption">Interactive Sex by QTL Effect</div>
</figure><p>In the figure above, female (red) and male (blue) phenotypes are
plotted versus the three genotypes. In females, there is no QTL effect
because the mean values in each genotype group are not different. In
males, there is a QTL effect because the mean in each genotype group
changes.</p>
<p>You should always include an interactive covariate as an additive
covariate as well. In this case, we only have sex as a covariate, so we
can use the additive covariate matrix for the interactive covariate. For
clarity, we will make a copy and name it for the interactive
covariates.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">intcovar</span> <span class="op">=</span> <span class="va">addcovar</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_int</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                 pheno     <span class="op">=</span> <span class="va">insulin</span>, </span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                 intcovar  <span class="op">=</span> <span class="va">intcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_int</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           main <span class="op">=</span> <span class="st">'log(insulin): 10 weeks: Sex Interactive'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/perform-genome-scan-rendered-plot_int-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>It is difficult to tell if there is a difference in LOD scores
between the additive and interactive scans. To resolve this, we can plot
both genome scans in the same plot using the <code>add = TRUE</code>
argument. We will also color the additive scan in blue.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_int</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           main <span class="op">=</span> <span class="st">'log(insulin): 10 weeks'</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_add</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           col  <span class="op">=</span> <span class="st">'blue'</span>, </span>
<span>           add  <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span>x <span class="op">=</span> <span class="fl">1000</span>, y <span class="op">=</span> <span class="fl">7.6</span>, legend <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'Additive'</span>, <span class="st">'Interactive'</span><span class="op">)</span>, </span>
<span>       col <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'blue'</span>, <span class="st">'black'</span><span class="op">)</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/perform-genome-scan-rendered-plot_add_int_lod-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>It is still difficult to tell whether any peaks differ by sex.
Another way to view the plot is to plot the difference between the
interactive and additive scans.</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_int</span> <span class="op">-</span> <span class="va">lod_add</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>           main <span class="op">=</span> <span class="st">'log10(insulin): Interactive - Additive'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/perform-genome-scan-rendered-plot_add_int_diff_lod-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>While it was important to look at the effect of sex on the trait, in
this experiment, there do not appear to be any sex-specific peaks.
Without performing a formal test, we usually look for peaks with a LOD
greater than 3 and there do not appear to be any in this scan.</p>
<div id="challenge-4-why-didnt-we-map-and-plot-males-and-females-separately" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-why-didnt-we-map-and-plot-males-and-females-separately" class="callout-inner">
<h3 class="callout-title">Challenge 4: Why didn’t we map and plot males and females separately?</h3>
<div class="callout-content">
<p>Map and plot the males and females separately. Now explain why we
didn’t do things this way.</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>A qtl2 genome scan requires genotype probabilities and a phenotype
matrix.</li>
<li>The output from a genome scan contains a LOD score matrix, map
positions, and phenotypes.</li>
<li>LOD curve plots for a genome scan can be viewed with
plot_scan1().</li>
<li>A genome scan using sex as an additive covariate searches for QTL
which affect both sexes.</li>
<li>A genome scan using sex as an interactive covariate searches for QTL
which affect each sex differently.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-calc-kinship"><p>Content from <a href="calc-kinship.html">Calculating A Kinship Matrix</a></p>
<hr>
<p>Last updated on 2024-11-13 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/calc-kinship.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 30 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>Why would I calculate kinship between individuals?</li>
<li>How do I calculate kinship between individuals?</li>
<li>What does a kinship matrix look like?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Explain why and when kinship calculation matters in mapping.</li>
<li>Create a kinship matrix for individuals.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Population structure and kinship are common confounding factors in
genome-wide association studies (GWAS), case-control studies, and other
study types in genetics. They create false positive associations between
genotype and phenotype at genetic markers that differ in genotype
frequencies between subpopulations due to genetic relatedness between
samples. Simple association tests assume statistical independence
between individuals. Population structure and kinship confound
associations when phenotype covariance between individuals results from
genetic similarity. Accounting for relatedness between individuals helps
to distinguish true associations from false positives generated by
population structure or kinship.</p>
<p>As an example see the table below for phenotype and genotype
frequencies between two subpopulations in a case-control study.</p>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="center">subpop1</th>
<th align="center">subpop2</th>
<th align="center">overall pop</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">frequency</td>
<td align="center">0.5</td>
<td align="center">0.5</td>
<td align="center">1</td>
</tr>
<tr class="even">
<td align="left">probability of AA genotype</td>
<td align="center">0.1</td>
<td align="center">0.9</td>
<td align="center">0.5</td>
</tr>
<tr class="odd">
<td align="left">probability of disease</td>
<td align="center">0.9</td>
<td align="center">0.1</td>
<td align="center">0.5</td>
</tr>
<tr class="even">
<td align="left">probability of disease &amp; AA</td>
<td align="center">0.09</td>
<td align="center">0.09</td>
<td align="center">0.09</td>
</tr>
</tbody>
</table>
<p>The full population consists of two equally represented
subpopulations. In the overall population, the probability of the AA
genotype is 0.5, and the probability of disease is also 0.5. The joint
probability of both disease and AA genotype in the population (0.09) is
less than either the probability of disease (0.5) or the probability of
the AA genotype (0.5) alone, and is considerably less than the joint
probability of 0.25 that would be calculated if subpopulations weren’t
taken into account. In a case-control study that fails to recognize
subpopulations, most of the cases will come from subpopulation 1 since
this subpopulation has a disease probability of 0.9. However, this
subpopulation also has a low probability of the AA genotype. So a false
association between AA genotype and disease would occur because only
overall population probabilities would be considered.</p>
<section><h2 class="section-heading" id="kinship-calculation">Kinship Calculation<a class="anchor" aria-label="anchor" href="#kinship-calculation"></a>
</h2>
<hr class="half-width">
<p>In the B6/BTBR cross, we have three possible genotypes: BB, BR, and
RR. Suppose that we look at the genotypes of two mice and estimate their
kinship. To do this, we select the first allele from mouse 1 and get the
probability of picking the same allele from mouse 2. Then we do the same
procedure with the second allele and take the mean. This calculates the
mean allele-sharing at that marker. We do this for all markers and take
the overall mean.</p>
<p>Let’s look at an example. In the table below, we have listed four
markers and their genotypes in mouse 1 (M1) and mouse 2 (M2). At each
marker, we applied the procedure above and recorded the allele-sharing.
Then we took the mean and found that these two mice have a kinship value
of 0.5.</p>
<table class="table">
<thead><tr class="header">
<th>marker</th>
<th>M1</th>
<th>M2</th>
<th>allele-sharing</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>BB</td>
<td>BB</td>
<td>1.0</td>
</tr>
<tr class="even">
<td>2</td>
<td>BB</td>
<td>BR</td>
<td>0.5</td>
</tr>
<tr class="odd">
<td>3</td>
<td>BB</td>
<td>RR</td>
<td>0.0</td>
</tr>
<tr class="even">
<td>4</td>
<td>BR</td>
<td>BR</td>
<td>0.5</td>
</tr>
<tr class="odd">
<td>—</td>
<td>–</td>
<td>–</td>
<td>—</td>
</tr>
<tr class="even">
<td>All</td>
<td>–</td>
<td>–</td>
<td>0.5</td>
</tr>
</tbody>
</table>
<div id="challenge-1-calculate-the-kinship-of-the-following-two-sets-of-mice." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-calculate-the-kinship-of-the-following-two-sets-of-mice." class="callout-inner">
<h3 class="callout-title">Challenge 1: Calculate the kinship of the following two sets of mice.</h3>
<div class="callout-content">
<ol style="list-style-type: decimal"><li>
</li></ol>
<table class="table">
<thead><tr class="header">
<th>marker</th>
<th>M1</th>
<th>M2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>BB</td>
<td>RR</td>
</tr>
<tr class="even">
<td>2</td>
<td>BB</td>
<td>RR</td>
</tr>
<tr class="odd">
<td>3</td>
<td>BB</td>
<td>RR</td>
</tr>
<tr class="even">
<td>4</td>
<td>BB</td>
<td>BR</td>
</tr>
</tbody>
</table>
<ol start="2" style="list-style-type: decimal"><li>
</li></ol>
<table class="table">
<thead><tr class="header">
<th>marker</th>
<th>M1</th>
<th>M2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>BB</td>
<td>BR</td>
</tr>
<tr class="even">
<td>2</td>
<td>BB</td>
<td>BR</td>
</tr>
<tr class="odd">
<td>3</td>
<td>RR</td>
<td>RR</td>
</tr>
<tr class="even">
<td>4</td>
<td>RR</td>
<td>RR</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<ol style="list-style-type: decimal"><li>
</li></ol>
<table class="table">
<thead><tr class="header">
<th>marker</th>
<th>M1</th>
<th>M2</th>
<th>allele-sharing</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>BB</td>
<td>RR</td>
<td>0.0</td>
</tr>
<tr class="even">
<td>2</td>
<td>BB</td>
<td>RR</td>
<td>0.0</td>
</tr>
<tr class="odd">
<td>3</td>
<td>BB</td>
<td>RR</td>
<td>0.0</td>
</tr>
<tr class="even">
<td>4</td>
<td>BB</td>
<td>BR</td>
<td>0.5</td>
</tr>
<tr class="odd">
<td>—</td>
<td>–</td>
<td>–</td>
<td>—</td>
</tr>
<tr class="even">
<td>All</td>
<td>–</td>
<td>–</td>
<td>0.125</td>
</tr>
</tbody>
</table>
<ol start="2" style="list-style-type: decimal"><li>
</li></ol>
<table class="table">
<thead><tr class="header">
<th>marker</th>
<th>M1</th>
<th>M2</th>
<th>allele-sharing</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>BB</td>
<td>BR</td>
<td>0.5</td>
</tr>
<tr class="even">
<td>2</td>
<td>BB</td>
<td>BR</td>
<td>0.5</td>
</tr>
<tr class="odd">
<td>3</td>
<td>RR</td>
<td>RR</td>
<td>1.0</td>
</tr>
<tr class="even">
<td>4</td>
<td>RR</td>
<td>RR</td>
<td>1.0</td>
</tr>
<tr class="odd">
<td>—</td>
<td>–</td>
<td>–</td>
<td>—</td>
</tr>
<tr class="even">
<td>All</td>
<td>–</td>
<td>–</td>
<td>0.75</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<p>Linear mixed models (LMMs) consider genome-wide similarity between
all pairs of individuals to account for population structure, known
kinship and unknown relatedness. They model the covariance between
individuals. Linear mixed models in association mapping studies can
successfully correct for genetic relatedness between individuals in a
population by incorporating kinship into the model. To perform a genome
scan by a linear mixed model, accounting for the relationships among
individuals (in other words, including a random polygenic effect),
you’ll need to calculate a kinship matrix for the individuals. This is
accomplished with the <code>calc_kinship()</code> function. It takes the
genotype probabilities as input.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kinship</span> <span class="op">&lt;-</span> <span class="fu">calc_kinship</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span><span class="op">)</span></span></code></pre>
</div>
<p>Take a look at the kinship values calculated for the first 5
individuals.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kinship</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          Mouse3051 Mouse3551 Mouse3430 Mouse3476 Mouse3414
Mouse3051     0.737     0.501     0.535     0.520     0.510
Mouse3551     0.501     0.758     0.515     0.512     0.499
Mouse3430     0.535     0.515     0.749     0.458     0.493
Mouse3476     0.520     0.512     0.458     0.749     0.434
Mouse3414     0.510     0.499     0.493     0.434     0.696</code></pre>
</div>
<p>We can also look at the first 50 mice in the kinship matrix.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_samples</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="fu">heatmap</span><span class="op">(</span><span class="va">kinship</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span><span class="op">]</span>, symm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/calc-kinship-rendered-plot_kinship-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The mice are listed in the same order on both sides of the matrix.
The comb-like structures are called “dendrograms” and they indicate how
the mice are clustered together. Each cell represents the degree of
allele sharing between mice. Red colors indicate higher kinship and
yellow colors indicate lower kinship. Each mouse is closely related to
itself, so the cells along the diagonal tend to be darker than the other
cells. You can see some evidence of related mice, possibly siblings, in
the orange-shaded blocks along the diagonal.</p>
<p>By default, the genotype probabilities are converted to allele
probabilities, and the kinship matrix is calculated as the proportion of
shared alleles. Also by default we omit the X chromosome and only use
the autosomes. To include the X chromosome, use
<code>omit_x=FALSE</code>.</p>
<p>On a multi-core machine, you can get some speed-up via the
<code>cores</code> argument, as with <code>calc_genoprob()</code>.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kinship</span> <span class="op">&lt;-</span> <span class="fu">calc_kinship</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                        cores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-2-what-is-a-kinship-matrix" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-what-is-a-kinship-matrix" class="callout-inner">
<h3 class="callout-title">Challenge 2: What is a Kinship Matrix?</h3>
<div class="callout-content">
<p>Think about what a kinship matrix is and what it represents. Share
your understanding with a neighbor. Write your explanation in the
collaborative document or in your own personal notes.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div id="challenge-3-mean-value-in-kinship-matrix" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-mean-value-in-kinship-matrix" class="callout-inner">
<h3 class="callout-title">Challenge 3: Mean Value in Kinship Matrix</h3>
<div class="callout-content">
<p>What is the mean kinship value of all mice? Can you explain why the
value is the number that you get? Think about the three possible
genotypes in the cross and how they compare to each other.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mean</span><span class="op">(</span><span class="va">kinship</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 0.501</code></pre>
</div>
<p>The mean kinship between the mice in the cross is 0.5. All of the
mice were derived from F1 parents created from two inbred strains. The
kinship calculation compares the number of identical alleles between
mice at each marker. On average, each mouse carries 50% B alleles and
50% R alleles. So the mean allele sharing is 0.5.</p>
</div>
</div>
</div>
</div>
<div id="challenge-4-kinship-of-a-mouse-with-itself" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-kinship-of-a-mouse-with-itself" class="callout-inner">
<h3 class="callout-title">Challenge 4: Kinship of a Mouse with Itself</h3>
<div class="callout-content">
<p>Look at the values in the kinship matrix. What is the kinship value
of each mouse with itself? Why isn’t this equal to 1?</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<p>The kinship of a mouse with itself will only be 1 if the mouse is
homozygous at every marker. Since the mice in this study are F2s, they
will have genomes that are around 50% heterozygous. The variation in
heterozygosity affects the actual kinship values, but, on average, we
expect self-kinship to be 0.75 because half of the genome is homozygous
and contributes values of 1, and half of the genome is heterozygous and
contributes values of 0.5. So <span class="math inline">\((0.5 \times 1)
+ (0.5 \times 0.5) = 0.75\)</span>.</p>
<p>This is shown in the plot below of homozygosity versus kinship. There
is a clear linear relationship between the two values and at 0.5
homozygosity, the self-kinship values is 0.75. The slope of the line is
0.5.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1 &amp; 3 are the encodings for the two homozygous genotypes</span></span>
<span><span class="va">homo</span> <span class="op">&lt;-</span> <span class="fu">sapply</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">geno</span>, </span>
<span>               <span class="kw">function</span><span class="op">(</span><span class="va">z</span><span class="op">)</span> <span class="op">{</span></span>
<span>                 <span class="fu">rowSums</span><span class="op">(</span><span class="va">z</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">|</span> <span class="va">z</span> <span class="op">==</span> <span class="fl">3</span><span class="op">)</span></span>
<span>               <span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="va">homo</span> <span class="op">&lt;-</span> <span class="fu">rowSums</span><span class="op">(</span><span class="va">homo</span><span class="op">)</span> <span class="op">/</span> <span class="fu">sum</span><span class="op">(</span><span class="fu">n_mar</span><span class="op">(</span><span class="va">cross</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">df</span>   <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>id      <span class="op">=</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">geno</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                   homo    <span class="op">=</span> <span class="va">homo</span>,</span>
<span>                   kinship <span class="op">=</span> <span class="fu">diag</span><span class="op">(</span><span class="va">kinship</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">df</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="va">homo</span>, <span class="va">kinship</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Homozygosity versus Self-Kinship"</span>,</span>
<span>         x     <span class="op">=</span> <span class="st">"Homozygosity"</span>,</span>
<span>         y     <span class="op">=</span> <span class="st">"Self-Kinship"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/calc-kinship-rendered-challenge4-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rm</span><span class="op">(</span><span class="va">homo</span>, <span class="va">df</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Kinship matrices account for relationships among individuals.</li>
<li>Kinship is calculated as the proportion of shared alleles between
individuals.</li>
<li>Kinship calculation is a precursor to a genome scan via a linear
mixed model.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-perform-genome-scan-lmm"><p>Content from <a href="perform-genome-scan-lmm.html">Performing a genome scan with a linear mixed model</a></p>
<hr>
<p>Last updated on 2024-11-13 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/perform-genome-scan-lmm.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 45 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I use a linear mixed model in a genome scan?</li>
<li>How do different mapping and kinship calculation methods
differ?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Create a genome scan with a linear mixed model.</li>
<li>Compare LOD plots for Haley-Knott regression and linear mixed model
methods.</li>
<li>Compare LOD plots for the standard kinship matrix with the
leave-one-chromosome-out (LOCO) method.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Genetic mapping in mice presents a good example of why accounting for
population structure is important.</p>
<figure><img src="https://www.informatics.jax.org/silver/images/figure2-3.gif" alt="A world map showing the origins and ranges of five Mus musculus subspecies: domesticus, musculus, castaneus, bactrianus and molossinus" class="figure mx-auto d-block"><div class="figcaption">Geographical ranges of subspecies in the Mus
musculus group. From <a href="https://www.informatics.jax.org/silverbook/" class="external-link"><em>Mouse Genetics:
Concepts and Applications</em></a> by Lee M. Silver, Oxford University
Press, 1995.</div>
</figure><p>Laboratory mouse strains are descended from a small number of
founders (fancy mice) and went through several population
bottlenecks.</p>
<figure><img src="https://www.informatics.jax.org/morsebook/images/1-10as.jpg" alt="A genealogy chart showing the origin of commonly used inbred strains as Chinese, Japanese and English mouse fanciers. Abby Lathrop of Granby Mouse Farm supplied fancy mice to researchers including Marsh, C.C. Little, MacDowell, Castle and Strong in the early 1900s. These mice were developed into common inbred strains such as C57BL, 129, DBA and C3H strains." class="figure mx-auto d-block"><div class="figcaption">The genealogy of inbred mice. From <a href="https://www.informatics.jax.org/morsebook/" class="external-link"><em>Origins of Inbred
Mice</em></a> edited by Herbert C. Morse III, Academic Press, 1978.
Permission from Elsevier.</div>
</figure><p>Wild-derived strains are not descended from fancy mice and don’t
share the same history as laboratory strains.</p>
<p>Issues of population structure and differing degrees of genetic
relatedness between mice are addressed using linear mixed models (LMM).
LMMs consider genome-wide similarity between all pairs of individuals to
account for population structure, known kinship and unknown relatedness.
LMMs in mapping studies can successfully adjust for genetic relatedness
between individuals in a population by incorporating kinship into the
model. Earlier we <a href="https://smcclatchy.github.io/qtl-mapping/instructor/calc-kinship.html" class="external-link">calculated
a kinship matrix</a> for input to a linear mixed model to account for
relationships among individuals. For a current review of mixed models in
genetics, see <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC6307707/" class="external-link">Sul et al, PLoS
Genetics, 2018</a>.</p>
<p>Simple linear regression takes the form:</p>
<p><span class="math inline">\(y = \mu + \beta G + \epsilon\)</span></p>
<p>which describes a line with slope <span class="math inline">\(\beta\)</span> and y-intercept <span class="math inline">\(\mu\)</span>. The error (or residual) is
represented by <span class="math inline">\(\epsilon\)</span>.</p>
<p>To model the relationship between the phenotype <span class="math inline">\(y\)</span> and the genotypes at one marker, we
use:</p>
<p><span class="math inline">\(y_j = \mu + \beta_k G_{jk} +
\epsilon_j\)</span></p>
<p>where <span class="math inline">\(y_j\)</span> is the phenotype of
the <span class="math inline">\(j\)</span>th individual, <span class="math inline">\(\mu\)</span> is the mean phenotype value, <span class="math inline">\(\beta_k\)</span> is the effect of the <span class="math inline">\(k\)</span>th genotype, <span class="math inline">\(G_{jk}\)</span> is the genotype for individual
<span class="math inline">\(j\)</span>, and <span class="math inline">\(\epsilon_j\)</span> is the error for the <span class="math inline">\(j\)</span>th individual. In the figure below,
<span class="math inline">\(\mu\)</span> equals 94.6 and <span class="math inline">\(\beta\)</span> equals 15 for the alternative
hypothesis (QTL exists). This linear model is <span class="math inline">\(y\)</span> = 94.6 + 15<span class="math inline">\(G\)</span> + <span class="math inline">\(\epsilon\)</span>. The model intersects the
genotype groups at their group means. In contrast, the null hypothesis
would state that there is no difference in group means (no QTL
anywhere). The linear model for the null hypothesis would be <span class="math inline">\(y\)</span> = 94.6 + 0<span class="math inline">\(G\)</span> + <span class="math inline">\(\epsilon\)</span>. This states that the phenotype
is equal to the combined mean (94.6), plus some error (<span class="math inline">\(\epsilon\)</span>). In other words, genotype
doesn’t affect the phenotype.</p>
<figure><img src="../fig/nullvalt.png" alt="Null versus alternative hypotheses" class="figure mx-auto d-block"><div class="figcaption">Null and Alternative Hypotheses</div>
</figure><p>The linear models above describe the relationship between genotype
and phenotype but are inadequate for describing the relationship between
genotype and phenotype in large datasets. They don’t account for
relatedness among individuals. In real populations, the effect of a
single genotype is influenced by many other genotypes that affect the
phenotype. A true genetic model takes into account the effect of all
variants on the phenotype.</p>
<p>To model the phenotypes of all individuals in the data, we can adapt
the simple linear model to include all individuals and their variants so
that we capture the effect of all variants shared by individuals on
their phenotypes.</p>
<p><span class="math inline">\(y=\mu+\sum_{i=1}^M{\beta_iG_i}+\epsilon\)</span></p>
<p>Now, <span class="math inline">\(y\)</span> represents the phenotypes
of all individuals. The effect of the <span class="math inline">\(i\)</span>th genotype on the phenotype is <span class="math inline">\(\beta_i\)</span>, the mean is <span class="math inline">\(\mu\)</span> and the error is <span class="math inline">\(\epsilon\)</span>. Here, the number of genotypes
is M.</p>
<p>To model the effect of all genotypes and to account for relatedness,
we test the effect of a single genotype while bringing all other
genotypes into the model.</p>
<p><span class="math inline">\(y=\mu + \beta_kG_k +\sum_{i\neq
k}^{M}\beta_iG_i + \epsilon\)</span></p>
<p><span class="math inline">\(\beta_k\)</span> is the effect of the
genotype <span class="math inline">\(G_k\)</span>, and <span class="math inline">\(\sum_{i\neq k}^{M}\beta_iG_i\)</span> sums the
effects across all markers (M) of all other genotypes except genotype
<span class="math inline">\(k\)</span>. For the leave one chromosome out
(LOCO) method, <span class="math inline">\(\beta_kG_k\)</span> is the
effect of genotypes on chromosome <span class="math inline">\(k\)</span>, and <span class="math inline">\(\beta_iG_i\)</span> represents the effects of
genotypes on all other chromosomes.</p>
<p>If the sample contains divergent subpopulations, SNPs on different
chromosomes will be correlated because of the difference in allele
frequencies between subpopulations caused by relatedness. To correct for
correlations between chromosomes, we model all genotypes on the other
chromosomes when testing for the association of a SNP.</p>
<p>First, we will create a single kinship matrix using all of the
genotype probabilities on all chromosomes.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kinship_all</span> <span class="op">&lt;-</span> <span class="fu">calc_kinship</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                            type <span class="op">=</span> <span class="st">"overall"</span><span class="op">)</span></span></code></pre>
</div>
<p>To perform a genome scan using a linear mixed model you also use the
function <code>scan1</code>; you just need to provide the argument
<code>kinship</code>, a kinship matrix (or, for the LOCO method, a list
of kinship matrices).</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_add_all</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                     pheno     <span class="op">=</span> <span class="va">insulin</span>, </span>
<span>                     kinship   <span class="op">=</span> <span class="va">kinship_all</span>, </span>
<span>                     addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>Again, on a multi-core machine, you can get some speed-up using the
<code>cores</code> argument.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_add_all</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                     pheno     <span class="op">=</span> <span class="va">insulin</span>, </span>
<span>                     kinship   <span class="op">=</span> <span class="va">kinship_all</span>, </span>
<span>                     addcovar  <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                     cores     <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s plot the insulin genome scan using the overall kinship
matrix.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x     <span class="op">=</span> <span class="va">lod_add</span>,  </span>
<span>           map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>           col   <span class="op">=</span> <span class="st">'black'</span>,</span>
<span>           main  <span class="op">=</span> <span class="st">"log10(insulin): Kinship Comparison"</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x     <span class="op">=</span> <span class="va">lod_add_all</span>,  </span>
<span>           map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>           col   <span class="op">=</span> <span class="st">'blue'</span>,</span>
<span>           add   <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span>x <span class="op">=</span> <span class="fl">1000</span>, y <span class="op">=</span> <span class="fu">max</span><span class="op">(</span><span class="va">lod_add</span><span class="op">)</span>, legend <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'No kinship'</span>, <span class="st">'Overall kinship'</span><span class="op">)</span>, </span>
<span>       col <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'black'</span>, <span class="st">'blue'</span><span class="op">)</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/perform-genome-scan-lmm-rendered-plot_lod_all-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>How did the LOD scores change? Did they increase or decrease? In
general, the LOD scores decreased. In some cases, the LOD score
decreased drastically. Why would this happen?</p>
<p>To answer this, we have to think back to the mapping model that we
are using.</p>
<p><span class="math inline">\(y=\mu + \beta_kG_k +\sum_{i\neq
k}^{M}\beta_iG_i + \epsilon\)</span></p>
<p>In this model, we explicitly excluded the current marker from the
kinship calculation. However, when we calculated the kinship matrix
above, we used <strong>ALL</strong> of the markers. This means that we
have put the current marker <span class="math inline">\(k\)</span> into
the model <em>twice</em>. This causes a loss of power which has been
reported in <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC3989144/" class="external-link">Yang et al.,
Nat. Gen., 2014</a>.</p>
<p>In order to address this issue, we could remove the current marker
from the calculation. But this would mean that we would have to
calculate a different kinship matrix for every marker. As a compromise,
researchers create a different kinship matrix for each chromosome which
uses all of the markers except those on the current chromosome. For
example, to create a kinship matrix to use on chromosome 1, we would use
the markers on chromosomes 2 through X. This is called the <em>Leave One
Chromosome Out</em> (LOCO) method.</p>
<p>In order to use the LOCO method (scan each chromosome using a kinship
matrix that is calculated using data from all other chromosomes), use
<code>type="loco"</code> in the call to <code>calc_kinship()</code>.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kinship_loco</span> <span class="op">&lt;-</span> <span class="fu">calc_kinship</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                             type  <span class="op">=</span> <span class="st">"loco"</span><span class="op">)</span></span></code></pre>
</div>
<p>For the LOCO (leave one chromosome out) method, provide the list of
kinship matrices as obtained from <code>calc_kinship()</code> with
<code>type="loco"</code>.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_add_loco</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                      pheno     <span class="op">=</span> <span class="va">insulin</span>, </span>
<span>                      kinship   <span class="op">=</span> <span class="va">kinship_loco</span>, </span>
<span>                      addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>To plot the results, we again use <code>plot_scan1()</code>.</p>
<p>Here is a plot of the LOD scores by Haley-Knott regression and the
linear mixed model using either the standard kinship matrix or the LOCO
method.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x     <span class="op">=</span> <span class="va">lod_add</span>,      </span>
<span>           map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           col   <span class="op">=</span> <span class="st">'black'</span>,</span>
<span>           ylim  <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0</span>, <span class="fl">7.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x     <span class="op">=</span> <span class="va">lod_add_all</span>,  </span>
<span>           map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           col   <span class="op">=</span> <span class="st">'blue'</span>,   </span>
<span>           add   <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x     <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>           map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           col   <span class="op">=</span> <span class="st">'darkgreen'</span>, </span>
<span>           add   <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span>x <span class="op">=</span> <span class="fl">1500</span>, y <span class="op">=</span> <span class="fl">7.5</span>, legend <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"No kinship"</span>, <span class="st">"All kinship"</span>, <span class="st">"LOCO kinship"</span><span class="op">)</span>,</span>
<span>       lwd <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'black'</span>, <span class="st">'blue'</span>, <span class="st">'darkgreen'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/perform-genome-scan-lmm-rendered-plot_lod_add_loco_all-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>For circulating insulin, the three methods give quite different
results. The linear mixed model with an overall kinship matrix (blue)
produces much lower LOD scores than the other two methods. On
chromosomes with some evidence of a QTL, the LOCO method gives higher
LOD scores than Haley-Knott, except on chromosome 6 where it gives lower
LOD scores.</p>
<p>Let’s plot the difference between the LOCO-kinship genome scan and
the no-kinship genome scan.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_add_loco</span> <span class="op">-</span> <span class="va">lod_add</span>, </span>
<span>     map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>     ylim <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">1.5</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>     main <span class="op">=</span> <span class="st">"LOCO - no kinship Genome Scan"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/perform-genome-scan-lmm-rendered-plot_loco_none-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>As you can see from the plot above, the LOCO-kinship scan generally
produces higher LOD scores. While higher LOD scores don’t always mean
that the model is better, in this case, we expect that accounting for
the correlation in residual errors using kinship matrices will produce
more correct results.</p>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p>What are the benefits and disadvantages of the three methods for
genome scanning (Haley-Knott regression, kinship matrix, and
leave-one-chromosome out (LOCO)?)<br>
Which method would you use to scan, and why?<br>
Think about the advantages and disadvantages of each, discuss with a
neighbor, and share your thoughts in the collaborative document.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>Haley-Knott regression does not consider the kinship between mice.
This may increase or decrease the LOD score and least to false or missed
discoveries. Including a kinship matrix which uses all of the markers in
the mapping model causes a loss of power because the marker is fit twice
in the model. A compromise approach is to create a separate kinship
matrix for each chromosome which uses all of the markers expect those on
the current chromosome. This estimates kinship while preventing the
current marker from being in the model twice. The LOCO method leads to
better power to identify QTL.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2-pair-programming" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-pair-programming" class="callout-inner">
<h3 class="callout-title">Challenge 2: Pair programming</h3>
<div class="callout-content">
<p>With a partner, review and carry out all of the steps in QTL mapping
that we have covered so far, using a new data set. One of you types the
code, the other explains what needs to happen next, finds the relevant
code in the lesson, suggests new names for objects (i.e. NOT the ones
you’ve already used, such as <code>map</code>, <code>pr</code>,
<code>out</code>, <code>prob</code>, etc.).</p>
<ol style="list-style-type: decimal">
<li>Load the <a href="https://github.com/rqtl/qtl2data/tree/main/Gough" class="external-link">Gough data</a>
into an object called <code>gough</code>. You can find code to load the
data at the bottom of the page.</li>
<li>Calculate genotype probabilities.</li>
<li>Run a genome scan for the first 16 phenotypes using
<code>pheno = gough$pheno[, 1:16]</code>. These represent 16 weeks of
body weight measurements. The remaining phenotypes are derivations of
these measurements.</li>
<li>Calculate a kinship matrix.</li>
<li>Calculate a list of kinship matrices with the LOCO method.</li>
<li>Run genome scans with the regular kinship matrix and with the list
of LOCO matrices.</li>
<li>Plot the 3 different genome scans in a single plot in different
colors. By default, the <code>plot()</code> function will give you the
first lod column which is for week 1. You can add the
<code>lodcolumn =</code> argument to the call to plot and insert
additional lod column numbers up to 16.</li>
<li>Which chromosomes appear to have peaks with a LOD score greater than
7? Which methods identify these peaks? Which don’t?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p><code>file &lt;- paste0("https://raw.githubusercontent.com/rqtl/", "qtl2data/main/Gough/gough.zip")</code><br><code>gough &lt;- read_cross2(file)</code><br><code>summary(gough)</code><br><code>head(goughpheno)</code><br><code>colnames(gough$pheno)</code><br><code>goughmap &lt;- gough$pmap</code><br><code>prgough &lt;- calc_genoprob(cross=gough, map=goughmap, error_prob=0.002)</code><br><code>goughaddcovar &lt;- get_x_covar(gough)</code><br><code>goughscan &lt;- scan1(genoprobs = prgough, pheno = gough$pheno[, 1:16], addcovar=goughaddcovar)</code><br><code>plot(goughscan, map = goughmap)</code><br><code>goughkinship &lt;- calc_kinship(probs = prgough)</code><br><code>out_pg_gough &lt;- scan1(prgough, bgough$pheno, kinship=goughkinship, addcovar=goughaddcovar)</code><br><code>kinship_loco_gough &lt;- calc_kinship(prgough, "loco")</code><br><code>out_pg_loco_gough &lt;- scan1(prgough, gough$pheno, kinship_loco_gough, addcovar=goughaddcovar)</code><br><code>plot_scan1(out_pg_loco_gough, map = goughmap, lodcolumn = 2, col = "black")</code><br><code>plot_scan1(out_pg_gough, map = goughmap, lodcolumn = 2, col = "blue", add = TRUE)</code><br><code>plot_scan1(goughscan, map = goughmap, lodcolumn = 2, col = "green", add = TRUE)</code></p>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>To perform a genome scan with a linear mixed model, supply a kinship
matrix.</li>
<li>Different mapping and kinship calculation methods give different
results.</li>
<li>Using a set of Leave-One-Chromosome-Out kinship matrices generally
produces higher LOD scores than other methods.</li>
</ul>
</div>
</div>
</div></section><section id="aio-perform-genome-scan-bin"><p>Content from <a href="perform-genome-scan-bin.html">Performing a Genome Scan with Binary Traits</a></p>
<hr>
<p>Last updated on 2024-11-13 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/perform-genome-scan-bin.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 50 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I perform a genome scan for binary traits?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Convert phenotypes to binary values.</li>
<li>Use logistic regression for genome scans with binary traits.</li>
<li>Plot and compare genome scans for binary traits.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="binary-phenotypes">Binary Phenotypes<a class="anchor" aria-label="anchor" href="#binary-phenotypes"></a>
</h2>
<hr class="half-width">
<p>The genome scans in the previous episode were performed assuming that
the residual variation followed a normal distribution. This will often
provide reasonable results even if the residuals are not normal, but an
important special case is that of a binary trait, with values 0 and 1,
which is best treated differently. The <code>scan1</code> function can
perform a genome scan with binary traits by logistic regression, using
the argument <code>model="binary"</code>. (The default value for the
<code>model</code> argument is <code>"normal"</code>). At present, we
<em>cannot</em> account for kinship relationships among individuals in
this analysis.</p>
<p>Let’s look at the phenotypes in the cross again.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          log10_insulin_10wk agouti_tan tufted
Mouse3051              1.399          1      0
Mouse3551              0.369          1      1
Mouse3430              0.860          0      1
Mouse3476              0.800          1      0
Mouse3414              1.370          0      0
Mouse3145              1.783          1      0</code></pre>
</div>
<p>There are two binary traits called <code>agouti_tan</code> and
<code>tufted</code> which are related to coat color and shape.</p>
<p>The <code>agouti_tan</code> phenotype is <code>1</code> if a mouse
has an agouti or tan coat and <code>0</code> if a mouse has a black
coat. The founder strains have different coat colors. C57BL/6J has a
black coat.</p>
<figure><img src="../fig/C57BL_6J_000664.png" alt="C57BL/6J mouse" class="figure mx-auto d-block"><div class="figcaption">C57BL/6J</div>
</figure><p>BTBR appears to have a black coat, but this coat color is actually
called “black and tan” because their bellies are tan.</p>
<figure><img src="../fig/BTBR_002282.png" atl="BTBR mouse" alt="BTBR" class="figure mx-auto d-block"><div class="figcaption">BTBR</div>
</figure><p>Agouti mice appear to be tan or brown, but their hair is a mix of
brown and black coloring. As an example, <a href="">C3H/HeJ</a> mice
have agouti coats.</p>
<figure><img src="../fig/C3H_HeJ_000659.png" alt="C3H/HeJ mouse" class="figure mx-auto d-block"><div class="figcaption">C3H/HeJ</div>
</figure><p>There is more information about mouse coat colors in the <a href="https://jackson.jax.org/rs/444-BUH-304/images/Poster_Mouse_Coat_Color.pdf" class="external-link">JAX
Coat Color guide</a>.</p>
<p>The tufted phenotype has to do with progressive hair loss in a
“tufted” pattern. Mice with a value of <code>1</code> have the tufted
phenotype and mice with a value of <code>0</code> do not.</p>
<figure><img src="../fig/Ellis_JHered_2013.jpg" alt="Image showing tufted hair loss in BTBR mouse" class="figure mx-auto d-block"><div class="figcaption">Tufted hair loss pattern</div>
</figure><p>Photo Credit: <a href="https://pubmed.ncbi.nlm.nih.gov/23100490/" class="external-link">Ellis et al., J Hered,
2013</a></p>
</section><section><h2 class="section-heading" id="performing-a-binary-genome-scan">Performing a Binary Genome Scan<a class="anchor" aria-label="anchor" href="#performing-a-binary-genome-scan"></a>
</h2>
<hr class="half-width">
<p>We perform a binary genome scan in a manner similar to mapping
continuous traits by using <code>scan1</code>. When we mapped insulin,
there was a hidden argument called <code>model</code> which told
<code>qtl2</code> which mapping model to use. There are two options:
<code>normal</code>, the default, and <code>binary</code>. The
<code>normal</code> argument tells <code>qtl2</code> to use a
<em>normal</em> (least squares) linear model. To map a binary trait, we
will include the <code>model = "binary"</code> argument to indicate that
the phenotype is a binary trait with values 0 and 1.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_agouti</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                    pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">'agouti_tan'</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                    addcovar  <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                    model     <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s plot the result and see if there is a peak.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_agouti</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           main <span class="op">=</span> <span class="st">'Agouti'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/perform-genome-scan-bin-rendered-plot_bin_scan-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Yes! There is a big peak on chromosome 2. Let’s zoom in on chromosome
2.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_agouti</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           chr  <span class="op">=</span> <span class="st">"2"</span>,</span>
<span>           main <span class="op">=</span> <span class="st">"Agouti"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/perform-genome-scan-bin-rendered-plot_bin_scan_chr2-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>We can use <code>find_peaks</code> to find the position of the
highest LOD score.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_agouti</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex  lodcolumn chr pos  lod
1        1 agouti_tan   2 157 61.9</code></pre>
</div>
<p>This turns out to be a well-known coat color locus for agouti coat
color which contains the <a href="https://www.informatics.jax.org/marker/MGI:87853" class="external-link">nonagouti</a>
gene. Mice carrying two black alleles will have a black coat, and mice
carrying one or no black alleles will have agouti coats.</p>
<div id="challenge-1-how-many-mice-have-black-coats" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-how-many-mice-have-black-coats" class="callout-inner">
<h3 class="callout-title">Challenge 1: How many mice have black coats?</h3>
<div class="callout-content">
<p>Look at the frequency of the black (0) and agouti (1) phenotypes.
What proportion of the mice are black? Can you use what you learned
about how the <code>nonagouti</code> locus works and the cross design to
explain the frequency of black mice?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>First, get the number of black and agouti mice.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl</span> <span class="op">&lt;-</span> <span class="fu">table</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"agouti_tan"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">tbl</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
  0   1
125 356 </code></pre>
</div>
<p>Then use the number of mice to calculate the proportion with each
coat color.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl</span> <span class="op">/</span> <span class="fu">sum</span><span class="op">(</span><span class="va">tbl</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>
   0    1
0.26 0.74 </code></pre>
</div>
<p>We can see that the black (0) mice occur about 25 % of the time. If
the <code>B</code> allele causes mice to have black coats when it is
recessive, and if <code>R</code> is the agouti allele, then, when
breeding two heterozygous (<code>BR</code>) mice together, we expect the
following genotypes in the progeny:</p>
<table class="table">
<thead><tr class="header">
<th>/</th>
<th>B</th>
<th>R</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>B</td>
<td>BB</td>
<td>BR</td>
</tr>
<tr class="even">
<td>R</td>
<td>BR</td>
<td>RR</td>
</tr>
</tbody>
</table>
<p>Hence, we expect mean allele frequencies and coat colors as
follows:</p>
<table class="table">
<thead><tr class="header">
<th>Allele</th>
<th>Frequency</th>
<th>Coat Color</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>BB</td>
<td>0.25</td>
<td>black</td>
</tr>
<tr class="even">
<td>BR</td>
<td>0.50</td>
<td>agouti</td>
</tr>
<tr class="odd">
<td>RR</td>
<td>0.25</td>
<td>agouti</td>
</tr>
</tbody>
</table>
<p>From this, we can see that about 25% of the mice should have black
coats.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2-map-the-tufted-phenotype." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-map-the-tufted-phenotype." class="callout-inner">
<h3 class="callout-title">Challenge 2: Map the “tufted” phenotype.</h3>
<div class="callout-content">
<p>Map the tufted phenotype an determine if there are any tall peaks for
this trait.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>First, map the trait.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_tufted</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                    pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"tufted"</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                    addcovar  <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                    model     <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span></code></pre>
</div>
<p>Then, plot the LOD score.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_tufted</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           main <span class="op">=</span> <span class="st">"Tufted"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/perform-genome-scan-bin-rendered-challenge2b-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Finally, use <code>find_peaks</code> to get the peak LOD
location.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_tufted</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr  pos  lod
1        1    tufted  17 27.3 62.2</code></pre>
</div>
<p>There is a large peak on chromosome 17. This is a <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3570182/" class="external-link">known
locus</a> associated with the <a href="https://www.informatics.jax.org/marker/MGI:96624" class="external-link">Itpr3</a> gene
near 27.3 Mb on chromosome 17.</p>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>A genome scan for binary traits (0 and 1) requires special handling;
scans for non-binary traits assume normal variation of the
residuals.</li>
<li>A genome scan for binary traits is performed using logistic
regression.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-perform-perm-test"><p>Content from <a href="perform-perm-test.html">Finding Significant Peaks via Permutation</a></p>
<hr>
<p>Last updated on 2024-11-13 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/perform-perm-test.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 30 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can I evaluate the statistical significance of genome scan
results?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Run a permutation test to establish LOD score thresholds.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" aria-labelledby="headingInstructor1" data-bs-parent="#accordionInstructor1">
<div class="accordion-body">
<p>We need the following block for the site to build on Github. The
students do not need to see or run the next block.</p>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="using-permutations-to-determine-significance-thresholds">Using Permutations to Determine Significance Thresholds<a class="anchor" aria-label="anchor" href="#using-permutations-to-determine-significance-thresholds"></a>
</h2>
<hr class="half-width">
<p>To establish the statistical significance of the results of a genome
scan, a permutation test can identify the maximum LOD score that can
occur by random chance. A permutation tests shuffles genotypes and
phenotypes, essentially breaking the relationship between the two. The
genome-wide maximum LOD score is then calculated on the permuted data,
and this score used as a threshold of statistical significance. A
genome-wide maximum LOD on shuffled, or permuted, data serves as the
threshold because it represents the highest LOD score generated by
random chance.</p>
<p>The <code>scan1perm()</code> function takes the same arguments as
<code>scan1()</code>, plus additional arguments to control the
permutations:</p>
<ul>
<li>
<code>n_perm</code> is the number of permutation replicates.</li>
<li>
<code>perm_Xsp</code> controls whether to perform autosome/X
chromosome specific permutations (with <code>perm_Xsp=TRUE</code>) or
not (the default is FALSE).</li>
<li>
<code>perm_strata</code> is a vector that defines the strata for a
stratified permutation test.</li>
<li>
<code>chr_lengths</code> is a vector of chromosome lengths, used in
the case that <code>perm_Xsp=TRUE</code>.</li>
</ul>
<p>As with <code>scan1()</code>, you may provide a kinship matrix (or
vector of kinship matrices, for the “leave one chromosome out” (loco)
approach), in order to fit linear mixed models. If <code>kinship</code>
is unspecified, the function performs ordinary Haley-Knott
regression.</p>
<p>To perform a permutation test with the insulin phenotype, we run
<code>scan1perm()</code>, provide it with the genotype probabilities,
the phenotype data, X covariates and number of permutations.</p>
<div id="accordionInstructor2" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor2" aria-expanded="false" aria-controls="collapseInstructor2">
  <h3 class="accordion-header" id="headingInstructor2">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor2" class="accordion-collapse collapse" aria-labelledby="headingInstructor2" data-bs-parent="#accordionInstructor2">
<div class="accordion-body">
<p>Replace number of permutations (1000) with 100 for expediency. 1000
permutations takes about 2 minutes.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perm_add</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                      pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                      addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                      Xcovar    <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                      n_perm    <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span> </span></code></pre>
</div>
<p>Note the need to specify special covariates for the X chromosome (via
<code>Xcovar</code>), to be included under the null hypothesis of no
QTL. And note that when these are provided, the default is to perform a
stratified permutation test, using strata defined by the rows in
<code>Xcovar</code>. In general, when the X chromosome is considered,
one will wish to stratify at least by sex.</p>
<p>Also note that, as with <code>scan1()</code>, you can speed up the
calculations on a multi-core machine by specifying the argument
<code>cores</code>. With <code>cores=0</code>, the number of available
cores will be detected via <code>parallel::detectCores()</code>.
Otherwise, specify the number of cores as a positive integer. For large
data sets, be mindful of the amount of memory that will be needed; you
may need to use fewer than the maximum number of cores, to avoid going
beyond the available memory.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perm_add</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                      pheno     <span class="op">=</span> <span class="va">insulin</span>, </span>
<span>                      addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                      Xcovar    <span class="op">=</span> <span class="va">Xcovar</span>, </span>
<span>                      n_perm    <span class="op">=</span> <span class="fl">1000</span>, </span>
<span>                      cores     <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre>
</div>
<p><code>perm_add</code> now contains the maximum LOD score for each
permutation for the phenotypes. There should be 1000 values for each
phenotype. We can view the insulin permutation LOD scores by making a
histogram.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">perm_add</span>, </span>
<span>     breaks <span class="op">=</span> <span class="fl">50</span>, </span>
<span>     xlab   <span class="op">=</span> <span class="st">"LOD"</span>, </span>
<span>     las    <span class="op">=</span> <span class="fl">1</span>,</span>
<span>     main   <span class="op">=</span> <span class="st">"Empirical distribution of maximum LOD scores under permutation"</span><span class="op">)</span></span>
<span><span class="fu">abline</span><span class="op">(</span>v <span class="op">=</span> <span class="fu">summary</span><span class="op">(</span><span class="va">perm_add</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">'red'</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/perform-perm-test-rendered-hist_perm-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the histogram above, you can see that most of the maximum LOD
scores fall between 1 and 3.5. This means that we expect LOD scores less
than 3.5 to occur by chance fairly often. The red line indicates the
alpha = 0.05 threshold, which means that we only see LOD values by
chance this high or higher, 5% of the time. This is <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1206241/" class="external-link">one way</a>
of estimating a significance threshold for QTL plots.</p>
<p>To get estimated significance thresholds, use the function
<code>summary()</code>.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perm_add</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (1000 permutations)
     log10_insulin_10wk
0.05               3.87</code></pre>
</div>
<p>The default is to return the 5% significance threshold. Thresholds
for other (or for multiple) significance levels can be obtained via the
<code>alpha</code> argument.</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perm_add</span>, </span>
<span>        alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (1000 permutations)
     log10_insulin_10wk
0.2                3.18
0.05               3.87</code></pre>
</div>
</section><section><h2 class="section-heading" id="estimating-an-x-chromosome-specific-threshold">Estimating an X Chromosome Specific Threshold<a class="anchor" aria-label="anchor" href="#estimating-an-x-chromosome-specific-threshold"></a>
</h2>
<hr class="half-width">
<p>To obtain autosome/X chromosome-specific significance thresholds,
specify <code>perm_Xsp=TRUE</code>. In this case, you need to provide
chromosome lengths, which may be obtained with the function
<code>chr_lengths()</code>.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perm_add2</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs   <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                       pheno       <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"log10_insulin_10wk"</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                       addcovar    <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                       n_perm      <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>                       perm_Xsp    <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                       chr_lengths <span class="op">=</span> <span class="fu">chr_lengths</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Separate permutations are performed for the autosomes and X
chromosome, and considerably more permutation replicates are needed for
the X chromosome. The computations take about twice as much time. See <a href="https://www.ncbi.nlm.nih.gov/pubmed/17028340" class="external-link">Broman et al. (2006)
Genetics 174:2151-2158</a>.</p>
<p>The significance thresholds are again derived via
<code>summary()</code>:</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perm_add2</span>, </span>
<span>        alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Autosome LOD thresholds (1000 permutations)
     log10_insulin_10wk
0.2                3.12
0.05               3.76

X chromosome LOD thresholds (14369 permutations)
     log10_insulin_10wk
0.2                3.14
0.05               3.84</code></pre>
</div>
</section><section><h2 class="section-heading" id="estimating-significance-thresholds-with-the-kinship-matrix">Estimating Significance Thresholds with the Kinship Matrix<a class="anchor" aria-label="anchor" href="#estimating-significance-thresholds-with-the-kinship-matrix"></a>
</h2>
<hr class="half-width">
<p>You may have noticed above that we did not include the kinship matrix
as an argument to <code>scan1perm</code>. We can include the LOCO
kinship matrices in our permutations, since this is how we mapped
insulin previously.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perm_add_loco</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                           pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                           kinship   <span class="op">=</span> <span class="va">kinship_loco</span>,</span>
<span>                           addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                           n_perm    <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span> </span></code></pre>
</div>
<p>How does this affect the significance threshold estimates?</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perm_add_loco</span>, </span>
<span>        alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (1000 permutations)
     log10_insulin_10wk
0.2                3.21
0.05               3.94</code></pre>
</div>
<p>There is not a large difference in the thresholds. Currently, we are
on the fence about using the kinship matrices to estimate significance
thresholds. In principle, the kinship matrices should be used because we
used them when mapping the phenotype. However, in practice, we often
find that the significance thresholds are similar to those obtained
without including the kinship matrices. Given that it also takes more
time to run the permutations with the kinship matrices, it seems
reasonable to exclude them.</p>
<p>We ran 1000 permutations of the insulin phenotype and estimated the
0.05 significance threshold with and without the kinship matrices. We
repeated this process 100 times and plotted the thresholds below.</p>
<figure><img src="../fig/perm_kinship.png" alt="Comparison of significance thresholds" class="figure mx-auto d-block"><div class="figcaption">Significance Thresholds with/without
Kinship</div>
</figure><p>The plot shows that the median significance threshold is the same.
The lines connecting the points denote matched simulations in which the
permutation order was the same. While the exact value of the LOD
threshold is different, the median value and the variance are
similar.</p>
</section><section><h2 class="section-heading" id="estimating-binary-model-significance-thresholds">Estimating Binary Model Significance Thresholds<a class="anchor" aria-label="anchor" href="#estimating-binary-model-significance-thresholds"></a>
</h2>
<hr class="half-width">
<p>As with <code>scan1</code>, we can use <code>scan1perm</code> with
binary traits, using the argument <code>model="binary"</code>. Again,
this can’t be used with a kinship matrix, but all of the other arguments
can be applied.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perm_bin</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                      pheno     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">[</span>,<span class="st">"agouti_tan"</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                      addcovar  <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                      n_perm    <span class="op">=</span> <span class="fl">1000</span>, </span>
<span>                      perm_Xsp  <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                      chr_lengths <span class="op">=</span> <span class="fu">chr_lengths</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">)</span>,</span>
<span>                      model     <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span></code></pre>
</div>
<p>Here are the estimated 5% and 20% significance thresholds.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perm_bin</span>, </span>
<span>        alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Autosome LOD thresholds (1000 permutations)
     agouti_tan
0.2        3.19
0.05       3.84

X chromosome LOD thresholds (14369 permutations)
     agouti_tan
0.2        3.13
0.05       3.89</code></pre>
</div>
</section><section><h2 class="section-heading" id="selecting-the-number-of-permutations">Selecting the Number of Permutations<a class="anchor" aria-label="anchor" href="#selecting-the-number-of-permutations"></a>
</h2>
<hr class="half-width">
<p>How do we know how many permutations to perform in order to obtain a
good estimate of the significance threshold? Could we get a good
estimate with 10 permutations? 100? 1000?</p>
<p>When we run more permutations, we decrease the variance of the
threshold estimate.</p>
<figure><img src="../fig/permutation_simulations.png" alt="Figure showing decreasing variance of significance threshold estimates with increasing permutations" class="figure mx-auto d-block"><div class="figcaption">Significance Threshold Variance</div>
</figure><p>In the figure above, we performed 10, 100, or 1000 permutations 1000
times and recorded the 0.05 significance threshold each time. We plotted
the significance threshold versus the number of permutations and
overlaid violin plots showing the median value. Note that the variance
of the significance threshold estimate is higher at lower numbers of
permutations. With 1000 permutations, the variance decreases. The table
below shows the number of permutations and the mean and standard
deviation of the significance threshold. With 1000 permutations, the
estimate is 3.86 and the standard deviation is 0.064, which is an
acceptable value.</p>
<table class="table">
<thead><tr class="header">
<th>Num. Perm.</th>
<th>Mean</th>
<th>Std. Dev.</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>10</td>
<td>3.63</td>
<td>0.492</td>
</tr>
<tr class="even">
<td>100</td>
<td>3.81</td>
<td>0.195</td>
</tr>
<tr class="odd">
<td>1000</td>
<td>3.86</td>
<td>0.064</td>
</tr>
</tbody>
</table>
<!-- DMG: We need to review these challenges. --><p>The code below shuffles the phenotypes so that they no longer match
up with the genotypes. The purpose of this is to find out how high the
LOD score can be due to random chance alone.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">shuffled_order</span> <span class="op">&lt;-</span> <span class="fu">sample</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pheno_permuted</span> <span class="op">&lt;-</span> <span class="va">cross</span><span class="op">$</span><span class="va">pheno</span></span>
<span><span class="fu">rownames</span><span class="op">(</span><span class="va">pheno_permuted</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">shuffled_order</span></span>
<span><span class="va">xcovar_permuted</span> <span class="op">&lt;-</span> <span class="va">addcovar</span></span>
<span><span class="fu">rownames</span><span class="op">(</span><span class="va">xcovar_permuted</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">shuffled_order</span></span>
<span><span class="va">out_permuted</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                          pheno <span class="op">=</span> <span class="va">pheno_permuted</span>, </span>
<span>                         Xcovar <span class="op">=</span> <span class="va">xcovar_permuted</span><span class="op">)</span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">out_permuted</span>, map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">shuffled_order</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1:</h3>
<div class="callout-content">
<p>Run the preceding code to shuffle the phenotype data and plot a
genome scan with this shuffled (permuted) data.</p>
<p>What is the maximum LOD score in the scan from this permuted
data?<br>
How does it compare to the maximum LOD scores obtained from the earlier
scan?<br>
How does it compare to the 5% and 20% LOD thresholds obtained
earlier?<br>
Paste the maximum LOD score in the scan from your permuted data into the
etherpad.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">

</div>
</div>
</div>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Find the 1% and 10% significance thresholds for the first set of
permutations contained in the object <code>perm_add_loco</code>.<br>
</li>
<li>What do the 1% and 10% significance thresholds say about LOD
scores?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>Use the <code>alpha</code> argument to supply the desired
significance thresholds.</li>
</ol>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perm_add_loco</span>, alpha <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.01</span>, <span class="fl">0.10</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (1000 permutations)
     log10_insulin_10wk
0.01               4.86
0.1                3.57</code></pre>
</div>
<ol start="2" style="list-style-type: decimal">
<li>These LOD thresholds indicate maximum LOD scores that can be
obtained by random chance at the 1% and 10% significance levels. We
expect to see LOD values this high or higher 1% and 10% of the time
respectively.</li>
</ol>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>A permutation test establishes the statistical significance of a
genome scan.</li>
<li>1,000 permutations provides a good estimate of the significance
threshold.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-find-lod-peaks"><p>Content from <a href="find-lod-peaks.html">Finding QTL peaks</a></p>
<hr>
<p>Last updated on 2024-11-13 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/find-lod-peaks.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 60 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I locate QTL peaks above a certain LOD threshold value?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Locate QTL peaks above a LOD threshold value throughout the
genome.</li>
<li>Identify the Bayesian credible interval for a QTL peak.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor1" aria-labelledby="headingInstructor1">
<div class="accordion-body">
<p>We need the following block for the site to build on Github. The
students do not need to see or run the next block.</p>
</div>
</div>
</div>
</div>
<p>Once we have LOD scores from a genome scan and a significance
threshold, we can look for QTL peaks associated with the phenotype. High
LOD scores indicate the neighborhood of a QTL but don’t give its precise
position. To find the exact position of a QTL, we define an interval
that is likely to hold the QTL.</p>
<p>We’ll use the Bayesian credible interval, which is a method for
defining QTL intervals. It describes the probability that the interval
contains the true value. Credible intervals make a probabilistic
statement about the true value, for example, a 95% credible interval
states that there is a 95% chance that the true value lies within the
interval.</p>
<p>Let’s remind ourselves how the genome scan for insulin looks.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">thr</span> <span class="op">&lt;-</span> <span class="fu">summary</span><span class="op">(</span><span class="va">perm_add_loco</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_add_loco</span>,</span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>           main <span class="op">=</span> <span class="st">"Insulin"</span><span class="op">)</span></span>
<span><span class="fu">add_threshold</span><span class="op">(</span>map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, thresholdA <span class="op">=</span> <span class="va">thr</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/find-lod-peaks-rendered-plot_scan1-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>To find peaks above a given threshold LOD value, use the function
<code>find_peaks()</code>. It can also provide Bayesian credible
intervals by using the argument <code>prob</code> (the nominal coverage
for the Bayes credible intervals). Set the argument
<code>expand2markers = FALSE</code> to keep from expanding the interval
out to genotyped markers, or exclude this argument if you’d like to
include flanking markers.</p>
<p>You need to provide both the <code>scan1()</code> output, the marker
map and a threshold. We will use the 95% threshold from the permutations
in the previous lesson.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           threshold    <span class="op">=</span> <span class="va">thr</span>, </span>
<span>           prob         <span class="op">=</span> <span class="fl">0.95</span>, </span>
<span>           expand2markers <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr       pos      lod      ci_lo     ci_hi
1        1 log10_insulin_10wk   2 138.94475 7.127351  64.949395 149.57739
2        1 log10_insulin_10wk   7 144.18230 5.724018 139.368290 144.18230
3        1 log10_insulin_10wk  12  25.14494 4.310493  15.834815  29.05053
4        1 log10_insulin_10wk  14  22.24292 3.974322   6.241951  45.93876
5        1 log10_insulin_10wk  16  80.37433 4.114024  10.238134  80.37433
6        1 log10_insulin_10wk  19  54.83012 5.476587  48.370980  55.15007</code></pre>
</div>
<p>In the table above, we have one peak per chromosome because that is
the default behavior of <code>find_peaks()</code>. The
<code>find_peaks()</code> function can also pick out multiple peaks on a
chromosome: each peak must exceed the chosen threshold, and the argument
<code>peakdrop</code> indicates the amount that the LOD curve must drop
between the lowest of two adjacent peaks. Use this feature with
caution.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           threshold    <span class="op">=</span> <span class="va">thr</span>, </span>
<span>           peakdrop     <span class="op">=</span> <span class="fl">1.8</span>, </span>
<span>           prob         <span class="op">=</span> <span class="fl">0.95</span>, </span>
<span>           expand2markers <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr       pos      lod      ci_lo     ci_hi
1        1 log10_insulin_10wk   2 138.94475 7.127351  64.949395 149.57739
2        1 log10_insulin_10wk   7 144.18230 5.724018 139.368290 144.18230
3        1 log10_insulin_10wk  12  25.14494 4.310493  15.834815  29.05053
4        1 log10_insulin_10wk  14  22.24292 3.974322   6.241951  45.93876
5        1 log10_insulin_10wk  16  17.48123 3.995627   5.604167  37.99110
6        1 log10_insulin_10wk  16  80.37433 4.114024  74.250773  80.37433
7        1 log10_insulin_10wk  19  54.83012 5.476587  48.370980  55.15007</code></pre>
</div>
<p>Each row shows a different peak; the columns show the peak location,
LOD score and the lower and upper interval endpoints. Note that we now
have two peaks on chromosome 16, one at 17.5 Mb and one at 80.4 Mb.</p>
<div id="challenge-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1" class="callout-inner">
<h3 class="callout-title">Challenge 1</h3>
<div class="callout-content">
<p>Find peaks in the genome scan object called <code>lod_add_loco</code>
that meet a threshold of 3 and are in the interval described by a 2
point LOD drop from the peak. How many peaks meet the LOD threshold of 3
and lie within the interval defined by a 2 point LOD drop from the
maximum peaks on each chromosome?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>           threshold    <span class="op">=</span> <span class="fl">3</span>, </span>
<span>           drop         <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr       pos      lod      ci_lo     ci_hi
1        1 log10_insulin_10wk   2 138.94475 7.127351  63.943187 156.83772
2        1 log10_insulin_10wk   5 103.41486 3.130862  41.967549 132.28428
3        1 log10_insulin_10wk   7 144.18230 5.724018 129.414016 144.18230
4        1 log10_insulin_10wk   9  83.67606 3.865635  17.504307 111.02206
5        1 log10_insulin_10wk  12  25.14494 4.310493   9.998200  34.23274
6        1 log10_insulin_10wk  14  22.24292 3.974322   6.241951  68.04655
7        1 log10_insulin_10wk  16  80.37433 4.114024   3.804882  96.52406
8        1 log10_insulin_10wk  19  54.83012 5.476587  47.361847  56.37100</code></pre>
</div>
<p>This produces 8 peaks on 8 different chromosomes that meet a LOD
threshold of 3 and are within the interval defined by a 2-LOD drop from
the maximum peak on each chromosome.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<p>1). Calculate the 90% Bayes credible interval. For chromosome 2, what
is the range of the interval that has a 90% chance of containing the
true QTL position?<br>
2). Compare with the 95% Bayes credible interval calculated earlier. How
does the interval change as you increase the probability? Why?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>1). This produces a range from 118.0 to 149.6 Mb.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pks</span> <span class="op">=</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>                 map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>                 prob         <span class="op">=</span> <span class="fl">0.90</span>, </span>
<span>                 expand2markers <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">subset</span><span class="op">(</span><span class="va">pks</span>, <span class="va">chr</span> <span class="op">==</span> <span class="st">'2'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr      pos      lod    ci_lo    ci_hi
1        1 log10_insulin_10wk   2 138.9448 7.127351 117.9557 149.5774</code></pre>
</div>
<p>2). This produces a range from 64.9 to 149.6 Mb, which is much
broader than the 90% interval. The interval widens because the
probability that the interval contains the true QTL position has
increased.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pks</span> <span class="op">=</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>                 map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>                 prob         <span class="op">=</span> <span class="fl">0.95</span>, </span>
<span>                 expand2markers <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu">subset</span><span class="op">(</span><span class="va">pks</span>, <span class="va">chr</span> <span class="op">==</span> <span class="st">'2'</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr      pos      lod    ci_lo    ci_hi
1        1 log10_insulin_10wk   2 138.9448 7.127351 64.94939 149.5774</code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>LOD peaks and support intervals can be identified with
<code>find_peaks()</code>.</li>
<li>The Bayesian Credible Interval estimates the width of the support
interval around a QTL peak.</li>
<li>Using a higher <code>prob</code> value for the Bayesian Credible
Interval results in a wider support interval.</li>
</ul>
</div>
</div>
</div></section><section id="aio-est-qtl-effects"><p>Content from <a href="est-qtl-effects.html">Estimating QTL effects</a></p>
<hr>
<p>Last updated on 2024-11-13 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/est-qtl-effects.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 30 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I find the founder allele effects at a QTL peak?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Estimate the founder allele effects at a QTL peak.</li>
<li>Plot the estimated founder allele effects.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="estimating-founder-allele-effects">Estimating Founder Allele Effects<a class="anchor" aria-label="anchor" href="#estimating-founder-allele-effects"></a>
</h2>
<hr class="half-width">
<p>Recall that to model data from a cross, we use</p>
<p><span class="math inline">\(y_j = \mu + \beta_k G_{jk} +
\epsilon_j\)</span></p>
<p>where <span class="math inline">\(y_{ij}\)</span> is the phenotype of
the <span class="math inline">\(j\)</span>th individual, <span class="math inline">\(\mu\)</span> is the mean phenotype value, <span class="math inline">\(\beta_k\)</span> is the effect of the <span class="math inline">\(k\)</span>th genotype, <span class="math inline">\(G_{jk}\)</span> is the genotype for individual
<span class="math inline">\(j\)</span>, and <span class="math inline">\(\epsilon_j\)</span> is the error for the <span class="math inline">\(j\)</span>th individual. In the figure below,
<span class="math inline">\(\mu\)</span> equals 94.6 and <span class="math inline">\(\beta\)</span> equals 15 for the alternative
hypothesis (QTL exists).</p>
<figure><img src="../fig/nullvalt.png" alt="Null and alternative hypotheses" class="figure mx-auto d-block"></figure><p>This linear model is <span class="math inline">\(y\)</span> = 94.6 +
15<span class="math inline">\(G\)</span> + <span class="math inline">\(\epsilon\)</span>. The model intersects the
genotype groups at their group means, and is based <span class="math inline">\(\mu\)</span> and <span class="math inline">\(\beta_k\)</span>.</p>
<p>The effect of genotype RR (the <span class="math inline">\(\beta\)</span> for the RR genotype) at the marker
is 15, while the effect of the BB genotype is -15 on the phenotype. The
effect of the BR genotype is 0 relative to <span class="math inline">\(\mu\)</span> equals 94.6.</p>
<p>The <code>scan1()</code> function returns only LOD scores. To obtain
estimated QTL effects, use the function <code>scan1coef()</code>. This
function takes a single phenotype and the genotype probabilities for a
single chromosome and returns a matrix with the estimated coefficients
at each putative QTL location along the chromosome.</p>
<p>For example, to get the estimated QTL effects on chromosome 19 for
the insulin phenotype, we would provide the chromosome 19 genotype
probabilities and the insulin phenotype to the function
<code>scan1coef()</code> as follows:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>       <span class="op">&lt;-</span> <span class="st">"19"</span></span>
<span><span class="va">eff_chr19</span> <span class="op">&lt;-</span> <span class="fu">scan1coef</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                       pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                       kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                       addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>The result is a matrix of 50 positions <span class="math inline">\(\times\)</span> 3 genotypes. Additional columns
contain the sex and intercept values (<span class="math inline">\(\mu\)</span>).</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">eff_chr19</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                BB      BR     RR SexMale intercept
rs4232073  -0.0387 -0.0275 0.0661  -0.169      1.01
rs13483548 -0.0386 -0.0272 0.0658  -0.168      1.01
rs13483549 -0.0388 -0.0272 0.0660  -0.169      1.01
rs13483550 -0.0413 -0.0275 0.0688  -0.169      1.01
rs13483554 -0.0374 -0.0325 0.0698  -0.170      1.01
rs13483555 -0.0367 -0.0288 0.0655  -0.170      1.01</code></pre>
</div>
</section><section><h2 class="section-heading" id="plotting-founder-allele-effects-along-a-chromosome">Plotting Founder Allele Effects Along a Chromosome<a class="anchor" aria-label="anchor" href="#plotting-founder-allele-effects-along-a-chromosome"></a>
</h2>
<hr class="half-width">
<p>To plot the QTL effects, use the <code>plot_coef()</code> function.
Add the LOD plot to the <code>scan1_output</code> argument to include a
LOD plot at the bottom.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x            <span class="op">=</span> <span class="va">eff_chr19</span>, </span>
<span>          map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>          legend       <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/est-qtl-effects-rendered-plot_coef-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The plot shows effect values on the y-axis and cM values on the
x-axis. The value of the intercept (<span class="math inline">\(\mu\)</span>) appears at the top. The effect of
the BR genotype is centered around zero, with the effects of the other
two genotypes above and below. We are usually not directly interested in
how the additive covariates change across the genome, but rather, how
the founder allele effects change.</p>
<p>To plot only the founder allele effects, use the argument
<code>columns</code> to indicate which coefficients to plot. Let’s look
at the columns which contain the founder allele effects.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">eff_chr19</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>                BB      BR     RR SexMale intercept
rs4232073  -0.0387 -0.0275 0.0661  -0.169      1.01
rs13483548 -0.0386 -0.0272 0.0658  -0.168      1.01
rs13483549 -0.0388 -0.0272 0.0660  -0.169      1.01
rs13483550 -0.0413 -0.0275 0.0688  -0.169      1.01
rs13483554 -0.0374 -0.0325 0.0698  -0.170      1.01
rs13483555 -0.0367 -0.0288 0.0655  -0.170      1.01</code></pre>
</div>
<p>We would like to plot the columns “BB”, “BR”, and “RR”, which are in
columns 1 through 3. This is what we pass into the <code>columns</code>
argument.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">eff_chr19</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>          main    <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="va">chr</span>, <span class="st">"QTL effects and LOD scores"</span><span class="op">)</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/est-qtl-effects-rendered-plot_effects_chr19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="challenge-1-founder-allele-effects" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-founder-allele-effects" class="callout-inner">
<h3 class="callout-title">Challenge 1: Founder Allele Effects</h3>
<div class="callout-content">
<p>Looking at the plot above, which founder allele contributes to higher
insulin levels?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>The BTBR allele contributes to higher insulin levels because the “RR”
allele effect (green line) is higher than the other allele effect
lines.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="estimating-founder-allele-effects-using-blups">Estimating Founder Allele Effects using BLUPs<a class="anchor" aria-label="anchor" href="#estimating-founder-allele-effects-using-blups"></a>
</h2>
<hr class="half-width">
<p>Another option for estimating the founder allele effects is to treat
them as <a href="https://stats.stackexchange.com/questions/4700/what-is-the-difference-between-fixed-effect-random-effect-and-mixed-effect-mode#151800" class="external-link">random
effects</a> and calculate <a href="https://en.wikipedia.org/wiki/Best_linear_unbiased_prediction" class="external-link">Best
Linear Unbiased Predictors</a> (BLUPs). This is particularly valuable
for multi-parent populations such as the Collaborative Cross and
Diversity Outbred mice, where the large number of possible genotypes at
a QTL leads to considerable variability in the effect estimates. To
calculate BLUPs, use <code>scan1blup()</code>; it takes the same
arguments as <code>scan1coef()</code>, including the option of a kinship
matrix to account for a residual polygenic effect.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blup_chr19</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                        pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                        kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                        addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>We can plot the BLUP effects using <code>plot_coef</code> as
before.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">blup_chr19</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>          main    <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="va">chr</span>, <span class="st">"QTL BLUP effects and LOD scores"</span><span class="op">)</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/est-qtl-effects-rendered-plot_blup_chr19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot below, we plotted the founder allele effects (solid
lines) and the BLUPs (dashed lines). In this case, the effects are not
greatly different, but the effects are “shrunken” toward zero.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">eff_chr19</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          main    <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="va">chr</span>, <span class="st">"QTL BLUP effects and LOD scores"</span><span class="op">)</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span>
<span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">blup_chr19</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,</span>
<span>          lty     <span class="op">=</span> <span class="fl">2</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span>,</span>
<span>          add     <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/est-qtl-effects-rendered-plot_blup_chr19_again-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure></section><section><h2 class="section-heading" id="plotting-allele-effects-at-one-marker">Plotting Allele Effects at One Marker<a class="anchor" aria-label="anchor" href="#plotting-allele-effects-at-one-marker"></a>
</h2>
<hr class="half-width">
<p>You may also want plot the founder allele effects at the marker with
the highest LOD. To do this, you first need to get the position of the
marker from the peak list.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>,</span>
<span>                    map          <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>                    threshold    <span class="op">=</span> <span class="va">thr</span><span class="op">)</span></span>
<span><span class="va">peaks</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr   pos  lod
1        1 log10_insulin_10wk   2 138.9 7.13
2        1 log10_insulin_10wk   7 144.2 5.72
3        1 log10_insulin_10wk  12  25.1 4.31
4        1 log10_insulin_10wk  14  22.2 3.97
5        1 log10_insulin_10wk  16  80.4 4.11
6        1 log10_insulin_10wk  19  54.8 5.48</code></pre>
</div>
<p>The position of the maximum LOD on chromosome 19 is 54.83 Mb. We can
pass this value into the <code>qtl2</code> function
<code>pull_genoprobpos</code> to get the genoprobs at this marker.</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">max_pos</span> <span class="op">&lt;-</span> <span class="fu">subset</span><span class="op">(</span><span class="va">peaks</span>, <span class="va">chr</span> <span class="op">==</span> <span class="st">'19'</span><span class="op">)</span><span class="op">$</span><span class="va">pos</span></span>
<span><span class="va">max_mkr</span> <span class="op">&lt;-</span> <span class="fu">find_marker</span><span class="op">(</span>map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>                       chr <span class="op">=</span> <span class="va">chr</span>, </span>
<span>                       pos <span class="op">=</span> <span class="va">max_pos</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pr</span>      <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                            marker    <span class="op">=</span> <span class="va">max_mkr</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-2-structure-of-genoprobs-at-one-marker" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-structure-of-genoprobs-at-one-marker" class="callout-inner">
<h3 class="callout-title">Challenge 2: Structure of Genoprobs at One Marker</h3>
<div class="callout-content">
<p>What does the structure of <code>pr</code> look like? How many rows
and columns does it have?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>The <a href="https://www.rdocumentation.org/packages/utils/versions/3.6.2/topics/str" class="external-link">str</a>
function provides the structure of an R object.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">str</span><span class="op">(</span><span class="va">pr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> num [1:490, 1:3] 1.09e-08 1.09e-08 3.06e-14 1.09e-08 1.09e-08 ...
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:490] "Mouse3051" "Mouse3551" "Mouse3430" "Mouse3476" ...
  ..$ : chr [1:3] "BB" "BR" "RR"</code></pre>
</div>
<p>In this case, we can see that <code>pr</code> is a numeric matrix
because the first line starts with “num”. The next item on the first
line provides the dimensions of the object. <code>pr</code> is a matrix
because there are two dimensions, rows and columns. There are 490 rows
and 3 columns in <code>pr</code>.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3-genoprobs-at-one-marker" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-genoprobs-at-one-marker" class="callout-inner">
<h3 class="callout-title">Challenge 3: Genoprobs at One Marker</h3>
<div class="callout-content">
<p>Look at the top few rows of <code>pr</code>. You may want to round
the values to make them easier to read. What is the genotype of
<code>Mouse3051</code>? What about <code>Mouse3430</code>?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>We can use the <a href="">head</a> function to see the top few rows
of <code>pr</code>.</p>
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="fu">round</span><span class="op">(</span><span class="va">pr</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          BB BR RR
Mouse3051  0  1  0
Mouse3551  0  1  0
Mouse3430  0  0  1
Mouse3476  0  1  0
Mouse3414  0  1  0
Mouse3145  0  1  0</code></pre>
</div>
<p>From the listing above, the probability that Mouse3051 carries the
“BR” genotype at this marker is close to 1, so this is the most likely
genotype.</p>
<p>the probability that Mouse3430 carries the “RR” genotype at this
marker is close to 1.</p>
</div>
</div>
</div>
</div>
<p><code>pr</code> is a numeric matrix with 490 rows and 3 columns. The
rownames contain the mouse IDs and the column names contain the
genotypes.</p>
<p>We can then pass <code>pr</code> as an argument into the
<code>fit1</code> function, which fits the mapping model at a single
marker.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod</span> <span class="op">=</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr</span>,</span>
<span>           pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>           kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>           addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>Then, we can plot the founder allele effects and their standard
error.</p>
<div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" aria-labelledby="headingInstructor1" data-bs-parent="#accordionInstructor1">
<div class="accordion-body">
<p>Don’t have the students type all of this. Copy and paste or show this
code in a web browser.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod_eff</span> <span class="op">=</span> <span class="fu">data.frame</span><span class="op">(</span>eff <span class="op">=</span> <span class="va">mod</span><span class="op">$</span><span class="va">coef</span>, </span>
<span>                     se  <span class="op">=</span> <span class="va">mod</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"genotype"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">filter</span><span class="op">(</span><span class="va">genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"BB"</span>, <span class="st">"BR"</span>, <span class="st">"RR"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data    <span class="op">=</span> <span class="va">mod_eff</span>, </span>
<span>       mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">genotype</span>, y <span class="op">=</span> <span class="va">eff</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_pointrange</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>ymin <span class="op">=</span> <span class="va">eff</span> <span class="op">-</span> <span class="va">se</span>, </span>
<span>                                ymax <span class="op">=</span> <span class="va">eff</span> <span class="op">+</span> <span class="va">se</span><span class="op">)</span>,</span>
<span>                  size       <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>                  linewidth  <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Founder Allele Effects on Chr"</span>, <span class="va">chr</span><span class="op">)</span>,</span>
<span>       x     <span class="op">=</span> <span class="st">"Genotype"</span>, y <span class="op">=</span> <span class="st">"Founder Allele Effects"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/est-qtl-effects-rendered-plot_fit1-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="challenge-4-allele-effects" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-allele-effects" class="callout-inner">
<h3 class="callout-title">Challenge 4: Allele Effects</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>In the plot above, which founder allele contributes to higher
insulin levels?</li>
<li>Is that consistent with the plot created using
<code>plot_coef</code> above?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>The R (BTBR) allele contributes to higher insulin levels on
chromosome 19.</li>
<li>Yes, the BTBR allele also contributed to higher insulin level in the
founder allele effects plot.</li>
</ol>
</div>
</div>
</div>
</div>
<div id="challenge-5-allele-effects-on-chromosome-7" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-5-allele-effects-on-chromosome-7" class="callout-inner">
<h3 class="callout-title">Challenge 5: Allele Effects on Chromosome 7</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Use <code>scan1_blup</code> and <code>plot_coef</code> to look at
the founder allele effects at the chromosome 7 QTL peak. Does the
pattern of allele effects look different? Which founder allele
contributes to higher insulin levels?</li>
<li>Get the genoprobs at the marker with the highest LOD on chromosome 7
and use <code>fit1</code> and the plotting code for <code>fit1</code>
above to examine the founder allele effects at the peak marker. Again,
how do these allele effects differ from the allele effects on chromosome
19? Which founder allele contributes to higher insulin levels?</li>
</ol>
<p>Be careful to rename the ouput objects so that you don’t overwrite
the values from chromosome 19.</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>We will use <code>scan1_blup</code> and <code>plot_coef</code> to
estimate the founder allele effects on chromosome 7.</li>
</ol>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blup_chr7</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="st">"7"</span><span class="op">]</span>,</span>
<span>                       pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                       kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="st">"7"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                       addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">blup_chr7</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>          main    <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="st">"7"</span>, <span class="st">"QTL BLUP effects and LOD scores"</span><span class="op">)</span>,</span>
<span>          legend  <span class="op">=</span> <span class="st">"topleft"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/est-qtl-effects-rendered-challenge5a-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>On chromosome 19, the BB and BR genotypes seems to have the same
effect. On chromosome 7, there is a clear split between the BB, BR, and
RR genotypes. Each additional B allele raises insulin levels. The
C57BL/6J (B) allele contributes to higher insulin levels.</p>
<ol start="2" style="list-style-type: decimal">
<li>We will get the marker with the highest LOD on chromosome 7, then
get the genoprobs at that marker and use them in <code>fit1</code> to
estimate the founder allele effects.</li>
</ol>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">max_pos</span> <span class="op">&lt;-</span> <span class="fu">subset</span><span class="op">(</span><span class="va">peaks</span>, <span class="va">chr</span> <span class="op">==</span> <span class="st">"7"</span><span class="op">)</span><span class="op">$</span><span class="va">pos</span></span>
<span><span class="va">max_mkr</span> <span class="op">&lt;-</span> <span class="fu">find_marker</span><span class="op">(</span>map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>                       chr <span class="op">=</span> <span class="st">"7"</span>, </span>
<span>                       pos <span class="op">=</span> <span class="va">max_pos</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pr7</span>     <span class="op">&lt;-</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                            marker    <span class="op">=</span> <span class="va">max_mkr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb22">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod7</span> <span class="op">=</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr7</span>,</span>
<span>            pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>            kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="st">"7"</span><span class="op">]</span><span class="op">]</span>, </span>
<span>            addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod_eff7</span> <span class="op">=</span> <span class="fu">data.frame</span><span class="op">(</span>eff <span class="op">=</span> <span class="va">mod7</span><span class="op">$</span><span class="va">coef</span>, </span>
<span>                      se  <span class="op">=</span> <span class="va">mod7</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"genotype"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">filter</span><span class="op">(</span><span class="va">genotype</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="st">"BB"</span>, <span class="st">"BR"</span>, <span class="st">"RR"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data    <span class="op">=</span> <span class="va">mod_eff7</span>, </span>
<span>       mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">genotype</span>, y <span class="op">=</span> <span class="va">eff</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_pointrange</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>ymin <span class="op">=</span> <span class="va">eff</span> <span class="op">-</span> <span class="va">se</span>, </span>
<span>                                ymax <span class="op">=</span> <span class="va">eff</span> <span class="op">+</span> <span class="va">se</span><span class="op">)</span>,</span>
<span>                  size       <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>                  linewidth  <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Founder Allele Effects on Chr"</span>, <span class="st">"7"</span><span class="op">)</span>,</span>
<span>       x     <span class="op">=</span> <span class="st">"Genotype"</span>, y <span class="op">=</span> <span class="st">"Founder Allele Effects"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/est-qtl-effects-rendered-challenge5d-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>On chromosome 19, mice carrying the BB and BR genotypes had almost
the same levels of insulin. On chromosome 7, the BB genotype has the
highest insulin levels, the BR genotype has the next highest, and the RR
genotype has the lowest insulin levels.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="estimating-additive-and-dominant-allele-effects">Estimating Additive and Dominant Allele Effects<a class="anchor" aria-label="anchor" href="#estimating-additive-and-dominant-allele-effects"></a>
</h2>
<hr class="half-width">
<p>You may have noticed the difference in the pattern of allele effects
between chromosome 7 and 19. On chromosome 7, adding each “B” allele
increased insulin levels by the same amount. On chromosome 19, adding
one “R” allele did not change insulin levels. We saw an increase in
insulin only when we added two “R” alleles.</p>
<p>What we are seeing between these two QTL are “additive” and
“dominant” effects. At an additive QTL (which is different from additive
covariates), adding (or subtracting) one allele produces the same effect
between each genotype. At a dominant QTL, adding (or subtracting) two
alleles is required to see the effect. You can also have QTL which are a
mixture of additive and dominant effects.</p>
<p>We can map the additive and dominance effects by providing a matrix
of <em>contrasts</em> to indicate the different effects. Let’s look at
the additive and dominant effects on chromosome 7 first.</p>
<figure><img src="../fig/add_dom_qtl_effect.png" alt="Figure showing additive &amp; dominance QTL effects" class="figure mx-auto d-block"><div class="figcaption">Additive &amp; Dominance Effects</div>
</figure><p>In the figure above, we have simulated a QTL with a dominant effect.
Genotypes are shown on the X-axis and the phenotype on the Y-axis. Each
point represents the phenotype value for one mouse. The grey line is the
mean of the two homozygote groups. The additive effect is defined as the
difference between the homozygote mean (grey line) and the individual
homozygote group means (red lines). The blue line shows the heterozygote
mean. The dominance effect is the difference between the homozygote mean
(grey line) and the heterozygote mean (blue line).</p>
<p>First, we create a <em>contrasts</em> matrix to indicate the mean,
additive, and dominance effects that we want to estimate.</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">add_dom_contr</span> <span class="op">&lt;-</span> <span class="fu">cbind</span><span class="op">(</span>mu <span class="op">=</span> <span class="fu">c</span><span class="op">(</span> <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                       a  <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                       d  <span class="op">=</span> <span class="fu">c</span><span class="op">(</span> <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">c7effB</span> <span class="op">&lt;-</span> <span class="fu">scan1coef</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="st">"7"</span><span class="op">]</span>, </span>
<span>                    pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                    kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="st">"7"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                    addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                    contrasts <span class="op">=</span> <span class="va">add_dom_contr</span><span class="op">)</span></span></code></pre>
</div>
<p>The result will then contain the estimates of <code>mu</code>,
<code>a</code> (the additive effect), and <code>d</code> (the dominance
effect).</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">c7effB</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 109   4</code></pre>
</div>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">c7effB</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             mu       a        d SexMale
rs8252589  1.01 -0.0267 -0.00378  -0.175
rs13479104 1.01 -0.0228  0.00409  -0.175
rs13479112 1.01 -0.0234  0.00991  -0.175
rs13479114 1.00 -0.0226  0.01148  -0.175
rs13479120 1.01 -0.0238  0.00899  -0.175
rs13479124 1.01 -0.0219  0.00375  -0.175</code></pre>
</div>
<p>For marker rs13479570, <code>mu</code>, <code>a</code>, and
<code>d</code> are 1.016, -0.111, -0.03, -0.164.</p>
<p>Here’s a plot of the chromosome 7 additive and dominance effects,
which are in the second and third columns.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">c7effB</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">[</span><span class="st">"7"</span><span class="op">]</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          col     <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,</span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>,</span>
<span>          main    <span class="op">=</span> <span class="st">"log10(insulin): Chromosome 7"</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span><span class="st">'bottomleft'</span>, lwd <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, legend <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"additive"</span>, <span class="st">"dominance"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/est-qtl-effects-rendered-plot_add_dom_contrasts-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot above, the dominance effects remain near zero across the
chromosome. The additive effects decrease as we move</p>
<div id="challenge-6-estimating-additive-and-dominance-effects-on-chromsome-19" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-estimating-additive-and-dominance-effects-on-chromsome-19" class="callout-inner">
<h3 class="callout-title">Challenge 6: Estimating Additive and Dominance Effects on Chromsome 19</h3>
<div class="callout-content">
<p>Use the code above to estimate and plot the additive and dominance
effects along chromosome 19. You can use the same contrasts matrix as
before. How dow they differ from the effects on chromosome 7?</p>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">
<p>First we estimate the effects using <code>scan1coef</code>.</p>
<div class="codewrapper sourceCode" id="cb31">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">c19effB</span> <span class="op">&lt;-</span> <span class="fu">scan1coef</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="st">"19"</span><span class="op">]</span>, </span>
<span>                     pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                     kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="st">"19"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                     addcovar  <span class="op">=</span> <span class="va">addcovar</span>,</span>
<span>                     contrasts <span class="op">=</span> <span class="va">add_dom_contr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coef</span><span class="op">(</span>x       <span class="op">=</span> <span class="va">c19effB</span>, </span>
<span>          map     <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">[</span><span class="st">"19"</span><span class="op">]</span>, </span>
<span>          columns <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, </span>
<span>          col     <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span>,</span>
<span>          scan1_output <span class="op">=</span> <span class="va">lod_add_loco</span>,</span>
<span>          main    <span class="op">=</span> <span class="st">"log10(insulin): Chromosome 19"</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span><span class="st">'bottomleft'</span>, lwd <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span>, </span>
<span>       legend <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"additive"</span>, <span class="st">"dominance"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/est-qtl-effects-rendered-challenge6b-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In this case, we see that the additive effect adds about 0.094 and
the dominance effect adds about -0.113.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="plotting-phenotypes-versus-genotypes">Plotting Phenotypes versus Genotypes<a class="anchor" aria-label="anchor" href="#plotting-phenotypes-versus-genotypes"></a>
</h2>
<hr class="half-width">
<p>Finally, to plot the raw phenotypes against the genotypes at a single
putative QTL position, you can use the function <code>plot_pxg()</code>.
This takes a vector of genotypes as produced by the
<code>maxmarg()</code> function, which picks the most likely genotype
from a set of genotype probabilities, provided it is greater than some
specified value (the argument <code>minprob</code>). Note that the
<code>marg</code> in <code>maxmarg</code> stands for “marginal”, as this
function is selecting the genotype at each position that has maximum
marginal probability.</p>
<p>For example, we could get inferred genotypes at the chr 19 QTL for
the insulin phenotype (at 28.6 cM) as follows:</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu">maxmarg</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>             map   <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>, </span>
<span>             chr   <span class="op">=</span> <span class="va">chr</span>, </span>
<span>             pos   <span class="op">=</span> <span class="va">max_pos</span>, </span>
<span>             return_char <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<p>We use <code>return_char = TRUE</code> to have <code>maxmarg()</code>
return a vector of character strings with the genotype labels.</p>
<p>We then plot the insulin phenotype against these genotypes as
follows:</p>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_pxg</span><span class="op">(</span>geno   <span class="op">=</span> <span class="va">g</span>, </span>
<span>         pheno  <span class="op">=</span> <span class="va">insulin</span>, </span>
<span>         SEmult <span class="op">=</span> <span class="fl">2</span>,</span>
<span>         main   <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Insulin vs Chr"</span>, <span class="va">chr</span> ,<span class="st">"Genotype"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/est-qtl-effects-rendered-plot_pheno_geno_se-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Estimated founder allele effects can be plotted from the mapping
model coefficients.</li>
<li>Additive and dominance effects can be plotted using contrasts.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-integrating-gene-expression"><p>Content from <a href="integrating-gene-expression.html">Integrating Gene Expression Data</a></p>
<hr>
<p>Last updated on 2024-11-13 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/integrating-gene-expression.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 80 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How can I use gene expression data to identify candidate genes?</li>
<li>What is expression QTL mapping?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Find genes which are correlated with a physiological phenotype.</li>
<li>Find genes which have expression QTL in the same position as a
physiological phenotype.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>Once you have QTL peak, the next step if often to identify genes
which may be causal candidates. However, there are often hundreds of
genes within the QTL support interval. How do you select candidate
genes?</p>
<p>Often, there is a tissue which is related to the phenotype that you
are measuring. For example, if you measure circulating cardiac troponin,
then gene expression in the heart may be relevant. For some complex
phenotypes, it may not be easy to select a tissue. For example, if the
liver creates a metabolite that causes kidney injury, which is then
exacerbated by the immune system, in which tissue would you measure gene
expression?</p>
<p>Susceptibility to type 2 diabetes involves a complex interaction
between several tissues. However, for the purposes of this tutorial in
which we mapped circulating insulin levels, it is reasonable to look at
gene expression in the pancreas.</p>
</section><section><h2 class="section-heading" id="reading-in-gene-expression-data">Reading in Gene Expression Data<a class="anchor" aria-label="anchor" href="#reading-in-gene-expression-data"></a>
</h2>
<hr class="half-width">
<p>Gene expression data consists of two parts: expression measurements
and gene annotation. In this study, gene expression was measured via
two-color microarray. The expression values represent a ratio of two
sets of RNA: a control set and the sample set. The expression values
represent the log-normalized ratio of these two sets.</p>
<p>We have prepared two files containing normalized gene expression data
and the gene annotation.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annot</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/attie_b6btbr_grcm39/attie_gene_annot.csv"</span>,</span>
<span>                  show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">mutate</span><span class="op">(</span>a_gene_id <span class="op">=</span> <span class="fu">as.character</span><span class="op">(</span><span class="va">a_gene_id</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">expr</span>  <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/attie_b6btbr_grcm39/attie_gene_expr.csv"</span>,</span>
<span>                  show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-1-size-of-data-structures." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-size-of-data-structures." class="callout-inner">
<h3 class="callout-title">Challenge 1: Size of data structures.</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>How many rows and columns are in <code>annot</code>?</li>
<li>What are the column names off <code>annot</code>?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>Dimensions of <code>annot</code>.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">annot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 29860    13</code></pre>
</div>
<p>Column names of <code>annot</code>.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colnames</span><span class="op">(</span><span class="va">annot</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code> [1] "a_gene_id"       "chr"             "start"           "end"
 [5] "width"           "strand"          "gene_id"         "gene_name"
 [9] "gene_biotype"    "description"     "gene_id_version" "symbol"
[13] "entrezid"       </code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="challenge-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2" class="callout-inner">
<h3 class="callout-title">Challenge 2</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>What are the dimensions of <code>expr</code>?</li>
<li>Look at the top 5 row by 5 column block of <code>expr</code>.</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>Dimensions of <code>expr</code>.</li>
</ol>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">expr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1]   490 29861</code></pre>
</div>
<ol start="2" style="list-style-type: decimal">
<li>Top-left block of <code>expr</code>.</li>
</ol>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code># A tibble: 5 × 5
  MouseNum  `497628` `497629` `497630` `497632`
  &lt;chr&gt;        &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
1 Mouse3051  -0.0790 -0.0114  -0.0790    0.0461
2 Mouse3551   0.0544 -0.0693  -0.0721   -0.408
3 Mouse3430   0.154  -0.0468  -0.00900  -0.298
4 Mouse3476   0.144  -0.0451  -0.00505   0.12
5 Mouse3414   0.266   0.00496 -0.0746   -0.127 </code></pre>
</div>
</div>
</div>
</div>
</div>
<p>Let’s look at the relationship between the annotation and the
expression data.</p>
<p>The annotation data has 29860 rows and the expression data has 29861
columns. The first column in <code>expr</code> contains the mouse ID and
the remaining columns contain the expression values for each gene. The
gene IDs are in the column names. These are Agilent gene IDs. They are
also in the <code>a_gene_id</code> column in the annotation.</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">all</span><span class="op">(</span><span class="va">annot</span><span class="op">$</span><span class="va">a_gene_id</span> <span class="op">==</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">expr</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] TRUE</code></pre>
</div>
<p>Now we know that the genes are aligned between the annotation and the
expression data. When you receive your own expression data, it is
critical that you align the genes in your expression data with your
annotation data.</p>
<p>We must also align the mouse IDs between the physiological phenotypes
and the expression data. We saw above that the mouse IDs are in the
first column of the expression data. Let’s see where the mouse IDs are
in the phenotype data.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          log10_insulin_10wk agouti_tan tufted
Mouse3051              1.399          1      0
Mouse3551              0.369          1      1
Mouse3430              0.860          0      1
Mouse3476              0.800          1      0
Mouse3414              1.370          0      0
Mouse3145              1.783          1      0</code></pre>
</div>
<p><code>cross$pheno</code> is a matrix which contains the mouse IDs in
the rownames. Let’s convert the expression data to a matrix as well.</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr</span> <span class="op">&lt;-</span> <span class="va">expr</span> <span class="op">|&gt;</span></span>
<span>         <span class="fu">column_to_rownames</span><span class="op">(</span>var <span class="op">=</span> <span class="st">"MouseNum"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>         <span class="fu">as.matrix</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<p>Now let’s check whether the mouse IDs are aligned between
<code>cross$pheno</code> and <code>expr</code>. Again, when you assemble
your phenotype and expression data, you will need to make sure that the
sample IDs are aligned between the data sets.</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">all</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">cross</span><span class="op">$</span><span class="va">pheno</span><span class="op">)</span> <span class="op">==</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">expr</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] TRUE</code></pre>
</div>
</section><section><h2 class="section-heading" id="identifying-genes-in-a-qtl-support-interval">Identifying Genes in a QTL Support Interval<a class="anchor" aria-label="anchor" href="#identifying-genes-in-a-qtl-support-interval"></a>
</h2>
<hr class="half-width">
<p>In previous episodes, we found significant QTL peaks using the
<code>find_peaks</code> function. Let’s look at those peaks again.</p>
<div class="codewrapper sourceCode" id="cb17">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peaks</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr   pos  lod  ci_lo ci_hi
1        1 log10_insulin_10wk   2 138.9 7.13  64.95 149.6
2        1 log10_insulin_10wk   7 144.2 5.72 139.37 144.2
3        1 log10_insulin_10wk  12  25.1 4.31  15.83  29.1
4        1 log10_insulin_10wk  14  22.2 3.97   6.24  45.9
5        1 log10_insulin_10wk  16  80.4 4.11  10.24  80.4
6        1 log10_insulin_10wk  19  54.8 5.48  48.37  55.2</code></pre>
</div>
<p>We looked at the QTL peak on chromosome 19 in a previous lesson. The
QTL interval is 6.779 Mb wide. This is quite wide. Let’s get the genes
expressed in the pancreas within this region.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>         <span class="op">&lt;-</span> <span class="st">'19'</span></span>
<span><span class="va">peaks_chr19</span> <span class="op">&lt;-</span> <span class="fu">filter</span><span class="op">(</span><span class="va">peaks</span>, <span class="va">chr</span> <span class="op">==</span> <span class="st">'19'</span><span class="op">)</span></span>
<span><span class="va">annot_chr19</span> <span class="op">&lt;-</span> <span class="fu">filter</span><span class="op">(</span><span class="va">annot</span>, <span class="va">chr</span> <span class="op">==</span> <span class="st">'19'</span> <span class="op">&amp;</span> <span class="va">start</span> <span class="op">&gt;</span> <span class="va">peaks_chr19</span><span class="op">$</span><span class="va">ci_lo</span> <span class="op">&amp;</span> <span class="va">end</span> <span class="op">&lt;</span> <span class="va">peaks_chr19</span><span class="op">$</span><span class="va">ci_hi</span><span class="op">)</span></span>
<span><span class="va">expr_chr19</span>  <span class="op">&lt;-</span> <span class="va">expr</span><span class="op">[</span>,<span class="va">annot_chr19</span><span class="op">$</span><span class="va">a_gene_id</span><span class="op">]</span></span></code></pre>
</div>
<p>There are 22 genes! How can we start to narrow down which ones may be
candidate genes?</p>
<div id="challenge-3-how-can-we-narrow-down-the-candidate-gene-list" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-how-can-we-narrow-down-the-candidate-gene-list" class="callout-inner">
<h3 class="callout-title">Challenge 3: How can we narrow down the candidate gene list?</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Take a moment to think of ways that you could narrow down the gene
list to select the most promising candidate genes that regulate insulin
levels.</li>
<li>Turn to your neighbor and exchange your ideas.</li>
<li>Share your ideas with the group.</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<!-- DMG: Correlation, eQTL, annotation? -->
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="identifying-coding-snps-in-qtl-intervals">Identifying Coding SNPs in QTL Intervals<a class="anchor" aria-label="anchor" href="#identifying-coding-snps-in-qtl-intervals"></a>
</h2>
<hr class="half-width">
<p>When you have a QTL peak, you can search for SNPs which lie within
the coding regions of genes. These SNPs may cause amino acid
substitutions which will affect protein structure. The most
comprehensive SNP resource is the <a href="https://www.sanger.ac.uk/data/mouse-genomes-project/" class="external-link">Mouse
Genomes Project</a>. They have sequenced 52 inbred strains, including
BTBR, and have made the data available in <a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf" class="external-link">Variant Call
Format</a> (VCF). The full VCF files are available on the <a href="https://ftp.ebi.ac.uk/pub/databases/mousegenomes/REL-2112-v8-SNPs_Indels/" class="external-link">EBI
FTP site</a>. However, the full SNP file is 22 GB! When you need other
strains, you can download and query that file. However, for this
workshop, we only need SNPs for the BTBR strain. We have created a VCF
file containing only SNPs which differ from C57BL/6J and BTBR. Let’s
read in this file now using the <a href="https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html" class="external-link">VariantAnnotation</a>
function <a href="https://rdrr.io/bioc/VariantAnnotation/man/readVcf-methods.html" class="external-link">readVcf</a>.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vcf</span> <span class="op">&lt;-</span> <span class="fu">readVcf</span><span class="op">(</span><span class="fu">file.path</span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"btbr_snps_grcm39.vcf.gz"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>The <code>vcf</code> object contains both the SNP allele calls and
information about the sequencing depth and SNP consequences. Let’s see
how many SNPs are in the VCF.</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 4791000       1</code></pre>
</div>
<p>There are about 4.8 million SNPs in <code>vcf</code>. There are many
fields in the VCF file and there are <a href="https://thejacksonlaboratory.box.com/shared/static/gwt3o0gh6pgaeyxuuym58zpur0ktvkz5.tbi" class="external-link">vignettes</a>
which document the more advanced features. Here, we will search for SNPs
within QTL support intervals and filter them to retain missense or stop
coding SNPs.</p>
<p>In order to filter the SNPs by location, we need to create a <a href="https://www.bioconductor.org/packages/release/bioc/html/GenomicRanges.html" class="external-link">GenomicRanges</a>
object which contains the coordinates of the QTL support interval on
chromosome 19. Note that the support interval is reported by
<code>qtl2</code> in Mb and the GRanges object must be given bp
positions, so we need to multiply these by 1e6.</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr19_gr</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">peaks_chr19</span><span class="op">$</span><span class="va">chr</span>, </span>
<span>                    ranges   <span class="op">=</span> <span class="fu">IRanges</span><span class="op">(</span><span class="va">peaks_chr19</span><span class="op">$</span><span class="va">ci_lo</span> <span class="op">*</span> <span class="fl">1e6</span>, <span class="va">peaks_chr19</span><span class="op">$</span><span class="va">ci_hi</span> <span class="op">*</span> <span class="fl">1e6</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">chr19_gr</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>GRanges object with 1 range and 0 metadata columns:
      seqnames            ranges strand
         &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;
  [1]       19 48370979-55150074      *
  -------
  seqinfo: 20 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
<p>Next, we will filter the <code>vcf</code> object to retain SNPs
within the support interval.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vcf_chr19</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">vcf</span>, ranges <span class="op">=</span> <span class="va">chr19_gr</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s see how many SNPs there are.</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">vcf_chr19</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 19065     1</code></pre>
</div>
<p>There are still 19,000 SNPs! But many of them may not be in coding
regions or may be synonymous. Next, we will search for SNPs that have
missense or stop coding changes. These are likely to have the most
severe effects. Note that synonymous SNPs and SNPs in the untranslated
region of genes may affect RNA folding, RNA stability, and translation.
But it is more difficult to predict those effects.</p>
<p>Let’s plot the density of the SNPs to see where they fall. We would
not expect genes which lie in regions without variation between B6 and
BTBR to be involved in regulating insulin levels.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the SNP positions.</span></span>
<span><span class="va">pos</span> <span class="op">=</span> <span class="fu">start</span><span class="op">(</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">vcf_chr19</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span> <span class="fl">1e6</span></span>
<span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="fu">density</span><span class="op">(</span><span class="va">pos</span><span class="op">)</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">2</span>, main <span class="op">=</span> <span class="st">"Density of SNPs"</span><span class="op">)</span></span>
<span><span class="fu">rug</span><span class="op">(</span><span class="va">pos</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/integrating-gene-expression-rendered-snp_density-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>From the plot above, we can see that the density of SNPS between B6
and BTBR is not uniform. Most of the SNPs lie between 50 and 52 Mb, with
less dense regions extending past 54 Mb.</p>
<p>The <code>vcf</code> object contains predicted SNP consequences in a
field called <code>CSQ</code>. The format of this field is challenging
to parse, so we will get it, search for missense and stop SNPs, and then
do some data-wrangling to may the output readable.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">csq</span> <span class="op">&lt;-</span> <span class="fu">info</span><span class="op">(</span><span class="va">vcf_chr19</span><span class="op">)</span><span class="op">$</span><span class="va">CSQ</span></span>
<span><span class="va">csq</span> <span class="op">&lt;-</span> <span class="fu">as.list</span><span class="op">(</span><span class="va">csq</span><span class="op">)</span></span>
<span></span>
<span><span class="va">wh</span>  <span class="op">&lt;-</span> <span class="fu">grep</span><span class="op">(</span><span class="st">'missense|stop'</span>, <span class="va">csq</span><span class="op">)</span></span>
<span><span class="va">csq</span> <span class="op">&lt;-</span> <span class="va">csq</span><span class="op">[</span><span class="va">wh</span><span class="op">]</span></span>
<span><span class="va">rr</span>  <span class="op">&lt;-</span> <span class="fu">rowRanges</span><span class="op">(</span><span class="va">vcf_chr19</span><span class="op">)</span><span class="op">[</span><span class="va">wh</span><span class="op">]</span></span></code></pre>
</div>
<p>Let’s see how many SNPs had missense or stop codon consequences.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">length</span><span class="op">(</span><span class="va">csq</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 8</code></pre>
</div>
<p>There were only 8 SNPs out of 19,000 SNPs with missense or stop codon
consequences. Next, let’s do the data wrangling to reformat the
consequences of these SNPs.</p>
<div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" data-bs-parent="#accordionInstructor1" aria-labelledby="headingInstructor1">
<div class="accordion-body">
<p>This may be a good code chunk to have the students copy and
paste.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">csq</span> <span class="op">&lt;-</span> <span class="fu">lapply</span><span class="op">(</span><span class="va">csq</span>, <span class="va">strsplit</span>, split <span class="op">=</span> <span class="st">"\\|"</span><span class="op">)</span></span>
<span><span class="va">csq</span> <span class="op">&lt;-</span> <span class="fu">lapply</span><span class="op">(</span><span class="va">csq</span>, <span class="kw">function</span><span class="op">(</span><span class="va">z</span><span class="op">)</span> <span class="op">{</span></span>
<span>                     <span class="fu">matrix</span><span class="op">(</span><span class="fu">unlist</span><span class="op">(</span><span class="va">z</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fu">length</span><span class="op">(</span><span class="va">z</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>                   <span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu">seq_along</span><span class="op">(</span><span class="va">csq</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">csq</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">data.frame</span><span class="op">(</span>snp_id <span class="op">=</span> <span class="fu">names</span><span class="op">(</span><span class="va">rr</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>,</span>
<span>                         <span class="va">csq</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span>  </span>
<span><span class="op">}</span> <span class="co"># for(i)</span></span>
<span></span>
<span><span class="va">csq</span> <span class="op">=</span> <span class="fu">do.call</span><span class="op">(</span><span class="va">rbind</span>, <span class="va">csq</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s look at the top of <code>csq</code>.</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">csq</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>           snp_id X1               X2       X3     X4                 X5
1 19:50141282_A/G  G   intron_variant MODIFIER Sorcs1 ENSMUSG00000043531
2 19:50141282_A/G  T   intron_variant MODIFIER Sorcs1 ENSMUSG00000043531
3 19:50141282_A/G  C   intron_variant MODIFIER Sorcs1 ENSMUSG00000043531
4 19:50141282_A/G  G missense_variant MODERATE Sorcs1 ENSMUSG00000043531
5 19:50141282_A/G  T missense_variant MODERATE Sorcs1 ENSMUSG00000043531
6 19:50141282_A/G  C missense_variant MODERATE Sorcs1 ENSMUSG00000043531
          X6                 X7             X8    X9   X10 X11 X12  X13  X14
1 Transcript ENSMUST00000072685 protein_coding       26/26
2 Transcript ENSMUST00000072685 protein_coding       26/26
3 Transcript ENSMUST00000072685 protein_coding       26/26
4 Transcript ENSMUST00000111756 protein_coding 27/27               3465 3448
5 Transcript ENSMUST00000111756 protein_coding 27/27               3465 3448
6 Transcript ENSMUST00000111756 protein_coding 27/27               3465 3448
   X15 X16     X17 X18 X19 X20 X21 X22 X23 X24               X25 X26 X27 X28
1                           -1     SNV MGI
2                           -1     SNV MGI
3                           -1     SNV MGI
4 1150 S/P Tct/Cct          -1     SNV MGI       tolerated(0.09)
5 1150 S/T Tct/Act          -1     SNV MGI       tolerated(0.26)
6 1150 S/A Tct/Gct          -1     SNV MGI     deleterious(0.03)
  X29
1
2
3
4
5
6    </code></pre>
</div>
<p>There are a lot of columns and we may not need all of them. Next,
let’s reduce the number of columns to keep the ones that we need.</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">csq</span> <span class="op">&lt;-</span> <span class="va">csq</span><span class="op">[</span>,<span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">7</span>,<span class="fl">9</span>,<span class="fl">17</span>,<span class="fl">18</span>,<span class="fl">26</span><span class="op">)</span><span class="op">]</span></span></code></pre>
</div>
<p>Next, we will subset the consequences to retain the unique rows and
then retain rows which contain “missense” and “stop”.</p>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">csq</span> <span class="op">&lt;-</span> <span class="fu">distinct</span><span class="op">(</span><span class="va">csq</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>         <span class="fu">filter</span><span class="op">(</span><span class="fu">str_detect</span><span class="op">(</span><span class="va">X2</span>, <span class="st">"missense|stop"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s look at the top of <code>csq</code> now.</p>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">csq</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>           snp_id X1               X2       X3     X4                 X5
1 19:50141282_A/G  G missense_variant MODERATE Sorcs1 ENSMUSG00000043531
2 19:50141282_A/G  T missense_variant MODERATE Sorcs1 ENSMUSG00000043531
3 19:50141282_A/G  C missense_variant MODERATE Sorcs1 ENSMUSG00000043531
4 19:50141444_G/A  A missense_variant MODERATE Sorcs1 ENSMUSG00000043531
5 19:50141444_G/A  T missense_variant MODERATE Sorcs1 ENSMUSG00000043531
6 19:50141444_G/A  C missense_variant MODERATE Sorcs1 ENSMUSG00000043531
          X6             X8 X16     X17                              X25
1 Transcript protein_coding S/P Tct/Cct                  tolerated(0.09)
2 Transcript protein_coding S/T Tct/Act                  tolerated(0.26)
3 Transcript protein_coding S/A Tct/Gct                deleterious(0.03)
4 Transcript protein_coding S/F tCc/tTc deleterious_low_confidence(0.01)
5 Transcript protein_coding S/Y tCc/tAc deleterious_low_confidence(0.01)
6 Transcript protein_coding S/C tCc/tGc deleterious_low_confidence(0.01)</code></pre>
</div>
<p>Column <code>X4</code> contains gene names. Let’s get the unique gene
names.</p>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">csq</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">distinct</span><span class="op">(</span><span class="va">X4</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>      X4
1 Sorcs1
2  Rbm20</code></pre>
</div>
<p>There are only two genes in the QTL support interval which contain
missense SNPs.</p>
<div id="challenge-4-search-genes-using-pubmed" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-search-genes-using-pubmed" class="callout-inner">
<h3 class="callout-title">Challenge 4: Search Genes using Pubmed</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Go to Pubmed and search for papers involving the two genes and
insulin or diabetes.</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<p>Both genes return published papers relating to insulin:</p>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/?term=Sorcs1+insulin" class="external-link">Sorcs1</a></p>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/?term=rbm20+insulin" class="external-link">Rbm2</a></p>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/16682971/" class="external-link">Sorcs1</a> was
identified as a gene which relates to obesity-induced type 2 diabetes by
the same group that provided this data set. In fact, it was identified
in the same type of mouse cross between C57BL/6J and BTBR.</p>
</div>
</div>
</div>
</div>
<p>In this case, we used published SNPs data to identify two potential
candidate genes. If we had not found publications that had already
associated these genes with diabetes, these would be genes that you
would follow up on in the lab.</p>
</section><section><h2 class="section-heading" id="using-expression-qtl-mapping">Using Expression QTL Mapping<a class="anchor" aria-label="anchor" href="#using-expression-qtl-mapping"></a>
</h2>
<hr class="half-width">
<p>Another method of searching for candidate genes is to look for genes
which have QTL peaks in the same location as the insulin QTL. Genes
which have QTL that are co-located with insulin QTL will also be
strongly correlated with insulin levels. These genes may be correlated
with insulin because they control insulin levels, respond to insulin
levels, or are correlated by chance. However, genes which have QTL
within the insulin QTL support interval are reasonable candidate genes
since we expect the genotype to influence expression levels.</p>
<p>Let’s perform QTL mapping on the genes within the chromosome 19
insulin QTL support interval. We can do this by passing in the
expression values as phenotypes into <code>scan1</code>.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eqtl_chr19</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>,</span>
<span>                    pheno     <span class="op">=</span> <span class="va">expr_chr19</span>,</span>
<span>                    kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                    addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>Let’s look at the top of the results.</p>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">head</span><span class="op">(</span><span class="va">eqtl_chr19</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>           498370 500695 501344 504488 505754 506116
rs4232073    2.38   2.02  0.766  0.279  0.918 0.1567
rs13483548   2.34   2.01  0.778  0.279  0.919 0.1567
rs13483549   2.33   2.00  0.784  0.278  0.913 0.1489
rs13483550   2.23   1.72  0.865  0.245  0.765 0.0618
rs13483554   1.94   1.69  0.886  0.249  0.819 0.1127
rs13483555   2.08   1.62  0.862  0.255  0.867 0.1289</code></pre>
</div>
<p>The eQTL results have one row for each marker and one column for each
gene. Each column represents the LOD plot for one gene. Let’s plot the
genome scan for one gene.</p>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="st">'10002678668'</span></span>
<span><span class="va">symbol</span>  <span class="op">&lt;-</span> <span class="va">annot_chr19</span><span class="op">$</span><span class="va">gene_name</span><span class="op">[</span><span class="va">annot_chr19</span><span class="op">$</span><span class="va">a_gene_id</span> <span class="op">==</span> <span class="va">gene_id</span><span class="op">]</span></span>
<span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x   <span class="op">=</span> <span class="va">eqtl_chr19</span>, </span>
<span>           map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>           lodcolumn <span class="op">=</span> <span class="va">gene_id</span>,</span>
<span>           main      <span class="op">=</span> <span class="va">symbol</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/integrating-gene-expression-rendered-plot_1_eqtl-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>This gene seems to have a QTL on chromosome 19 near 52 Mb with a LOD
of 34. The insulin QTL is closer to 55 Mb, so this may be a good
candidate gene.</p>
<p>We can harvest the significant QTL peaks on chromosome 19 using
<code>find_peaks</code>. We will use the default threshold of LOD =
3.</p>
<div class="codewrapper sourceCode" id="cb45">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eqtl_chr19_peaks</span> <span class="op">&lt;-</span> <span class="fu">find_peaks</span><span class="op">(</span><span class="va">eqtl_chr19</span>, map <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>                      <span class="fu">arrange</span><span class="op">(</span><span class="va">pos</span><span class="op">)</span></span>
<span><span class="va">eqtl_chr19_peaks</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex   lodcolumn chr  pos   lod
1       21 10004035475  19 47.4  4.66
2       17 10002936879  19 51.4  7.66
3       16 10002928587  19 51.8  7.12
4       11 10002678668  19 52.4 34.67
5        1      498370  19 53.0 12.39
6       14 10002910800  19 54.8  4.23</code></pre>
</div>
<p>We can see that there are 6 significant QTL on chromosome 19. All of
these genes are within the support interval of the insulin QTL.</p>
<p>Let’s filter the eQTL genes and add in gene annotation for them.</p>
<div class="codewrapper sourceCode" id="cb47">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eqtl_chr19_peaks</span> <span class="op">&lt;-</span> <span class="va">eqtl_chr19_peaks</span> <span class="op">|&gt;</span></span>
<span>                     <span class="fu">filter</span><span class="op">(</span><span class="va">pos</span> <span class="op">&gt;</span> <span class="va">peaks_chr19</span><span class="op">$</span><span class="va">ci_lo</span> <span class="op">&amp;</span> <span class="va">pos</span> <span class="op">&lt;</span> <span class="va">peaks_chr19</span><span class="op">$</span><span class="va">ci_hi</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>                     <span class="fu">left_join</span><span class="op">(</span><span class="va">annot</span>, by <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'lodcolumn'</span> <span class="op">=</span> <span class="st">'a_gene_id'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">eqtl_chr19_peaks</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex   lodcolumn chr.x  pos   lod chr.y start  end  width strand
1       17 10002936879    19 51.4  7.66    19  53.7 53.9 189775      +
2       16 10002928587    19 51.8  7.12    19  50.1 50.7 535348      -
3       11 10002678668    19 52.4 34.67    19  52.9 53.0 108289      -
4        1      498370    19 53.0 12.39    19  53.4 53.4  11189      -
5       14 10002910800    19 54.8  4.23    19  52.3 52.3   1180      +
             gene_id gene_name   gene_biotype
1 ENSMUSG00000043639     Rbm20 protein_coding
2 ENSMUSG00000043531    Sorcs1 protein_coding
3 ENSMUSG00000025027   Xpnpep1 protein_coding
4 ENSMUSG00000025024    Smndc1 protein_coding
5 ENSMUSG00000035804      Ins1 protein_coding
                                                                                description
1                          RNA binding motif protein 20 [Source:MGI Symbol;Acc:MGI:1920963]
2   sortilin-related VPS10 domain containing receptor 1 [Source:MGI Symbol;Acc:MGI:1929666]
3 X-prolyl aminopeptidase (aminopeptidase P) 1, soluble [Source:MGI Symbol;Acc:MGI:2180003]
4             survival motor neuron domain containing 1 [Source:MGI Symbol;Acc:MGI:1923729]
5                                               insulin I [Source:MGI Symbol;Acc:MGI:96572]
        gene_id_version  symbol entrezid
1 ENSMUSG00000043639.16   Rbm20       NA
2 ENSMUSG00000043531.17  Sorcs1       NA
3 ENSMUSG00000025027.19 Xpnpep1       NA
4  ENSMUSG00000025024.8  Smndc1       NA
5  ENSMUSG00000035804.6    Ins1       NA</code></pre>
</div>
<p>Let’s plot the insulin QTL again to remind ourselves what the peak on
chromosome 19 look like. We will also plot the genome scan for one of
the genes on chromosome 19.</p>
<div class="codewrapper sourceCode" id="cb49">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="st">"10002936879"</span></span>
<span><span class="va">symbol</span>  <span class="op">&lt;-</span> <span class="va">annot_chr19</span><span class="op">$</span><span class="va">gene_name</span><span class="op">[</span><span class="va">annot_chr19</span><span class="op">$</span><span class="va">a_gene_id</span> <span class="op">==</span> <span class="va">gene_id</span><span class="op">]</span></span>
<span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">eqtl_chr19</span>,</span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>           lodcolumn <span class="op">=</span> <span class="va">gene_id</span>,</span>
<span>           main <span class="op">=</span> <span class="st">"Insulin"</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_add_loco</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>           chr  <span class="op">=</span> <span class="st">'19'</span>,</span>
<span>           col  <span class="op">=</span> <span class="st">'blue'</span>,</span>
<span>           add  <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span><span class="st">"topleft"</span>, legend <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"insluin"</span>, <span class="va">symbol</span><span class="op">)</span>, </span>
<span>       col <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'blue'</span>, <span class="st">'black'</span><span class="op">)</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/integrating-gene-expression-rendered-plot_insulin_qtl-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Why would we prioritize genes with an eQTL in the same location as an
insulin QTL? If a gene has an eQTL in the same location, there are
genetic variants which influence it’s expression at that location. These
variants may be distinct from the ones which influence the phenotype or
they may influence gene expression which then influences the phenotype.
We call this causal relationship of a SNP changing expression, which
then changes the phenotype, “mediation”.</p>
</section><section><h2 class="section-heading" id="mediation-analysis">Mediation Analysis<a class="anchor" aria-label="anchor" href="#mediation-analysis"></a>
</h2>
<hr class="half-width">
<p>If a gene is correlated with insulin levels and if it has an eQTL in
the same location as insulin, then we could add it into the insulin QTL
model and see if it reduces the LOD on chromosome 19.</p>
<div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">stopifnot</span><span class="op">(</span><span class="fu">rownames</span><span class="op">(</span><span class="va">addcovar</span><span class="op">)</span> <span class="op">==</span> <span class="fu">rownames</span><span class="op">(</span><span class="va">expr_chr19</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">addcovar_chr19</span> <span class="op">&lt;-</span> <span class="fu">cbind</span><span class="op">(</span><span class="va">addcovar</span>, <span class="va">expr_chr19</span><span class="op">[</span>,<span class="st">"10002936879"</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">lod_med</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>,</span>
<span>                 pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>                 kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar_chr19</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will plot the original insulin genome scan and overlay the
genome scan with gene <code>10002936879</code> in the model.</p>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="st">"10002936879"</span></span>
<span><span class="va">symbol</span>  <span class="op">&lt;-</span> <span class="va">annot_chr19</span><span class="op">$</span><span class="va">gene_name</span><span class="op">[</span><span class="va">annot_chr19</span><span class="op">$</span><span class="va">a_gene_id</span> <span class="op">==</span> <span class="va">gene_id</span><span class="op">]</span></span>
<span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_add_loco</span>,</span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>           chr  <span class="op">=</span> <span class="st">'19'</span>,</span>
<span>           main <span class="op">=</span> <span class="fu">paste</span><span class="op">(</span><span class="st">"Insulin with/without"</span>, <span class="va">symbol</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_med</span>,</span>
<span>           map  <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>           chr  <span class="op">=</span> <span class="st">'19'</span>,</span>
<span>           col  <span class="op">=</span> <span class="st">"blue"</span>,</span>
<span>           add  <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">legend</span><span class="op">(</span><span class="st">"topleft"</span>, legend <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">"insluin"</span>, <span class="st">"mediation"</span><span class="op">)</span>, </span>
<span>       col <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="st">'black'</span>, <span class="st">'blue'</span><span class="op">)</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/integrating-gene-expression-rendered-med_plot_10002936879-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The LOD dropped from 5.477 to 2.02. This is a difference of
3.456.</p>
<p>It would be slow to run a genome scan for each gene on chromosome 19
and look at the LOD drop. Also, we don’t really need the LOD drop across
the entire chromosome. We only need the LOD drop at the location of the
peak LOD. To do this, we can use the <code>fit1</code> function to get
the LOD at the marker with the highest LOD in the insulin genome
scan.</p>
<div class="codewrapper sourceCode" id="cb52">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the probs at the maximum insulin QTL on Chr 19.</span></span>
<span><span class="va">pr_chr19</span> <span class="op">=</span> <span class="fu">pull_genoprobpos</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>,</span>
<span>                            map       <span class="op">=</span> <span class="va">cross</span><span class="op">$</span><span class="va">pmap</span>,</span>
<span>                            chr       <span class="op">=</span> <span class="va">chr</span>,</span>
<span>                            pos       <span class="op">=</span> <span class="va">peaks_chr19</span><span class="op">$</span><span class="va">pos</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create data sructure for results.</span></span>
<span><span class="va">lod_drop</span> <span class="op">=</span> <span class="fu">data.frame</span><span class="op">(</span>a_gene_id <span class="op">=</span> <span class="fu">colnames</span><span class="op">(</span><span class="va">expr_chr19</span><span class="op">)</span>,</span>
<span>                      lod       <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu">ncol</span><span class="op">(</span><span class="va">expr_chr19</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span></span>
<span>  <span class="co"># Make new covariates.</span></span>
<span>  <span class="va">curr_covar</span> <span class="op">=</span> <span class="fu">cbind</span><span class="op">(</span><span class="va">addcovar</span>, <span class="va">expr_chr19</span><span class="op">[</span>,<span class="va">i</span><span class="op">]</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">mod</span> <span class="op">=</span> <span class="fu">fit1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">pr_chr19</span>,</span>
<span>             pheno     <span class="op">=</span> <span class="va">insulin</span>,</span>
<span>             kinship   <span class="op">=</span> <span class="va">kinship_loco</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>,</span>
<span>             addcovar  <span class="op">=</span> <span class="va">curr_covar</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">lod_drop</span><span class="op">$</span><span class="va">lod</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">=</span> <span class="va">mod</span><span class="op">$</span><span class="va">lod</span></span>
<span>  </span>
<span><span class="op">}</span> <span class="co"># for(i)</span></span>
<span></span>
<span><span class="va">lod_drop</span><span class="op">$</span><span class="va">lod_drop</span> <span class="op">=</span> <span class="va">lod_drop</span><span class="op">$</span><span class="va">lod</span> <span class="op">-</span> <span class="va">peaks_chr19</span><span class="op">$</span><span class="va">lod</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb53">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_drop</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">lod_drop</span>, <span class="va">annot_chr19</span>, by <span class="op">=</span> <span class="st">'a_gene_id'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot</span><span class="op">(</span><span class="va">lod_drop</span><span class="op">$</span><span class="va">start</span>, <span class="va">lod_drop</span><span class="op">$</span><span class="va">lod_drop</span>, col <span class="op">=</span> <span class="cn">NA</span>, </span>
<span>     main <span class="op">=</span> <span class="st">"Mediation Analysis on Chr 19"</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"Posiiton (Mb)"</span>, ylab <span class="op">=</span> <span class="st">"LOD Drop"</span><span class="op">)</span></span>
<span><span class="fu">text</span><span class="op">(</span><span class="va">lod_drop</span><span class="op">$</span><span class="va">start</span>, <span class="va">lod_drop</span><span class="op">$</span><span class="va">lod_drop</span>, labels <span class="op">=</span> <span class="va">lod_drop</span><span class="op">$</span><span class="va">gene_name</span><span class="op">)</span></span>
<span><span class="fu">abline</span><span class="op">(</span>v <span class="op">=</span> <span class="va">peaks_chr19</span><span class="op">$</span><span class="va">pos</span>, col <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/integrating-gene-expression-rendered-plot_mediation-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot above, we plotted the position of each gene versus the
decrease in the LOD score (i.e. LOD drop). Genes with the lowest LOD
drop are the best candidate genes based on mediation analysis.</p>
<p>Let’s look at the annotation for the genes with LOD drop less than
-2.</p>
<div class="codewrapper sourceCode" id="cb54">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_drop</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter</span><span class="op">(</span><span class="va">lod_drop</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">2</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">gene_id</span>, <span class="va">symbol</span>, <span class="va">chr</span>, <span class="va">start</span>, <span class="va">end</span>, <span class="va">lod</span>, <span class="va">lod_drop</span>, <span class="va">description</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>             gene_id  symbol chr start  end  lod lod_drop
1 ENSMUSG00000025027 Xpnpep1  19  52.9 53.0 2.85    -2.63
2 ENSMUSG00000035804    Ins1  19  52.3 52.3 2.60    -2.88
3 ENSMUSG00000043639   Rbm20  19  53.7 53.9 2.02    -3.46
                                                                                description
1 X-prolyl aminopeptidase (aminopeptidase P) 1, soluble [Source:MGI Symbol;Acc:MGI:2180003]
2                                               insulin I [Source:MGI Symbol;Acc:MGI:96572]
3                          RNA binding motif protein 20 [Source:MGI Symbol;Acc:MGI:1920963]</code></pre>
</div>
<p>Do you see any good candidate genes which might regulate insulin?</p>
</section><section><h2 class="section-heading" id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<hr class="half-width">
<p>In this episode, we learned how to identify candidate genes under a
QTL peak. In a perfect world, there would be exactly one gene implicated
by these analyses. In most cases, you will have a set of candidate genes
and you will need to study each gene and prioritize some for laboratory
follow-up.</p>
<p>There are two ways of searching for candidate genes: using SNPs in
the QTL interval, and looking for genes with eQTL which are co-located
with the phenotype QTL. We learned how to query a VCF file and how to
perform mediation analysis. After this step, you will have a set of
genes which you can test for association with your phenotype.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>There will be many genes under a QTL peak.</li>
<li>You can search for genes with SNPs that produce coding changes by
querying a VCF file.</li>
<li>You can search for genes with expression changes that may influence
your phenotype by performing mediation analysis with expression data
from the same mice.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-do_qtl_mapping"><p>Content from <a href="do_qtl_mapping.html">QTL Mapping in Diversity Outbred Mice</a></p>
<hr>
<p>Last updated on 2024-11-13 |

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/episodes/do_qtl_mapping.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 150 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How do I map traits in Diversity Outbred mice?</li>
<li>How do I interpret the founder allele effects at a QTL peak?</li>
<li>How do I perform association mapping in Diversity Outbred mice?</li>
<li>How do I narrow down the set of candidate genes under a QTL?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the data objects required for QTL mapping.</li>
<li>Use the scan1 family of functions to perform different types of QTL
scans.</li>
<li>Interpret the founder allele effects.</li>
<li>Find significant QTL peaks and identify candidate peaks under a QTL
peak.</li>
<li>Use auxiliary data to identify candidate genes under a QTL
peak.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<p>In the Diversity Outbred (DO) and Collaborative Cross (CC), alleles A
to H represent haplotypes of the 8 founder strains.</p>
<figure><img src="../fig/cc-founder-alleles.png" alt="Figure showing the colors and letter codes of the CC/DO founders." class="figure mx-auto d-block"><div class="figcaption">CC and DO Founder Alleles</div>
</figure><p>CC lines have very low heterozygosity throughout their genomes. For
most loci, CC lines will be homozygous for one of the founder strains
A-H above, and as such will have one of only 8 genotypes (<em>e.g.</em>
AA, BB, CC, …). In contrast, DO <em>mice</em> (not lines) have high
heterozygosity throughout their genomes. Each locus will have one of 36
possible genotypes (<em>e.g.</em> AA, AB, AC, …, BB, BC, BD,…).</p>
<figure><img src="../fig/cc-do-genome-comparison.png" alt="Figure showing CC and DO genomes" class="figure mx-auto d-block"><div class="figcaption">Example Collaborative Cross and Diversity
Outbred genomes</div>
</figure><p>This tutorial will take you through the process of mapping a QTL and
searching for candidate genes for DNA damage from benzene exposure. The
adverse health effects of benzene, including leukemia and aplastic
anemia, were first studied in occupational settings in which workers
were exposed to high benzene concentrations. Environmental sources of
benzene exposure typically come from the petrochemical industry,
however, a person’s total exposure can be increased from cigarettes,
consumer products, gas stations, and gasoline powered engines or tools
stored at home.</p>
<p>Exposure thresholds for toxicants are often determined using animal
models that have limited genetic diversity, including standard inbred
and outbred rats and mice. These animal models fail to capture the
influence of genetic diversity on toxicity response, an important
component of human responses to chemicals such as benzene. The Diversity
Outbred (DO) mice reflect a level of genetic diversity similar to that
of humans.</p>
<p>The data comes from a toxicology study in which Diversity Outbred
(DO) mice were exposed to benzene via inhalation for 6 hours a day, 5
days a week for 4 weeks <a href="https://ehp.niehs.nih.gov/1408202/" class="external-link">(French, J. E., et al. (2015)
<i>Environ Health Perspect</i> 123(3): 237-245)</a>. The study was
conducted in two equally sized cohort of 300 male mice each, for a total
of 600 mice. They were then sacrificed and reticulocytes (red blood cell
precursors) were isolated from bone marrow. The number of
micro-nucleated reticulocytes, a measure of DNA damage, was then
measured in each mouse. The goal is to map gene(s) that influence the
level of DNA damage in the bone marrow.</p>
<p><img src="../fig/benzene_study_design.png" alt="Benzene Study Dosing" class="figure">{alt=‘Benzene study dosing showing 6 hours
per day, 5 days per week of inhalation.’,width=‘100%’}</p>
</section><section><h2 class="section-heading" id="do-reference-data">DO Reference Data<a class="anchor" aria-label="anchor" href="#do-reference-data"></a>
</h2>
<hr class="half-width">
<p>As you work with DO data, you may need different reference files.
These are complied on a JAX website at: <a href="https://www.jax.org/research-and-faculty/genetic-diversity-initiative/tools-data/diversity-outbred-reference-data" class="external-link uri">https://www.jax.org/research-and-faculty/genetic-diversity-initiative/tools-data/diversity-outbred-reference-data</a>.
There are links to the markers and founder genotypes as well as the
standard colors that we use for the eight founder strains.</p>
</section><section><h2 class="section-heading" id="preparing-your-diversity-outbred-do-data-for-qtl2">Preparing your Diversity Outbred (DO) data for qtl2<a class="anchor" aria-label="anchor" href="#preparing-your-diversity-outbred-do-data-for-qtl2"></a>
</h2>
<hr class="half-width">
<p>Karl Broman provides detailed instructions for <a href="https://kbroman.org/qtl2/pages/prep_do_data.html" class="external-link">preparing DO
mouse data for R/qtl2</a>.</p>
</section><section><h2 class="section-heading" id="open-the-qtl_mapping-project-">Open the qtl_mapping Project.<a class="anchor" aria-label="anchor" href="#open-the-qtl_mapping-project-"></a>
</h2>
<hr class="half-width">
<p>If you have not opened the qtl_mapping project, do this now so that
your script and console will run in the correct directory.</p>
<ol style="list-style-type: decimal">
<li>From the File menu, select “Open Project…”.</li>
<li>Navigate to the project that you created in your Desktop/qtl_mapping
folder and open it.</li>
</ol>
<p>This should reset your RStudio window and close any files.</p>
</section><section><h2 class="section-heading" id="create-a-new-markdown-file">Create a New Markdown File<a class="anchor" aria-label="anchor" href="#create-a-new-markdown-file"></a>
</h2>
<hr class="half-width">
<p>Create a new R Markdown file and save it in the “Desktop/scripts”
folder as “do_qtl_mapping.Rmd”.</p>
</section><section><h2 class="section-heading" id="load-libraries">Load Libraries<a class="anchor" aria-label="anchor" href="#load-libraries"></a>
</h2>
<hr class="half-width">
<p>The primary library that we will use is <a href="https://github.com/kbroman/qtl2" class="external-link">qtl2</a>. You should have
<code>qtl2</code> installed from the Setup that you performed before the
workshop.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">tidyverse</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">ggbeeswarm</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">suppressPackageStartupMessages</span><span class="op">(</span><span class="kw">library</span><span class="op">(</span><span class="va">qtl2convert</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="load-and-explore-the-data">Load and Explore the Data<a class="anchor" aria-label="anchor" href="#load-and-explore-the-data"></a>
</h2>
<hr class="half-width">
<p>The data for this tutorial has been saved as an R binary file that
contains several data objects. Load it in now by running the following
command in your new script.</p>
<div id="accordionInstructor1" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor1" aria-expanded="false" aria-controls="collapseInstructor1">
  <h3 class="accordion-header" id="headingInstructor1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor1" class="accordion-collapse collapse" aria-labelledby="headingInstructor1" data-bs-parent="#accordionInstructor1">
<div class="accordion-body">
<p>The code below is needed to load in the data objects to build the
lesson on Github. The <code>*.Rdata</code> file is over 200 Mb and
Github has a file size limit of 100 Mb. The students should
<strong>NOT</strong> run the next block.</p>
</div>
</div>
</div>
</div>
<div id="accordionInstructor2" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor2" aria-expanded="false" aria-controls="collapseInstructor2">
  <h3 class="accordion-header" id="headingInstructor2">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor2" class="accordion-collapse collapse" aria-labelledby="headingInstructor2" data-bs-parent="#accordionInstructor2">
<div class="accordion-body">
<p>This is the only block that the students need to run.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">load</span><span class="op">(</span><span class="st">"./data/qtl2_demo_grcm39.Rdata"</span><span class="op">)</span></span></code></pre>
</div>
<p>We loaded in four data objects from the *.Rdata file. Look at the
Environment tab to see what was loaded. You should see an object called
<code>pheno</code> which contains the phenotypes. There are two objects
called <code>gmap</code> and <code>pmap</code>, which contain the marker
positions in genetic and physical coordinates. Finally, there is an
object called <code>probs</code> which contains the founder allele
probabilities for each mouse at each marker.</p>
<div class="section level3">
<h3 id="phenotypes">Phenotypes<a class="anchor" aria-label="anchor" href="#phenotypes"></a>
</h3>
<p><code>pheno</code> is a data frame containing the phenotype data.
Click on the blue circle to the left of <code>pheno</code> in the
Environment pane to view its contents.</p>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>The sample IDs <strong>must</strong> be in the rownames of
<code>pheno</code> for qtl2 to work.</p>
</div>
</div>
</div>
<p><code>pheno</code> contains the sample ID, the study cohort, the
concentration of benzene and several blood cell measurements. Note that
the sample IDs are also stored in the rownames of
<code>pheno</code>.</p>
<p>In order to save time for this tutorial, we will only map with 149
samples from the 100 ppm dosing group.</p>
<div id="challenge-1-how-many-mice-are-there-in-the-phenotype-data" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-1-how-many-mice-are-there-in-the-phenotype-data" class="callout-inner">
<h3 class="callout-title">Challenge 1: How many mice are there in the phenotype data?</h3>
<div class="callout-content">
<p>Look in the Environment tab or use the Console to figure out how many
mice there are in the phenotype data.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>You can look at the number of observations in the Environment tab or
you can get the number of rows in <code>pheno</code> in the Console.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 598</code></pre>
</div>
<p>There are 598 mice in the phenotype data.</p>
</div>
</div>
</div>
</div>
<div id="challenge-2-how-many-mice-are-there-from-each-sex" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-2-how-many-mice-are-there-from-each-sex" class="callout-inner">
<h3 class="callout-title">Challenge 2: How many mice are there from each sex?</h3>
<div class="callout-content">
<p>Use an R command to determine how many mice there are from each
sex.</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>You can look at the number of observations in the Environment tab or
you can get the number of rows in <code>pheno</code> in the Console.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">count</span><span class="op">(</span><span class="va">pheno</span>, <span class="va">sex</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>   sex   n
1 male 598</code></pre>
</div>
<p>All of the mice are male.</p>
</div>
</div>
</div>
</div>
<div id="challenge-3-how-is-the-prop.bm.mn.ret-phenotype-distributed" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3-how-is-the-prop.bm.mn.ret-phenotype-distributed" class="callout-inner">
<h3 class="callout-title">Challenge 3: How is the “prop.bm.mn.ret” phenotype distributed?</h3>
<div class="callout-content">
<p>Make histogram of the “prop.bm.mn.ret” column and assess whether it
is normally distributed.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>You can look at the number of observations in the Environment tab or
you can get the number of rows in <code>pheno</code> in the Console.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">pheno</span>, mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span><span class="va">prop.bm.mn.ret</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Histogram of Bone Marrow MN-RETs"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-6-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The bone marrow MN-RET values do not look normally distributed.</p>
</div>
</div>
</div>
</div>
<div id="challenge-4-how-is-the-logprop.bm.mn.ret-phenotype-distributed" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-4-how-is-the-logprop.bm.mn.ret-phenotype-distributed" class="callout-inner">
<h3 class="callout-title">Challenge 4: How is the log(“prop.bm.mn.ret”) phenotype distributed?</h3>
<div class="callout-content">
<p>Take the log of the “prop.bm.mn.ret” column and make a
histogram.Assess whether it is normally distributed.</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" aria-labelledby="headingSolution4" data-bs-parent="#accordionSolution4">
<div class="accordion-body">
<p>You can look at the number of observations in the Environment tab or
you can get the number of rows in <code>pheno</code> in the Console.</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>log_mnret <span class="op">=</span> <span class="fu">log</span><span class="op">(</span><span class="va">prop.bm.mn.ret</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span><span class="va">log_mnret</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_histogram</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Histogram of log(Bone Marrow MN-RETs)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-7-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The log of the bone marrow MN-RET values look more normally
distributed.</p>
</div>
</div>
</div>
</div>
<p>Let’s add a column containing the log-transformed phenotype to our
<code>pheno</code> object.</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="va">pheno</span> <span class="op">|&gt;</span></span>
<span>           <span class="fu">mutate</span><span class="op">(</span>log_mnret <span class="op">=</span> <span class="fu">log</span><span class="op">(</span><span class="va">prop.bm.mn.ret</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Some researchers are concerned about the reproducibility of DO
studies. The argument is that each DO mouse is unique and therefore can
never be reproduced. But this misses the point of using the DO. While
mice are the sampling unit, in QTL mapping we are sampling the founder
alleles at each locus. An average of 1/8th of the alleles should come
from each founder at any given locus. Also, DO mice are a
<em>population</em> of mice, not a single strain. While it is true that
results in an individual DO mouse may not be reproducible, results at
the population level should be reproducible. This is similar to the
human population in that results from one individual may not represent
all humans, but results at the population level should be
reproducible.</p>
<p>The benzene inhalation study was conducted in two separate cohorts
(termed <em>study</em> in the <code>pheno</code> object). Let’s plot the
proportion of micronucleated reticulocytes in bone marrow versus the
benzene concentration for each study cohort.</p>
<div id="accordionInstructor3" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor3" aria-expanded="false" aria-controls="collapseInstructor3">
  <h3 class="accordion-header" id="headingInstructor3">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor3" class="accordion-collapse collapse" aria-labelledby="headingInstructor3" data-bs-parent="#accordionInstructor3">
<div class="accordion-body">
<p>Remember, you just need to show the basic plot. Just type out the
ggplot, geom, log scale, and facet wrap lines in class.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>conc <span class="op">=</span> <span class="fu">factor</span><span class="op">(</span><span class="va">conc</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span><span class="va">conc</span>, <span class="va">prop.bm.mn.ret</span> <span class="op">*</span> <span class="fl">1000</span>, color <span class="op">=</span> <span class="va">conc</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_violin</span><span class="op">(</span>draw_quantiles <span class="op">=</span> <span class="fl">0.5</span>, linewidth <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_beeswarm</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_color_brewer</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Reds"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">study</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Bone Marrow MN-RET by Study Cohort"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span>text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>, </span>
<span>          legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-9-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>As you can see, while individual mice have varying micronucleated
reticulocytes, there is a dose-dependent increase in micronucleated
reticulocytes in both cohorts. This is an example of how results in the
DO reproduce at the population level.</p>
</div>
<div class="section level3">
<h3 id="markers">Markers<a class="anchor" aria-label="anchor" href="#markers"></a>
</h3>
<p>The markers are the SNPs on the Mouse Universal Genotyping Array (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4751547/" class="external-link">MUGA</a>)
that were used to reconstruct the haplotypes of the DO mice. This
version of the array had about 8,000 SNPs on it. The latest version
contains over 140,000 SNPs. We are using this smaller data set so that
we can finish the analysis in a timely manner during class.</p>
<p>The markers have been mapped to the latest mouse genome build
(GRCm39) and are provided by Karl Broman on Github at: <a href="https://github.com/kbroman/MUGAarrays/tree/main/UWisc" class="external-link uri">https://github.com/kbroman/MUGAarrays/tree/main/UWisc</a>.
There are four versions of the MUGA platforms. In this study, we used
the MUGA and those marker files are named as “muga_uwisc_vN.csv”, where
“N” is a version number. We will use version 4, which you downloaded
into your “data” directory during the Setup before the workshop.</p>
<p>Open the marker file now:</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers</span> <span class="op">=</span> <span class="fu">read.csv</span><span class="op">(</span><span class="st">"./data/muga_uwisc_v4.csv"</span><span class="op">)</span></span></code></pre>
</div>
<div id="challenge-5-which-columns-contain-the-marker-positions" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-5-which-columns-contain-the-marker-positions" class="callout-inner">
<h3 class="callout-title">Challenge 5: Which columns contain the marker positions?</h3>
<div class="callout-content">
<p>Look at the structure of the genetic or physical map in the
Environment tab and see which columns contain marker positions in some
coordinate system.</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" aria-labelledby="headingSolution5" data-bs-parent="#accordionSolution5">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">str</span><span class="op">(</span><span class="va">markers</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>'data.frame':	7854 obs. of  11 variables:
 $ marker        : chr  "JAX00240603" "UNC010001397" "UNC010515443" "UNC010001943" ...
 $ chr           : chr  "1" "1" "1" "1" ...
 $ bp_mm10       : int  3252796 3336839 3668628 3977130 4430623 4531029 5045931 5840130 6020779 6378154 ...
 $ bp_grcm39     : int  3323019 3407062 3738851 4047353 4500846 4601252 5116154 5910353 6091003 6448378 ...
 $ cM_cox        : num  0.15 0.172 0.194 0.197 0.202 ...
 $ strand        : chr  "minus" "minus" "minus" "plus" ...
 $ snp           : chr  "TC" "AC" "TC" "AC" ...
 $ unique        : logi  TRUE TRUE TRUE TRUE TRUE TRUE ...
 $ unmapped      : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...
 $ probe         : chr  "AAAATATGATTCTTTTAATTAATGCATCAGTGATGAGAAGTGAGAGTGGC" "CATAGTGTCTGGTGAGAAGTCTGGAGTTATTCTAATAGGCCTGCCTTTAT" "CAGGAAATGATGCTGAGAAAGTGAGAAGTAGGAAAACGTGGAGAAAAATA" "TCTATTCCTATCACCTTGTACAAAGCTCAAGTCTTGTAAACCCCCCCCCC" ...
 $ strand_flipped: logi  TRUE FALSE FALSE FALSE FALSE FALSE ...</code></pre>
</div>
<p>The columns labelled “bp_mm10”, “bp_grcm39”, and “cM_cox” contain
marker positions.</p>
</div>
</div>
</div>
</div>
<div id="challenge-6-which-chromosomes-are-in-the-markers" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-which-chromosomes-are-in-the-markers" class="callout-inner">
<h3 class="callout-title">Challenge 6: Which chromosomes are in the markers?</h3>
<div class="callout-content">
<p>Use an R command to determine which chromosomes are in
<code>markers</code>.</p>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" aria-labelledby="headingSolution6" data-bs-parent="#accordionSolution6">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">count</span><span class="op">(</span><span class="va">markers</span>, <span class="va">chr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>    chr   n
1     1 493
2    10 375
3    11 335
4    12 318
5    13 327
6    14 331
7    15 265
8    16 266
9    17 227
10   18 216
11   19 162
12    2 474
13    3 438
14    4 412
15    5 382
16    6 414
17    7 379
18    8 341
19    9 342
20    M   3
21    X 414
22    Y   5
23 &lt;NA&gt; 935</code></pre>
</div>
<p>There are 23 chromosomes, including some markers with “NA”. These are
markers that could not be uniquely mapped to one chromosome.</p>
</div>
</div>
</div>
</div>
<div id="challenge-6-what-do-you-think-the-difference-is-between-teh-bp_mm10-and" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-6-what-do-you-think-the-difference-is-between-teh-bp_mm10-and" class="callout-inner">
<h3 class="callout-title">Challenge 6: What do you think the difference is between teh “bp_mm10” and</h3>
<div class="callout-content">
<p>“bp_grcm39” columns?</p>
<p>Turn to your neighbor and discuss what the differences between the
two column are. Then share your ideas with the class.</p>
</div>
</div>
</div>
<div id="accordionSolution7" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution7" aria-expanded="false" aria-controls="collapseSolution7">
  <h4 class="accordion-header" id="headingSolution7"> Show me the solution </h4>
</button>
<div id="collapseSolution7" class="accordion-collapse collapse" aria-labelledby="headingSolution7" data-bs-parent="#accordionSolution7">
<div class="accordion-body">
<p>There are different builds of the mouse genome. The Genome Resource
Consortium (GRC) named mouse builds from build 1 to 39. The University
of California at Santa Cruz developed their own naming system which
conflicted with the GRC names. “GRCm38” was equivalent to “mm10”. We are
using the GRC naming conventions for genome builds.</p>
</div>
</div>
</div>
</div>
<p><a href="https://github.com/kbroman/qtl2" class="external-link">qtl2</a> needs the markers
to be in a specific format. <code>qtl2</code> requires that the markers
be in a list. The <a href="https://github.com/kbroman/qtl2convert" class="external-link">qtl2convert</a> package
provides functions to make this easier.</p>
<p>First, we will filter the markers to retain ones on the autosomes and
Chr X. We don’t use the mitochondrial or Y chromosomes in mapping, but
they may be important covariates. The MUGA did not contain enough
markers on Chr M and Y to identify the contributing founder strain, but
later versions of the MUGA, such as the GigaMUGA, do contain enough
markers to identify founder contributions.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers</span> <span class="op">&lt;-</span> <span class="va">markers</span> <span class="op">|&gt;</span></span>
<span>             <span class="fu">filter</span><span class="op">(</span><span class="va">chr</span> <span class="op">%in%</span> <span class="fu">c</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>, <span class="st">'X'</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we will create a column with GRCm39 Mb positions.</p>
<div class="codewrapper sourceCode" id="cb19">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers</span> <span class="op">&lt;-</span> <span class="va">markers</span> <span class="op">|&gt;</span></span>
<span>             <span class="fu">mutate</span><span class="op">(</span>pos <span class="op">=</span> <span class="va">bp_grcm39</span> <span class="op">/</span> <span class="fl">1e6</span><span class="op">)</span></span></code></pre>
</div>
<p>Finally, we will use the <a href="">qtl2convert::map_df_to_list()</a>
function to create the list of markers that <code>qtl2</code> requires.
We will call this object the marker “map” to distinguish it from the
marker data.frame.</p>
<div class="codewrapper sourceCode" id="cb20">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">map</span> <span class="op">=</span> <span class="fu">qtl2convert</span><span class="fu">::</span><span class="fu">map_df_to_list</span><span class="op">(</span>map <span class="op">=</span> <span class="va">markers</span>, pos_column <span class="op">=</span> <span class="st">"pos"</span><span class="op">)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="founder-allele-probabilities-genoprobs">Founder Allele Probabilities (Genoprobs)<a class="anchor" aria-label="anchor" href="#founder-allele-probabilities-genoprobs"></a>
</h3>
<p>The <code>probs</code> object is a list with 20 elements. Each
element of <code>probs</code> is a three-dimensional array containing
the founder allele dosages for each sample at each marker on one
chromosome. These probabilities have been pre-calculated for you, so you
can skip the step for calculating allele probabilities.</p>
<p>Let’s look at the dimensions of <code>probs</code> for chromosome
1:</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dim</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 590   8 479</code></pre>
</div>
<p><code>probs[[1]]</code> is a three-dimensional array containing the
proportion of each founder haplotype at each marker for each DO sample.
The 590 samples are in the first dimension, the 8 founders in the second
and the 479 markers along chromosome 1 are in the third dimension.</p>
<p>The figure below illustrates the structure of the <code>probs</code>
object. <code>prob</code> is a list with one element per chromosome.
Each element is a three-dimensional array which contains the founder
allele probabilities for eahch sample at each marker.</p>
<p><img src="../fig/genoprobs_3D.png" alt="Genoprobs Object" class="figure">{alt=‘Three-dimensional figure of genoprobs
list.’,width=’100%}</p>
<p>Let’s return to the <code>probs</code> object. Look at the contents
of one sample on chromosome 1.</p>
<div id="accordionInstructor4" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor4" aria-expanded="false" aria-controls="collapseInstructor4">
  <h3 class="accordion-header" id="headingInstructor4">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor4" class="accordion-collapse collapse" aria-labelledby="headingInstructor4" data-bs-parent="#accordionInstructor4">
<div class="accordion-body">
<p>This code is intended to throw an error. It is used to illustrate the
point that you have to align the markers between <code>probs</code> and
<code>map</code>.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_genoprob</span><span class="op">(</span><span class="va">probs</span>, <span class="va">map</span>, ind <span class="op">=</span> <span class="fl">1</span>, chr <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre>
</div>
<p>Uh oh! We have an error which says “Different numbers of positions in
probs and map”. This means that the number of markers in the genoprobs
and the marker map is different. In fact, this is quite important. The
markers in <code>probs</code> and <code>map</code> must match
<strong>exactly</strong> on every chromosome.</p>
<div id="challenge-7-align-the-markers-in-probs-and-map." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-7-align-the-markers-in-probs-and-map." class="callout-inner">
<h3 class="callout-title">Challenge 7: Align the markers in
<code>probs</code> and <code>map</code>.</h3>
<div class="callout-content">
<p>Work with the person next to you to write a short script that will
get the intersection of the marker names in the <code>probs</code> and
<code>map</code> for each chromosome, subset the markers in each object
to only contain the common markers, and make sure that the markers in
<code>probs</code> are in the same order as in <code>map</code>.</p>
<p><strong>Hint</strong>: The markers names for each chromosome in
<code>probs</code> are in dimention 3. i.e. dimnames(probs[[1]])[[3]]
contains the marker names for chromosome 1.</p>
</div>
</div>
</div>
<div id="accordionSolution8" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution8" aria-expanded="false" aria-controls="collapseSolution8">
  <h4 class="accordion-header" id="headingSolution8"> Show me the solution </h4>
</button>
<div id="collapseSolution8" class="accordion-collapse collapse" aria-labelledby="headingSolution8" data-bs-parent="#accordionSolution8">
<div class="accordion-body">
<p>We will write a loop that gets the common markers between probs and
map for each chromosome and then subsets them.</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Verify that probs and map are the same length.</span></span>
<span><span class="fu">stopifnot</span><span class="op">(</span><span class="fu">length</span><span class="op">(</span><span class="va">probs</span><span class="op">)</span> <span class="op">==</span> <span class="fu">length</span><span class="op">(</span><span class="va">map</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Verify that probs and map have their chromsomes in the same order.</span></span>
<span><span class="fu">stopifnot</span><span class="op">(</span><span class="fu">names</span><span class="op">(</span><span class="va">probs</span><span class="op">)</span> <span class="op">==</span> <span class="fu">names</span><span class="op">(</span><span class="va">map</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Loop through each chromosome...</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu">seq_along</span><span class="op">(</span><span class="va">probs</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="co"># Get the intersection of the markers in probs and map.</span></span>
<span>  <span class="va">common_markers</span> <span class="op">&lt;-</span> <span class="fu">intersect</span><span class="op">(</span><span class="fu">dimnames</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span>, <span class="fu">names</span><span class="op">(</span><span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Subset the map to contain the common markers.</span></span>
<span>  <span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">common_markers</span><span class="op">]</span></span>
<span>  </span>
<span>  <span class="co"># Subset the probs to contain the common markers.</span></span>
<span>  <span class="va">probs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">probs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,,<span class="fu">names</span><span class="op">(</span><span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">]</span></span>
<span>  </span>
<span>  <span class="co"># Verify that the markers are identical in probs and map.</span></span>
<span>  <span class="fu">stopifnot</span><span class="op">(</span><span class="fu">dimnames</span><span class="op">(</span><span class="va">probs</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="fu">names</span><span class="op">(</span><span class="va">map</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span><span class="op">}</span> <span class="co"># for(i)</span></span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>Now that we have aligned the markers in <code>probs</code> and
<code>map</code>, let’s plot the allele probabilities for one sample on
chromosome 1.</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_genoprob</span><span class="op">(</span><span class="va">probs</span>, <span class="va">map</span>, ind <span class="op">=</span> <span class="fl">1</span>, chr <span class="op">=</span> <span class="fl">1</span>, main <span class="op">=</span> <span class="st">"Founder Allele Probabilities"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In the plot above, the founder contributions, which range between 0
and 1, are colored from white (= 0) to black (= 1.0). A value of ~0.5 is
grey. The markers are on the X-axis and the eight founders (denoted by
the letters A through H) on the Y-axis. Starting at the left, we see
that this sample has genotype BB because the row for B is black,
indicating values of 1.0. Moving along the genome to the right, the
genotype becomes CE where rows C and E are gray, followed by CD, FH, AG,
GH, etc. The values at each marker sum to 1.0.</p>
<p>The figure below shows genotype and allele probabilities for 3
samples. In the Diversity Outbred, there are 36 possible genotype states
(AA, AB, AC, …, BB, BC, BD, …, CC, CD, CE, …, DD, DE, DF, …, EE,…) or 8
+ 7 + 6 + 5 + 4 + 3 + 2 + 1. The first SNP below has genotype BB. In the
table describing alleles (8 state founder probabilities), the
probability that this SNP has a B allele is 1. The 2nd SNP has genotype
BH, so the allele table shows a probability of 0.5 for B and 0.5 for H.
The third SNP is either BG or BH, and has a probability of 0.5 for each
of these genotypes. The allele table shows a probability of 0.5 for
allele B, and 0.25 for both G and H.</p>
<figure><img src="../fig/geno-to-allele-probs.png" alt="a table showing the probabilities for each of 36 genotypes in the Diversity Outbred followed by a second table showing probabilities for each of the 8 founder alleles" class="figure mx-auto d-block"><div class="figcaption">Genotype and allele probabilities</div>
</figure>
</div>
<div class="section level3">
<h3 id="calculating-a-kinship-matrix">Calculating A Kinship Matrix<a class="anchor" aria-label="anchor" href="#calculating-a-kinship-matrix"></a>
</h3>
<p>Next, we need to create a matrix that accounts for the kinship
relationships between the mice. We do this by looking at the correlation
between the founder haplotypes for each sample at each SNP. For each
chromosome, we create a kinship matrix using all markers <em>except</em>
the ones on the current chromosome using the “loco”
(leave-one-chromosome-out) method. Simulations suggest that mapping
using this approach increases the power to detect QTL.</p>
<div id="callout2" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>The sample IDs must be in the rownames of <code>probs</code>. The
sample IDs will be copied to the row and column names in the kinship
matrices.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">K</span> <span class="op">&lt;-</span> <span class="fu">calc_kinship</span><span class="op">(</span>probs <span class="op">=</span> <span class="va">probs</span>, type <span class="op">=</span> <span class="st">"loco"</span><span class="op">)</span></span></code></pre>
</div>
<p>Kinship values between pairs of samples range between 0 (no
relationship) and 1.0 (completely identical). Let’s look at the kinship
matrix for the first 50 samples.</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n_samples</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="fu">heatmap</span><span class="op">(</span><span class="va">K</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span>, <span class="fl">1</span><span class="op">:</span><span class="va">n_samples</span><span class="op">]</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-kinship_probs-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The figure above shows kinship between all pairs of samples. Light
yellow indicates low kinship and dark red indicates higher kinship.
Orange values indicate varying levels of kinship between 0 and 1. The
dark red diagonal of the matrix indicates that each sample is identical
to itself. The orange blocks along the diagonal may indicate close
relatives (i.e. siblings or cousins).</p>
</div>
<div class="section level3">
<h3 id="covariates">Covariates<a class="anchor" aria-label="anchor" href="#covariates"></a>
</h3>
<p>Next, we need to create additive covariates that will be used in the
mapping model. We will use study cohort as a covariate in the mapping
model. This is the same as outbreeding generation since each cohort was
purchased from successive generations. If we were mapping with
<em>all</em> mice, we would also add benzene concentration to the model.
This study contained only male mice, but in most cases, you would
include sex as an additive covariate as well.</p>
<p>To recap, you would normally add sex and outbreeding generation to
the model. In this case, we have one sex and are using
<code>study</code> instead of generation.</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">addcovar</span> <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">study</span>, data <span class="op">=</span> <span class="va">pheno</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span></code></pre>
</div>
<p>The code above copies the <code>rownames(pheno)</code> to
<code>rownames(addcovar)</code> as a side-effect.</p>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>The sample IDs <strong>must</strong> be in the rownames of
<code>pheno</code>, <code>addcovar</code>, <code>genoprobs</code> and
<code>K</code>. <code>qtl2</code> uses the sample IDs to align the
samples between objects. For more information about data file format,
see <a href="https://kbroman.org/qtl2/assets/vignettes/input_files.html" class="external-link">Karl
Broman’s vignette on input file format</a>.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="performing-a-genome-scan">Performing a Genome Scan<a class="anchor" aria-label="anchor" href="#performing-a-genome-scan"></a>
</h2>
<hr class="half-width">
<p>Before we perform our first genome scan, let’s look at the mapping
model. At each marker on the genotyping array, we will fit a model that
regresses the phenotype (micronucleated reticulocytes) on covariates and
the founder allele proportions.</p>
<p><span class="math inline">\(y_i =
\beta_ss_i+\sum_{j=1}^8(\beta_jg_{ij}) + \lambda_i +
\epsilon_i\)</span></p>
<p>where:</p>
<p><span class="math inline">\(y_i\)</span> is the phenotype for mouse
<span class="math inline">\(i\)</span>, <span class="math inline">\(\beta_s\)</span> is the effect of study cohort,
<span class="math inline">\(s_i\)</span> is the study cohort for mouse
<span class="math inline">\(i\)</span>, <span class="math inline">\(\beta_j\)</span> is the effect of founder allele
<span class="math inline">\(j\)</span>, <span class="math inline">\(g_{ij}\)</span> is the probability that mouse
<span class="math inline">\(i\)</span> carries an allele from founder
<span class="math inline">\(j\)</span>, <span class="math inline">\(\lambda_;&lt;sub&gt;_i\)</span> is an adjustment
for kinship-induced correlated errors for mouse <span class="math inline">\(i\)</span>, <span class="math inline">\(\epsilon_i\)</span> is the residual error for
mouse <span class="math inline">\(i\)</span>.</p>
<p>Note that this model will give us an estimate of the effect of each
of the eight founder alleles at each marker. This will be important when
we estimate the founder allele effects below.</p>
<p>There are almost 600 samples in this data set and it may take several
minutes to map one trait. In order to save some time, we will map using
only the samples in the 100 ppm concentration group. We will create a
smaller phenotype data.frame.</p>
<div class="codewrapper sourceCode" id="cb29">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno_100</span> <span class="op">&lt;-</span> <span class="va">pheno</span><span class="op">[</span><span class="va">pheno</span><span class="op">$</span><span class="va">conc</span> <span class="op">==</span> <span class="fl">100</span>,<span class="op">]</span></span></code></pre>
</div>
<div id="challenge-8-how-many-mice-are-we-mapping-with" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-8-how-many-mice-are-we-mapping-with" class="callout-inner">
<h3 class="callout-title">Challenge 8: How many mice are we mapping with?</h3>
<div class="callout-content">
<p>Get the number of rows in <code>pheno_100</code>.</p>
</div>
</div>
</div>
<div id="accordionSolution9" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution9" aria-expanded="false" aria-controls="collapseSolution9">
  <h4 class="accordion-header" id="headingSolution9"> Show me the solution </h4>
</button>
<div id="collapseSolution9" class="accordion-collapse collapse" aria-labelledby="headingSolution9" data-bs-parent="#accordionSolution9">
<div class="accordion-body">
<p>You can do this by looking at the number of observations for
<code>pheno_100</code> in the Environment tab or in the Console.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">nrow</span><span class="op">(</span><span class="va">pheno_100</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>[1] 149</code></pre>
</div>
<p>There are 149 mice in <code>pheno_100</code>.</p>
</div>
</div>
</div>
</div>
<p>In order to map the proportion of bone marrow reticulocytes that were
micro-nucleated, you will use the <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_scan1.R" class="external-link">scan1</a>
function. To see the arguments for <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_scan1.R" class="external-link">scan1</a>,
you can type <code>help(scan1)</code>. First, let’s map the
<em>untransformed</em> phenotype. (Recall that we log-transformed it
above).</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">pheno_100</span><span class="op">)</span> <span class="op">==</span> <span class="st">"prop.bm.mn.ret"</span><span class="op">)</span></span>
<span><span class="va">lod</span>   <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>               pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>               kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>               addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>Next, we plot the genome scan.</p>
<div class="codewrapper sourceCode" id="cb33">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">map</span>, </span>
<span>           main <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-qtl_plot-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>It looks like we have a large peak on chromosome 10.</p>
<div id="challenge-9-how-does-a-log-tranformation-change-the-qtl-plot" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-9-how-does-a-log-tranformation-change-the-qtl-plot" class="callout-inner">
<h3 class="callout-title">Challenge 9: How does a log-tranformation change the QTL plot?</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Perform a genome scan on the column called <code>log_mnret</code>.
(Hint: set <code>index</code> to the column index in
<code>pheno</code>.)<br>
</li>
<li>How does the LOD score for the peak on Chr 10 change?</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution10" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution10" aria-expanded="false" aria-controls="collapseSolution10">
  <h4 class="accordion-header" id="headingSolution10"> Show me the solution </h4>
</button>
<div id="collapseSolution10" class="accordion-collapse collapse" aria-labelledby="headingSolution10" data-bs-parent="#accordionSolution10">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">pheno_100</span><span class="op">)</span> <span class="op">==</span> <span class="st">"log_mnret"</span><span class="op">)</span></span>
<span><span class="va">lod2</span>  <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>               pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>               kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>               addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod2</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">map</span>, </span>
<span>           main <span class="op">=</span> <span class="st">"(log(Proportion of Micro-nucleated Bone Marrow Reticulocytes)"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-22-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Using a log transformation increases the LOD increase from about 17
to over 25.</p>
</div>
</div>
</div>
</div>
<p>The challenge shows the importance of looking at your data and
transforming it to meet the expectations of the mapping model. In this
case, a log transformation improved the model fit and increased the LOD
score. We will continue the rest of this lesson using the
log-transformed data. Set your <code>index</code> variable equal to the
column index of <code>log.MN.RET</code>. Redo the genome scan with the
log-transformed reticulocytes.</p>
<div class="codewrapper sourceCode" id="cb35">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span> <span class="op">==</span> <span class="st">"log_mnret"</span><span class="op">)</span></span>
<span><span class="va">lod</span>   <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>               pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>, <span class="va">index</span>,  drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>               kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>               addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="assessing-significance-of-lod-scores">Assessing Significance of LOD Scores<a class="anchor" aria-label="anchor" href="#assessing-significance-of-lod-scores"></a>
</h2>
<hr class="half-width">
<p>There is clearly a large peak on Chr 10. But is it significant? In
other words, could we see a LOD score over 25 by chance? And if so, what
is the probability of seeing a LOD of 25 or higher?</p>
<p>These questions are most commonly answered via <a href="https://www.genetics.org/content/178/1/609.long" class="external-link">permutation</a>
of the sample labels and recording the maximum LOD score in each
permutation. This procedure breaks the connection between the phenotypes
and the genotypes of the mice, so the results represent the expected LOD
by chance.</p>
<p>Imagine that you shuffle the sample labels in the phenotype file and
map the resampled trait. Then you record the maximum LOD across the
genome. You would get a table that looks like this:</p>
<table class="table">
<thead><tr class="header">
<th>Permutation</th>
<th>Max LOD</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>1.7</td>
</tr>
<tr class="even">
<td>2</td>
<td>3.2</td>
</tr>
<tr class="odd">
<td>3</td>
<td>2.2</td>
</tr>
<tr class="even">
<td>4</td>
<td>1.8</td>
</tr>
<tr class="odd">
<td>5</td>
<td>2.3</td>
</tr>
<tr class="even">
<td>…</td>
<td>…</td>
</tr>
</tbody>
</table>
<p>Once you ran 1,000 permutations, you could plot a histogram of the
maximum LOD values.</p>
<div class="codewrapper sourceCode" id="cb36">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perms</span>   <span class="op">&lt;-</span> <span class="fu">readRDS</span><span class="op">(</span><span class="st">'./data/sim_perm1000.rds'</span><span class="op">)</span></span>
<span><span class="va">max_lod</span> <span class="op">&lt;-</span> <span class="fu">apply</span><span class="op">(</span><span class="va">perms</span>, <span class="fl">2</span>, <span class="va">max</span><span class="op">)</span></span>
<span><span class="fu">hist</span><span class="op">(</span><span class="va">max_lod</span>, breaks <span class="op">=</span> <span class="fl">100</span>, main <span class="op">=</span> <span class="st">'Histogram of Max. LOD'</span><span class="op">)</span></span>
<span><span class="va">q95</span>     <span class="op">&lt;-</span> <span class="fu">quantile</span><span class="op">(</span><span class="va">max_lod</span>, probs <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span>
<span><span class="fu">abline</span><span class="op">(</span>v <span class="op">=</span> <span class="va">q95</span>, col <span class="op">=</span> <span class="st">'red'</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-23-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>In this case, the maximum LOD score is around 9. Most of the values
fall between 5 and 6. We have drawn a red line at the 95th percentile of
the distribution. LOD values above this are likely to occur 5% of the
time. That means 1 in 20 times. We often use this 95th percentile as our
significance threshold for one trait.</p>
<p>We advise running at least 1,000 permutations to obtain significance
thresholds. In the interest of time, we perform 100 permutations here
using the <a href="">scan1perm</a> function.</p>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perms</span> <span class="op">&lt;-</span> <span class="fu">scan1perm</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                   pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                   kinshp    <span class="op">=</span> <span class="va">K</span>,</span>
<span>                   addcovar  <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                   n_perm    <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></code></pre>
</div>
<p>The <code>perms</code> object contains the maximum LOD score from
each genome scan of permuted data.</p>
<div id="challenge-10-how-does-a-log-tranformation-change-the-qtl-plot" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-10-how-does-a-log-tranformation-change-the-qtl-plot" class="callout-inner">
<h3 class="callout-title">Challenge 10: How does a log-tranformation change the QTL plot?</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Create a histogram of the LOD scores <code>perms</code>. Hint: use
the <code>hist()</code> function.<br>
</li>
<li>Estimate the value of the LOD score at the 95th percentile.<br>
</li>
<li>Then find the value of the LOD score at the 95th percentile using
the <code>summary()</code> function.</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution11" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution11" aria-expanded="false" aria-controls="collapseSolution11">
  <h4 class="accordion-header" id="headingSolution11"> Show me the solution </h4>
</button>
<div id="collapseSolution11" class="accordion-collapse collapse" aria-labelledby="headingSolution11" data-bs-parent="#accordionSolution11">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb38">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span>x <span class="op">=</span> <span class="va">perms</span>, breaks <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-24-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">summary</span><span class="op">(</span><span class="va">perms</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>LOD thresholds (100 permutations)
     log_mnret
0.05      7.87</code></pre>
</div>
<p>Note that this summary function returns the 95th percentile value of
the LOD distribution.</p>
</div>
</div>
</div>
</div>
<p>We can now add thresholds to the previous QTL plot. We use a
significance threshold of p &lt; 0.05. To do this, we select the 95th
percentile of the permutation LOD distribution.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">map</span>,  </span>
<span>           main <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span><span class="op">)</span></span>
<span><span class="va">thr</span> <span class="op">&lt;-</span> <span class="fu">summary</span><span class="op">(</span><span class="va">perms</span><span class="op">)</span></span>
<span><span class="fu">add_threshold</span><span class="op">(</span>map <span class="op">=</span> <span class="va">map</span>, thresholdA <span class="op">=</span> <span class="va">thr</span>, col <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-25-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The peak on Chr 10 is well above the red significance line.</p>
</section><section><h2 class="section-heading" id="finding-lod-peaks">Finding LOD Peaks<a class="anchor" aria-label="anchor" href="#finding-lod-peaks"></a>
</h2>
<hr class="half-width">
<p>We can find all of the peaks above the significance threshold using
the <a href="https://github.com/rqtl/qtl2/blob/master/R/find_peaks.R" class="external-link">find_peaks</a>
function.</p>
<div class="codewrapper sourceCode" id="cb42">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod</span>,</span>
<span>           map          <span class="op">=</span> <span class="va">map</span>, </span>
<span>           threshold    <span class="op">=</span> <span class="va">thr</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr      pos      lod
1        1 log_mnret  10 33.52438 27.01701</code></pre>
</div>
<div id="challenge-11-how-does-a-log-tranformation-change-the-qtl-plot" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-11-how-does-a-log-tranformation-change-the-qtl-plot" class="callout-inner">
<h3 class="callout-title">Challenge 11: How does a log-tranformation change the QTL plot?</h3>
<div class="callout-content">
<p>Find all peaks for this scan whether or not they meet the 95%
significance threshold.</p>
</div>
</div>
</div>
<div id="accordionSolution12" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution12" aria-expanded="false" aria-controls="collapseSolution12">
  <h4 class="accordion-header" id="headingSolution12"> Show me the solution </h4>
</button>
<div id="collapseSolution12" class="accordion-collapse collapse" aria-labelledby="headingSolution12" data-bs-parent="#accordionSolution12">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">map</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>   lodindex lodcolumn chr        pos       lod
1         1 log_mnret   1   4.601252  3.594686
2         1 log_mnret   2 141.066196  5.351645
3         1 log_mnret   3 124.914616  3.904994
4         1 log_mnret   4 132.636314  5.501382
5         1 log_mnret   6  85.246537  5.965923
6         1 log_mnret   7  44.136809  4.325292
7         1 log_mnret   8  26.176972  3.073792
8         1 log_mnret   9   8.115893  4.027077
9         1 log_mnret  10  33.524378 27.017008
10        1 log_mnret  11  48.605368  3.563292
11        1 log_mnret  12  28.480475  3.549043
12        1 log_mnret  13 109.752043  3.111400
13        1 log_mnret  16  72.091039  5.263606
14        1 log_mnret  17  12.528523  3.659285
15        1 log_mnret  18  65.977753  3.894702
16        1 log_mnret   X   8.770202  4.059789</code></pre>
</div>
<p>Notice that some peaks are missing because they don’t meet the
default threshold value of 3. See <code>help(find_peaks)</code> for more
information about this function.</p>
</div>
</div>
</div>
</div>
<p>The support interval is determined using the<br><a href="https://www.ncbi.nlm.nih.gov/pubmed/11560912" class="external-link">Bayesian Credible
Interval</a> and represents the region most likely to contain the
causative polymorphism(s). We can obtain this interval by adding a
<code>prob</code> argument to <a href="https://github.com/rqtl/qtl2/blob/master/R/find_peaks.R" class="external-link">find_peaks</a>.
We pass in a value of <code>0.95</code> to request a support interval
that contains the causal SNP 95% of the time.</p>
<div class="codewrapper sourceCode" id="cb46">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span>scan1_output <span class="op">=</span> <span class="va">lod</span>, </span>
<span>           map          <span class="op">=</span> <span class="va">map</span>, </span>
<span>           threshold    <span class="op">=</span> <span class="va">thr</span>, </span>
<span>           prob         <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex lodcolumn chr      pos      lod   ci_lo    ci_hi
1        1 log_mnret  10 33.52438 27.01701 31.7442 34.05311</code></pre>
</div>
<p>From the output above, you can see that the support interval is 2.3
Mb wide (31.7442 to 34.05311 Mb). The location of the maximum LOD score
is at 33.52438 Mb.</p>
</section><section><h2 class="section-heading" id="estimated-founder-allele-effects">Estimated Founder Allele Effects<a class="anchor" aria-label="anchor" href="#estimated-founder-allele-effects"></a>
</h2>
<hr class="half-width">
<p>We will now zoom in on Chr 10 and look at the contribution of each of
the eight founder alleles to the proportion of bone marrow reticulocytes
that were micro-nucleated. Remember, the mapping model above estimates
the effect of each of the eight DO founders. We can plot these effects
(also called ‘coefficients’) across Chr 10 using <a href="https://github.com/rqtl/qtl2/blob/master/R/scan1coef.R" class="external-link">scan1coef</a>.</p>
<div class="codewrapper sourceCode" id="cb48">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>    <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">coef10</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                    pheno     <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                    kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                    addcovar  <span class="op">=</span> <span class="va">addcovar</span><span class="op">)</span></span></code></pre>
</div>
<p>This produces an object containing estimates of each of the eight DO
founder allele effect. These are the <i>β<sub>j</sub></i> values in the
mapping equation above.</p>
<div class="codewrapper sourceCode" id="cb49">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_coefCC</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">coef10</span>, </span>
<span>            map  <span class="op">=</span> <span class="va">map</span>, </span>
<span>            scan1_output <span class="op">=</span> <span class="va">lod</span>, </span>
<span>            main <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-27-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>The top panel shows the eight founder allele effects (or model
coefficients) along Chr 10. The founder allele effects are centered at
zero and the units are the same as the phenotype. You can see that DO
mice containing the CAST/EiJ allele near 34 Mb have lower levels of
micro-nucleated reticulocytes. This means that the CAST allele is
associated with less DNA damage and has a protective effect. The bottom
panel shows the LOD score, with the support interval for the peak shaded
blue.</p>
</section><section><h2 class="section-heading" id="snp-association-mapping">SNP Association Mapping<a class="anchor" aria-label="anchor" href="#snp-association-mapping"></a>
</h2>
<hr class="half-width">
<p>At this point, we have a 2.3 Mb wide support interval that contains
polymorphism(s) that influence benzene-induced DNA damage. Next, we will
impute the DO founder sequences onto the DO genomes. The <a href="https://www.mousegenomes.org/" class="external-link">Mouse Genomes Project</a> has
sequenced the eight DO founders and provides SNP, insertion-deletion
(indel), and structural variant files for the strains (see <a href="https://www.nature.com/ng/journal/v45/n7/full/ng.2644.html" class="external-link">Baud
et.al., Nat. Gen., 2013</a>). We can impute these SNPs onto the DO
genomes and then perform association mapping. The process involves
several steps and I have provided a function to encapsulate the steps.
To access the Sanger SNPs, we use a SQLlite database provided by <a href="https://github.com/kbroman" class="external-link">Karl Broman</a>. You should have
downloaded this during Setup. It is available from<br><a href="https://figshare.com/ndownloader/files/40157572" class="external-link">figshare</a>,
but the file is 10 GB, so it may take too long to download right
now.</p>
<p><img src="../fig/DO.impute.founders.sm.png" alt="SNP Imputation" class="figure">{alt=‘Figure showing haplotypes being used to
impute founder SNPs onto DO genomes’,width=‘100%’}</p>
<p>Association mapping involves imputing the founder SNPs onto each DO
genome and fitting the mapping model at each SNP. At each marker, we fit
the following model:</p>
<p><span class="math inline">\(y_{im} = \beta_ss_i+\beta_mg_{im} +
\lambda_i + \epsilon_i\)</span></p>
<p>where:</p>
<p><span class="math inline">\(y_i\)</span> is the phenotype for mouse
<span class="math inline">\(i\)</span>, <span class="math inline">\(\beta_s\)</span> is the effect of study cohort,
<span class="math inline">\(s_i\)</span> is the study cohort for mouse
<span class="math inline">\(i\)</span>, <span class="math inline">\(\beta_m\)</span> is the effect of adding one
allele at marker <span class="math inline">\(m\)</span>, <span class="math inline">\(g_{im}\)</span> is the allele call for mouse <span class="math inline">\(i\)</span> at marker <span class="math inline">\(m\)</span>&gt;, <span class="math inline">\(\lambda_i\)</span> is an adjustment for
kinship-induced correlated errors for mouse <span class="math inline">\(i\)</span>, <span class="math inline">\(\epsilon_i\)</span> is the residual error for
mouse <span class="math inline">\(i\)</span>.</p>
<p>We can call <a href="https://github.com/rqtl/qtl2/blob/master/R/scan1snps.R" class="external-link">scan1snps</a>
to perform association mapping in the QTL interval on Chr 10. We first
create variables for the chromosome and support interval where we are
mapping. We then create a function to get the SNPs from the founder SNP
database.</p>
<div id="callout4" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>It is important to use the <code>keep_all_snps = TRUE</code> in order
to return all SNPs.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb50">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>   <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">start</span> <span class="op">&lt;-</span> <span class="fl">30</span></span>
<span><span class="va">end</span>   <span class="op">&lt;-</span> <span class="fl">36</span></span>
<span></span>
<span><span class="co"># Create function to query founder SNP database.</span></span>
<span><span class="va">query_snps</span>  <span class="op">&lt;-</span> <span class="fu">create_variant_query_func</span><span class="op">(</span>dbfile   <span class="op">=</span> <span class="st">"./data/fv.2021.snps.db3"</span>,</span>
<span>                                         id_field <span class="op">=</span> <span class="st">"variants_id"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">assoc</span>      <span class="op">&lt;-</span> <span class="fu">scan1snps</span><span class="op">(</span>genoprobs  <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                        map        <span class="op">=</span> <span class="va">map</span>, </span>
<span>                        pheno      <span class="op">=</span> <span class="va">pheno_100</span><span class="op">[</span>,<span class="va">index</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                        kinship    <span class="op">=</span> <span class="va">K</span>, </span>
<span>                        addcovar   <span class="op">=</span> <span class="va">addcovar</span>, </span>
<span>                        query_func <span class="op">=</span> <span class="va">query_snps</span>,</span>
<span>                        chr        <span class="op">=</span> <span class="va">chr</span>, </span>
<span>                        start      <span class="op">=</span> <span class="va">start</span>, </span>
<span>                        end        <span class="op">=</span> <span class="va">end</span>, </span>
<span>                        keep_all_snps <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<p>The <code>assoc</code> object is a list containing two objects: the
LOD scores for each unique SNP and a <code>snpinfo</code> object that
maps the LOD scores to each SNP. To plot the association mapping, we
need to provide both objects to the <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_snpasso.R" class="external-link">plot_snpasso</a>
function.</p>
<div class="codewrapper sourceCode" id="cb51">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#png('./fig/bm_mnret_assoc_fig1.png', width = 2000, height = 2000, res = 256)</span></span>
<span><span class="fu">plot_snpasso</span><span class="op">(</span>scan1output <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">lod</span>, </span>
<span>             snpinfo     <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">snpinfo</span>, </span>
<span>             main        <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span><span class="op">)</span></span>
<span><span class="co">#dev.off()</span></span></code></pre>
</div>
<p><img src="../fig/bm_mnret_assoc_fig1.png" alt="Bone Marrow MN-RET Association Mapping" class="figure">{alt=‘Plot showing LOD of
each SNP in QTL interval’,width=‘100%’}</p>
<p>This plot shows the LOD score for each SNP in the QTL interval. The
SNPs occur in “shelves” because all of the SNPs in a haplotype block
have the same founder strain pattern. The SNPs with the highest LOD
scores are the ones for which CAST/EiJ contributes the alternate
allele.</p>
<p>You might naturally ask what genes are under this QTL peak. We will
query the SNP/gens database for the genes in the interval. As before, we
will create a function to query the genes in a specific interval of the
genome.</p>
<div class="codewrapper sourceCode" id="cb52">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create function to query gene database.</span></span>
<span><span class="va">query_genes</span> <span class="op">&lt;-</span> <span class="fu">create_gene_query_func</span><span class="op">(</span>dbfile      <span class="op">=</span> <span class="st">"./data/fv.2021.snps.db3"</span>,</span>
<span>                                      chr_field   <span class="op">=</span> <span class="st">"chromosome"</span>, </span>
<span>                                      name_field  <span class="op">=</span> <span class="st">"symbol"</span>,</span>
<span>                                      start_field <span class="op">=</span> <span class="st">"start_position"</span>, </span>
<span>                                      stop_field  <span class="op">=</span> <span class="st">"end_position"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu">query_genes</span><span class="op">(</span><span class="va">chr</span>, <span class="va">start</span>, <span class="va">end</span><span class="op">)</span></span>
<span><span class="fu">head</span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>          ensembl_id ensembl_version    Name                     name chromosome start_position end_position strand    start
1 ENSMUSG00000000295              14   Hddc2   HD domain containing 2         10       31189379     31204200      1 31.18938
2 ENSMUSG00000000296               9 Tpd52l1 tumor protein D52-like 1         10       31208372     31321954     -1 31.20837
3 ENSMUSG00000019779              16     Frk       fyn-related kinase         10       34359395     34487274      1 34.35939
4 ENSMUSG00000019782              11   Rwdd1  RWD domain containing 1         10       33872551     33895620     -1 33.87255
5 ENSMUSG00000019785              15   Clvs2               clavesin 2         10       33388282     33500765     -1 33.38828
6 ENSMUSG00000019787              10    Trdn                  triadin         10       32956550     33352705      1 32.95655
      stop
1 31.20420
2 31.32195
3 34.48727
4 33.89562
5 33.50077
6 33.35271</code></pre>
</div>
<p>The <code>genes</code> object contains annotation information for
each gene in the interval.</p>
<p>Next, we will create a plot with two panels: one containing the
association mapping LOD scores and one containing the genes in the QTL
interval. We do this by passing in the <code>genes</code> argument to <a href="https://github.com/rqtl/qtl2/blob/master/R/plot_snpasso.R" class="external-link">plot_snpasso</a>.
We can also adjust the proportion of the plot allocated for SNPs (on the
top) and genes (on the bottom) using the ‘top_panel_prop’ argument.</p>
<div id="accordionInstructor5" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor5" aria-expanded="false" aria-controls="collapseInstructor5">
  <h3 class="accordion-header" id="headingInstructor5">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor5" class="accordion-collapse collapse" aria-labelledby="headingInstructor5" data-bs-parent="#accordionInstructor5">
<div class="accordion-body">
<p>Note that you have to add the “genes”, “colors”, and “top_panel_prop”
arguments.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb54">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#png('./fig/bm_mnret_assoc_fig2.png', width = 2000, height = 2000, res = 256)</span></span>
<span><span class="fu">plot_snpasso</span><span class="op">(</span>scan1output <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">lod</span>, </span>
<span>             snpinfo     <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">snpinfo</span>, </span>
<span>             main        <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span>, </span>
<span>             genes       <span class="op">=</span> <span class="va">genes</span>,</span>
<span>             colors      <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>             top_panel_prop <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span>
<span><span class="co">#dev.off()</span></span></code></pre>
</div>
<p><img src="../fig/bm_mnret_assoc_fig2.png" alt="Bone Marrow MN-RET Association Mapping" class="figure">{alt=‘Plot showing LOD of
each SNP in QTL interval with genes below’,width=‘100%’}</p>
<p>Now that we have the genes in this interval, we would like to know
which founders have the minor allele for the SNPs with the highest LOD
scores. To do this, we will highlight SNPs that are withing a 1 LOD drop
of the highest LOD and we will add an argument to show which founder
contributes the minor allele at the highlighted SNPs.</p>
<div id="accordionInstructor6" class="accordion instructor-note accordion-flush">
<div class="accordion-item">
<button class="accordion-button instructor-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseInstructor6" aria-expanded="false" aria-controls="collapseInstructor6">
  <h3 class="accordion-header" id="headingInstructor6">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="edit-2"></i></div> Instructor Note </h3>
</button>
<div id="collapseInstructor6" class="accordion-collapse collapse" aria-labelledby="headingInstructor6" data-bs-parent="#accordionInstructor6">
<div class="accordion-body">
<p>Note that you have to change the “top_panel_prop” argument to
“panel_prop” and add the “drop_hilit”, “col_hilit”, and “sdp_panel”
arguments.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb55">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#png('./fig/bm_mnret_assoc_fig3.png', width = 2000, height = 2000, res = 256)</span></span>
<span><span class="fu">plot_snpasso</span><span class="op">(</span>scan1output <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">lod</span>, </span>
<span>             snpinfo     <span class="op">=</span> <span class="va">assoc</span><span class="op">$</span><span class="va">snpinfo</span>, </span>
<span>             main        <span class="op">=</span> <span class="st">"Proportion of Micro-nucleated Bone Marrow Reticulocytes"</span>, </span>
<span>             genes       <span class="op">=</span> <span class="va">genes</span>,</span>
<span>             colors      <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>             panel_prop <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>             drop_hilit  <span class="op">=</span> <span class="fl">1.0</span>,</span>
<span>             col_hilit   <span class="op">=</span> <span class="st">'red'</span>,</span>
<span>             sdp_panel   <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#dev.off()</span></span></code></pre>
</div>
<p><img src="../fig/bm_mnret_assoc_fig3.png" alt="Bone Marrow MN-RET Association Mapping" class="figure">{alt=‘Plot showing LOD of
each SNP in QTL interval with genes below and strain distribution
pattern above’,width=‘100%’}</p>
<div class="section level3">
<h3 id="searching-for-candidate-genes">Searching for Candidate Genes<a class="anchor" aria-label="anchor" href="#searching-for-candidate-genes"></a>
</h3>
<p>One strategy for finding genes related to a phenotype is to search
for genes with expression QTL (eQTL) in the same location. Ideally, we
would have liver and bone marrow gene expression data in the DO mice
from this experiment. Unfortunately, we did not collect this data.
However, we have liver gene expression for a separate set of untreated
DO mice <a href="https://churchilllab.jax.org/qtlviewer/svenson/DOHFD" class="external-link">Liver eQTL
Viewer</a>. We searched for genes in the QTL interval that had an eQTL
in the same location. Then, we looked at the pattern of founder effects
to see if CAST stood out. We found two genes that met these
criteria.</p>
<p><img src="../fig/French.et.al.Figure3.png" alt="Sult3a1/Sult3a2 Liver eQTL" class="figure">{alt=‘Figure showing LOD plots of
Sult3a1 &amp; 2 and CAST-specific haplotype effects’,width=‘75%’}</p>
<p>The plot above shows the genome scane for two genes: <em>Sult3a1</em>
and <em>Gm4794</em>. <em>Gm4794</em> has been renamed to
<em>Sult3a2</em>. As you can see, both <em>Sult3a1</em> <em>Sult3a2</em>
have eQTL in the same location at the MN-RET QTL on Chr 10. Mice
carrying the CAST allele (in green) express these genes more highly.
<em>Sult3a1</em> is a <a href="https://en.wikipedia.org/wiki/Sulfotransferase" class="external-link">sulfotransferase</a>
that may be involved in adding a sulfate group to phenol, one of the
metabolites of benzene.</p>
<div id="challenge-12-look-for-sult3a1-sult3a2-qtl-in-do-bone-expression" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-12-look-for-sult3a1-sult3a2-qtl-in-do-bone-expression" class="callout-inner">
<h3 class="callout-title">Challenge 12: Look for Sult3a1 &amp; Sult3a2 QTL in DO bone expression</h3>
<div class="callout-content">
<p>Go to the <a href="https://churchilllab.jax.org/qtlviewer/DO/bone" class="external-link">DO
Bone eQTL Viewer</a>, type in <em>Sult3a1</em> and click on the search
button. Then search on <em>Sult3a2</em>.</p>
</div>
</div>
</div>
<div id="accordionSolution13" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution13" aria-expanded="false" aria-controls="collapseSolution13">
  <h4 class="accordion-header" id="headingSolution13"> Show me the solution </h4>
</button>
<div id="collapseSolution13" class="accordion-collapse collapse" aria-labelledby="headingSolution13" data-bs-parent="#accordionSolution13">
<div class="accordion-body">
<p>Neither <em>Sult3a1</em> or <em>Sult3a2</em> has an eQTL in the DO
bone marrow data set.</p>
</div>
</div>
</div>
</div>
<p>Next, we will look at the relationship between <em>Sult3a1</em> and
<em>Sult3a2</em>. Go to the Ensembl web page for <a href="https://useast.ensembl.org/Mus_musculus/Gene/Summary?db=core;g=ENSMUSG00000090298;r=10:33642420-33662700" class="external-link">Sult3a2</a>.
In the menu on the left, click on the “Gene Tree (image)” link.</p>
<p><img src="../fig/Ensembl_Mouse_Sult3a1_Sult3a2.png" alt="Ensembl gene tree of Sult3a1 &amp; Sult3a2" class="figure">{alt=‘Figure showing
that Sult3a1 &amp; 2 are paralogs’,width=‘100%’}</p>
<p>As you can see, <em>Sult3a2</em> is a paralog of <em>Sult3a1</em> and
hence both genes are sulfotransferases. These genes encode enzymes that
attach a sulfate group to other compounds.</p>
<div id="challenge-13-look-for-structural-variants-near-sult3a1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-13-look-for-structural-variants-near-sult3a1" class="callout-inner">
<h3 class="callout-title">Challenge 13: Look for structural variants near Sult3a1</h3>
<div class="callout-content">
<p>Go to the <a href="https://useast.ensembl.org/index.html" class="external-link">Ensembl
website</a> and select “Mouse” from the species dropdown. Then type
“10:33300000-33800000” into the search box and press “Go”. Scroll down
to the detailed view and look for structural variants in this
region.</p>
</div>
</div>
</div>
<div id="accordionSolution14" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution14" aria-expanded="false" aria-controls="collapseSolution14">
  <h4 class="accordion-header" id="headingSolution14"> Show me the solution </h4>
</button>
<div id="collapseSolution14" class="accordion-collapse collapse" aria-labelledby="headingSolution14" data-bs-parent="#accordionSolution14">
<div class="accordion-body">
<p>You should see a plot similar to the one below.</p>
<p><img src="../fig/Ensembl_SV_Chr10.png" alt="Ensembl Structural Variants" class="figure">{alt=‘Ensembl viewer showing genes
and structural variants under that chromosome 10 QTL
peak’,width=‘100%’}</p>
</div>
</div>
</div>
</div>
<p><img src="../fig/Ensembl_SV_Chr10.png" alt="Ensembl Structural Variants" class="figure">{alt=‘Ensembl viewer showing genes
and structural variants under that chromosome 10 QTL
peak’,width=‘100%’}</p>
<p>In order to visualize the size of the copy number gain, we queried
the Mouse Genomes Project <a href="https://www.mousegenomes.org/mouse-strains-sequenced/" class="external-link">alignment
files</a> for the eight founders. We piled up the reads at each base
(which is beyond the scope of this tutorial) and made the figure
below.</p>
<p><img src="../fig/French.et.al.Sup.Figure3.png" alt="DO Founder BAM file pileups" class="figure">{alt=‘Figure showing CAST with a
duplication at the Sult3a1/2 locus.’,width=‘100%’}</p>
<p>As you can see, there appears to be a duplicatation in the CAST
founders that covers four genes: <em>Clvs2</em>, <em>Gm15939</em>,
<em>Sult3a1</em> and <em>Sult3a2</em>. <em>Clvs2</em> is expressed in
neurons and <em>Gm15939</em> is a predicted gene that may not produce a
transcript.</p>
<p>Hence, we have three pieces of evidence that narrows our candidate
gene list to <em>Sult3a1</em> and <em>Sult3a2</em>:</p>
<ol style="list-style-type: decimal">
<li>Both genes have a liver eQTL in the same location as the
micronucleated reticulocytes QTL.</li>
<li>Among genes in the micronucleated reticulocytes QTL interval, only
<em>Sult3a1</em> and <em>Sult3a2</em> have differential expression of
the CAST allele in the liver.</li>
<li>There is a copy number gain of these two genes in CAST.</li>
</ol>
<p>Sulfation is a prominent detoxification mechanism for benzene as
well. The diagram below shows the metabolism pathway for benzene <a href="https://europepmc.org/articles/PMC4414400" class="external-link">(Monks, T. J., et
al. (2010). Chem Biol Interact 184(1-2): 201-206.)</a> Hydroquinone,
phenol and catechol are all sulfated and excreted from the body.</p>
<p><img src="../fig/Monks_ChemBiolInter_2010_Fig1.jpg" alt="Benezene metabolism pathways" class="figure">{alt=‘Figure showing benezene
metabolism pathways’,width=‘100%’}</p>
<p>This analysis has led us to the following hypothesis. Inhaled benzene
is absorbed by the lungs into the bloodstream and transported to the
liver. There, it is metabolized, and some metabolites are transported to
the bone marrow. One class of genes that is involved in toxicant
metabolism are sulfotransferases. <a href="https://www.informatics.jax.org/marker/MGI:1931469" class="external-link"><i>Sult3a1</i></a>
is a phase II enzyme that conjugates compounds (such as phenol, which is
a metabolite of benzene) with a sulfate group before transport into the
bile. It is possible that a high level of <em>Sult3a1</em> expression
could remove benzene by-products and be protective. Our hypothesis is
that the copy number gain in the CAST allele increases liver gene
expression of <em>Sult3a1</em> and <em>Sult3a2</em>. High liver
expression of these genes allows mice containing the CAST allele to
rapidly conjugate harmful benzene metabolites and excrete them from the
body before they can reach the bone marrow and cause DNA damage. Further
experimental validation is required, but this is a plausible
hypothesis.</p>
<p><img src="../fig/benzene_hypothesis.png" alt="Benzene metabolism hypothesis" class="figure">{alt=‘Figure showing mice carrying
the CAST allele being protected from benezene
toxicity’,width=‘100%’}</p>
<p>We hope that this tutorial has shown you how the DO can be used to
map QTL and use the founder effects and bioinformatics resources to
narrow down the candidate gene list. Here, we made used of external gene
expression databases and the founder sequence data to build a case for a
pair of genes.</p>
<div id="challenge-14-map-another-trait." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-14-map-another-trait." class="callout-inner">
<h3 class="callout-title">Challenge 14: Map another trait.</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Make a histogram of the column <code>pre.prop.mn.ret</code> in
<code>pheno</code>. Does it look like it should be log transformed? If
so, add a column to <code>pheno</code> that contains the log of
<code>pre.prop.mn.ret</code>.<br>
</li>
<li>Perform a genome scan <em>using all samples</em> on the column
called <code>pre.prop.mn.ret</code>. Since you’re using all samples, the
scan will take longer. You will also need to add the benezene
concentration to the covariates. (Hint: set <code>index</code> to the
column index in <code>pheno</code>.)<br>
</li>
<li>Which chromosome has the highest peak? Use <code>find_peaks</code>
to get the location and support interval for the highest peak.<br>
</li>
<li>Calculate and plot the BLUPs for the chromosome with the highest
peak. (This may take a few minutes.) Which founder contributes an allele
that makes MN-RET values higher?</li>
<li>Perform association mapping in the support interval for the QTL
peak, plot the results and plot the genes beneath the association
mapping plot.</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution15" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution15" aria-expanded="false" aria-controls="collapseSolution15">
  <h4 class="accordion-header" id="headingSolution15"> Show me the solution </h4>
</button>
<div id="collapseSolution15" class="accordion-collapse collapse" aria-labelledby="headingSolution15" data-bs-parent="#accordionSolution15">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>Plot the distribution of</li>
</ol>
<div class="codewrapper sourceCode" id="cb56">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">hist</span><span class="op">(</span><span class="va">pheno</span><span class="op">$</span><span class="va">pre.prop.mn.ret</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-33-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><p>Log transform the pre-dose proportion of MN-RETs.</p>
<div class="codewrapper sourceCode" id="cb57">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span><span class="op">$</span><span class="va">log_pre_prop_mnret</span> <span class="op">&lt;-</span> <span class="fu">log</span><span class="op">(</span><span class="va">pheno</span><span class="op">$</span><span class="va">pre.prop.mn.ret</span><span class="op">)</span></span></code></pre>
</div>
<ol start="2" style="list-style-type: decimal">
<li>Set the mapping index equal to the <code>pre.prop.mn.ret</code>
column.</li>
</ol>
<div class="codewrapper sourceCode" id="cb58">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu">which</span><span class="op">(</span><span class="fu">colnames</span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span> <span class="op">==</span> <span class="st">"log_pre_prop_mnret"</span><span class="op">)</span></span></code></pre>
</div>
<p>Create new covariates which include study and benzene
concentration.</p>
<div class="codewrapper sourceCode" id="cb59">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">addcovar2</span> <span class="op">&lt;-</span> <span class="fu">model.matrix</span><span class="op">(</span><span class="op">~</span><span class="va">study</span> <span class="op">+</span> <span class="va">conc</span>, data <span class="op">=</span> <span class="va">pheno</span><span class="op">)</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb60">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lod_pre</span> <span class="op">&lt;-</span> <span class="fu">scan1</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span>, </span>
<span>                 pheno     <span class="op">=</span> <span class="va">pheno</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                 kinship   <span class="op">=</span> <span class="va">K</span>, </span>
<span>                 addcovar  <span class="op">=</span> <span class="va">addcovar2</span><span class="op">)</span></span>
<span><span class="fu">plot_scan1</span><span class="op">(</span>x    <span class="op">=</span> <span class="va">lod_pre</span>, </span>
<span>           map  <span class="op">=</span> <span class="va">map</span>, </span>
<span>           main <span class="op">=</span> <span class="st">"Log-Transformed Pre-dose micronucleated reticulocytes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-37-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><ol start="3" style="list-style-type: decimal">
<li>Find peaks above a significance threshold.</li>
</ol>
<div class="codewrapper sourceCode" id="cb61">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">find_peaks</span><span class="op">(</span><span class="va">lod_pre</span>, <span class="va">map</span>, threshold <span class="op">=</span> <span class="fl">8</span>, prob <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>  lodindex          lodcolumn chr       pos       lod     ci_lo     ci_hi
1        1 log_pre_prop_mnret   4 132.77875 13.145968 132.77875 134.27938
2        1 log_pre_prop_mnret  12  75.45433  8.404242  74.14949  77.30552</code></pre>
</div>
<ol start="4" style="list-style-type: decimal">
<li>Calculate and plot BLUPs for the highest peak.</li>
</ol>
<div class="codewrapper sourceCode" id="cb63">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr</span>   <span class="op">&lt;-</span> <span class="fl">4</span></span>
<span><span class="va">coef4</span> <span class="op">&lt;-</span> <span class="fu">scan1blup</span><span class="op">(</span>genoprobs <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                   pheno     <span class="op">=</span> <span class="va">pheno</span><span class="op">[</span>,<span class="va">index</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                   kinship   <span class="op">=</span> <span class="va">K</span><span class="op">[[</span><span class="va">chr</span><span class="op">]</span><span class="op">]</span>, </span>
<span>                   addcovar  <span class="op">=</span> <span class="va">addcovar2</span><span class="op">)</span></span>
<span><span class="fu">plot_coefCC</span><span class="op">(</span>x            <span class="op">=</span> <span class="va">coef4</span>, </span>
<span>            map          <span class="op">=</span> <span class="va">map</span>, </span>
<span>            scan1_output <span class="op">=</span> <span class="va">lod_pre</span>, </span>
<span>            legend       <span class="op">=</span> <span class="st">"topleft"</span>,</span>
<span>            main         <span class="op">=</span> <span class="st">"Log-Transformed Pre-dose micronucleated reticulocytes"</span><span class="op">)</span></span></code></pre>
</div>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-39-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><ol start="5" style="list-style-type: decimal">
<li>Perform association mapping in the QTL interval for the highest
peak.</li>
</ol>
<div class="codewrapper sourceCode" id="cb64">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">start</span>  <span class="op">&lt;-</span> <span class="fl">132.8</span></span>
<span><span class="va">end</span>    <span class="op">&lt;-</span> <span class="fl">134.3</span></span>
<span><span class="va">assoc4</span> <span class="op">&lt;-</span> <span class="fu">scan1snps</span><span class="op">(</span>genoprobs  <span class="op">=</span> <span class="va">probs</span><span class="op">[</span>,<span class="va">chr</span><span class="op">]</span>, </span>
<span>                    map        <span class="op">=</span> <span class="va">map</span>, </span>
<span>                    pheno      <span class="op">=</span> <span class="va">pheno</span><span class="op">[</span>,<span class="va">index</span>,drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>, </span>
<span>                    kinship    <span class="op">=</span> <span class="va">K</span>, </span>
<span>                    addcovar   <span class="op">=</span> <span class="va">addcovar2</span>, </span>
<span>                    query_func <span class="op">=</span> <span class="va">query_snps</span>, </span>
<span>                    chr        <span class="op">=</span> <span class="va">chr</span>,</span>
<span>                    start      <span class="op">=</span> <span class="va">start</span>, </span>
<span>                    end        <span class="op">=</span> <span class="va">end</span>, </span>
<span>                    keep_all_snps <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu">query_genes</span><span class="op">(</span><span class="va">chr</span>, <span class="va">start</span>, <span class="va">end</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper sourceCode" id="cb65">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#png('./fig/pre_mnret_assoc.png', width = 2000, height = 2000, res = 256)</span></span>
<span><span class="fu">plot_snpasso</span><span class="op">(</span>scan1output <span class="op">=</span> <span class="va">assoc4</span><span class="op">$</span><span class="va">lod</span>, </span>
<span>             snpinfo     <span class="op">=</span> <span class="va">assoc4</span><span class="op">$</span><span class="va">snpinfo</span>, </span>
<span>             main        <span class="op">=</span> <span class="st">"Log-Transformed Pre-dose micronucleated reticulocytes"</span>, </span>
<span>             genes       <span class="op">=</span> <span class="va">genes</span>,</span>
<span>             colors      <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>             panel_prop <span class="op">=</span> <span class="fu">c</span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>             drop_hilit  <span class="op">=</span> <span class="fl">1.0</span>,</span>
<span>             col_hilit   <span class="op">=</span> <span class="st">'red'</span>,</span>
<span>             sdp_panel   <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#dev.off()</span></span></code></pre>
</div>
<p><img src="../fig/pre_mnret_assoc.png" alt="Pre-dose MN-RET Association Mapping" class="figure">{alt=‘Figure showing
pre-dose MN-RET association mapping with WSB carrying the alternate
allele’,width=‘100%’}</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb66">
<h3 class="code-label">R<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessionInfo</span><span class="op">(</span><span class="op">)</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C

time zone: UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
 [1] qtl2convert_0.30 qtl2_0.36        ggbeeswarm_0.7.2 lubridate_1.9.3
 [5] forcats_1.0.0    stringr_1.5.1    dplyr_1.1.4      purrr_1.0.2
 [9] readr_2.1.5      tidyr_1.3.1      tibble_3.2.1     ggplot2_3.5.1
[13] tidyverse_2.0.0

loaded via a namespace (and not attached):
 [1] utf8_1.2.4          generics_0.1.3      renv_1.0.11
 [4] RSQLite_2.3.7       stringi_1.8.4       hms_1.1.3
 [7] magrittr_2.0.3      RColorBrewer_1.1-3  evaluate_0.24.0
[10] grid_4.4.1          timechange_0.3.0    fastmap_1.2.0
[13] blob_1.2.4          DBI_1.2.3           BiocManager_1.30.25
[16] fansi_1.0.6         scales_1.3.0        qtl_1.70
[19] cli_3.6.3           rlang_1.1.4         bit64_4.5.2
[22] munsell_0.5.1       withr_3.0.2         cachem_1.1.0
[25] yaml_2.3.10         parallel_4.4.1      tools_4.4.1
[28] tzdb_0.4.0          memoise_2.0.1       colorspace_2.1-1
[31] vctrs_0.6.5         R6_2.5.1            lifecycle_1.0.4
[34] bit_4.5.0           vipor_0.4.7         pkgconfig_2.0.3
[37] beeswarm_0.4.0      pillar_1.9.0        gtable_0.3.6
[40] glue_1.7.0          data.table_1.16.2   Rcpp_1.0.13-1
[43] highr_0.11          xfun_0.46           tidyselect_1.2.1
[46] knitr_1.48          farver_2.1.2        labeling_0.4.3
[49] compiler_4.4.1     </code></pre>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>There are generally five steps to QTL mapping in DO mice:
<ul>
<li>map the trait,</li>
<li>perform permutations,</li>
<li>find significant peaks,</li>
<li>calculate founder allele effects at the QTL peak,</li>
<li>perform association mapping to narrow the gene candidates.</li>
</ul>
</li>
<li>You may need to bring in outside resources to help narrow your
candidate gene list.</li>
<li>You will need the 10 GB SNP database to perform association
mapping.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.10" class="external-link">sandpaper (0.16.10)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.7" class="external-link">pegboard (0.7.7)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.5" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries.github.io/workbench-template-rmd/instructor/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries, genetics, statistics, genome, genetic mapping",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/instructor/aio.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/instructor/aio.html",
  "dateCreated": "2024-09-06",
  "dateModified": "2024-11-13",
  "datePublished": "2024-11-13"
}

  </script><script>
		feather.replace();
	</script>
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

