<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Quantitative Trait Mapping: All Images</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/incubator/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicons/incubator/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicons/incubator/favicon-16x16.png">
<link rel="manifest" href="../favicons/incubator/site.webmanifest">
<link rel="mask-icon" href="../favicons/incubator/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="../assets/images/incubator-logo.svg"><span class="badge text-bg-info">
          <abbr title="This lesson is in the beta phase, which means that it is ready for teaching by instructors outside of the original author team.">
            <a href="https://docs.carpentries.org/resources/curriculum/lesson-life-cycle.html" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-circle" style="border-radius: 5px"></i>
              Beta
            </a>
            <span class="visually-hidden">This lesson is in the beta phase, which means that it is ready for teaching by instructors outside of the original author team.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../images.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="../assets/images/incubator-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Quantitative Trait Mapping
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Quantitative Trait Mapping
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Quantitative Trait Mapping
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../images.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="intro_data_set.html">1. Introduction to the Data Set</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="input-file-format.html">2. Input File Format</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="calc-genoprob.html">3. Calculating Genotype Probabilities</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="perform-genome-scan.html">4. Performing a Genome Scan</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="calc-kinship.html">5. Calculating A Kinship Matrix</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="perform-genome-scan-lmm.html">6. Performing a genome scan with a linear mixed model</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="perform-genome-scan-bin.html">7. Performing a Genome Scan with Binary Traits</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="perform-perm-test.html">8. Finding Significant Peaks via Permutation</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="find-lod-peaks.html">9. Finding QTL peaks</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="est-qtl-effects.html">10. Estimating QTL effects</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="integrating-gene-expression.html">11. Integrating Gene Expression Data</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush13">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading13">
        <a href="do_qtl_mapping.html">12. QTL Mapping in Diversity Outbred Mice</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="intro_data_set"><h2 class="section-heading"><a href="intro_data_set.html">Introduction to the Data Set</a></h2>
<hr class="half-width">
<h3 id="intro_data_set-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing Type 2 diabetes &amp; insulin.’</p>
<p><img src="../fig/healthy-vs-T2D.png" alt="Figure showing Type 2 diabetes &amp; insulin." class="figure"> Created in
BioRender.com</p>
<hr>
<h3 id="intro_data_set-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing intercross breeding design.’</p>
<figure><img src="../fig/intercross.png" alt="Figure showing intercross breeding design." class="figure mx-auto d-block"></figure><hr></section><section id="input-file-format"><h2 class="section-heading"><a href="input-file-format.html">Input File Format</a></h2>
<hr class="half-width">
<h3 id="input-file-format-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘Table showing the mouse genotypes as BB, BR, and RR.’</p>
<figure><img src="../fig/attie_geno_sample.png" alt="Table showing the mouse genotypes as BB, BR, and RR." class="figure mx-auto d-block"><div class="figcaption">Attie Sample Genotypes</div>
</figure><hr>
<h3 id="input-file-format-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘Table showing the marker, chromosome, and centimorgan position for five markers’</p>
<figure><img src="../fig/attie_geno_map_sample.png" alt="Table showing the marker, chromosome, and centimorgan position for five markers" class="figure mx-auto d-block"><div class="figcaption">Attie Genetic Map</div>
</figure><hr>
<h3 id="input-file-format-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘Table showing top five rows of physical marker map.’</p>
<figure><img src="../fig/attie_phys_map_sample.png" alt="Table showing top five rows of physical marker map." class="figure mx-auto d-block"><div class="figcaption">Attie Physical Map</div>
</figure><hr>
<h3 id="input-file-format-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘Table showing top five rows of phenotype table, including insulin’</p>
<figure><img src="../fig/attie_pheno_sample.png" alt="Table showing top five rows of phenotype table, including insulin" class="figure mx-auto d-block"><div class="figcaption">Attie Phenotypes</div>
</figure><hr>
<h3 id="input-file-format-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘Table showing the top five rows of covariates table’</p>
<figure><img src="../fig/attie_covar_sample.png" alt="Table showing the top five rows of covariates table" class="figure mx-auto d-block"><div class="figcaption">Attie Covariates</div>
</figure><hr>
<h3 id="input-file-format-figure-6">Figure 6</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing the qtl2 control file’</p>
<figure><img src="../fig/attie_control_file_sample.png" alt="Figure showing the qtl2 control file" class="figure mx-auto d-block"><div class="figcaption">Attie Control File</div>
</figure><hr></section><section id="calc-genoprob"><h2 class="section-heading"><a href="calc-genoprob.html">Calculating Genotype Probabilities</a></h2>
<hr class="half-width">
<h3 id="calc-genoprob-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘a chromosome with two typed markers labeled BR and RR with a locus of unknown genotype between them’</p>
<figure><img src="../fig/unknown_genotype.png" alt="a chromosome with two typed markers labeled BR and RR with a locus of unknown genotype between them" class="figure mx-auto d-block"><div class="figcaption">Genotype probabilities must be calculated
between typed markers; adapted from Broman &amp; Sen, 2009</div>
</figure><hr>
<h3 id="calc-genoprob-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing three-dimensional array of genotype probabilities (genoprobs)’</p>
<figure><img src="../fig/threeD_array.png" alt="Figure showing three-dimensional array of genotype probabilities (genoprobs)" class="figure mx-auto d-block"><div class="figcaption">Three-dimensional genotype probabilities
array</div>
</figure><hr>
<h3 id="calc-genoprob-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘a web page showing R data structures including one-dimensional vectors and lists, two dimensional dataframes and matrices, and n-dimensional arrays’</p>
<p><img src="https://devopedia.org/images/article/46/9213.1526125966.png" alt="a web page showing R data structures including one-dimensional vectors and lists, two dimensional dataframes and matrices, and n-dimensional arrays" class="figure">
Notice that arrays in R require data to be all of the same type - all
numeric, all character, all Boolean, etc. If you are familiar with data
frames in R you know that you can mix different kinds of data in that
structure. The first column might contain numeric data, the second
column character data, the third Boolean (True / False), and so on.
Arrays won’t accept mixed data types though.</p>
<hr>
<h3 id="calc-genoprob-figure-4">Figure 4</h3>
<figure><img src="../fig/calc-genoprob-rendered-plot_genoprob-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="calc-genoprob-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘a workflow diagram showing genotypes and genetic maps used to calculate genotype probabilities’</p>
<figure><img src=".././fig/mapping-workflow-calc-genoprob.png" alt="a workflow diagram showing genotypes and genetic maps used to calculate genotype probabilities" class="figure mx-auto d-block"><div class="figcaption">We have completed the first steps of the mapping
workflow.</div>
</figure><hr></section><section id="perform-genome-scan"><h2 class="section-heading"><a href="perform-genome-scan.html">Performing a Genome Scan</a></h2>
<hr class="half-width">
<h3 id="perform-genome-scan-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘Null and alternative hypotheses’</p>
<figure><img src="../fig/nullvalt.png" alt="Null and alternative hypotheses" class="figure mx-auto d-block"><div class="figcaption">Linear regression can identify QTL; adapted from
Broman &amp; Sen, 2009</div>
</figure><hr>
<h3 id="perform-genome-scan-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 2: ‘Alternative hypothesis’</p>
<p aria-hidden="true">Image 2 of 2: ‘Null hypothesis’</p>
<p><img src="../fig/althypothesis.png" alt="Alternative hypothesis" class="figure"><img src="../fig/nullhypothesis.png" alt="Null hypothesis" class="figure"></p>
<hr>
<h3 id="perform-genome-scan-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘Haley Knott regression’</p>
<figure><img src="../fig/hk-regress.png" alt="Haley Knott regression" class="figure mx-auto d-block"><div class="figcaption">Haley-Knott regression for missing
genotypes</div>
</figure><hr>
<h3 id="perform-genome-scan-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing sex and additive QTL effects.’</p>
<figure><img src="../fig/additive_qtl_effect_w_sex.png" alt="Figure showing sex and additive QTL effects." class="figure mx-auto d-block"><div class="figcaption">Additive Sex Effect</div>
</figure><hr>
<h3 id="perform-genome-scan-figure-5">Figure 5</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing sex and additive QTL effects.’</p>
<figure><img src="../fig/additive_qtl_effect_wo_sex.png" alt="Figure showing sex and additive QTL effects." class="figure mx-auto d-block"><div class="figcaption">Additive Sex Effect</div>
</figure><hr>
<h3 id="perform-genome-scan-figure-6">Figure 6</h3>
<figure><img src="../fig/perform-genome-scan-rendered-plot_add_lod-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="perform-genome-scan-figure-7">Figure 7</h3>
<p aria-hidden="true">Image 1 of 1: ‘A workflow diagram showing genotypes and genetic map used to calculate genotype probabilities which are then used to perform a genome scan along with phenotypes, covariates, and an optional physical map’</p>
<figure><img src=".././fig/mapping-workflow-genome-scan.png" alt="A workflow diagram showing genotypes and genetic map used to calculate genotype probabilities which are then used to perform a genome scan along with phenotypes, covariates, and an optional physical map" class="figure mx-auto d-block"><div class="figcaption">We have completed these steps in the mapping
workflow.</div>
</figure><hr>
<h3 id="perform-genome-scan-figure-8">Figure 8</h3>
<figure><img src="../fig/perform-genome-scan-rendered-challenge3-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="perform-genome-scan-figure-9">Figure 9</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing interactive sex by QTL effects’</p>
<figure><img src="../fig/interactive_qtl_effect_w_sex.png" alt="Figure showing interactive sex by QTL effects" class="figure mx-auto d-block"><div class="figcaption">Interactive Sex by QTL Effect</div>
</figure><hr>
<h3 id="perform-genome-scan-figure-10">Figure 10</h3>
<figure><img src="../fig/perform-genome-scan-rendered-plot_int-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="perform-genome-scan-figure-11">Figure 11</h3>
<figure><img src="../fig/perform-genome-scan-rendered-plot_add_int_lod-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="perform-genome-scan-figure-12">Figure 12</h3>
<figure><img src="../fig/perform-genome-scan-rendered-plot_add_int_diff_lod-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr></section><section id="calc-kinship"><h2 class="section-heading"><a href="calc-kinship.html">Calculating A Kinship Matrix</a></h2>
<hr class="half-width">
<h3 id="calc-kinship-figure-1">Figure 1</h3>
<figure><img src="../fig/calc-kinship-rendered-plot_kinship-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="calc-kinship-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘We have completed these steps in the mapping workflow.’</p>
<p><img src=".././fig/mapping-workflow-calc-kinship.png" alt="We have completed these steps in the mapping workflow." class="figure">{alt=“Part
of the mapping workflow starting with genotypes and a genetic map used
to calculate genotype probabilities, which are then used to calculate
kinship between individuals in the data.} By default, the genotype
probabilities are converted to allele probabilities, and the kinship
matrix is calculated as the proportion of shared alleles. Also by
default we omit the X chromosome and only use the autosomes. To include
the X chromosome, use <code>omit_x=FALSE</code>.</p>
<hr>
<h3 id="calc-kinship-figure-3">Figure 3</h3>
<figure><img src="../fig/calc-kinship-rendered-challenge4-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr></section><section id="perform-genome-scan-lmm"><h2 class="section-heading"><a href="perform-genome-scan-lmm.html">Performing a genome scan with a linear mixed model</a></h2>
<hr class="half-width">
<h3 id="perform-genome-scan-lmm-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘A world map showing the origins and ranges of five Mus musculus subspecies: domesticus, musculus, castaneus, bactrianus and molossinus’</p>
<figure><img src="https://www.informatics.jax.org/silver/images/figure2-3.gif" alt="A world map showing the origins and ranges of five Mus musculus subspecies: domesticus, musculus, castaneus, bactrianus and molossinus" class="figure mx-auto d-block"><div class="figcaption">Geographical ranges of subspecies in the Mus
musculus group. From <a href="https://www.informatics.jax.org/silverbook/" class="external-link"><em>Mouse Genetics:
Concepts and Applications</em></a> by Lee M. Silver, Oxford University
Press, 1995.</div>
</figure><hr>
<h3 id="perform-genome-scan-lmm-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘A genealogy chart showing the origin of commonly used inbred strains as Chinese, Japanese and English mouse fanciers. Abby Lathrop of Granby Mouse Farm supplied fancy mice to researchers including Marsh, C.C. Little, MacDowell, Castle and Strong in the early 1900s. These mice were developed into common inbred strains such as C57BL, 129, DBA and C3H strains.’</p>
<figure><img src="https://www.informatics.jax.org/morsebook/images/1-10as.jpg" alt="A genealogy chart showing the origin of commonly used inbred strains as Chinese, Japanese and English mouse fanciers. Abby Lathrop of Granby Mouse Farm supplied fancy mice to researchers including Marsh, C.C. Little, MacDowell, Castle and Strong in the early 1900s. These mice were developed into common inbred strains such as C57BL, 129, DBA and C3H strains." class="figure mx-auto d-block"><div class="figcaption">The genealogy of inbred mice. From <a href="https://www.informatics.jax.org/morsebook/" class="external-link"><em>Origins of Inbred
Mice</em></a> edited by Herbert C. Morse III, Academic Press, 1978.
Permission from Elsevier.</div>
</figure><hr>
<h3 id="perform-genome-scan-lmm-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘Null versus alternative hypotheses’</p>
<figure><img src="../fig/nullvalt.png" alt="Null versus alternative hypotheses" class="figure mx-auto d-block"><div class="figcaption">Null and Alternative Hypotheses</div>
</figure><hr>
<h3 id="perform-genome-scan-lmm-figure-4">Figure 4</h3>
<figure><img src="../fig/perform-genome-scan-lmm-rendered-plot_lod_all-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="perform-genome-scan-lmm-figure-5">Figure 5</h3>
<figure><img src="../fig/perform-genome-scan-lmm-rendered-plot_lod_add_loco_all-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="perform-genome-scan-lmm-figure-6">Figure 6</h3>
<figure><img src="../fig/perform-genome-scan-lmm-rendered-plot_loco_none-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr></section><section id="perform-genome-scan-bin"><h2 class="section-heading"><a href="perform-genome-scan-bin.html">Performing a Genome Scan with Binary Traits</a></h2>
<hr class="half-width">
<h3 id="perform-genome-scan-bin-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘C57BL/6J mouse’</p>
<figure><img src="../fig/C57BL_6J_000664.png" alt="C57BL/6J mouse" class="figure mx-auto d-block"><div class="figcaption">C57BL/6J</div>
</figure><hr>
<h3 id="perform-genome-scan-bin-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘BTBR’</p>
<figure><img src="../fig/BTBR_002282.png" atl="BTBR mouse" alt="BTBR" class="figure mx-auto d-block"><div class="figcaption">BTBR</div>
</figure><hr>
<h3 id="perform-genome-scan-bin-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘C3H/HeJ mouse’</p>
<figure><img src="../fig/C3H_HeJ_000659.png" alt="C3H/HeJ mouse" class="figure mx-auto d-block"><div class="figcaption">C3H/HeJ</div>
</figure><hr>
<h3 id="perform-genome-scan-bin-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘Image showing tufted hair loss in BTBR mouse’</p>
<figure><img src="../fig/Ellis_JHered_2013.jpg" alt="Image showing tufted hair loss in BTBR mouse" class="figure mx-auto d-block"><div class="figcaption">Tufted hair loss pattern</div>
</figure><hr>
<h3 id="perform-genome-scan-bin-figure-5">Figure 5</h3>
<figure><img src="../fig/perform-genome-scan-bin-rendered-plot_bin_scan-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="perform-genome-scan-bin-figure-6">Figure 6</h3>
<figure><img src="../fig/perform-genome-scan-bin-rendered-plot_bin_scan_chr2-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="perform-genome-scan-bin-figure-7">Figure 7</h3>
<figure><img src="../fig/perform-genome-scan-bin-rendered-challenge2b-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr></section><section id="perform-perm-test"><h2 class="section-heading"><a href="perform-perm-test.html">Finding Significant Peaks via Permutation</a></h2>
<hr class="half-width">
<h3 id="perform-perm-test-figure-1">Figure 1</h3>
<figure><img src="../fig/perform-perm-test-rendered-hist_perm-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="perform-perm-test-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing decreasing variance of significance threshold estimates with increasing permutations’</p>
<figure><img src="../fig/permutation_simulations.png" alt="Figure showing decreasing variance of significance threshold estimates with increasing permutations" class="figure mx-auto d-block"><div class="figcaption">Significance Threshold Variance</div>
</figure><hr>
<h3 id="perform-perm-test-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘A diagram showing mapping steps including calculating genotype probabilities, calculating kinship, performing a genome scan, finding QTL peaks, and performing a permutation test.’</p>
<p><img src=".././fig/mapping-workflow-permutation.png" alt="A diagram showing mapping steps including calculating genotype probabilities, calculating kinship, performing a genome scan, finding QTL peaks, and performing a permutation test." class="figure">
As with <code>scan1()</code>, you can speed up the calculations on a
multi-core machine by specifying the argument <code>cores</code>. With
<code>cores=0</code>, the number of available cores will be detected via
<code>parallel::detectCores()</code>. Otherwise, specify the number of
cores as a positive integer. For large data sets, be mindful of the
amount of memory that will be needed; you may need to use fewer than the
maximum number of cores, to avoid going beyond the available memory.</p>
<hr>
<h3 id="perform-perm-test-figure-4">Figure 4</h3>
<p aria-hidden="true">Image 1 of 1: ‘Comparison of significance thresholds’</p>
<figure><img src="../fig/perm_kinship.png" alt="Comparison of significance thresholds" class="figure mx-auto d-block"><div class="figcaption">Significance Thresholds with/without
Kinship</div>
</figure><hr></section><section id="find-lod-peaks"><h2 class="section-heading"><a href="find-lod-peaks.html">Finding QTL peaks</a></h2>
<hr class="half-width">
<h3 id="find-lod-peaks-figure-1">Figure 1</h3>
<figure><img src="../fig/find-lod-peaks-rendered-plot_scan1-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr></section><section id="est-qtl-effects"><h2 class="section-heading"><a href="est-qtl-effects.html">Estimating QTL effects</a></h2>
<hr class="half-width">
<h3 id="est-qtl-effects-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘Null and alternative hypotheses’</p>
<figure><img src="../fig/nullvalt.png" alt="Null and alternative hypotheses" class="figure mx-auto d-block"></figure><hr>
<h3 id="est-qtl-effects-figure-2">Figure 2</h3>
<figure><img src="../fig/est-qtl-effects-rendered-plot_coef-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="est-qtl-effects-figure-3">Figure 3</h3>
<figure><img src="../fig/est-qtl-effects-rendered-plot_effects_chr19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="est-qtl-effects-figure-4">Figure 4</h3>
<figure><img src="../fig/est-qtl-effects-rendered-plot_blup_chr19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="est-qtl-effects-figure-5">Figure 5</h3>
<figure><img src="../fig/est-qtl-effects-rendered-plot_blup_chr19_again-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="est-qtl-effects-figure-6">Figure 6</h3>
<figure><img src="../fig/est-qtl-effects-rendered-plot_fit1-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="est-qtl-effects-figure-7">Figure 7</h3>
<p aria-hidden="true">Image 1 of 1: ‘A diagram showing mapping steps up to estimating QTL effects.’</p>
<p><img src=".././fig/mapping-workflow-estimate-effects.png" alt="A diagram showing mapping steps up to estimating QTL effects." class="figure">
::::::::::::::::::::::::::::::::::::: challenge</p>
<hr>
<h3 id="est-qtl-effects-figure-8">Figure 8</h3>
<figure><img src="../fig/est-qtl-effects-rendered-challenge5a-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="est-qtl-effects-figure-9">Figure 9</h3>
<figure><img src="../fig/est-qtl-effects-rendered-challenge5d-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="est-qtl-effects-figure-10">Figure 10</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing additive &amp; dominance QTL effects’</p>
<figure><img src="../fig/add_dom_qtl_effect.png" alt="Figure showing additive &amp; dominance QTL effects" class="figure mx-auto d-block"><div class="figcaption">Additive &amp; Dominance Effects</div>
</figure><hr>
<h3 id="est-qtl-effects-figure-11">Figure 11</h3>
<figure><img src="../fig/est-qtl-effects-rendered-plot_add_dom_contrasts-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="est-qtl-effects-figure-12">Figure 12</h3>
<figure><img src="../fig/est-qtl-effects-rendered-challenge6b-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="est-qtl-effects-figure-13">Figure 13</h3>
<figure><img src="../fig/est-qtl-effects-rendered-plot_pheno_geno_se-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr></section><section id="integrating-gene-expression"><h2 class="section-heading"><a href="integrating-gene-expression.html">Integrating Gene Expression Data</a></h2>
<hr class="half-width">
<h3 id="integrating-gene-expression-figure-1">Figure 1</h3>
<figure><img src="../fig/integrating-gene-expression-rendered-snp_density-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="integrating-gene-expression-figure-2">Figure 2</h3>
<figure><img src="../fig/integrating-gene-expression-rendered-plot_1_eqtl-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="integrating-gene-expression-figure-3">Figure 3</h3>
<figure><img src="../fig/integrating-gene-expression-rendered-plot_insulin_qtl-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="integrating-gene-expression-figure-4">Figure 4</h3>
<figure><img src="../fig/integrating-gene-expression-rendered-med_plot_10002936879-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="integrating-gene-expression-figure-5">Figure 5</h3>
<figure><img src="../fig/integrating-gene-expression-rendered-plot_mediation-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr></section><section id="do_qtl_mapping"><h2 class="section-heading"><a href="do_qtl_mapping.html">QTL Mapping in Diversity Outbred Mice</a></h2>
<hr class="half-width">
<h3 id="do_qtl_mapping-figure-1">Figure 1</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing the colors and letter codes of the CC/DO founders.’</p>
<figure><img src="../fig/cc-founder-alleles.png" alt="Figure showing the colors and letter codes of the CC/DO founders." class="figure mx-auto d-block"><div class="figcaption">CC and DO Founder Alleles</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-2">Figure 2</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing CC and DO genomes’</p>
<figure><img src="../fig/cc-do-genome-comparison.png" alt="Figure showing CC and DO genomes" class="figure mx-auto d-block"><div class="figcaption">Example Collaborative Cross and Diversity
Outbred genomes</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-3">Figure 3</h3>
<p aria-hidden="true">Image 1 of 1: ‘Benzene study dosing showing 6 hours per day, 5 days per week of inhalation.’</p>
<figure><img src="../fig/benzene_study_design.png" alt="Benzene study dosing showing 6 hours per day, 5 days per week of inhalation." class="figure mx-auto d-block"><div class="figcaption">Benzene Study Dosing</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-4">Figure 4</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-6-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-5">Figure 5</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-7-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-6">Figure 6</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-9-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-7">Figure 7</h3>
<p aria-hidden="true">Image 1 of 1: ‘Three-dimensional figure of genoprobs list.’</p>
<figure><img src="../fig/genoprobs_3D.png" alt="Three-dimensional figure of genoprobs list." class="figure mx-auto d-block"><div class="figcaption">Genoprobs Object</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-8">Figure 8</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-19-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-9">Figure 9</h3>
<p aria-hidden="true">Image 1 of 1: ‘a table showing the probabilities for each of 36 genotypes in the Diversity Outbred followed by a second table showing probabilities for each of the 8 founder alleles’</p>
<figure><img src="../fig/geno-to-allele-probs.png" alt="a table showing the probabilities for each of 36 genotypes in the Diversity Outbred followed by a second table showing probabilities for each of the 8 founder alleles" class="figure mx-auto d-block"><div class="figcaption">Genotype and allele probabilities</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-10">Figure 10</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-kinship_probs-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-11">Figure 11</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-qtl_plot-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-12">Figure 12</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-22-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-13">Figure 13</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-23-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-14">Figure 14</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-24-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-15">Figure 15</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-25-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-16">Figure 16</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-27-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-17">Figure 17</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing haplotypes being used to impute founder SNPs onto DO genomes’</p>
<figure><img src="../fig/DO.impute.founders.sm.png" alt="Figure showing haplotypes being used to impute founder SNPs onto DO genomes" class="figure mx-auto d-block"><div class="figcaption">SNP Imputation</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-18">Figure 18</h3>
<p aria-hidden="true">Image 1 of 1: ‘Plot showing LOD of each SNP in QTL interval’</p>
<p><img src="../fig/bm_mnret_assoc_fig1.png" alt="Plot showing LOD of each SNP in QTL interval" class="figure"> This plot shows
the LOD score for each SNP in the QTL interval. The SNPs occur in
“shelves” because all of the SNPs in a haplotype block have the same
founder strain pattern. The SNPs with the highest LOD scores are the
ones for which CAST/EiJ contributes the alternate allele.</p>
<hr>
<h3 id="do_qtl_mapping-figure-19">Figure 19</h3>
<p aria-hidden="true">Image 1 of 1: ‘Plot showing LOD of each SNP in QTL interval with genes below’</p>
<p><img src="../fig/bm_mnret_assoc_fig2.png" alt="Plot showing LOD of each SNP in QTL interval with genes below" class="figure">
Now that we have the genes in this interval, we would like to know which
founders have the minor allele for the SNPs with the highest LOD scores.
To do this, we will highlight SNPs that are within a 1 LOD drop of the
highest LOD and we will add an argument to show which founder
contributes the minor allele at the highlighted SNPs.</p>
<hr>
<h3 id="do_qtl_mapping-figure-20">Figure 20</h3>
<p aria-hidden="true">Image 1 of 1: ‘Plot showing LOD of each SNP in QTL interval with genes below and strain distribution pattern above’</p>
<figure><img src="../fig/bm_mnret_assoc_fig3.png" alt="Plot showing LOD of each SNP in QTL interval with genes below and strain distribution pattern above" class="figure mx-auto d-block"><div class="figcaption">Bone Marrow MN-RET Association Mapping</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-21">Figure 21</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing LOD plots of Sult3a1 &amp; 2 and CAST-specific haplotype effects’</p>
<p><img src="../fig/French.et.al.Figure3.png" alt="Figure showing LOD plots of Sult3a1 &amp; 2 and CAST-specific haplotype effects" class="figure">
The plot above shows the genome scane for two genes: <em>Sult3a1</em>
and <em>Gm4794</em>. <em>Gm4794</em> has been renamed to
<em>Sult3a2</em>. As you can see, both <em>Sult3a1</em> <em>Sult3a2</em>
have eQTL in the same location at the MN-RET QTL on chromosome 10. Mice
carrying the CAST allele (in green) express these genes more highly.
<em>Sult3a1</em> is a <a href="https://en.wikipedia.org/wiki/Sulfotransferase" class="external-link">sulfotransferase</a>
that may be involved in adding a sulfate group to phenol, one of the
metabolites of benzene.</p>
<hr>
<h3 id="do_qtl_mapping-figure-22">Figure 22</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing that Sult3a1 &amp; 2 are paralogs’</p>
<figure><img src="../fig/Ensembl_Mouse_Sult3a1_Sult3a2.png" alt="Figure showing that Sult3a1 &amp; 2 are paralogs" class="figure mx-auto d-block"><div class="figcaption">Ensembl gene tree of Sult3a1 &amp; Sult3a2</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-23">Figure 23</h3>
<p aria-hidden="true">Image 1 of 1: ‘Ensembl viewer showing genes and structural variants under that chromosome 10 QTL peak’</p>
<figure><img src="../fig/Ensembl_SV_Chr10.png" alt="Ensembl viewer showing genes and structural variants under that chromosome 10 QTL peak" class="figure mx-auto d-block"><div class="figcaption">Ensembl Structural Variants</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-24">Figure 24</h3>
<p aria-hidden="true">Image 1 of 1: ‘Ensembl viewer showing genes and structural variants under that chromosome 10 QTL peak’</p>
<figure><img src="../fig/Ensembl_SV_Chr10.png" alt="Ensembl viewer showing genes and structural variants under that chromosome 10 QTL peak" class="figure mx-auto d-block"><div class="figcaption">Ensembl Structural Variants</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-25">Figure 25</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing CAST with a duplication at the Sult3a1/2 locus.’</p>
<figure><img src="../fig/French.et.al.Sup.Figure3.png" alt="Figure showing CAST with a duplication at the Sult3a1/2 locus." class="figure mx-auto d-block"><div class="figcaption">DO Founder BAM file pileups</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-26">Figure 26</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing benezene metabolism pathways’</p>
<figure><img src="../fig/Monks_ChemBiolInter_2010_Fig1.jpg" alt="Figure showing benezene metabolism pathways" class="figure mx-auto d-block"><div class="figcaption">Benezene metabolism pathways</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-27">Figure 27</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing mice carrying the CAST allele being protected from benezene toxicity’</p>
<figure><img src="../fig/benzene_hypothesis.png" alt="Figure showing mice carrying the CAST allele being protected from benezene toxicity" class="figure mx-auto d-block"><div class="figcaption">Benzene metabolism hypothesis</div>
</figure><hr>
<h3 id="do_qtl_mapping-figure-28">Figure 28</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-33-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-29">Figure 29</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-37-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-30">Figure 30</h3>
<figure><img src="../fig/do_qtl_mapping-rendered-unnamed-chunk-39-1.png" style="display: block; margin: auto;" class="figure mx-auto d-block"></figure><hr>
<h3 id="do_qtl_mapping-figure-31">Figure 31</h3>
<p aria-hidden="true">Image 1 of 1: ‘Figure showing pre-dose MN-RET association mapping with WSB carrying the alternate allele’</p>
<figure><img src="../fig/pre_mnret_assoc.png" alt="Figure showing pre-dose MN-RET association mapping with WSB carrying the alternate allele" class="figure mx-auto d-block"><div class="figcaption">Pre-dose MN-RET Association Mapping</div>
</figure><hr></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries/workbench-template-rmd/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries/workbench-template-rmd/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries/workbench-template-rmd/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:susan.mcclatchy@jax.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.17.1" class="external-link">sandpaper (0.17.1)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.7" class="external-link">varnish (1.0.7)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "@id": "https://carpentries.github.io/workbench-template-rmd/instructor/images.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/LearningResource/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries, genetics, statistics, genome, genetic mapping",
  "name": "All Images",
  "creativeWorkStatus": "active",
  "url": "https://carpentries.github.io/workbench-template-rmd/instructor/images.html",
  "identifier": "https://carpentries.github.io/workbench-template-rmd/instructor/images.html",
  "dateCreated": "2024-09-06",
  "dateModified": "2025-08-12",
  "datePublished": "2025-08-12"
}

  </script><script>
		feather.replace();
	</script>
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

