---
title: "Estimating QTL effects"
teaching: 10
exercises: 20
---

:::::::::::::::::::::::::::::::::::::: questions 

- How do I find the founder allele effects at a QTL peak?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Estimate the founder allele effects at a QTL peak.
- Plot the estimated founder allele effects.

::::::::::::::::::::::::::::::::::::::::::::::::

```{r github_load_data,echo=FALSE,warning=FALSE,message=FALSE}
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(ggbeeswarm))
suppressPackageStartupMessages(library(qtl2))
cross <- read_cross2(file = 'https://thejacksonlaboratory.box.com/shared/static/svw7ivp5hhmd7vb8fy26tc53h7r85wez.zip')
probs <- calc_genoprob(cross = cross, map = cross$gmap, error_prob = 0.002)
kinship_loco <- calc_kinship(probs = probs, type = "loco")
# Covariates
cross$covar$Sex <- factor(cross$covar$Sex)
addcovar <- model.matrix(~Sex, data = cross$covar)[,-1, drop = FALSE]
# LOD scan for insulin.
lod_add_loco <- scan1(genoprobs = probs, 
                      pheno = cross$pheno[,"log10_insulin_10wk"], 
                      kinship = kinship_loco, addcovar = addcovar)
perm_add_loco = readRDS(url("https://thejacksonlaboratory.box.com/shared/static/fatud2gsjogbfg4xntyix05j6dg3pdnj.rds"))
thr = summary(perm_add_loco, alpha = 0.05)
```

## Estimating Founder Allele Effects

<!-- DMG: Sue, can you make the equations Latex? -->

Recall that to model data from a cross, we use

![](../fig/linear-genetic-model.png)  
 

where <i>y<sub>j</sub></i> is the phenotype of the <i>j</i>th individual, &mu; 
is the mean phenotype value, <i>&beta;<sub>k</sub></i> is the effect of the 
<i>kth</i> genotype, <i>X<sub>jk</sub></i> is the genotype for individual j, and &epsilon;<sub>j</sub> is the error for the <i>j</i>th individual. In the figure 
below, &mu; equals `r round(c2eff["D2Mit17", 4], 1)`, and &beta; equals 
`r round(abs(c2eff["D2Mit17", 1]), 1)` for the alternative hypothesis 
(QTL exists).

![](../fig/nullvalt.png)

This linear model is <i>y</i> = `r round(c2eff["D2Mit17",4], 1)` + 
`r round(abs(c2eff["D2Mit17",1]), 1)`X + &epsilon;. The model intersects the 
genotype groups at their group means, and is based on &mu; and 
<i>&beta;<sub>k</sub></i> for chromosome 2 marker D2Mit17 located at 56.8 cM.  

The effect of genotype `r dimnames(c2eff)[[2]][[3]]` (the &beta; for the 
`r dimnames(c2eff)[[2]][[3]]` genotype) at marker D2Mit17 is 
`r round(c2eff["D2Mit17", 3], 1)`, while the effect of the 
`r dimnames(c2eff)[[2]][[1]]` genotype is `r round(c2eff["D2Mit17", 1], 1)` on 
the liver phenotype. The effect of the `r dimnames(c2eff)[[2]][[2]]` genotype is 
`r round(c2eff["D2Mit17",2], 1)` relative to &mu; equals 
`r round(c2eff["D2Mit17",4], 1)`.

The `scan1()` function returns only LOD scores. To obtain estimated QTL effects,
use the function `scan1coef()`. This function takes a single phenotype and the 
genotype probabilities for a single chromosome and returns a matrix with the 
estimated coefficients at each putative QTL location along the chromosome.

For example, to get the estimated QTL effects on chromosome 2 for the liver 
phenotype, we would provide the chromosome 2 genotype probabilities and the 
liver phenotype to the function `scan1coef()` as follows:

```{r est_effects_chr2}
chr      <- "2"
eff_chr2 <- scan1coef(genoprobs = probs[,chr], 
                      pheno     = cross$pheno[,"log10_insulin_10wk", drop = FALSE],
                      kinship   = kinship_loco[[chr]],
                      addcovar  = addcovar)
```

The result is a matrix of `r nrow(eff_chr2)` positions &times; `r ncol(eff_chr2)-1` 
genotypes. An additional column contains the intercept values (&mu;).

```{r effects_matrix_dim}
dim(eff_chr2)
```

## Plotting Founder Allele Effects Along a Chromosome

To plot the QTL effects, use the `plot_coef()` function. Add the LOD plot to 
the `scan1_output` argument to include a LOD plot at the bottom.

```{r plot_coef}
plot_coef(x            = eff_chr2, 
          map          = cross$pmap,
          scan1_output = lod_add_loco, 
          legend       = "topright")
```

The plot shows effect values on the y-axis and cM values on the x-axis. The
value of the intercept (&mu;) appears at the top. The effect of the 
`r dimnames(eff_chr2)[[2]][[2]]` genotype is centered around zero, with the
effects of the other two genotypes above and below. We are usually not 
directly interested in how the additive covariates change across the genome, 
but rather, the the founder allele effects change.

To plot only the founder allele effects, use the argument `columns` to indicate
which coefficients to plot. Let's look at the columns which contain the founder 
allele effects.

```{r}
head(eff_chr2)
```

We would like to plot the columns "BB", "BR", and "RR", which are in columns
1 through 3. This is what we pass into the `columns` argument.

```{r plot_effects_chr 2}
plot_coef(x       = eff_chr2, 
          map     = cross$pmap, 
          columns = 1:3, 
          scan1_output = lod_add_loco, 
          main    = "Chromosome 2 QTL effects and LOD scores",
          legend  = "topleft")
```

Looking at the plot above, which founder allele contributes to higher 
insulin levels?

## Estimating Founder Allele Effects using BLUPs

Another option for estimating the founder allele effects is to treat them as 
[random effects](https://stats.stackexchange.com/questions/4700/what-is-the-difference-between-fixed-effect-random-effect-and-mixed-effect-mode#151800) 
and calculate 
[Best Linear Unbiased Predictors](https://en.wikipedia.org/wiki/Best_linear_unbiased_prediction) 
(BLUPs). This is particularly valuable for multi-parent populations such as the 
Collaborative Cross and Diversity Outbred mice, where the large number of 
possible genotypes at a QTL leads to considerable variability in the effect 
estimates. To calculate BLUPs, use `scan1blup()`; it takes the same arguments 
as `scan1coef()`, including the option of a kinship matrix to account for a 
residual polygenic effect.

```{r est_blups_chr2}
blup_chr2 <- scan1blup(genoprobs = probs[,chr], 
                       pheno     = cross$pheno[,"log10_insulin_10wk", drop = FALSE],
                       kinship   = kinship_loco[[chr]],
                       addcovar  = addcovar)
```

We can plot the BLUP effects using `plot_coef` as before.

```{r plot_blup_chr2}
plot_coef(x       = blup_chr2, 
          map     = cross$pmap, 
          columns = 1:3, 
          scan1_output = lod_add_loco, 
          main    = "Chromosome 2 QTL BLUP effects and LOD scores",
          legend  = "topleft")
```

In the plot below, we plotted the founder allele effects (solid lines) and the 
BLUPs (dashed lines). In this case, the effects are not greatly different, but 
the effects are "shrunken" toward zero. 

```{r plot_blup_chr2}
plot_coef(x       = eff_chr2, 
          map     = cross$pmap, 
          columns = 1:3, 
          main    = "Chromosome 2 QTL BLUP effects and LOD scores",
          legend  = "topleft")
plot_coef(x       = blup_chr2, 
          map     = cross$pmap, 
          columns = 1:3,
          lty     = 2,
          legend  = "topleft",
          add     = TRUE)
```

## Plotting Allele Effects at one Marker

You may also want plot the founder allele effects at the marker with the
highest LOD. To do this, you first need to get the position of the marker from
the peak list.

```{r find_peaks}
peaks <- find_peaks(scan1_output = lod_add_loco,
                   map          = cross$pmap,
                   threshold    = thr)
peaks
```

The position of the maximum LOD on chromosome 2 is `r subset(peaks, chr == '2')$pos`
Mb. We can pass this value into the `qtl2` function `pull_genoprobpos` to get
the genoprobs at this marker.

```{r pull_genoprobs}
max_pos <- subset(peaks, chr == '2')$pos
max_mkr <- find_marker(map = cross$pmap, 
                       chr = chr, 
                       pos = max_pos)

pr      <- pull_genoprobpos(genoprobs = probs, 
                            marker    = max_mkr)
```

<!-- DMG: Make this a challenge. -->

What does the structure of `pr` look like?

```{r str_pr}
str(pr)
```

`pr` is a numeric matrix with `r nrow(pr)` rows and `r ncol(pr)` columns. The
rownames contain the mouse IDs and the column names contain the genotypes.

We can then pass `pr` as an argument into the `fit1` function, which fits
the mapping model at a single marker.

```{r fit1}
mod = fit1(genoprobs = pr,
           pheno     = cross$pheno[,"log10_insulin_10wk", drop = FALSE],
           kinship   = kinship_loco[[chr]], 
           addcovar  = addcovar)
```

Then, we can plot the founder allele effects and their standard error.

```{r plot_fit1, fig.width=8,fig.height=6}
mod_eff = data.frame(eff = mod$coef, 
                     se  = mod$SE) |>
           rownames_to_column("genotype") |>
           filter(genotype %in% c("BB", "BR", "RR"))

ggplot(data    = mod_eff, 
       mapping = aes(x = genotype, y = eff)) +
  geom_beeswarm() +
  geom_pointrange(mapping = aes(ymin = eff - se, 
                                ymax = eff + se),
                  size       = 1.5,
                  linewidth  = 1.25) +
  labs(title = "Founder Allele Effects on Chr 2",
       x     = "Genotype", y = "Founder Allele Effects") +
  theme(text = element_text(size = 20))

```

In the plot above, which founder allele contributes to higher insulin levels?
Is that consistent with the plot created using `plot_coef` above?


<!-- DMG: STOPPED HERE -->


If instead you want additive and dominance effects, you can provide a square 
matrix of _contrasts_, as follows:

```{r est_effects_liver_c2_contr}
c2effB <- scan1coef(genoprobs = probs[,chr], 
                    pheno     = cross$pheno[,"log10_insulin_10wk", drop = FALSE],
                    kinship   = kinship_loco[[chr]],
                    addcovar  = addcovar,
                    contrasts = cbind(mu = c(1,1,1), 
                                      a  = c(-1, 0, 1), 
                                      d  = c(-0.5, 1, -0.5)))
```

The result will then contain the estimates of `mu`, `a` (the additive effect), 
and `d` (the dominance effect). 

```{r add_dom_matrix_dim}
dim(c2effB)
head(c2effB)
```
For marker `r max_mkr`, `mu`, `a`, and `d` are `r c2effB[max_mkr,]`.

Here's a plot of the chromosome 2 additive and dominance effects, which are in 
the second and third columns.

```{r add_dom_contrasts}
plot_coef(x       = c2effB, 
          map     = cross$pmap[chr], 
          columns = 2:3, 
          col     = 1:2)
```

## Plotting Phenotypes versus Genootypes

Finally, to plot the raw phenotypes against the genotypes at a single putative
QTL position, you can use the function `plot_pxg()`. This takes a vector of 
genotypes as produced by the `maxmarg()` function, which picks the most likely
genotype from a set of genotype probabilities, provided it is greater than some
specified value (the argument `minprob`). Note that the “marg” in “maxmarg” 
stands for “marginal”, as this function is selecting the genotype at each
position that has maximum marginal probability.

For example, we could get inferred genotypes at the chr 2 QTL for the liver 
phenotype (at 28.6 cM) as follows:

```{r get_inferred_genotypes}
g <- maxmarg(probs = probs, 
             map   = cross$pmap, 
             chr   = chr, 
             pos   = max_pos, 
             return_char = TRUE)
```

We use `return_char = TRUE` to have `maxmarg()` return a vector of character 
strings with the genotype labels.

We then plot the insulin phenotype against these genotypes as follows:

```{r plot_pheno_geno_se}
plot_pxg(geno   = g, 
         pheno  = cross$pheno[,"log10_insulin_10wk"], 
         SEmult = 2,
         main   = "Insulin vs Chr 2 Genotype")
```


::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 1

Calculate the insulin BLUP effects for chromosome 7.  
1) Create an object called `blup_chr7` to contain the effects.  
2) Plot the chromosome 7 BLUPs and add the LOD plot at bottom.
3) Which founder allele increases insulin levels?

:::::::::::::::::::::::: solution 

```{r challenge1}
chr <- '7'
blup_chr7 <- scan1blup(genoprobs = probs[,chr], 
                       pheno     = cross$pheno[,"log10_insulin_10wk"],
                       addcovar  = addcovar,
                       kinship   = kinship_loco[[chr]])  
plot_coef(x       = blup_chr7, 
          map     = cross$pmap,
          columns = 1:3,
          scan1_output = lod_add_loco,
          legend  = "topleft",
          main    = "Insulin")
```

The C57BL/6J allele on chromosome 7 increases insulin levels.

:::::::::::::::::::::::::::::::::

## Challenge 2

Calculate the insulin BLUP effects for chromosome 19.  
1.  Create an object called `blup_chr19` to contain the effects.  
2. Plot the chromosome 19 BLUPs and add the LOD plot at bottom.
3. Which founder allele increases insulin levels?
4. Plot insulin versus the genotype at the marker with the higest LOD on 
chromosome 19.

:::::::::::::::::::::::: solution 

```{r challenge2}
chr <- '19'
blup_chr19 <- scan1blup(genoprobs = probs[,chr], 
                        pheno     = cross$pheno[,"log10_insulin_10wk"],
                        addcovar  = addcovar,
                        kinship   = kinship_loco[[chr]])  
plot_coef(x       = blup_chr19, 
          map     = cross$pmap,
          columns = 1:3,
          scan1_output = lod_add_loco,
          legend  = "topleft",
          main    = "Insulin")
```

The BTBR allele on chromosome 19 increases insulin levels.

```{r challenge2_b}
max_pos19 = peaks$pos[peaks$chr == chr]
g <- maxmarg(probs = probs, 
             map   = cross$pmap, 
             chr   = chr, 
             pos   = max_pos19, 
             return_char = TRUE)

plot_pxg(geno   = g, 
         pheno  = cross$pheno[,"log10_insulin_10wk"], 
         SEmult = 2, 
         main   = "Insulin vs Chr 19 Genotype")
```



:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: keypoints 

- "Estimated founder allele effects can be plotted from the mapping model coefficients."
- "Additive and dominance effects can be plotted using contrasts." 

::::::::::::::::::::::::::::::::::::::::::::::::
