---
title: "Finding QTL peaks"
teaching: 30
exercises: 30
---

:::::::::::::::::::::::::::::::::::::: questions 

- "How do I locate QTL peaks above a certain LOD threshold value?"

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Locate QTL peaks above a LOD threshold value throughout the genome.
- Identify the Bayesian credible interval for a QTL peak.

::::::::::::::::::::::::::::::::::::::::::::::::


:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: instructor

We need the following block for the site to build on Github. The students do
not need to see or run the next block.

:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

```{r github_load_data,echo=FALSE,warning=FALSE,message=FALSE}
suppressPackageStartupMessages(library(qtl2))
cross <- read_cross2(file = 'https://thejacksonlaboratory.box.com/shared/static/svw7ivp5hhmd7vb8fy26tc53h7r85wez.zip')
probs <- calc_genoprob(cross = cross, map = cross$gmap, error_prob = 0.002)
kinship_loco <- calc_kinship(probs = probs, type = "loco")
# Covariates
cross$covar$Sex <- factor(cross$covar$Sex)
addcovar <- model.matrix(~Sex, data = cross$covar)[,-1, drop = FALSE]
# LOD scan for insulin.
insulin <- cross$pheno[,"log10_insulin_10wk", drop = FALSE]
lod_add_loco <- scan1(genoprobs = probs, 
                      pheno = insulin, 
                      kinship = kinship_loco, addcovar = addcovar)
perm_add_loco = readRDS(url("https://thejacksonlaboratory.box.com/shared/static/fatud2gsjogbfg4xntyix05j6dg3pdnj.rds"))
```

Once we have LOD scores from a genome scan and a significance threshold, we can 
look for QTL peaks associated with the phenotype. High LOD scores indicate the 
neighborhood of a QTL but don't give its precise position. To find the exact 
position of a QTL, we define an interval that is likely to hold the QTL.

We'll use the Bayesian credible interval, which is a method for defining QTL 
intervals. It describes the probability that the interval contains the true 
value. Credible intervals make a probabilistic statement about the true value, 
for example, a 95% credible interval states that there is a 95% chance that the 
true value lies within the interval.

Let's remind ourselves how the genome scan for insulin looks.

```{r plot_scan1,fig.width=8,fig.height=6}
thr <- summary(perm_add_loco)

plot_scan1(x    = lod_add_loco,
           map  = cross$pmap,
           main = "Insulin")
add_threshold(map = cross$pmap, thresholdA = thr, col = 'red')
```

To find peaks above a given threshold LOD value, use the function 
`find_peaks()`. It can also provide Bayesian credible intervals by using the 
argument `prob` (the nominal coverage for the Bayes credible intervals). Set the 
argument `expand2markers = FALSE` to keep from expanding the interval out to 
genotyped markers, or exclude this argument if you'd like to include flanking 
markers.

You need to provide both the `scan1()` output, the marker map and a 
threshold. We will use the 95% threshold from the permutations in the previous 
lesson.

```{r find_peaks}
find_peaks(scan1_output = lod_add_loco, 
           map          = cross$pmap, 
           threshold    = thr, 
           prob         = 0.95, 
           expand2markers = FALSE)
```

In the table above, we have one peak per chromosome because that is the default
behavior of `find_peaks()`. The `find_peaks()` function can also pick out 
multiple peaks on a chromosome: each peak must exceed the chosen threshold, and 
the argument `peakdrop` indicates the amount that the LOD curve must drop 
between the lowest of two adjacent peaks.  Use this feature with caution.

```{r find_multiple_peaks}
find_peaks(scan1_output = lod_add_loco, 
           map          = cross$pmap, 
           threshold    = thr, 
           peakdrop     = 1.8, 
           prob         = 0.95, 
           expand2markers = FALSE)
```

Each row shows a different peak; the columns show the peak location, LOD score 
and the lower and upper interval endpoints. Note that we now have two peaks
on chromosome 16, one at 17.5 Mb and one at 80.4 Mb.

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 1

Find peaks in the genome scan object called `lod_add_loco` that meet a threshold
of 3 and are in the interval described by a 2 point LOD drop from the peak. How
many peaks meet the LOD threshold of 3 and lie within the interval defined by a 
2 point LOD drop from the maximum peaks on each chromosome?

:::::::::::::::::::::::: solution 

```{r challenge1}
find_peaks(scan1_output = lod_add_loco, 
           map          = cross$pmap, 
           threshold    = 3, 
           drop         = 2)
```

This produces 8 peaks on 8 different chromosomes that meet a LOD threshold of 3 
and are within the interval defined by a 2-LOD drop from the maximum peak on 
each chromosome.

:::::::::::::::::::::::::::::::::

## Challenge 2

1). Calculate the 90% Bayes credible interval.
For chromosome 2, what is the range of the interval that has a 90% 
chance of containing the true QTL position?  
2). Compare with the 95% Bayes credible interval calculated earlier. How does 
the interval change as you increase the probability? Why?

:::::::::::::::::::::::: solution 

1). This produces a range from 118.0 to 149.6 Mb. 

```{r challenge2a}
pks = find_peaks(scan1_output = lod_add_loco, 
                 map          = cross$pmap,
                 prob         = 0.90, 
                 expand2markers = FALSE)
subset(pks, chr == '2')
```

2). This produces a range from 64.9 to 149.6 Mb, which is much broader than 
the 90% interval. The interval widens because the probability that the interval
contains the true QTL position has increased. 

```{r challenge2b}
pks = find_peaks(scan1_output = lod_add_loco, 
                 map          = cross$pmap,
                 prob         = 0.95, 
                 expand2markers = FALSE)
subset(pks, chr == '2')
```

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: keypoints 

- LOD peaks and support intervals can be identified with find_peaks().
- The Bayesian Credible Interval estimates the width of the support interval
around a QTL peak.
- Using a higher `prob` value for the Bayesian Credible Interval results in a
wider support interval.

::::::::::::::::::::::::::::::::::::::::::::::::

