---
title: "Integrating Gene Expression Data"
teaching: 60
exercises: 20
---

:::::::::::::::::::::::::::::::::::::: questions 

- How can I use gene expression data to identify candidate genes?
- What is expression QTL mapping?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Find genes which are correlated with a physiological phenotype.
- Find genes which have expression QTL in the same position as a physiological 
phenotype.

::::::::::::::::::::::::::::::::::::::::::::::::

```{r github_load_data,echo=FALSE,warning=FALSE,message=FALSE}
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(ggbeeswarm))
suppressPackageStartupMessages(library(qtl2))
cross <- read_cross2(file = 'https://thejacksonlaboratory.box.com/shared/static/svw7ivp5hhmd7vb8fy26tc53h7r85wez.zip')
probs <- calc_genoprob(cross = cross, map = cross$gmap, error_prob = 0.002)
kinship_loco <- calc_kinship(probs = probs, type = "loco")
# Covariates
cross$covar$Sex <- factor(cross$covar$Sex)
addcovar <- model.matrix(~Sex, data = cross$covar)[,-1, drop = FALSE]
# LOD scan for insulin.
lod_add_loco <- scan1(genoprobs = probs, 
                      pheno = cross$pheno[,"log10_insulin_10wk"], 
                      kinship = kinship_loco, addcovar = addcovar)
perm_add_loco <- readRDS(url("https://thejacksonlaboratory.box.com/shared/static/fatud2gsjogbfg4xntyix05j6dg3pdnj.rds"))
thr           <- summary(perm_add_loco, alpha = 0.05)
peaks         <- find_peaks(lod_add_loco, cross$pmap, threshold = thr, prob = 0.95)
```

## Introduction

Once you have QTL peak, the next step if often to identify genes which may be 
causal candidates. However, there are often hundreds of genes within the QTL
support interval. How do you select candidate genes?

Often, there is a tissue which is related to the phenotype that you are
measuring. For example, if you measure circulating cardiac troponin, then gene
expression in the heart may be relevant. For some complex phenotypes, it may
not be easy to select a tissue. For example, if the liver creates a metabolite
that causes kidney injury, which is then exacerbated by the immune system, 
in which tissue would you measure gene expression?

Susceptibility to type 2 diabetes involves a complex interaction between
several tissues. However, for the purposes of this tutorial in which we mapped
circulating insulin levels, it is reasonable to look at gene expression in the 
pancreas.

## Reading in Gene Expression Data

Gene expression data consists of two parts: expression measurements and gene
annotation. In this study, gene expression was measured via two-color 
microarray. The expression values represent a ratio of two sets of RNA: a 
control set and the sample set. The expression values represent the 
log-normalized ratio of these two sets.

We have prepared two files containing normalized gene expression data and the 
gene annotation.

```{r test_chunk}
Sys.getenv("VROOM_CONNECTION_SIZE")
```

```{r load_data_box,include=FALSE}
Sys.setenv("VROOM_CONNECTION_SIZE" = 1e6)
# We need this because the expression files are too big for Github.
annot <- read_csv("https://thejacksonlaboratory.box.com/shared/static/2hvsoa1bs64ic2zad8isy9qrurzr4svi.csv",
                 show_col_types = FALSE) |>
           mutate(a_gene_id = as.character(a_gene_id))
expr  <- read_csv("https://thejacksonlaboratory.box.com/shared/static/dytobpcmlupvoth4a510zoxbz3wd1v98.csv",
                  show_col_types = FALSE)
```


```{r load_data_workshop,eval=FALSE}
annot <- read_csv("data/attie_b6btbr_grcm39/attie_gene_annot.csv",
                  show_col_types = FALSE) |>
           mutate(a_gene_id = as.character(a_gene_id))

expr  <- read_csv("data/attie_b6btbr_grcm39/attie_gene_expr.csv",
                  show_col_types = FALSE)
```

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 1: Size of data structures.

1. How many rows and columns are in `annot`?
2. What are the column names off `annot`?

:::::::::::::::::::::::: solution 

Dimensions of `annot`.

```{r challenge1_a}
dim(annot)
```

Column names of `annot`.

```{r challenge1_b}
colnames(annot)
```

:::::::::::::::::::::::::::::::::

## Challenge 2

1. What are the dimensions of `expr`?
2. Look at the top 5 row by 5 column block of `expr`.

:::::::::::::::::::::::: solution

1. Dimensions of `expr`.

```{r challenge2_a}
dim(expr)
```

2. Top-left block of `expr`.

```{r topleft_expr}
expr[1:5,1:5]
```

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

Let's look at the relationship between the annotation and the expression data.

The annotation data has `r nrow(annot)` rows and the expression data has
`r ncol(expr)` columns. The first column in `expr` contains the mouse ID and
the remaining columns contain the expression values for each gene. The gene
IDs are in the column names. These are Agilent gene IDs. They are also in the
`a_gene_id` column in the annotation.

```{r verify_gene_order}
all(annot$a_gene_id == colnames(expr)[-1])
```

Now we know that the genes are aligned between the annotation and the 
expression data. When you receive your own expression data, it is critical 
that you align the genes in your expression data with your annotation data.

We must also align the mouse IDs between the physiological phenotypes and the
expression data. We saw above that the mouse IDs are in the first column of
the expression data. Let's see where the mouse IDs are in the phenotype data.

```{r head_pheno}
head(cross$pheno)
```

`cross$pheno` is a matrix which contains the mouse IDs in the rownames. Let's convert
the expression data to a matrix as well.

```{r expr2matrix}
expr <- expr |>
         column_to_rownames(var = "MouseNum") |>
         as.matrix()
```

Now let's check whether the mouse IDs are aligned between `cross$pheno` and 
`expr`. Again, when you assemble your phenotype and expression data, you will
need to make sure that the sample IDs are aligned between the data sets.

```{r verify_mouse_ids}
all(rownames(cross$pheno) == rownames(expr))
```

## Identifying Genes in a QTL Support Interval

In previous episodes, we found significant QTL peaks using the `find_peaks`
function. Let's look at those peaks again.

```{r view_peaks}
peaks
```

We looked at the QTL peak on chromosome 7 in a previous lesson. The QTL interval
is `r peaks$ci_hi[peaks$chr == '7'] - peaks$ci_lo[peaks$chr == '7']` Mb wide.
This is quite wide. Let's get the genes expressed in the pancreas within this
region. 

```{r}
chr <- '7'
peaks_chr7 <- filter(peaks, chr == '7')
annot_chr7 <- filter(annot, chr == '7' & start > peaks_chr7$ci_lo & end < peaks_chr7$ci_hi)
expr_chr7  <- expr[,annot_chr7$a_gene_id]
```

There are `r nrow(annot_chr7)` genes! How can we start to narrow down which 
ones may be candidate genes?

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 1: How can we narrow down the candidate gene list?

1. Take a moment to think of ways that you could narrow down the gene list to
select the most promising candidate genes that regulate insulin levels.
2. Turn to your neighbor and exchange your ideas.
3. Share your ideas with the group.

:::::::::::::::::::::::: solution 

<!-- DMG: Correlation, eQTL, annotation? -->

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

## Using Expression QTL Mapping

One method to search for candidate genes is to look for genes with expression 
that is correlated with the phenotype. Genes which are correlated with the 
insulin levels may be correlated because they control insulin levels, respond
to insulin levels, or are correlated by chance. Genes which are strongly
correlated with insulin levels should have QTL peaks in similar positions.

```{r}
cor_chr7 = cor(cross$pheno[,'log10_insulin_10wk'], expr_chr7, use = 'pairwise')
```

```{r}
hist(cor_chr7, breaks = 20)
```

```{r get_chr7_genes}
ids_chr7 <- colnames(cor_chr7)[abs(cor_chr7) > 0.5]
annot |>
  filter(a_gene_id %in% ids_chr7)
```

Mediation.

```{r mediation}
# Get the probs at the maximum insulin QTL on Chr 7.
pr_chr7 = pull_genoprobpos(genoprobs = probs,
                           map       = cross$pmap,
                           chr       = chr,
                           pos       = peaks_chr7$pos)

# Create data sructure for results.
lod_drop = data.frame(a_gene_id = colnames(expr_chr7),
                      lod       = 0)

for(i in 1:ncol(expr_chr7)) {

  # Make new covariates.
  curr_covar = cbind(addcovar, expr_chr7[,i])
  
  mod = fit1(genoprobs = pr_chr7,
             pheno     = cross$pheno[,'log10_insulin_10wk',drop = FALSE],
             kinship   = kinship_loco[[chr]],
             addcovar  = curr_covar)
  
  lod_drop$lod[i] = mod$lod
  
} # for(i)
```


```{r plot_mediation}
lod_drop <- left_join(lod_drop, annot_chr7, by = 'a_gene_id')
plot(lod_drop$start, lod_drop$lod)
abline(v = peaks_chr7$pos, col = 2)
```

```{r}
plot(cor_chr7, lod_drop$lod)
```



If you have 


:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: instructor

This takes about 2 minutes to run.

::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::


```{r eqtl_chr7}
eqtl_chr7 <- scan1(genoprobs = probs[,chr],
                   pheno     = expr_chr7,
                   kinship   = kinship_loco[[chr]],
                   addcovar  = addcovar)
```

```{r}
eqtl_chr7 <- apply(eqtl_chr7, 2, max, simplify = TRUE)
eqtl_chr7 <- eqtl_chr7[eqtl_chr7 > thr[1]]
```

```{r}
length(eqtl_chr7)
```

::::::::::::::::::::::::::::::::::::: keypoints 

- Use `.md` files for episodes when you want static content
- Use `.Rmd` files for episodes when you need to generate output
- Run `sandpaper::check_lesson()` to identify any issues with your lesson
- Run `sandpaper::build_lesson()` to preview your lesson locally

::::::::::::::::::::::::::::::::::::::::::::::::

[r-markdown]: https://rmarkdown.rstudio.com/
