---
title: "Integrating Gene Expression Data"
teaching: 60
exercises: 20
---

:::::::::::::::::::::::::::::::::::::: questions 

- How can I use gene expression data to identify candidate genes?
- What is expression QTL mapping?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Find genes which are correlated with a physiological phenotype.
- Find genes which have expression QTL in the same position as a physiological 
phenotype.

::::::::::::::::::::::::::::::::::::::::::::::::

```{r github_load_data,echo=FALSE,warning=FALSE,message=FALSE}
options(scipen = 1, digits = 3)
suppressPackageStartupMessages(library(VariantAnnotation))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(ggbeeswarm))
suppressPackageStartupMessages(library(qtl2))
cross <- read_cross2(file = 'https://thejacksonlaboratory.box.com/shared/static/svw7ivp5hhmd7vb8fy26tc53h7r85wez.zip')
probs <- calc_genoprob(cross = cross, map = cross$gmap, error_prob = 0.002)
kinship_loco <- calc_kinship(probs = probs, type = "loco")
# Covariates
cross$covar$Sex <- factor(cross$covar$Sex)
addcovar <- model.matrix(~Sex, data = cross$covar)[,-1, drop = FALSE]
# LOD scan for insulin.
insulin <- cross$pheno[,'log10_insulin_10wk',drop = FALSE]
lod_add_loco <- scan1(genoprobs = probs, pheno = insulin, 
                      kinship   = kinship_loco, addcovar = addcovar)
perm_add_loco <- readRDS(url("https://thejacksonlaboratory.box.com/shared/static/fatud2gsjogbfg4xntyix05j6dg3pdnj.rds"))
thr           <- summary(perm_add_loco, alpha = 0.05)
peaks         <- find_peaks(lod_add_loco, cross$pmap, threshold = thr, prob = 0.95)
```

## Introduction

Once you have QTL peak, the next step if often to identify genes which may be 
causal candidates. However, there are often hundreds of genes within the QTL
support interval. How do you select candidate genes?

Often, there is a tissue which is related to the phenotype that you are
measuring. For example, if you measure circulating cardiac troponin, then gene
expression in the heart may be relevant. For some complex phenotypes, it may
not be easy to select a tissue. For example, if the liver creates a metabolite
that causes kidney injury, which is then exacerbated by the immune system, 
in which tissue would you measure gene expression?

Susceptibility to type 2 diabetes involves a complex interaction between
several tissues. However, for the purposes of this tutorial in which we mapped
circulating insulin levels, it is reasonable to look at gene expression in the 
pancreas.

## Reading in Gene Expression Data

Gene expression data consists of two parts: expression measurements and gene
annotation. In this study, gene expression was measured via two-color 
microarray. The expression values represent a ratio of two sets of RNA: a 
control set and the sample set. The expression values represent the 
log-normalized ratio of these two sets.

We have prepared two files containing normalized gene expression data and the 
gene annotation.

```{r load_data_box,include=FALSE}
Sys.setenv("VROOM_CONNECTION_SIZE" = 1e6)
# We need this because the expression files are too big for Github.
annot <- read_csv("https://thejacksonlaboratory.box.com/shared/static/2hvsoa1bs64ic2zad8isy9qrurzr4svi.csv",
                 show_col_types = FALSE) |>
           mutate(a_gene_id = as.character(a_gene_id))
expr  <- read_csv("https://thejacksonlaboratory.box.com/shared/static/dytobpcmlupvoth4a510zoxbz3wd1v98.csv",
                  show_col_types = FALSE)
```


```{r load_data_workshop,eval=FALSE}
annot <- read_csv("data/attie_b6btbr_grcm39/attie_gene_annot.csv",
                  show_col_types = FALSE) |>
           mutate(a_gene_id = as.character(a_gene_id))

expr  <- read_csv("data/attie_b6btbr_grcm39/attie_gene_expr.csv",
                  show_col_types = FALSE)
```

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 1: Size of data structures.

1. How many rows and columns are in `annot`?
2. What are the column names off `annot`?

:::::::::::::::::::::::: solution 

Dimensions of `annot`.

```{r challenge1_a}
dim(annot)
```

Column names of `annot`.

```{r challenge1_b}
colnames(annot)
```

:::::::::::::::::::::::::::::::::

## Challenge 2

1. What are the dimensions of `expr`?
2. Look at the top 5 row by 5 column block of `expr`.

:::::::::::::::::::::::: solution

1. Dimensions of `expr`.

```{r challenge2_a}
dim(expr)
```

2. Top-left block of `expr`.

```{r topleft_expr}
expr[1:5,1:5]
```

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

Let's look at the relationship between the annotation and the expression data.

The annotation data has `r nrow(annot)` rows and the expression data has
`r ncol(expr)` columns. The first column in `expr` contains the mouse ID and
the remaining columns contain the expression values for each gene. The gene
IDs are in the column names. These are Agilent gene IDs. They are also in the
`a_gene_id` column in the annotation.

```{r verify_gene_order}
all(annot$a_gene_id == colnames(expr)[-1])
```

Now we know that the genes are aligned between the annotation and the 
expression data. When you receive your own expression data, it is critical 
that you align the genes in your expression data with your annotation data.

We must also align the mouse IDs between the physiological phenotypes and the
expression data. We saw above that the mouse IDs are in the first column of
the expression data. Let's see where the mouse IDs are in the phenotype data.

```{r head_pheno}
head(cross$pheno)
```

`cross$pheno` is a matrix which contains the mouse IDs in the rownames. Let's convert
the expression data to a matrix as well.

```{r expr2matrix}
expr <- expr |>
         column_to_rownames(var = "MouseNum") |>
         as.matrix()
```

Now let's check whether the mouse IDs are aligned between `cross$pheno` and 
`expr`. Again, when you assemble your phenotype and expression data, you will
need to make sure that the sample IDs are aligned between the data sets.

```{r verify_mouse_ids}
all(rownames(cross$pheno) == rownames(expr))
```

## Identifying Genes in a QTL Support Interval

In previous episodes, we found significant QTL peaks using the `find_peaks`
function. Let's look at those peaks again.

```{r view_peaks}
peaks
```

We looked at the QTL peak on chromosome 19 in a previous lesson. The QTL interval
is `r peaks$ci_hi[peaks$chr == '19'] - peaks$ci_lo[peaks$chr == '19']` Mb wide.
This is quite wide. Let's get the genes expressed in the pancreas within this
region. 

```{r get_chr19_expr}
chr         <- '19'
peaks_chr19 <- filter(peaks, chr == '19')
annot_chr19 <- filter(annot, chr == '19' & start > peaks_chr19$ci_lo & end < peaks_chr19$ci_hi)
expr_chr19  <- expr[,annot_chr19$a_gene_id]
```

There are `r nrow(annot_chr19)` genes! How can we start to narrow down which 
ones may be candidate genes?

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 3: How can we narrow down the candidate gene list?

1. Take a moment to think of ways that you could narrow down the gene list to
select the most promising candidate genes that regulate insulin levels.
2. Turn to your neighbor and exchange your ideas.
3. Share your ideas with the group.

:::::::::::::::::::::::: solution 

<!-- DMG: Correlation, eQTL, annotation? -->

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

## Identifying Coding SNPs in QTL Intervals

When you have a QTL peak, you can search for SNPs which lie within the coding
regions of genes. These SNPs may cause amino acid substitutions which will
affect protein structure. The most comprehensive SNP resource is the
[Mouse Genomes Project](https://www.sanger.ac.uk/data/mouse-genomes-project/).
They have sequenced 52 inbred strains, including BTBR, and have made the data
available in 
[Variant Call Format](https://samtools.github.io/hts-specs/VCFv4.2.pdf) (VCF). 
The full VCF files are available on the 
[EBI FTP site](https://ftp.ebi.ac.uk/pub/databases/mousegenomes/REL-2112-v8-SNPs_Indels/).
However, the full SNP file is 22 GB! When you need other strains, you can 
download and query that file. However, for this workshop, we only need SNPs for
the BTBR strain. We have created a VCF file containing only SNPs which 
differ from C57BL/6J and BTBR. Let's read in this file now using the 
[VariantAnnotation](https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html) 
function [readVcf](https://rdrr.io/bioc/VariantAnnotation/man/readVcf-methods.html).


```{r readvcf_github,include=FALSE}
vcf <- readVcf(VcfFile(file  = "https://thejacksonlaboratory.box.com/shared/static/dp0srdjojpqoml91fftpbasiwcq22vqi.gz",
                       index = "https://thejacksonlaboratory.box.com/shared/static/gwt3o0gh6pgaeyxuuym58zpur0ktvkz5.tbi"))
```


```{r readvcf,eval=FALSE}
vcf <- readVcf(file.path("data", "btbr_snps_grcm39.vcf.gz"))
```

The `vcf` object contains both the SNP allele calls and information about the
sequencing depth and SNP consequences. Let's see how many SNPs are in the VCF.

```{r}
dim(vcf)
```

There are about 4.8 million SNPs in `vcf`. There are many fields in the VCF
file and there are 
[vignettes](https://thejacksonlaboratory.box.com/shared/static/gwt3o0gh6pgaeyxuuym58zpur0ktvkz5.tbi)
which document the more advanced features. Here, we will search for SNPs within
QTL support intervals and filter them to retain missense or stop coding SNPs.

In order to filter the SNPs by location, we need to create a 
[GenomicRanges](https://www.bioconductor.org/packages/release/bioc/html/GenomicRanges.html) 
object which contains the coordinates of the QTL support interval on chromosome 
19. Note that the support interval is reported by `qtl2` in Mb and the GRanges
object must be given bp positions, so we need to multiply these by 1e6.

```{r make_chr19_granges}
chr19_gr <- GRanges(seqnames = peaks_chr19$chr, 
                    ranges   = IRanges(peaks_chr19$ci_lo * 1e6, peaks_chr19$ci_hi * 1e6))
chr19_gr
```

Next, we will filter the `vcf` object to retain SNPs within the support interval.

```{r subset_snps_chr19}
vcf_chr19 <- subsetByOverlaps(vcf, ranges = chr19_gr)
```

Let's see how many SNPs there are.

```{r dim_vcf_19}
dim(vcf_chr19)
```

There are still 19,000 SNPs! But many of them may not be in coding regions or
may be synonymous. Next, we will search for SNPs that have missense or stop
coding changes. These are likely to have the most severe effects. Note that
synonymous SNPs and SNPs in the untranslated region of genes may affect RNA
folding, RNA stability, and translation. But it is more difficult to predict
those effects.

Let's plot the density of the SNPs to see where they fall. We would not expect
genes which lie in regions without variation between B6 and BTBR to be involved
in regulating insulin levels.

```{r snp_density,fig.width=8,fig.height=6}
# Get the SNP positions.
pos = start(rowRanges(vcf_chr19)) / 1e6

plot(density(pos), las = 1, lwd = 2, main = "Density of SNPs")
rug(pos)
```

From the plot above, we can see that the density of SNPS between B6 and BTBR is
not uniform. Most of the SNPs lie between 50 and 52 Mb, with less dense 
regions extending past 54 Mb.

The `vcf` object contains predicted SNP consequences in a field called `CSQ`.
The format of this field is challenging to parse, so we will get it, search
for missense and stop SNPs, and then do some data-wrangling to may the output
readable.

```{r grep_csq}
csq <- info(vcf_chr19)$CSQ
csq <- as.list(csq)

wh  <- grep('missense|stop', csq)
csq <- csq[wh]
rr  <- rowRanges(vcf_chr19)[wh]
```

Let's see how many SNPs had missense or stop codon consequences.

```{r len_csq}
length(csq)
```

There were only `r length(csq)` SNPs out of 19,000 SNPs with missense or stop 
codon consequences. Next, let's do the data wrangling to reformat the
consequences of these SNPs.

::::::::::::::::::::::::::::::::::::::::::::: instructor

This may be a good code chunk to have the students copy and paste.

::::::::::::::::::::::::::::::::::::::::::::::::::::::::

```{r}
csq <- lapply(csq, strsplit, split = "\\|")
csq <- lapply(csq, function(z) {
                     matrix(unlist(z), ncol = length(z[[1]]), byrow = TRUE)
                   })

for(i in seq_along(csq)) {
  
  csq[[i]] <- data.frame(snp_id = names(rr)[i],
                         csq[[i]])
  
} # for(i)

csq = do.call(rbind, csq)
```

Let's look at the top of `csq`.

```{r head_csq}
head(csq)
```

There are a lot of columns and we may not need all of them. Next, let's reduce
the number of columns to keep the ones that we need.

```{r csq_subset}
csq <- csq[,c(1:7,9,17,18,26)]
```

Next, we will subset the consequences to retain the unique rows and then retain
rows which contain "missense" and "stop".

```{r distinct_csq}
csq <- distinct(csq) |>
         filter(str_detect(X2, "missense|stop"))
```

Let's look at the top of `csq` now.

```{r head_csq2}
head(csq)
```

Column `X4` contains gene names. Let's get the unique gene names.

```{r}
csq |>
  distinct(X4)
```

There are only two genes in the QTL support interval which contain missense
SNPs. 

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 4: Search Genes using Pubmed

1. Go to Pubmed and search for papers involving the two genes and insulin or
diabetes.

:::::::::::::::::::::::: solution 

Both genes return published papers relating to insulin:

[Sorcs1](https://pubmed.ncbi.nlm.nih.gov/?term=Sorcs1+insulin)

[Rbm2](https://pubmed.ncbi.nlm.nih.gov/?term=rbm20+insulin)

[Sorcs1](https://pubmed.ncbi.nlm.nih.gov/16682971/) was identified as a gene 
which relates to obesity-induced type 2 diabetes by the same group that
provided this data set. In fact, it was identified in the same type of mouse
cross between C57BL/6J and BTBR.

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

In this case, we used published SNPs data to identify two potential candidate
genes. If we had not found publications that had already associated these 
genes with diabetes, these would be genes that you would follow up on in the
lab.

## Using Expression QTL Mapping

Another method of searching for candidate genes is to look for genes which have
QTL peaks in the same location as the insulin QTL. Genes which have QTL that are
co-located with insulin QTL will also be strongly correlated with insulin
levels. These genes may be correlated with insulin because they control insulin 
levels, respond to insulin levels, or are correlated by chance. However, genes
which have QTL within the insulin QTL support interval are reasonable candidate
genes since we expect the genotype to influence expression levels.

Let's perform QTL mapping on the genes within the chromosome 19 insulin QTL
support interval. We can do this by passing in the expression values as 
phenotypes into `scan1`.

```{r eqtl_chr19}
eqtl_chr19 <- scan1(genoprobs = probs[,chr],
                    pheno     = expr_chr19,
                    kinship   = kinship_loco[[chr]],
                    addcovar  = addcovar)
```

Let's look at the top of the results.

```{r head_eqtl_chr19}
head(eqtl_chr19[,1:6])
```

The eQTL results have one row for each marker and one column for each gene.
Each column represents the LOD plot for one gene. Let's plot the genome scan
for one gene.

```{r plot_1_eqtl,fig.width=8,fig.height=6}
gene_id <- '10002678668'
symbol  <- annot_chr19$gene_name[annot_chr19$a_gene_id == gene_id]

plot_scan1(x   = eqtl_chr19, 
           map = cross$pmap,
           lodcolumn = gene_id,
           main      = symbol)
```

This gene seems to have a QTL on chromosome 19 near 52 Mb with a LOD of 34. 
The insulin QTL is closer to 55 Mb, so this may be a good candidate gene. 

We can harvest the significant QTL peaks on chromosome 19 using `find_peaks`.
We will use the default threshold of LOD = 3.

```{r eqtl_chr19_filter_peaks}
eqtl_chr19_peaks <- find_peaks(eqtl_chr19, map = cross$pmap) |>
                      arrange(pos)
eqtl_chr19_peaks
```

We can see that there are `r nrow(eqtl_chr19_peaks)` significant QTL on 
chromosome 19. All of these genes are within the support interval of the insulin
QTL.

Let's filter the eQTL genes and add in gene annotation for them.

```{r eqtl_chr19_add_annot}
eqtl_chr19_peaks <- eqtl_chr19_peaks |>
                     filter(pos > peaks_chr19$ci_lo & pos < peaks_chr19$ci_hi) |>
                     left_join(annot, by = c('lodcolumn' = 'a_gene_id'))
eqtl_chr19_peaks
```

Let's plot the insulin QTL again to remind ourselves what the peak on chromosome
19 look like. We will also plot the genome scan for one of the genes on
chromosome 19.

```{r plot_insulin_qtl,fig.width=8,fig.height=6}
gene_id <- "10002936879"
symbol  <- annot_chr19$gene_name[annot_chr19$a_gene_id == gene_id]

plot_scan1(x    = eqtl_chr19,
           map  = cross$pmap,
           lodcolumn = gene_id,
           main = "Insulin")
plot_scan1(x    = lod_add_loco, 
           map  = cross$pmap,
           chr  = '19',
           col  = 'blue',
           add  = TRUE)
legend("topleft", legend = c("insluin", symbol), 
       col = c('blue', 'black'), lwd = 2)
```

Why would we prioritize genes with an eQTL in the same location as an insulin
QTL? If a gene has an eQTL in the same location, there are genetic variants 
which influence it's expression at that location. These variants may be 
distinct from the ones which influence the phenotype or they may influence
gene expression which then influences the phenotype. We call this causal
relationship of a SNP changing expression, which then changes the phenotype,
"mediation". 

## Mediation Analysis

If a gene is correlated with insulin levels and if it has an eQTL in the same
location as insulin, then we could add it into the insulin QTL model and see if
it reduces the LOD on chromosome 19.

```{r add_gene_to}
stopifnot(rownames(addcovar) == rownames(expr_chr19))
addcovar_chr19 <- cbind(addcovar, expr_chr19[,"10002936879"])

lod_med <- scan1(genoprobs = probs[,chr],
                 pheno     = insulin,
                 kinship   = kinship_loco[[chr]],
                 addcovar  = addcovar_chr19)
```

Next, we will plot the original insulin genome scan and overlay the genome scan
with gene `10002936879` in the model.

```{r med_plot_10002936879,fig.width=8,fig.height=6}
gene_id <- "10002936879"
symbol  <- annot_chr19$gene_name[annot_chr19$a_gene_id == gene_id]

plot_scan1(x    = lod_add_loco,
           map  = cross$pmap,
           chr  = '19',
           main = paste("Insulin with/without", symbol))
plot_scan1(x    = lod_med,
           map  = cross$pmap,
           chr  = '19',
           col  = "blue",
           add  = TRUE)
legend("topleft", legend = c("insluin", "mediation"), 
       col = c('black', 'blue'), lwd = 2)
```

The LOD dropped from `r peaks_chr19$lod` to 
`r find_peaks(lod_med, cross$pmap, threshold = 1.9)$lod`. This is a difference
of `r peaks_chr19$lod - find_peaks(lod_med, cross$pmap, threshold = 1.9)$lod`.

It would be slow to run a genome scan for each gene on chromosome 19 and look
at the LOD drop. Also, we don't really need the LOD drop across the entire 
chromosome. We only need the LOD drop at the location of the peak LOD. To do
this, we can use the `fit1` function to get the LOD at the marker with the
highest LOD in the insulin genome scan.


```{r mediation}
# Get the probs at the maximum insulin QTL on Chr 19.
pr_chr19 = pull_genoprobpos(genoprobs = probs,
                            map       = cross$pmap,
                            chr       = chr,
                            pos       = peaks_chr19$pos)

# Create data sructure for results.
lod_drop = data.frame(a_gene_id = colnames(expr_chr19),
                      lod       = 0)

for(i in 1:ncol(expr_chr19)) {

  # Make new covariates.
  curr_covar = cbind(addcovar, expr_chr19[,i])
  
  # Fit the insulin QTL model with the gene as a covariate.
  mod = fit1(genoprobs = pr_chr19,
             pheno     = insulin,
             kinship   = kinship_loco[[chr]],
             addcovar  = curr_covar)
  
  # Save the LOD.
  lod_drop$lod[i] = mod$lod
  
} # for(i)

# Subtract the insulin LOD from the mediation LODs.
lod_drop$lod_drop = lod_drop$lod - peaks_chr19$lod
```


```{r plot_mediation,fig.width=8,fig.height=6}
lod_drop <- left_join(lod_drop, annot_chr19, by = 'a_gene_id')

plot(lod_drop$start, lod_drop$lod_drop, col = NA, 
     main = "Mediation Analysis on Chr 19",
     xlab = "Posiiton (Mb)", ylab = "LOD Drop")
text(lod_drop$start, lod_drop$lod_drop, labels = lod_drop$gene_name)
abline(v = peaks_chr19$pos, col = 2)
```

In the plot above, we plotted the position of each gene versus the decrease in
the LOD score (i.e. LOD drop). Genes with the lowest LOD drop are the best 
candidate genes based on mediation analysis.

Let's look at the annotation for the genes with LOD drop less than -2.

```{r eqlt_chr19_lod_drop_annot}
lod_drop |>
  filter(lod_drop < -2) |>
  select(gene_id, symbol, chr, start, end, lod, lod_drop, description)
```

Do you see any good candidate genes which might regulate insulin?

## Summary

In this episode, we learned how to identify candidate genes under a QTL peak.
In a perfect world, there would be exactly one gene implicated by these 
analyses. In most cases, you will have a set of candidate genes and you will
need to study each gene and prioritize some for laboratory follow-up. 

There are two ways of searching for candidate genes: using SNPs in the QTL
interval, and looking for genes with eQTL which are co-located with the 
phenotype QTL. We learned how to query a VCF file and how to perform mediation
analysis. After this step, you will have a set of genes which you can test
for association with your phenotype. 

::::::::::::::::::::::::::::::::::::: keypoints 

- There will be many genes under a QTL peak.
- You can search for genes with SNPs that produce coding changes by querying
a VCF file.
- You can search for genes with expression changes that may influence your
phenotype by performing mediation analysis with expression data from the same
mice.

::::::::::::::::::::::::::::::::::::::::::::::::

[r-markdown]: https://rmarkdown.rstudio.com/
