---
title: "Performing a genome scan with a linear mixed model"
teaching: 20
exercises: 10
---

:::::::::::::::::::::::::::::::::::::: questions 

- "How do I use a linear mixed model in a genome scan?"
- "How do different mapping and kinship calculation methods differ?"

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Create a genome scan with a linear mixed model.
- Compare LOD plots for Haley-Knott regression and linear mixed model methods.
- Compare LOD plots for the standard kinship matrix with the 
leave-one-chromosome-out (LOCO) method.

::::::::::::::::::::::::::::::::::::::::::::::::


```{r load_dependencies, include=FALSE, warning=FALSE}
options(scipen = 1, digits = 3)
suppressPackageStartupMessages(library(qtl2))
cross <- read_cross2(file = 'https://thejacksonlaboratory.box.com/shared/static/svw7ivp5hhmd7vb8fy26tc53h7r85wez.zip')
probs <- calc_genoprob(cross = cross, map = cross$gmap, error_prob = 0.002)
kinship_loco  <- calc_kinship(probs = probs, type = 'loco')
# Covariates.
cross$covar$Sex <- factor(cross$covar$Sex)
addcovar <- model.matrix(~Sex, data = cross$covar)[,-1, drop = FALSE]
# LOD scan for insulin.
lod_add <- scan1(genoprobs = probs, pheno = cross$pheno[,'log10_insulin_10wk',drop = FALSE], 
                 addcovar  = addcovar)
# Chr 2 BLUPs.
chr2_eff = readRDS(url('https://thejacksonlaboratory.box.com/shared/static/kxsizeu1papwb0q21f1zjde8nx5lhxvg.rds'))
# Marker with maximum LOD on Chr 2.
chr2_max_mkr = 'rs13476803'
```

Genetic mapping in mice presents a good example of why accounting for population 
structure is important. Laboratory mouse strains are descended from a small 
number of founders (fancy mice) and went through several population bottlenecks. 
Wild-derived strains are not descended from fancy mice and don't share the same 
history as laboratory strains. Linear mixed models were developed to solve 
problems with population structure created by differing ancestries, and to 
handle relatedness between individuals.  Linear mixed models (LMMs) consider 
genome-wide similarity between all pairs of individuals to account for 
population structure, known kinship and unknown relatedness. Linear mixed models 
in mapping studies can successfully correct for genetic relatedness between 
individuals in a population by incorporating kinship into the model. Earlier we 
[calculated a kinship matrix](https://smcclatchy.github.io/mapping/04-calc-kinship/) 
for input to a linear mixed model to account for relationships among 
individuals. For a current review of mixed models in genetics, see this 
[preprint of Martin and Eskin, 2017](https://www.biorxiv.org/content/early/2017/01/28/092106).

Simple linear regression takes the form:

$y = \mu + \beta G + \epsilon$

which describes a line with slope $\beta$ and y-intercept $\mu$. The error 
(or residual) is represented by $\epsilon$.

To model the relationship between the phenotype $y$ and the genotypes at one
marker, we use:

$y_j = \mu + \beta_k G_{jk} + \epsilon_j$
 

where $y_{ij} is the phenotype of the $j$th individual, $\mu$; 
is the mean phenotype value, $\beta_k$ is the effect of the 
$k$th genotype, $G_{jk} is the genotype for individual $j$, and 
$\epsilon_j$ is the error for the $j$th individual. In the figure 
below, $\mu$; equals `r round(chr2_eff["rs13476803",4], 1)`, and $\beta$; equals 
`r round(abs(chr2_eff["rs13476803",1]), 1)` for the alternative hypothesis 
(QTL exists). This linear model is $y$ = `r round(chr2_eff["rs13476803",4], 1)` + 
`r round(abs(chr2_eff["rs13476803",1]), 1)`X + $\epsilon$. The model intersects the 
genotype groups at their group means. In contrast, the null hypothesis would 
state that there is no difference in group means (no QTL anywhere). The linear 
model for the null hypothesis would be $y$ = 94.6 + 0$X$ + $\epsilon$. 
This states that the phenotype is equal to the combined mean (94.6), plus 
some error ($\epsilon$). In other words, genotype doesn't affect the phenotype.

<!-- DMG: Make this figure again using ggbeeswarm and the BTBR data on Chr 2 
at marker rs13476803. -->

![](fig/nullvalt.png)

The linear models above describe the relationship between genotype and phenotype 
but are inadequate for describing the relationship between genotype and 
phenotype in large datasets. They don't account for relatedness among 
individuals. In real populations, the effect of a single genotype is influenced 
by many other genotypes that affect the phenotype. A  true genetic model takes 
into account the effect of all variants on the phenotype. 

To model the phenotypes of all individuals in the data, we can adapt the simple 
linear model to include all individuals and their variants so that we capture 
the effect of all variants shared by individuals on their phenotypes.

$y=\mu+\sum_{i=1}^M{\beta_iX_i}+\epsilon$

Now, $y$ represents the phenotypes of all individuals. The effect of the 
$i$th genotype on the phenotype is $\beta_i$, the mean is $\mu$ and the error 
is $\epsilon$. Here, 
the number of genotypes is M.

To model the effect of all genotypes and to account for relatedness, we test the 
effect of a single genotype while bringing all other genotypes into the model.

![](fig/all-geno-model.png)

<i>&beta;<sub>k</sub></i> is the effect of the genotype <i>X<sub>k</sub></i>, 
and &Sigma;<sub>i&#8800;k</sub><i>&beta;<sub>i</sub>X<sub>i</sub></i> sums the 
effects of all other genotypes except genotype k. For the leave one chromosome 
out (LOCO) method, <i>&beta;<sub>k</sub></i><i>X<sub>k</sub></i> is the effect 
of genotypes on chromosome <i>k</i>, and 
<i>&beta;<sub>i</sub></i><i>X<sub>i</sub></i> represents effect of genotypes on 
all other chromosomes.

If the sample contains divergent subpopulations, SNPs on different chromosomes 
will be correlated because of the  difference in allele frequencies between 
subpopulations caused by relatedness. To correct for correlations between 
chromosomes, we model all genotypes on the other chromosomes when testing for 
the association of a SNP. 

First, we will create a single kinship matrix using all of the genoprobs on all
chromosomes.

```{r calc_kinship_all}
kinship_all <- calc_kinship(probs = probs, 
                            type = "overall")
```


To perform a genome scan using a linear mixed model you also use the function 
`scan1`; you just need to provide the argument `kinship`, a kinship matrix 
(or, for the LOCO method, a list of kinship matrices).

```{r scan1_pg}
lod_add_pg <- scan1(genoprobs = probs, 
                    pheno     = cross$pheno[,"log10_insulin_10wk",drop = FALSE], 
                    kinship   = kinship_all, 
                    addcovar  = addcovar)
```

Again, on a multi-core machine, you can get some speed-up using the `cores` 
argument.

```{r scan1_pg_multicore, eval=FALSE}
lod_add_all <- scan1(genoprobs = probs, 
                     pheno     = cross$pheno[,"log10_insulin_10wk",drop = FALSE], 
                     kinship   = kinship_all, 
                     addcovar  = addcovar, 
                     cores     = 4)
```

If, for your linear mixed model genome scan, you wish to use the "leave one 
chromosome out" (LOCO) method (scan each chromosome using a kinship matrix that 
is calculated using data from all other chromosomes), use `type="loco"` in the 
call to `calc_kinship()`.

<!-- DMG: We need more explanation here. We need to show that using the all-
chromosome kinship matrix puts the marker being tested in the mapping model
twice, and this erases the peak. -->

```{r calc_kinship_loco, eval=FALSE}
kinship_loco <- calc_kinship(probs = probs, 
                             type  = "loco")
```

For the LOCO (leave one chromosome out) method, provide the list of kinship 
matrices as obtained from `calc_kinship()` with `method="loco"`.

```{r scan1_pg_loco, eval=FALSE}
lod_add_loco <- scan1(genoprobs = probs, 
                      pheno     = cross$pheno[,"log10_insulin_10wk",drop = FALSE], 
                      kinship   = kinship_loco, 
                      addcovar  = addcovar)
```

To plot the results, we again use `plot_scan1()`.

Here is a plot of the LOD scores by Haley-Knott regression and the linear mixed 
model using either the standard kinship matrix or the LOCO method.

```{r plot_lod_add_loco_all,fig.width=8}
plot_scan1(x     = lod_add,      
           map   = cross$pmap, 
           col   = 'black',
           ylim  = c(0, 7.5))
plot_scan1(x     = lod_add_all,  
           map   = cross$pmap, 
           col   = 'blue',   
           add   = TRUE)
plot_scan1(x     = lod_add_loco, 
           map   = cross$pmap, 
           col   = 'orange', 
           add   = TRUE)
legend(x = 1500, y = 7.5, legend = c("No kinship", "All kinship", "LOCO kinship"),
       lwd = 2, col = c('black', 'blue', 'orange'))
```


For the circulating insulin, the three methods give quite different 
results. The linear mixed model with an overall kinship matrix gives much lower 
LOD scores than the other two methods. On chromosomes with some evidence of a 
QTL, the LOCO method gives higher LOD scores than Haley-Knott, except on 
chromosome 6 where it gives lower LOD scores.

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 1

What are the benefits and disadvantages of the three 
methods for genome scanning (Haley-Knott regression,
kinship matrix, and leave-one-chromosome out (LOCO)?)  
Which method would you use to scan, and why?  
Think about the advantages and disadvantages of each,
discuss with a neighbor, and share your thoughts in the
collaborative document.

:::::::::::::::::::::::: solution 


:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 2

<!-- DMG: Ha! we were using the BTBR data anyway. Let's not load a second copy
of the data. Maybe we canb brainstorm about another challend. -->

Pair programming exercise: with your partner, review and carry 
out all of the steps in QTL mapping that we have covered so far,
using a new data set. One of you types the code, the other
explains what needs to happen next, finds the relevant code in
the lesson, suggests new names for objects (i.e. NOT the ones
you've already used, such as "map", "pr", "out", etc.).

1. Run the code above to load the 
[B6 x BTBR intercross data](https://github.com/rqtl/qtl2data/tree/master/B6BTBR) 
into an object called b6btbr.
2. Insert pseudomarkers and calculate genotype probabilities.
3. Run a genome scan for the log10_insulin_10wk phenotype.
4. Calculate a kinship matrix.
5. Calculate a list of kinship matrices with the LOCO method.
6. Run genome scans with the regular kinship matrix and with the
list of LOCO matrices.
7. Plot the 3 different genome scans in a single plot in 
different colors.
8. Which chromosomes appear to have peaks with a LOD score greater than 4? 
Which methods identify these peaks? Which don't?

:::::::::::::::::::::::: solution 

`file <- paste0("https://raw.githubusercontent.com/rqtl/", "qtl2data/master/B6BTBR/b6btbr.zip")`  
`b6btbr <- read_cross2(file)`  
`summary(b6btbr)`  
`head(b6btbr$pheno)`  
`colnames(b6btbr$pheno)`  
`b6bmap <- insert_pseudomarkers(map=b6btbr$gmap, step=1)`  
`prb6b <- calc_genoprob(cross=b6btbr, map=b6bmap, error_prob=0.002)`  
`b6baddcovar <- get_x_covar(b6btbr)`  
`b6bout <- scan1(genoprobs = prb6b, pheno = b6btbr$pheno, addcovar=b6baddcovar)`  
`plot(b6bout, map = b6bmap)`  
`b6bkinship <- calc_kinship(probs = prb6b)`  
`out_pg_b6b <- scan1(prb6b, b6btbr$pheno, kinship=b6bkinship, addcovar=b6baddcovar)`  
`kinship_loco_b6b <- calc_kinship(prb6b, "loco")`  
`out_pg_loco_b6b <- scan1(prb6b, b6btbr$pheno, kinship_loco_b6b, addcovar=b6baddcovar)`  
`plot_scan1(out_pg_loco_b6b, map = b6bmap, lodcolumn = "log10_insulin_10wk", col = "black")`  
`plot_scan1(out_pg_b6b, map = b6bmap, lodcolumn = "log10_insulin_10wk", col = "blue", add = TRUE)`  
`plot_scan1(b6bout, map = b6bmap, lodcolumn = "log10_insulin_10wk", col = "green", add = TRUE)`  


:::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::::::::::::::

<!-- DMG: Not sure what this is. -->

```{r}
file <- paste0("https://raw.githubusercontent.com/rqtl/",
               "qtl2data/master/B6BTBR/b6btbr.zip")
b6btbr <- read_cross2(file)
```

::::::::::::::::::::::::::::::::::::: keypoints 

- "To perform a genome scan with a linear mixed model, supply a kinship matrix."
- "Different mapping and kinship calculation methods give different results."

::::::::::::::::::::::::::::::::::::::::::::::::
