---
title: "Finding Significant Peaks via Permutation"
teaching: 10
exercises: 20
---

:::::::::::::::::::::::::::::::::::::: questions 

- How can I evaluate the statistical significance of genome scan results?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Run a permutation test to establish LOD score thresholds.

::::::::::::::::::::::::::::::::::::::::::::::::


:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: instructor

We need the following block for the site to build on Github. The students do
not need to see or run the next block.

:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

```{r github_load_data,echo=FALSE,warning=FALSE,message=FALSE}
suppressPackageStartupMessages(library(qtl2))
cross <- read_cross2(file = 'https://thejacksonlaboratory.box.com/shared/static/svw7ivp5hhmd7vb8fy26tc53h7r85wez.zip')
probs <- calc_genoprob(cross = cross, map = cross$gmap, error_prob = 0.002)
kinship_loco  <- calc_kinship(probs = probs, type = 'loco')
# Covariates
cross$covar$Sex <- factor(cross$covar$Sex)
addcovar <- model.matrix(~Sex, data = cross$covar)[,-1, drop = FALSE]
insulin <- cross$pheno[,'log10_insulin_10wk',drop = FALSE]
```

## Using Permutations to Determine Significance Thresholds

To establish the statistical significance of the results of a genome scan, a 
permutation test can identify the maximum LOD score that can occur by random 
chance. A permutation tests shuffles genotypes and phenotypes, essentially 
breaking the relationship between the two. The genome-wide maximum LOD score is 
then calculated on the permuted data, and this score used as a threshold of 
statistical significance. A genome-wide maximum LOD on shuffled, or permuted, 
data serves as the threshold because it represents the highest LOD score 
generated by random chance. 

The `scan1perm()` function takes the same arguments as `scan1()`, plus 
additional arguments to control the permutations:

- `n_perm` is the number of permutation replicates.
- `perm_Xsp` controls whether to perform autosome/X chromosome specific 
permutations (with `perm_Xsp=TRUE`) or not (the default is FALSE).
- `perm_strata` is a vector that defines the strata for a stratified permutation 
test.
- `chr_lengths` is a vector of chromosome lengths, used in the case that 
`perm_Xsp=TRUE`.

As with `scan1()`, you may provide a kinship matrix (or vector of kinship 
matrices, for the "leave one chromosome out" (loco) approach), in order to fit 
linear mixed models. If `kinship` is unspecified, the function performs ordinary 
Haley-Knott regression.

To perform a permutation test with the insulin phenotype, we run `scan1perm()`, 
provide it with the genotype probabilities, the phenotype data, X covariates and 
number of permutations. 

:::::::::::::::::::::::::::::::::::::::::::::::: instructor

Replace number of permutations (1000) with 100 for expediency.
1000 permutations takes about 2 minutes.

:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

```{r scan1perm}
perm_add <- scan1perm(genoprobs = probs, 
                      pheno     = insulin,
                      addcovar  = addcovar,
                      Xcovar    = addcovar,
                      n_perm    = 1000) 
```

Note the need to specify special covariates for the X chromosome (via `Xcovar`), 
to be included under the null hypothesis of no QTL. And note that when these are 
provided, the default is to perform a stratified permutation test, using strata 
defined by the rows in `Xcovar`. In general, when the X chromosome is 
considered, one will wish to stratify at least by sex.

Also note that, as with `scan1()`, you can speed up the calculations on a 
multi-core machine by specifying the argument `cores`. With `cores=0`, the 
number of available cores will be detected via `parallel::detectCores()`. 
Otherwise, specify the number of cores as a positive integer. For large 
data sets, be mindful of the amount of memory that will be needed; you may need 
to use fewer than the maximum number of cores, to avoid going beyond the 
available memory.

```{r scan1perm_multicore, eval=FALSE}
perm_add <- scan1perm(genoprobs = probs, 
                      pheno     = insulin, 
                      addcovar  = addcovar,
                      Xcovar    = Xcovar, 
                      n_perm    = 1000, 
                      cores     = 0)
```

`perm_add` now contains the maximum LOD score for each permutation for the 
phenotypes. There should be 1000 values for each phenotype. We can view the 
insulin permutation LOD scores by making a histogram.

```{r hist_perm, fig.width=8, fig.height=6}
hist(perm_add, 
     breaks = 50, 
     xlab   = "LOD", 
     las    = 1,
     main   = "Empirical distribution of maximum LOD scores under permutation")
abline(v = summary(perm_add), col = 'red', lwd = 2)
```

In the histogram above, you can see that most of the maximum LOD scores fall 
between 1 and 3.5. This means that we expect LOD scores less than 3.5 to occur by 
chance fairly often. The red line indicates the alpha = 0.05 threshold, which 
means that we only see LOD values by chance this high or higher, 5% of 
the time. This is [one way](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1206241/) 
of estimating a significance threshold for QTL plots.

To get estimated significance thresholds, use the function `summary()`.

```{r summary_scan1perm}
summary(perm_add)
```

The default is to return the 5% significance threshold. Thresholds for other 
(or for multiple) significance levels can be obtained via the `alpha` argument.

```{r summary_scan1perm_B}
summary(perm_add, 
        alpha = c(0.2, 0.05))
```

## Estimating an X Chromosome Specific Threshold

To obtain autosome/X chromosome-specific significance thresholds, specify 
`perm_Xsp=TRUE`. In this case, you need to provide chromosome lengths, which may 
be obtained with the function `chr_lengths()`.

```{r scan1perm_Xsp}
perm_add2 <- scan1perm(genoprobs   = probs, 
                       pheno       = cross$pheno[,"log10_insulin_10wk",drop = FALSE], 
                       addcovar    = addcovar, 
                       n_perm      = 1000,
                       perm_Xsp    = TRUE, 
                       chr_lengths = chr_lengths(cross$pmap))
```

Separate permutations are performed for the autosomes and X chromosome, and 
considerably more permutation replicates are needed for the X chromosome. The 
computations take about twice as much time.
See 
[Broman et al. (2006) Genetics 174:2151-2158](https://www.ncbi.nlm.nih.gov/pubmed/17028340).

The significance thresholds are again derived via `summary()`:

```{r summary_scan1perm_C}
summary(perm_add2, 
        alpha = c(0.2, 0.05))
```

## Estimating Significance Thresholds with the Kinship Matrix

You may have noticed above that we did not include the kinship matrix as an
argument to `scan1perm`. We can include the LOCO kinship matrices in our
permutations, since this is how we mapped insulin previously.

```{r scan1perm_loco}
perm_add_loco <- scan1perm(genoprobs = probs, 
                           pheno     = insulin,
                           kinship   = kinship_loco,
                           addcovar  = addcovar,
                           n_perm    = 1000) 
```

How does this affect the significance threshold estimates?

```{r}
summary(perm_add_loco, 
        alpha = c(0.2, 0.05))
```

There is not a large difference in the thresholds. Currently, we are on the
fence about using the kinship matrices to estimate significance thresholds. In
principle, the kinship matrices should be used because we used them when mapping
the phenotype. However, in practice, we often find that the significance 
thresholds are similar to those obtained without including the kinship matrices.
Given that it also takes more time to run the permutations with the kinship
matrices, it seems reasonable to exclude them.

We ran 1000 permutations of the insulin phenotype and estimated the 0.05 
significance threshold with and without the kinship matrices. We repeated this
process 100 times and plotted the thresholds below.

![Significance Thresholds with/without Kinship](fig/perm_kinship.png){alt="Comparison of significance thresholds"}

The plot shows that the median significance threshold is the same. The lines
connecting the points denote matched simulations in which the permutation
order was the same. While the exact value of the LOD threshold is different, 
the median value and the variance are similar.

## Estimating Binary Model Significance Thresholds

As with `scan1`, we can use `scan1perm` with binary traits, using the argument 
`model="binary"`. Again, this can't be used with a kinship matrix, but all of 
the other arguments can be applied.

```{r scan1perm_binary}
perm_bin <- scan1perm(genoprobs = probs, 
                      pheno     = cross$pheno[,"agouti_tan",drop = FALSE], 
                      addcovar  = addcovar, 
                      n_perm    = 1000, 
                      perm_Xsp  = TRUE, 
                      chr_lengths = chr_lengths(cross$pmap),
                      model     = "binary")
```

Here are the estimated 5% and 20% significance thresholds.

```{r summary_scan1perm_binary}
summary(perm_bin, 
        alpha = c(0.2, 0.05))
```

## Selecting the Number of Permutations

How do we know how many permutations to perform in order to obtain a good 
estimate of the significance threshold? Could we get a good estimate with 10
permutations? 100? 1000? 

When we run more permutations, we decrease the variance of the threshold
estimate.

![Significance Threshold Variance](fig/permutation_simulations.png){alt="Figure showing decreasing variance of significance threshold estimates with increasing permutations"}

In the figure above, we performed 10, 100, or 1000 permutations 1000 times and
recorded the 0.05 significance threshold each time. We plotted the significance
threshold versus the number of permutations and overlaid violin plots showing
the median value. Note that the variance of the significance threshold estimate
is higher at lower numbers of permutations. With 1000 permutations, the
variance decreases. The table below shows the number of permutations and the 
mean and standard deviation of the significance threshold. With 1000 permutations,
the estimate is 3.86 and the standard deviation is 0.064, which is an 
acceptable value.


Num. Perm. | Mean | Std. Dev.
-----------+------+----------
     10    | 3.63 | 0.492 
    100    | 3.81 | 0.195 
   1000    | 3.86 | 0.064

<!-- DMG: We need to review these challenges. -->

The code below shuffles the phenotypes so that they no longer match up with the 
genotypes. The purpose of this is to find out how high the LOD score can be due 
to random chance alone.

```{r permute_data, eval=FALSE}
shuffled_order <- sample(rownames(cross$pheno))
pheno_permuted <- cross$pheno
rownames(pheno_permuted) <- shuffled_order
xcovar_permuted <- addcovar
rownames(xcovar_permuted) <- shuffled_order
out_permuted <- scan1(genoprobs = probs, 
                          pheno = pheno_permuted, 
                         Xcovar = xcovar_permuted)
plot(out_permuted, map = cross$pmap)
head(shuffled_order)
```

::::::::::::::::::::::::::::::::::::: challenge 

## Challenge 1:

Run the preceding code to shuffle the phenotype data and plot a genome scan with 
this shuffled (permuted) data.  

What is the maximum LOD score in the scan from this permuted data?  
How does it compare to the maximum LOD scores obtained from the earlier scan?  
How does it compare to the 5% and 20% LOD thresholds obtained earlier?  
Paste the maximum LOD score in the scan from your permuted data into the 
etherpad.

:::::::::::::::::::::::: solution 


:::::::::::::::::::::::::::::::::


## Challenge 2

1) Find the 1% and 10% significance thresholds for the first set of 
permutations contained in the object `perm_add_loco`.  
2) What do the 1% and 10% significance thresholds say about LOD scores?
 
:::::::::::::::::::::::: solution 

1. Use the `alpha` argument to supply the desired significance thresholds.

```{r}
summary(perm_add_loco, alpha = c(0.01, 0.10))
```


2. These LOD thresholds indicate maximum LOD scores that can be obtained by 
random chance at the 1% and 10% significance levels. We expect to see LOD values 
this high or higher 1% and 10% of the time respectively.

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::


::::::::::::::::::::::::::::::::::::: keypoints 

- A permutation test establishes the statistical significance of a genome 
scan.
- 1,000 permutations provides a good estimate of the significance threshold. 

::::::::::::::::::::::::::::::::::::::::::::::::
