---
title: "Performing a genome scan with binary traits"
teaching: 20
exercises: 30
---

:::::::::::::::::::::::::::::::::::::: questions 

- "How do I create a genome scan for binary traits?"

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Convert phenotypes to binary values.
- Use logistic regression for genome scans with binary traits.
- Plot and compare genome scans for binary traits.

::::::::::::::::::::::::::::::::::::::::::::::::

```{r load_dependencies, include=FALSE, warning=FALSE}
options(scipen = 1, digits = 3)
suppressPackageStartupMessages(library(qtl2))
cross <- read_cross2(file = 'https://thejacksonlaboratory.box.com/shared/static/svw7ivp5hhmd7vb8fy26tc53h7r85wez.zip')
probs <- calc_genoprob(cross = cross, map = cross$gmap, error_prob = 0.002)
# Covariates.
cross$covar$Sex <- factor(cross$covar$Sex)
addcovar <- model.matrix(~Sex, data = cross$covar)[,-1, drop = FALSE]
```

The genome scans above were performed assuming that the residual variation 
followed a normal distribution. This will often provide reasonable results even 
if the residuals are not normal, but an important special case is that of a 
binary trait, with values 0 and 1, which is best treated differently. The 
`scan1` function can perform a genome scan with binary traits by logistic 
regression, using the argument `model="binary"`. (The default value for the 
`model` argument is `"normal"`.) At present, we _can not_ account for 
relationships among individuals in this analysis.

Let's look at the phenotypes in the cross again.

```{r threshold_phenotypes}
head(cross$pheno)
```

There are two binary traits called "agouti_tan", and "tufted" which are related
to coat color and shape.

We perform a binary genome scan in a similar manner to mapping continuous traits
by using `scan1`. When we mapped insulin, there was a hidden argument called 
`model` which told `qtl2` which mapping model to use. There are two options:
`normal`, the default, and `binary`. The `normal` argument tells `qtl2 to ues a
"normal" (least squares) linear model. To map a binary trait, we will 
include `model = "binary"` to indicate that the phenotype is a binary trait 
with values 0 and 1.

```{r binary_trait_scan}
lod_agouti <- scan1(genoprobs = probs, 
                    pheno     = cross$pheno[,'agouti_tan'], 
                    addcovar  = addcovar, 
                    model     = "binary")
```

<!-- DMG: Explain why we don't use kinship. It's because we don't have 
a model that can handle correlated residuals. It uses logistic regression
and we don't know how to adjust for correlated residuals. -->

Let's plot the result and see if there is a peak.

```{r plot_bin_scan}
plot_scan1(x    = lod_agouti, 
           map  = cross$pmap, 
           main = 'Agouti')
```

Yes! There is a big peak on chromosome 2. Let's zoom in on chromosome 2.

```{r plot_bin_scan_chr2}
plot_scan1(x    = lod_agouti, 
           map  = cross$pmap, 
           chr  = 2,
           main = 'Agouti')
```

We can use `find_peaks` to find the position of the highest LOD score.

```{r find_peaks_bin_scan, eval=FALSE}
find_peaks(scan1_output = lod_agouti, 
           map          = cross$pmap)
```

This turns out to be a well-known coat color locus for agouti coat color which
contains the [nonagouti](https://www.informatics.jax.org/marker/MGI:87853) gene.
Mice carrying two black alleles will have a black coat, and mice carrying
one or no black alleles will have agouti coats.

::::::::::::::::::::::::::::::::::::: challenge

## Challenge 1: How many mice have black coats?

Look at the frequency of the black (0) and agouti (1) phenotypes. What 
proportion of the mice are black? Can you use what you learned about how
the `nonagouti` locus works and the cross design to explain the frequency of 
black mice?

::::::::::::::::::::::::::::::::: solution

First, get the number of black and agouti mice.

```{r}
tbl <- table(cross$pheno[,"agouti_tan"])
tbl
```

Then use the number of mice to calculate the proportion with each coat color.

```{r}
tbl / sum(tbl)
```

We can see that the black (0) mice occur about 25 % of the time. If the `A` allele
causes mice to have black coats when it is recessive, and if `a` is the agouti
allele, then, when breeding two heterozygous (`Aa`) mice together, we expect
mean allele frequencies of:

Allele | Frequency | Coat Color
-------+-----------+-----------
  AA   |    0.25   |   black
  Aa   |    0.5    |   agouti
  aa   |    0.25   |   agouti

From this, we can see that about 25% of the mice should have black coats.

<!-- DMG: Fill this in with a figure showing the expected frequency of 
two allele from F1s. AA = 25%, Aa = 50% & aa = 25%. -->


::::::::::::::::::::::::::::::::::::::::::

## Challenge 2: Map the "tufted" phenotype.

Map the tufted phenotype an determine if there are any tall peaks for this 
trait.

::::::::::::::::::::::::::::::::: solution

First, map the trait.

```{r}
lod_tufted <- scan1(genoprobs = probs, 
                    pheno     = cross$pheno[,"tufted"], 
                    addcovar  = addcovar, 
                    model     = "binary")
```

Then, plot the LOD score.

```{r}
plot_scan1(x    = lod_tufted, 
           map  = cross$pmap, 
           main = "Tufted")
```

There is a large peak on chromosome 17. This is a 
[known locus](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3570182/) 
associated with the [Itpr3](https://www.informatics.jax.org/marker/MGI:96624)
gene near 27.3 Mb on chromsome 17.

::::::::::::::::::::::::::::::::::::::::::

:::::::::::::::::::::::::::::::::::::::::::::::


::::::::::::::::::::::::::::::::::::: keypoints 

- "A genome scan for binary traits (0 and 1) requires special handling; scans 
for non-binary traits assume normal variation of the residuals."
- "A genome scan for binary traits  is performed with logistic regression."

::::::::::::::::::::::::::::::::::::::::::::::::
